roachtest: activerecord failed

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13840930?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13840930?buildTab=artifacts#/activerecord) on master @ [d272e9ef5589deff570efc023db6c70edfde311c](https://github.com/cockroachdb/cockroach/commits/d272e9ef5589deff570efc023db6c70edfde311c):


```
(orm_helpers.go:221).summarizeFailed: 
Tests run on Cockroach v24.1.0-alpha.00000000-dev-d272e9ef5589deff570efc023db6c70edfde311c
Tests run against activerecord 7.1.3
8108 Total Tests Run
8081 tests passed
27 tests failed
66 tests skipped
0 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: BasicsTest#test_default_values_are_deeply_dupped (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-13840930/activerecord/1707116046131/1707129625137)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118725 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14167096?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14167096?buildTab=artifacts#/activerecord) on master @ [7bb52a7d1c75d5adfdfa53e5fcff6f5e6497408f](https://github.com/cockroachdb/cockroach/commits/7bb52a7d1c75d5adfdfa53e5fcff6f5e6497408f):


```
--- FAIL: RelationTest#test_primary_key (unexpected)
--- FAIL: RelationTest#test_reverse_order_with_function_other_predicates (unexpected)
--- FAIL: CalculationsTest#test_count_selected_arel_attribute (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_existing_record_twice_should_add_records_twice (unexpected)
--- FAIL: ReflectionTest#test_active_record_primary_key (unexpected)
--- FAIL: RelationTest#test_reverse_arel_assoc_order_with_function (unexpected)
--- FAIL: RelationTest#test_pluck_with_from_includes_original_table_name (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_singleton_has_many_through (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_delete_through_belongs_to_with_dependent_nullify (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_destroy_all_on_association_clears_scope (unexpected)
--- FAIL: CalculationsTest#test_should_return_integer_average_if_db_returns_such (unexpected)
--- FAIL: RelationTest#test_find_or_create_by (unexpected)
--- FAIL: CalculationsTest#test_should_perform_joined_include_when_referencing_included_tables (unexpected)
--- FAIL: CalculationsTest#test_ids_with_scope (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_field (unexpected)
--- FAIL: RelationTest#test_distinct (unexpected)
--- FAIL: CalculationsTest#test_count_should_shortcut_with_limit_zero (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_push_with_invalid_join_record (unexpected)
--- FAIL: RelationTest#test_relations_limit_the_records_in_#pretty_print_at_10 (unexpected)
--- FAIL: ReflectionTest#test_non_existent_types_are_identity_types (unexpected)
--- FAIL: RelationTest#test_select_with_subquery_string_in_from_does_not_use_original_table_name (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_unscope_raise_errors (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_left_outer_joins_raise_errors (unexpected)
--- FAIL: ReflectionTest#test_column_null_not_null (unexpected)
--- FAIL: CalculationsTest#test_should_count_with_manual_distinct_select_and_distinct (unexpected)
--- FAIL: CalculationsTest#test_limit_is_kept (unexpected)
--- FAIL: CalculationsTest#test_maximum_with_from_option (unexpected)
--- FAIL: RelationTest#test_finding_with_subquery_with_eager_loading_in_where (unexpected)
--- FAIL: CalculationsTest#test_sum_takes_attribute_type_precedence_over_database_type (unexpected)
--- FAIL: RelationTest#test_finding_with_complex_order_and_limit (unexpected)
--- FAIL: RelationTest#test_size_with_eager_loading_and_custom_order (unexpected)
--- FAIL: CalculationsTest#test_distinct_joins_count_with_group_by (unexpected)
--- FAIL: CalculationsTest#test_no_limit_no_offset (unexpected)
--- FAIL: CalculationsTest#test_should_count_field_in_joined_table_with_group_by (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_build_then_save_with_has_many_inverse (unexpected)
--- FAIL: ReflectionTest#test_automatic_inverse_suppresses_name_error_for_association (unexpected)
--- FAIL: CalculationsTest#test_should_count_selected_field_with_include (unexpected)
--- FAIL: RelationTest#test_loaded_relations_cannot_be_mutated_by_single_value_methods (unexpected)
--- FAIL: RelationTest#test_first_or_initialize_with_block (unexpected)
--- FAIL: RelationTest#test_first_or_initialize (unexpected)
--- FAIL: RelationTest#test_size_with_distinct (unexpected)
--- FAIL: RelationTest#test_finding_with_desc_order_with_string (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_field_with_conditions (unexpected)
--- FAIL: ReflectionTest#test_class_for_source_type (unexpected)
--- FAIL: RelationTest#test_anonymous_extension (unexpected)
--- FAIL: RelationTest#test_create_or_find_by_with_non_unique_attributes (unexpected)
--- FAIL: RelationTest#test_reorder_deduplication (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14167096/activerecord/1709016791627/1709025985522)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14183723?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14183723?buildTab=artifacts#/activerecord) on master @ [fed794450894a7d1197c763ed26f75d773b93070](https://github.com/cockroachdb/cockroach/commits/fed794450894a7d1197c763ed26f75d773b93070):


```
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_primary_key_counter (unexpected)
--- FAIL: BasicsTest#test_set_table_name_symbol_converted_to_string (unexpected)
--- FAIL: BasicsTest#test_quote (unexpected)
--- FAIL: InheritanceComputeTypeTest#test_instantiation_doesnt_try_to_require_corresponding_file (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit_and_conditions_array (unexpected)
--- FAIL: FinderTest#test_find_by_with_non-hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: HasManyAssociationsTest#test_select_query_method (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_single_has_many_through_association_with_unpersisted_parent_instance (unexpected)
--- FAIL: HasOneAssociationsTest#test_proxy_assignment (unexpected)
--- FAIL: BasicsTest#test_new_record_returns_boolean (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_assignment_with_primary_key_updates_foreign_id_field_for_new_and_saved_records (unexpected)
--- FAIL: PostgresqlTimestampTest#test_timestamp_with_zone_values_without_rails_time_zone_support (unexpected)
--- FAIL: AttributeMethodsTest#test_typecast_attribute_from_select_to_false (unexpected)
--- FAIL: FinderTest#test_hash_condition_utc_time_interpolation_with_default_timezone_local (unexpected)
--- FAIL: ActiveRecord::Migration::ForeignKeyChangeColumnWithSuffixTest#test_remove_foreign_key_by_column_in_change_table (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_limit_and_conditions (unexpected)
--- FAIL: FinderTest#test_find_by_id_with_hash (unexpected)
--- FAIL: FinderTest#test_with_limiting_with_custom_select (unexpected)
--- FAIL: HasManyAssociationsTest#test_building_the_associated_object_with_an_invalid_type (unexpected)
--- FAIL: HasOneAssociationsTest#test_creation_failure_due_to_new_record_should_raise_error (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_through_a_polymorphic_association_doesn't_require_the_association_to_exist (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_invalid_adding_with_validate_false (unexpected)
--- FAIL: FinderTest#test_include_on_unloaded_relation_with_having_referencing_aliased_select (unexpected)
--- FAIL: HasOneAssociationsTest#test_build_association_dont_create_transaction (unexpected)
--- FAIL: FinderTest#test_find_by_one_attribute_with_conditions (unexpected)
--- FAIL: InheritanceTest#test_new_with_invalid_type (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_assignment_before_parent_saved (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_count_has_many_through_with_named_scope (unexpected)
--- FAIL: AttributeMethodsTest#test_#id_value_alias_returns_the_value_in_the_id_column,_when_id_column_exists (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_touch_option_on_touch_without_updated_at_attributes (unexpected)
--- FAIL: HasManyAssociationsTest#test_in_memory_replacement_maintains_order (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_default_scope (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_association_with_bang (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_index_sort_order (unexpected)
--- FAIL: AssociationsTest#test_nullify_composite_foreign_key_belongs_to_association (unexpected)
--- FAIL: InheritanceTest#test_where_new_with_invalid_type (unexpected)
--- FAIL: BasicsTest#test_#connected_to_many_cannot_be_called_with_classes_that_include_ActiveRecord::Base (unexpected)
--- FAIL: BasicsTest#test_switching_between_table_name (unexpected)
--- FAIL: ReflectionTest#test_non_existent_columns_return_null_object (unexpected)
--- FAIL: EagerAssociationTest#test_loading_association_with_intersection_joins (unexpected)
--- FAIL: FinderTest#test_custom_select_takes_precedence_over_original_value (unexpected)
--- FAIL: AttributeMethodsTest#test_read_overridden_attribute_with_predicate_respects_override (unexpected)
--- FAIL: HasManyAssociationsTest#test_depends_and_nullify (unexpected)
--- FAIL: ActiveRecord::Migration::ForeignKeyChangeColumnWithSuffixTest#test_rename_column_of_child_table (unexpected)
--- FAIL: QuerySerializedParamTest#test_query_serialized_active_record (unexpected)
--- FAIL: BelongsToAssociationsTest#test_building_the_belonging_object_with_explicit_sti_base_class (unexpected)
--- FAIL: HasManyAssociationsTest#test_counting_with_counter_sql (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14183723/activerecord/1709103551843/1709112380713)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14193434?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14193434?buildTab=artifacts#/activerecord) on master @ [9a494f4e4846a67ceea18c4b4993cefbeb19c6d9](https://github.com/cockroachdb/cockroach/commits/9a494f4e4846a67ceea18c4b4993cefbeb19c6d9):


```
--- FAIL: HasAndBelongsToManyAssociationsTest#test_get_ids_for_unloaded_associations_does_not_load_them (unexpected)
--- FAIL: BelongsToAssociationsTest#test_default_scope_on_relations_is_not_cached (unexpected)
--- FAIL: OptimisticLockingTest#test_update_without_attributes_does_not_only_update_lock_version (unexpected)
--- FAIL: EnumTest#test_unicode_characters_for_enum_names (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_names_on_a_new_record (unexpected)
--- FAIL: AttributeMethodsTest#test_boolean_attributes_writing_and_reading (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_setting_foreign_key_after_nil_target_loaded (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_join_table_alias (unexpected)
--- FAIL: PersistenceTest#test_save_invalid_record (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_respects_select (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_build_a_model_from_hm_through_association_with_where_clause (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_include_returns_false_for_non_matching_record_to_verify_scoping (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_push_with_invalid_join_record (unexpected)
--- FAIL: AttributeMethodsTest#test_overridden_read_attribute (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_symbol_join_table (unexpected)
--- FAIL: YamlSerializationTest#test_deserializing_rails_v1_mysql_yaml (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_belongs_to_with_dependent_destroy (unexpected)
--- FAIL: BelongsToAssociationsTest#test_reassigning_the_parent_id_updates_the_object (unexpected)
--- FAIL: PersistenceTest#test_create_many_through_factory_with_block (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_delete_all_for_with_dependent_option_nullify (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_adding_before_save (unexpected)
--- FAIL: PersistenceTest#test_delete_new_record (unexpected)
--- FAIL: BelongsToAssociationsTest#test_counter_cache (unexpected)
--- FAIL: PersistenceTest#test_update_columns_with_default_scope (unexpected)
--- FAIL: PersistenceTest#test_update_many_with_active_record_base_object (unexpected)
--- FAIL: PostgresqlDateTest#test_bc_date (unexpected)
--- FAIL: DefaultScopingTest#test_nilable_default_scope_with_all_queries_runs_on_create (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_is_precise_enough (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_through_scope_with_includes (unexpected)
--- FAIL: EnumTest#test_enum_on_custom_attribute_with_default (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_symbols_as_keys (unexpected)
--- FAIL: BelongsToAssociationsTest#test_destroying_child_with_unloaded_parent_and_foreign_key_and_touch_is_possible_with_has_many_inversing (unexpected)
--- FAIL: PersistenceTest#test_update_after_create (unexpected)
--- FAIL: YamlSerializationTest#test_types_of_virtual_columns_are_not_changed_on_round_trip (unexpected)
--- FAIL: IntegrationTest#test_cache_key_retains_version_when_custom_timestamp_is_used (unexpected)
--- FAIL: PersistenceTest#test_update_many! (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_through_record_is_built_when_created_with_where (unexpected)
--- FAIL: DefaultScopingTest#test_default_scope_with_all_queries_runs_on_update_columns (unexpected)
--- FAIL: PersistenceTest#test_update_many_with_active_record_base_object! (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_accept_string_and_hash_join (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_method_on_an_abstract_class_is_available_on_subclasses (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_replace_association (unexpected)
--- FAIL: PersistenceTest#test_populates_non_primary_key_autoincremented_column (unexpected)
--- FAIL: BelongsToAssociationsTest#test_failing_create! (unexpected)
--- FAIL: AttributeMethodsTest#test_global_methods_are_overwritten (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_adding_type_mismatch (unexpected)
--- FAIL: DefaultScopingTest#test_default_scope_with_all_queries_runs_on_destroy (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14193434/activerecord/1709189695457/1709198586134)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14204270?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14204270?buildTab=artifacts#/activerecord) on master @ [d4a51117deac22a2ffa280bac5404811cd694265](https://github.com/cockroachdb/cockroach/commits/d4a51117deac22a2ffa280bac5404811cd694265):


```
(orm_helpers.go:221).summarizeFailed: 
Tests run on Cockroach v24.1.0-alpha.2-dev-d4a51117deac22a2ffa280bac5404811cd694265
Tests run against activerecord 7.1.3
8108 Total Tests Run
8079 tests passed
29 tests failed
66 tests skipped
0 tests ignored
0 tests passed unexpectedly
3 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_create_fixtures_resets_sequences_when_not_cached (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_default_pk_and_sequence (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14204270/activerecord/1709275840449/1709283464562)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14214555?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14214555?buildTab=artifacts#/activerecord) on master @ [791969fa98711e681289fa4f61f3d58c8fe44263](https://github.com/cockroachdb/cockroach/commits/791969fa98711e681289fa4f61f3d58c8fe44263):


```
--- FAIL: BasicsTest#test_dup (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: RelationTest#test_dynamic_finder (unexpected)
--- FAIL: FinderTest#test_include_on_unloaded_relation_without_match (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_order_raise_errors (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_custom_join_table (unexpected)
--- FAIL: IntegrationTest#test_cache_key_retains_version_when_custom_timestamp_is_used (unexpected)
--- FAIL: FinderTest#test_first_have_determined_order_by_default (unexpected)
--- FAIL: FinderTest#test_select_values (unexpected)
--- FAIL: FinderTest#test_fourth_with_offset (unexpected)
--- FAIL: FinderTest#test_find_by_nil_and_not_nil_attributes (unexpected)
--- FAIL: IntegrationTest#test_to_param_for_a_composite_primary_key_model (unexpected)
--- FAIL: RelationTest#test_find_all_using_where_with_relation_does_not_alter_select_values (unexpected)
--- FAIL: FinderTest#test_find_only_some_columns (unexpected)
--- FAIL: BasicsTest#test_all_with_conditions (unexpected)
--- FAIL: RelationTest#test_scoped_all (unexpected)
--- FAIL: BasicsTest#test_cannot_call_connects_to_on_non-abstract_or_non-ActiveRecord::Base_classes (unexpected)
--- FAIL: RelationTest#test_group_with_subquery_string_in_from_does_not_use_original_table_name (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_include? (unexpected)
--- FAIL: FinderTest#test_find_by_doesn't_have_implicit_ordering (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_polymorphic_with_primary_key_option (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_has_and_belongs_to_many_associations_on_new_records_use_null_relations (unexpected)
--- FAIL: RelationTest#test_reverse_arel_assoc_order_with_function (unexpected)
--- FAIL: HasOneThroughDisableJoinsAssociationsTest#test_preload_on_disable_joins_through (unexpected)
--- FAIL: BasicsTest#test_singular_table_name_guesses_with_prefixes_and_suffixes (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_should_save_parent_but_not_invalid_child (unexpected)
--- FAIL: HasOneThroughDisableJoinsAssociationsTest#test_counting_on_disable_joins_through (unexpected)
--- FAIL: RelationTest#test_size_with_distinct (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_select_chosen_fields_only (unexpected)
--- FAIL: FinderTest#test_find_with_entire_select_statement (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_collection_singular_ids_setter (unexpected)
--- FAIL: FinderTest#test_model_class_responds_to_third_to_last_bang (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_destroy_all_on_composite_primary_key_model (unexpected)
--- FAIL: InheritanceAttributeMappingTest#test_sti_with_custom_type (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_is_precise_enough (unexpected)
--- FAIL: RelationTest#test_first_or_create (unexpected)
--- FAIL: RelationTest#test_reorder_with_take (unexpected)
--- FAIL: AssociationsTest#test_belongs_to_association_does_not_use_parent_query_constraints_if_not_configured_to (unexpected)
--- FAIL: FinderTest#test_last_with_irreversible_order (unexpected)
--- FAIL: ReflectionTest#test_never_validate_association_if_explicit (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_get_ids_for_has_many_through_with_conditions_should_not_preload (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_destroying (unexpected)
--- FAIL: ReflectionTest#test_belongs_to_reflection_with_array_fk_raises (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_with_extend_option (unexpected)
--- FAIL: FinderTest#test_find_by_one_attribute_that_is_an_alias (unexpected)
--- FAIL: FinderTest#test_exists_with_empty_table_and_no_args_given (unexpected)
--- FAIL: RelationTest#test_default_scope_order_with_scope_order (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14214555/activerecord/1709362221176/1709370906625)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14220839?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14220839?buildTab=artifacts#/activerecord) on master @ [7f45cb61df77419ded37619f701993ee410f6951](https://github.com/cockroachdb/cockroach/commits/7f45cb61df77419ded37619f701993ee410f6951):


```
--- FAIL: RelationTest#test_select_takes_an_aliased_attribute (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_multiple_fields (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates_after_parameterize (unexpected)
--- FAIL: RelationTest#test_finding_with_cross_table_order_and_limit (unexpected)
--- FAIL: FinderTest#test_find_with_ids_returning_ordered (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_all (unexpected)
--- FAIL: CalculationsTest#test_should_sum_scoped_field (unexpected)
--- FAIL: RelationTest#test_build (unexpected)
--- FAIL: RelationTest#test_first_or_create_with_array (unexpected)
--- FAIL: RelationTest#test_reverse_order_with_nulls_first_or_last (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_with_create_and_no_options (unexpected)
--- FAIL: AttributeMethodsTest#test_inherited_custom_accessors (unexpected)
--- FAIL: PersistenceTest#test_update_column_should_not_modify_updated_at (unexpected)
--- FAIL: HasManyAssociationsTest#test_custom_named_counter_cache (unexpected)
--- FAIL: RelationTest#test_find_id (unexpected)
--- FAIL: BelongsToAssociationsTest#test_assignment_before_child_saved (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_reassignment_of_associated_type_updates_the_object (unexpected)
--- FAIL: PersistenceTest#test_update_column (unexpected)
--- FAIL: CalculationsTest#test_sum_uses_enumerable_version_when_block_is_given (unexpected)
--- FAIL: CalculationsTest#test_distinct_count_with_order_and_offset (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_inheritance (unexpected)
--- FAIL: ReflectionTest#test_join_table (unexpected)
--- FAIL: AssociationValidationTest#test_validates_associated_one (unexpected)
--- FAIL: BasicsTest#test_default_values (unexpected)
--- FAIL: RelationTest#test_relations_limit_the_records_in_#pretty_print_at_10 (unexpected)
--- FAIL: RelationTest#test_find_or_initialize_by_with_block (unexpected)
--- FAIL: EagerAssociationTest#test_eager_load_has_many_with_string_keys (unexpected)
--- FAIL: CalculationsTest#test_distinct_count_all_with_custom_select_and_order (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_associations_sum_on_columns (unexpected)
--- FAIL: RelationTest#test_automatically_added_where_not_references (unexpected)
--- FAIL: QuerySerializedParamTest#test_query_serialized_active_record (unexpected)
--- FAIL: SerializedAttributeTestWithYamlSafeLoad#test_serialized_boolean_value_false (unexpected)
--- FAIL: EagerAssociationTest#test_scoping_with_a_circular_preload (unexpected)
--- FAIL: SerializedAttributeTestWithYamlSafeLoad#test_where_by_serialized_attribute_with_hash (unexpected)
--- FAIL: HasManyAssociationsTest#test_included_in_collection (unexpected)
--- FAIL: BelongsToAssociationsTest#test_tracking_polymorphic_changes (unexpected)
--- FAIL: HasManyAssociationsTest#test_create_the_association_with_an_array (unexpected)
--- FAIL: FinderTest#test_select_value (unexpected)
--- FAIL: SerializedAttributeTest#test_serialized_boolean_value_false (unexpected)
--- FAIL: RelationTest#test_order_with_blank_value (unexpected)
--- FAIL: CalculationsTest#test_pick_loaded_relation_multiple_columns (unexpected)
--- FAIL: RelationTest#test_only (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_with_custom_name_counter_cache (unexpected)
--- FAIL: AttributeMethodsTest#test_define_attribute_method_works_with_both_symbol_and_string (unexpected)
--- FAIL: HasManyAssociationsTest#test_building_the_associated_object_with_sti_subclass (unexpected)
--- FAIL: PersistenceTest#test_increment_attribute_by (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_indexes_for_existent_table (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14220839/activerecord/1709448743344/1709457902450)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14227315?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14227315?buildTab=artifacts#/activerecord) on master @ [660a3c8050a8ef7828b0151e44a86335639a8f9d](https://github.com/cockroachdb/cockroach/commits/660a3c8050a8ef7828b0151e44a86335639a8f9d):


```
--- FAIL: FinderTest#test_exists_with_distinct_association_includes_and_limit (unexpected)
--- FAIL: BasicsTest#test_find_reverse_ordered_last (unexpected)
--- FAIL: HasOneAssociationsTest#test_reload_association_with_query_cache (unexpected)
--- FAIL: FinderTest#test_find_by!_doesn't_have_implicit_ordering (unexpected)
--- FAIL: FinderTest#test_exists_uses_existing_scope (unexpected)
--- FAIL: HasOneAssociationsTest#test_association_enum_works_properly_with_nested_join (unexpected)
--- FAIL: FinderTest#test_#find_with_a_single_composite_primary_key (unexpected)
--- FAIL: PersistenceTest#test_save_destroyed_object (unexpected)
--- FAIL: FinderTest#test_find_with_prepared_select_statement (unexpected)
--- FAIL: FinderTest#test_find_with_ids_where_and_limit (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_field_through_association_and_having (unexpected)
--- FAIL: HasOneAssociationsTest#test_destroyed_by_association_set_in_child_destroy_callback_on_replace (unexpected)
--- FAIL: FinderTest#test_find_with_a_multiple_sets_of_composite_primary_key_wrapped_in_an_array (unexpected)
--- FAIL: BasicsTest#test_toggle_attribute (unexpected)
--- FAIL: ReflectionTest#test_reflection_klass_not_found_with_pointer_to_non_existent_class_name (unexpected)
--- FAIL: ReflectionTest#test_join_table_can_be_overridden (unexpected)
--- FAIL: BasicsTest#test_marshalling_new_record_round_trip_with_associations (unexpected)
--- FAIL: BasicsTest#test_.columns_hash_raises_an_error_if_the_record_has_an_empty_table_name (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_fields_with_table_alias (unexpected)
--- FAIL: HasOneAssociationsTest#test_build_and_create_should_not_happen_within_scope (unexpected)
--- FAIL: BasicsTest#test_abstract_class_table_name (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_includes_limit_and_empty_result (unexpected)
--- FAIL: FinderTest#test_find_by_ids_with_limit_and_offset (unexpected)
--- FAIL: BasicsTest#test_using_table_name_qualified_column_names_unless_having_SELECT_list_explicitly (unexpected)
--- FAIL: ModulesTest#test_module_spanning_has_and_belongs_to_many_associations (unexpected)
--- FAIL: CalculationsTest#test_ids_with_includes_and_non_primary_key_order (unexpected)
--- FAIL: ReflectionTest#test_non_existent_columns_return_null_object (unexpected)
--- FAIL: FinderTest#test_find_with_ids_with_id_out_of_range (unexpected)
--- FAIL: BasicsTest#test_equality_of_new_records (unexpected)
--- FAIL: ReflectionTest#test_human_name (unexpected)
--- FAIL: CalculationsTest#test_count_with_eager_loading_and_custom_order (unexpected)
--- FAIL: ReflectionTest#test_chain (unexpected)
--- FAIL: CalculationsTest#test_count_should_shortcut_with_limit_zero (unexpected)
--- FAIL: FinderTest#test_named_bind_variables_with_quotes (unexpected)
--- FAIL: BasicsTest#test_count_with_join (unexpected)
--- FAIL: ReflectionTest#test_never_validate_association_if_explicit (unexpected)
--- FAIL: FinderTest#test_bind_variables_with_quotes (unexpected)
--- FAIL: BasicsTest#test_utc_as_time_zone_and_new (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_field_with_conditions (unexpected)
--- FAIL: BasicsTest#test_hashing (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_multiple_ranges (unexpected)
--- FAIL: BasicsTest#test_table_name_guesses_with_prefixes_and_suffixes (unexpected)
--- FAIL: FinderTest#test_select_rows (unexpected)
--- FAIL: BasicsTest#test_set_table_name_with_inheritance (unexpected)
--- FAIL: CalculationsTest#test_minimum_and_maximum_on_non_numeric_type (unexpected)
--- FAIL: IntegrationTest#test_to_param_for_a_composite_primary_key_model (unexpected)
--- FAIL: FinderTest#test_sole_failing_many (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14227315/activerecord/1709535215224/1709544770422)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14238618?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14238618?buildTab=artifacts#/activerecord) on master @ [bf013ea0a5311726e65d37e8f047ce39ea2d5f10](https://github.com/cockroachdb/cockroach/commits/bf013ea0a5311726e65d37e8f047ce39ea2d5f10):


```
(orm_helpers.go:221).summarizeFailed: 
Tests run on Cockroach v24.1.0-alpha.2-dev-bf013ea0a5311726e65d37e8f047ce39ea2d5f10
Tests run against activerecord 7.1.3
8108 Total Tests Run
8079 tests passed
29 tests failed
66 tests skipped
0 tests ignored
0 tests passed unexpectedly
3 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_default_pk_and_sequence (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_create_fixtures_resets_sequences_when_not_cached (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14238618/activerecord/1709621515199/1709631146767)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14256896?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14256896?buildTab=artifacts#/activerecord) on master @ [ce5f34ea97475f45fa354e58aacf424779d0de49](https://github.com/cockroachdb/cockroach/commits/ce5f34ea97475f45fa354e58aacf424779d0de49):


```
--- FAIL: SignedIdTest#test_fail_to_find_record_from_that_has_since_been_destroyed (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_aggregate_having_three_mappings_array (unexpected)
--- FAIL: DefaultScopingTest#test_unscope_comparison_where_clauses (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_associations_sum_on_columns (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_yaml_loads_5_1_dump_without_indexes_still_queries_for_indexes (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_alternate_database (unexpected)
--- FAIL: UniquenessValidationTest#test_validates_uniqueness_with_newline_chars (unexpected)
--- FAIL: DefaultScopingTest#test_default_scope_as_class_method (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_invalid_dependent_option_raises_exception (unexpected)
--- FAIL: JsonAttributeTest#test_duplication_with_store_accessors (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_proper_usage_of_primary_keys_and_join_table (unexpected)
--- FAIL: BasicsTest#test_equality (unexpected)
--- FAIL: EagerAssociationTest#test_finding_with_includes_on_empty_polymorphic_type_column (unexpected)
--- FAIL: PostgresqlTimestampWithAwareTypesTest#test_timestamp_with_zone_values_with_rails_time_zone_support_and_time_zone_set (unexpected)
--- FAIL: BelongsToAssociationsTest#test_multiple_counter_cache_with_after_create_update (unexpected)
--- FAIL: AttributeMethodsTest#test_#define_attribute_methods_brings_back_undefined_aliases (unexpected)
--- FAIL: AttributeMethodsTest#test_integers_as_nil (unexpected)
--- FAIL: BelongsToAssociationsTest#test_reflect_the_most_recent_change (unexpected)
--- FAIL: CalculationsTest#test_ids_with_includes_and_non_primary_key_order (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_where_first_level_returns_nil (unexpected)
--- FAIL: ActiveRecord::SelectTest#test_select_with_hash_and_table_alias (unexpected)
--- FAIL: AttributeMethodsTest#test_user-defined_date_attribute_predicate (unexpected)
--- FAIL: ActiveRecord::Type::IntegerTest#test_values_which_are_out_of_range_can_be_re-assigned (unexpected)
--- FAIL: BelongsToAssociationsTest#test_natural_assignment (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_through_has_one_association_with_order (unexpected)
--- FAIL: BasicsTest#test_#connected_to_many_with_a_multiple_classes_without_brackets_works (unexpected)
--- FAIL: CalculationsTest#test_should_count_scoped_select_with_options (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_distinct_option_prevents_duplicate_push (unexpected)
--- FAIL: FinderTest#test_member_when_non_AR_object_passed_on_loaded_relation (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_custom_join_table (unexpected)
--- FAIL: ActiveRecord::DupTest#test_dup_not_destroyed (unexpected)
--- FAIL: BelongsToAssociationsTest#test_creating_the_belonging_object_from_new_record (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_belongs_to_association_SQL (unexpected)
--- FAIL: AttributeMethodsTest#test_time_zone-aware_attribute_saved (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_existing_with_nonstandard_primary_key_on_belongs_to (unexpected)
--- FAIL: JsonAttributeTest#test_yaml_round_trip_with_store_accessors (unexpected)
--- FAIL: FinderTest#test_find_by_id_with_conditions_with_or (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_an_overridden_original_method_along_with_an_overridden_alias_method_doesn't_issue_a_deprecation (unexpected)
--- FAIL: AssociationCallbacksTest#test_has_and_belongs_to_many_after_add_called_after_save (unexpected)
--- FAIL: CalculationsTest#test_ids_for_a_composite_primary_key_with_scope (unexpected)
--- FAIL: CalculationsTest#test_calculation_with_query_cache (unexpected)
--- FAIL: PostGISTest#test_set_and_get_point (unexpected)
--- FAIL: FinderTest#test_first_have_determined_order_by_default (unexpected)
--- FAIL: ModulesTest#test_table_name (unexpected)
--- FAIL: ModulesTest#test_nested_models_should_not_raise_exception_when_using_delete_all_dependency_on_association (unexpected)
--- FAIL: BelongsToAssociationsTest#test_building_the_belonging_object_with_primary_key (unexpected)
--- FAIL: TimestampTest#test_touching_a_record_touches_polymorphic_record (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14256896/activerecord/1709707755077/1709714940731)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14274864?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14274864?buildTab=artifacts#/activerecord) on master @ [f4dc2b59541e11cf06e2b948c7c37bf1aed21868](https://github.com/cockroachdb/cockroach/commits/f4dc2b59541e11cf06e2b948c7c37bf1aed21868):


```
63 tests failed
66 tests skipped
0 tests ignored
0 tests passed unexpectedly
37 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_marshal_dump_and_load_with_ignored_tables (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_has_many_sti (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_two_levels_with_two_has_many_associations (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_three_levels_by_ping_pong (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_two_levels (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_recursive_cascading_four_levels_has_many_through (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_preloading_across_has_one_constrains_loaded_records (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_cascaded_eager_association_loading_with_duplicated_includes (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_multiple_stis_and_order (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_join_for_count (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_has_many_sti_and_subclasses (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_grafts_stashed_associations_to_correct_parent (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_yaml_loads_5_1_dump_without_indexes_still_queries_for_indexes (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_recursive_cascading_four_levels_has_and_belongs_to_many (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_two_levels_with_condition (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_of_stis_with_multiple_references (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_two_levels_and_one_level (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_preloading_across_has_one_through_constrains_loaded_records (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_missing_first_record (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_marshal_dump_and_load_with_gzip (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_hmt_does_not_table_name_collide_when_joining_associations (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_cascaded_eager_association_loading_with_twice_includes_edge_cases (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_create_fixtures_resets_sequences_when_not_cached (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_preloaded_records_are_not_duplicated (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_two_levels_and_self_table_reference (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_where_first_level_returns_nil (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_marshal_dump_and_load (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_cascaded_eager_association_loading_with_join_for_count (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_belongs_to_sti (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_indexes_for_existent_table (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_interdependent_one_level_and_two_levels (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_default_pk_and_sequence (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_preload_through_missing_records (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_yaml_dump_and_load (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_marshal_dump_and_load_via_disk (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_yaml_dump_and_load_with_gzip (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14274864/activerecord/1709793945662/1709803876019)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14291292?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14291292?buildTab=artifacts#/activerecord) on master @ [559994e230ac3776dd1188e66678cce9205ab345](https://github.com/cockroachdb/cockroach/commits/559994e230ac3776dd1188e66678cce9205ab345):


```
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_assign_ids (unexpected)
--- FAIL: HasOneAssociationsTest#test_polymorphic_has_one_with_touch_option_on_update_will_touch_record_by_fetching_from_database_if_needed (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_get_ids_for_unloaded_associations_does_not_load_them (unexpected)
--- FAIL: PolymorphicFullClassNamesTest#test_class_names_with_find_by (unexpected)
--- FAIL: RelationTest#test_optimizer_hints_with_blank_value (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_collection_size_from_build (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_attributes_are_being_set_when_initialized_from_hm_through_association_with_multiple_where_clauses (unexpected)
--- FAIL: FinderTest#test_find_by_bang_on_relation_with_large_number (unexpected)
--- FAIL: BelongsToAssociationsTest#test_create_with_block (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_delete_all_for_with_dependent_option_destroy (unexpected)
--- FAIL: StrictLoadingTest#test_raises_on_lazy_loading_a_belongs_to_relation_if_strict_loading_by_default (unexpected)
--- FAIL: TransactionInstrumentationTest#test_transaction_instrumentation_with_restart_savepoint_parent_transactions_on_commit (unexpected)
--- FAIL: FinderTest#test_find_with_prepared_select_statement (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependent_association_respects_optional_hash_conditions_on_delete (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_one_dependent_does_not_destroy_dependent (unexpected)
--- FAIL: PostGISTest#test_spatial_factory_attrs_parsing (unexpected)
--- FAIL: FinderTest#test_find_by_one_attribute_that_is_an_alias (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_default_scope_as_callable (unexpected)
--- FAIL: IntegrationTest#test_param_delimiter_is_defined_per_class (unexpected)
--- FAIL: BelongsToAssociationsTest#test_association_assignment_sticks (unexpected)
--- FAIL: RelationTest#test_finding_with_reorder_by_aliased_attributes (unexpected)
--- FAIL: BasicsTest#test_table_name_with_2_abstract_subclasses (unexpected)
--- FAIL: ValidationsTest#test_save_without_validation (unexpected)
--- FAIL: InheritanceTest#test_inheritance_new_with_default_class (unexpected)
--- FAIL: RelationTest#test_reverse_order_with_function_other_predicates (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_through_empty_belongs_to (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_recursive_cascading_four_levels_has_many_through (unexpected)
--- FAIL: BelongsToAssociationsTest#test_natural_assignment (unexpected)
--- FAIL: HasManyAssociationsTest#test_updates_counter_cache_when_default_scope_is_given (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_with_scope_with_joins (unexpected)
--- FAIL: RelationTest#test_order_doesnt_trigger_eager_loading_when_ordering_using_the_owner_table_and_hash_syntax (unexpected)
--- FAIL: InheritanceTest#test_becomes_and_change_tracking_for_inheritance_columns (unexpected)
--- FAIL: BasicsTest#test_column_types_typecast (unexpected)
--- FAIL: RelationTest#test_distinct (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_empty_belongs_to_polymorphic (unexpected)
--- FAIL: BasicsTest#test_bignum_pk (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_preloaded_records_are_not_duplicated (unexpected)
--- FAIL: RelationTest#test_destroy_by (unexpected)
--- FAIL: HasManyAssociationsTest#test_joining_through_a_polymorphic_association_with_a_where_clause (unexpected)
--- FAIL: DelegatedTypeTest#test_association_id (unexpected)
--- FAIL: RelationTest#test_reverse_order_with_reorder_nil_removes_the_order (unexpected)
--- FAIL: DirtyTest#test_changes_is_correct_for_subclass (unexpected)
--- FAIL: BasicsTest#test_reload (unexpected)
--- FAIL: EagerAssociationTest#test_eager_load_has_many_through_quotes_table_and_column_names (unexpected)
--- FAIL: EagerAssociationTest#test_including_associations_with_extensions_and_an_instance_dependent_scope_is_supported (unexpected)
--- FAIL: RelationTest#test_count_explicit_columns (unexpected)
--- FAIL: IntegrationTest#test_cache_version_format_is_precise_enough (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14291292/activerecord/1709880552704/1709889310174)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14306124?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14306124?buildTab=artifacts#/activerecord) on master @ [72646a555214c0705781e440b9df585d5eea9511](https://github.com/cockroachdb/cockroach/commits/72646a555214c0705781e440b9df585d5eea9511):


```
--- FAIL: EagerAssociationTest#test_has_many_through_with_order (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_the_same_association_twice_works (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_does_not_use_order_by (unexpected)
--- FAIL: YamlSerializationTest#test_yaml_encoding_keeps_false_values (unexpected)
--- FAIL: RelationTest#test_group_with_subquery_string_in_from_does_not_use_original_table_name (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_two_levels_and_one_level (unexpected)
--- FAIL: RelationTest#test_find_with_preloaded_associations (unexpected)
--- FAIL: DefaultScopingTest#test_default_scope_runs_on_select (unexpected)
--- FAIL: RelationTest#test_find_by_returns_nil_if_the_record_is_missing (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_association_replaces_existing_with_dependent_option (unexpected)
--- FAIL: EagerAssociationTest#test_limited_eager_with_order (unexpected)
--- FAIL: RelationTest#test_blank_like_arguments_to_query_methods_dont_raise_errors (unexpected)
--- FAIL: DefaultScopingTest#test_default_scope_with_conditions_hash (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_get_ids_for_loaded_associations (unexpected)
--- FAIL: CallbacksTest#test_before_save_throwing_abort (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_touch_option_on_update (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_has_many_through_polymorphic_has_manys_works (unexpected)
--- FAIL: RelationTest#test_finding_with_reorder_by_aliased_attributes (unexpected)
--- FAIL: BasicsTest#test_create_without_prepared_statement (unexpected)
--- FAIL: BelongsToAssociationsTest#test_with_polymorphic_and_condition (unexpected)
--- FAIL: TestDestroyAsPartOfAutosaveAssociation#test_should_run_remove_callback_methods_for_habtm (unexpected)
--- FAIL: PrimaryKeysTest#test_id_column_that_is_not_primary_key (unexpected)
--- FAIL: BasicsTest#test_cannot_call_connects_to_on_non-abstract_or_non-ActiveRecord::Base_classes (unexpected)
--- FAIL: DefaultScopingTest#test_nilable_default_scope_with_all_queries_runs_on_update_columns (unexpected)
--- FAIL: BasicsTest#test_previously_changed_dup (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_assignment_before_either_saved (unexpected)
--- FAIL: RelationTest#test_finding_with_hash_conditions_on_joined_table (unexpected)
--- FAIL: EagerAssociationTest#test_preload_has_one_using_primary_key (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_valid_adding_with_validate_false (unexpected)
--- FAIL: BasicsTest#test_can_call_connected_to_with_role_and_shard_on_abstract_classes (unexpected)
--- FAIL: BelongsToAssociationsTest#test_build_with_conditions (unexpected)
--- FAIL: RelationTest#test_already-loaded_relations_don't_perform_a_new_query_in_#pretty_print (unexpected)
--- FAIL: BelongsToAssociationsTest#test_natural_assignment_to_nil (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_touch_option_on_touch_and_reassigned_parent (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_with_validate_false_does_not_run_associated_validation_callbacks_on_update (unexpected)
--- FAIL: RelationTest#test_reverse_arel_assoc_order_with_function (unexpected)
--- FAIL: AttributeMethodsTest#test_inherited_custom_accessors_with_reserved_names (unexpected)
--- FAIL: PersistenceTest#test_destroy_record_with_associations (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_reassign_with_namespaced_models_and_counters (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_collection_size_from_build (unexpected)
--- FAIL: BelongsToAssociationsTest#test_self_referential_belongs_to_with_counter_cache_assigning_nil (unexpected)
--- FAIL: PrimaryKeysTest#test_customized_primary_key_can_be_get_before_saving (unexpected)
--- FAIL: RelationTest#test_size_with_eager_loading_and_custom_order (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_assignment_dont_trigger_save_on_change_of_same_object (unexpected)
--- FAIL: AssociationValidationTest#test_validates_associated_one (unexpected)
--- FAIL: PersistenceTest#test_build_many_through_factory_with_block (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14306124/activerecord/1709966810582/1709976998393)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14316621?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14316621?buildTab=artifacts#/activerecord) on master @ [41da9eccc216413c09cc66ad4c6bfa5f92c506e3](https://github.com/cockroachdb/cockroach/commits/41da9eccc216413c09cc66ad4c6bfa5f92c506e3):


```
--- FAIL: RelationTest#test_joins_with_order_by_custom_attribute (unexpected)
--- FAIL: BasicsTest#test_connection_in_local_time (unexpected)
--- FAIL: RelationTest#test_group_with_select_and_includes (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_invalid_adding (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_method? (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit_and_conditions_on_the_eagers (unexpected)
--- FAIL: RelationTest#test_find_all_using_where_with_relation_with_joins (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_left_joins_raise_errors (unexpected)
--- FAIL: I18nGenerateMessageValidationTest#test_translation_for_'taken'_can_be_overridden (unexpected)
--- FAIL: MultiParameterAttributeTest#test_multiparameter_attributes_on_date_with_empty_month (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_get_ids_for_unloaded_associations_does_not_load_them (unexpected)
--- FAIL: RelationTest#test_order_with_hash_and_symbol_generates_the_same_sql (unexpected)
--- FAIL: BasicsTest#test_marshalling_with_associations (unexpected)
--- FAIL: UniquenessValidationWithIndexTest#test_uniqueness_on_relation (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_preloading_empty_through_association_via_joins (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_a_time_zone_conversion_for_attributes_should_write_the_value_on_a_class_variable (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_yaml_loads_5_1_dump_without_indexes_still_queries_for_indexes (unexpected)
--- FAIL: IntegrationTest#test_cache_key_for_existing_record_is_not_timezone_dependent (unexpected)
--- FAIL: BasicsTest#test_default_values_are_deeply_dupped (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attribute (unexpected)
--- FAIL: EagerAssociationTest#test_eager_habtm_with_association_inheritance (unexpected)
--- FAIL: RelationTest#test_loading_with_one_association_with_non_preload (unexpected)
--- FAIL: ActiveRecord::RelationTest#test_selecting_aliased_attribute_quotes_column_name_when_from_is_used (unexpected)
--- FAIL: EagerAssociationTest#test_preconfigured_includes_with_habtm (unexpected)
--- FAIL: BasicsTest#test_can_call_connected_to_with_role_and_shard_on_abstract_classes (unexpected)
--- FAIL: ModulesTest#test_assign_ids (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_assignment_before_parent_saved (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_for_single_table_inheritance_(STI_Models) (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_with_a_bang_on_relation (unexpected)
--- FAIL: ActiveRecord::Migration::ForeignKeyChangeColumnWithSuffixTest#test_remove_reference_column_of_child_table (unexpected)
--- FAIL: ActiveRecord::Migration::ColumnsTest#test_rename_column_using_symbol_arguments (unexpected)
--- FAIL: EnumTest#test_validate_inclusion_of_value_in_array (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_belongs_to_with_cpk (unexpected)
--- FAIL: DirtyTest#test_in_place_mutation_for_binary (unexpected)
--- FAIL: StrictLoadingTest#test_does_not_raise_on_eager_loading_a_habtm_relation_if_strict_loading_by_default (unexpected)
--- FAIL: AttributeMethodsTest#test_hashes_are_not_mangled (unexpected)
--- FAIL: RelationTest#test_loaded_relations_cannot_be_mutated_by_single_value_methods (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_replace_association_with_duplicates (unexpected)
--- FAIL: RelationTest#test_find_or_create_by_with_block (unexpected)
--- FAIL: BasicsTest#test_load (unexpected)
--- FAIL: RelationTest#test_loaded_relations_cannot_be_mutated_by_merge! (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_preload_with_nested_association (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_through_association_with_through_scope_and_nested_where (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_with_a_bang_with_custom_primary_key (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_replace_on_new_object (unexpected)
--- FAIL: SerializedAttributeTestWithYamlSafeLoad#test_serialized_attributes_from_database_on_subclass (unexpected)
--- FAIL: BasicsTest#test_readonly_attributes_in_abstract_class_descendant (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14316621/activerecord/1710053372553/1710061865792)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14327141?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14327141?buildTab=artifacts#/activerecord) on master @ [41da9eccc216413c09cc66ad4c6bfa5f92c506e3](https://github.com/cockroachdb/cockroach/commits/41da9eccc216413c09cc66ad4c6bfa5f92c506e3):


```
--- FAIL: HasOneAssociationsTest#test_dependence (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_does_not_use_order_by (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroying_a_collection (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_assignment_dont_trigger_save_on_change_of_same_object (unexpected)
--- FAIL: ReloadAssociationCacheTest#test_reload_sets_correct_owner_for_association_cache (unexpected)
--- FAIL: AttributeMethodsTest#test_bulk_updates_respect_access_control (unexpected)
--- FAIL: HasManyAssociationsTest#test_collection_not_empty_after_building (unexpected)
--- FAIL: HasManyAssociationsTest#test_building_the_associated_object_with_an_invalid_type (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attribute (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroying_by_string_id (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_with_reflection_using_class_name_and_fixtures (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_associations_with_conditions (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_with_touch_option_on_create (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_first_or_last_with_integer_on_association_should_not_load_association (unexpected)
--- FAIL: AttributeMethodsTest#test_inherited_custom_accessors_with_reserved_names (unexpected)
--- FAIL: HasManyAssociationsTest#test_reload_with_query_cache (unexpected)
--- FAIL: HasManyAssociationsTest#test_clearing_updates_counter_cache (unexpected)
--- FAIL: HasManyAssociationsTest#test_can_unscope_the_default_scope_of_the_associated_model (unexpected)
--- FAIL: HasOneAssociationsTest#test_building_the_associated_object_with_explicit_sti_base_class (unexpected)
--- FAIL: AttributeMethodsTest#test_write_attribute (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_models_with_composite_keys (unexpected)
--- FAIL: AttributeMethodsTest#test_#id_value_alias_returns_id_column_only_for_composite_primary_key_models (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_assign_ids (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_assignment_before_either_saved (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_by_integer_id (unexpected)
--- FAIL: HasManyAssociationsTest#test_clear_collection_should_not_change_updated_at (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_build_by_new_record (unexpected)
--- FAIL: HasManyAssociationsTest#test_unscopes_the_default_scope_of_associated_model_when_used_with_include (unexpected)
--- FAIL: HasManyAssociationsTest#test_adding (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_include_returns_false_for_non_matching_record_to_verify_scoping (unexpected)
--- FAIL: HasManyAssociationsTest#test_has_many_associations_on_new_records_use_null_relations (unexpected)
--- FAIL: HasManyAssociationsTest#test_has_many_with_out_of_range_value (unexpected)
--- FAIL: HasOneAssociationsTest#test_save_of_record_with_loaded_has_one (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_with_touch_option_on_update (unexpected)
--- FAIL: HasManyAssociationsTest#test_first_or_create!_adds_the_record_to_the_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_via_block (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_and_create_from_association_should_respect_unscope_over_default_scope (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_many (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_caching_of_columns (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_proxy_should_not_respond_to_private_methods (unexpected)
--- FAIL: HasManyAssociationsTest#test_taking_not_found (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependence_with_transaction_support_on_failure (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_replace_on_new_object (unexpected)
--- FAIL: HasManyAssociationsTest#test_depends_and_nullify_on_polymorphic_assoc (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_size_on_an_association_that_has_not_been_loaded_performs_a_query (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_callbacks_firing_order_on_create (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_adding_before_save (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14327141/activerecord/1710136331914/1710145986425)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14366914?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14366914?buildTab=artifacts#/activerecord) on master @ [2ac5657aef0fd878973191079b6f3aadbc9c9266](https://github.com/cockroachdb/cockroach/commits/2ac5657aef0fd878973191079b6f3aadbc9c9266):


```
--- FAIL: HasAndBelongsToManyAssociationsTest#test_updating_attributes_on_non_rich_associations (unexpected)
--- FAIL: FinderTest#test_take_bang_missing (unexpected)
--- FAIL: EagerAssociationTest#test_preconfigured_includes_with_habtm (unexpected)
--- FAIL: DefaultScopingTest#test_default_scope_runs_on_create (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_with_polymorphic_has_many_does_not_allow_to_override_type_and_id (unexpected)
--- FAIL: InheritanceTest#test_compute_type_success (unexpected)
--- FAIL: HasOneAssociationsTest#test_successful_build_association (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_nested_has_many_through_association_with_unpersisted_parent_instance (unexpected)
--- FAIL: AttributeMethodsTest#test_time_attributes_are_retrieved_in_the_current_time_zone (unexpected)
--- FAIL: ActiveRecord::SelectTest#test_select_with_hash_argument (unexpected)
--- FAIL: BasicsTest#test_can_call_connected_to_with_role_and_shard_on_abstract_classes (unexpected)
--- FAIL: BelongsToAssociationsTest#test_create_bang_with_block (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_singleton_has_many_through (unexpected)
--- FAIL: InheritanceTest#test_inheritance_new_with_base_class (unexpected)
--- FAIL: EnumTest#test_unicode_characters_for_enum_names (unexpected)
--- FAIL: AttributeMethodsTest#test_set_attributes_with_a_block (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_alternate_database (unexpected)
--- FAIL: BasicsTest#test_.columns_hash_raises_an_error_if_the_record_has_an_empty_table_name (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_create_has_many_through_with_default_scope_on_join_model (unexpected)
--- FAIL: RelationTest#test_automatically_added_where_references (unexpected)
--- FAIL: RelationTest#test_using_a_custom_table_with_joins_affects_the_joins (unexpected)
--- FAIL: CalculationsTest#test_pluck_multiple_columns (unexpected)
--- FAIL: AttributeMethodsTest#test_write_attribute_does_not_raise_when_the_attribute_isn't_selected (unexpected)
--- FAIL: BelongsToAssociationsTest#test_counter_cache_double_destroy (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attribute_with_nil_should_not_asplode (unexpected)
--- FAIL: AssociationsTest#test_query_constraints_that_dont_include_the_primary_key_raise (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_all_with_include_and_conditions (unexpected)
--- FAIL: PersistenceTest#test_update_column_should_raise_exception_if_new_record (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_update_ids_with_conditions (unexpected)
--- FAIL: ActiveRecord::DupTest#test_dup (unexpected)
--- FAIL: HasOneAssociationsTest#test_clearing_an_association_clears_the_associations_inverse (unexpected)
--- FAIL: FinderTest#test_find_with_hash_conditions_on_joined_table (unexpected)
--- FAIL: RelationTest#test_joins_with_select_custom_attribute (unexpected)
--- FAIL: RelationTest#test_having_with_binds_for_both_where_and_having (unexpected)
--- FAIL: EagerAssociationTest#test_association_loading_notification (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_composite_primary_key_join_table (unexpected)
--- FAIL: RelationTest#test_multi_where_ands_queries (unexpected)
--- FAIL: ActiveRecord::Migration::CheckConstraintTest#test_not_valid_check_constraint (unexpected)
--- FAIL: RelationTest#test_find_all_using_where_with_relation (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_for_single_table_inheritance_(STI_Models) (unexpected)
--- FAIL: RelationTest#test_select_with_from_includes_quoted_original_table_name (unexpected)
--- FAIL: CalculationsTest#test_count_with_too_many_parameters_raises (unexpected)
--- FAIL: SerializedAttributeTestWithYamlSafeLoad#test_nil_serialized_attribute_without_class_constraint (unexpected)
--- FAIL: StoreTest#test_object_initialization_with_not_nullable_column_encoded_with_JSON (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_joining_has_many_through_with_distinct (unexpected)
--- FAIL: UpdateAllTest#test_update_on_relation (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14366914/activerecord/1710308840393/1710317622667)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14385491?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14385491?buildTab=artifacts#/activerecord) on master @ [455b16592df7d8efd121b3ba1256fb477e227564](https://github.com/cockroachdb/cockroach/commits/455b16592df7d8efd121b3ba1256fb477e227564):


```
--- FAIL: CalculationsTest#test_pluck_loaded_relation_sql_fragment (unexpected)
--- FAIL: HasOneAssociationsTest#test_with_polymorphic_has_one_with_custom_columns_name (unexpected)
--- FAIL: InheritanceTest#test_compute_type_argument_error (unexpected)
--- FAIL: BasicsTest#test_find_by_slug (unexpected)
--- FAIL: HasManyAssociationsTest#test_regular_create_on_has_many_when_parent_is_new_raises (unexpected)
--- FAIL: PersistenceTest#test_create_columns_not_equal_attributes (unexpected)
--- FAIL: SerializedAttributeTest#test_where_by_serialized_attribute_with_array (unexpected)
--- FAIL: HasManyThroughDisableJoinsAssociationsTest#test_polymophic_disable_joins_through_ordering (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_attributes_not_backed_by_database_columns_are_always_initialized (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_first_after_reset_scope (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_association_replaces_existing_without_dependent_option (unexpected)
--- FAIL: InheritanceTest#test_becomes_and_change_tracking_for_inheritance_columns (unexpected)
--- FAIL: AttributeMethodsTest#test_case-sensitive_attributes_hash (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_setting_foreign_key_after_nil_target_loaded (unexpected)
--- FAIL: BasicsTest#test_initialize_with_attributes (unexpected)
--- FAIL: CounterCacheTest#test_decrement_counter (unexpected)
--- FAIL: EagerAssociationTest#test_base_messages (unexpected)
--- FAIL: BasicsTest#test_marshal_new_record_round_trip (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_primary_key_joins_on_correct_column (unexpected)
--- FAIL: InverseHasOneTests#test_parent_instance_should_be_shared_with_newly_created_child (unexpected)
--- FAIL: InheritanceTest#test_compute_type_on_undefined_method (unexpected)
--- FAIL: CalculationsTest#test_should_count_scoped_select_with_options (unexpected)
--- FAIL: AssociationValidationTest#test_validates_associated_marked_for_destruction (unexpected)
--- FAIL: HasManyAssociationsTest#test_prevent_double_insertion_of_new_object_when_the_parent_association_loaded_in_the_after_save_callback (unexpected)
--- FAIL: BasicsTest#test_ignored_columns_have_no_attribute_methods (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_empty_in (unexpected)
--- FAIL: HasManyAssociationsTest#test_adding_array_and_collection (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_proxy_should_not_respond_to_private_methods (unexpected)
--- FAIL: BasicsTest#test_attribute_names_on_table_not_exists (unexpected)
--- FAIL: FinderTest#test_find_by_ids (unexpected)
--- FAIL: EagerAssociationTest#test_eager_loaded_has_one_association_with_references_does_not_run_additional_queries (unexpected)
--- FAIL: CalculationsTest#test_having_with_strong_parameters (unexpected)
--- FAIL: OptimisticLockingTest#test_lock_new (unexpected)
--- FAIL: FinderTest#test_find_passing_active_record_object_is_not_permitted (unexpected)
--- FAIL: BelongsToAssociationsTest#test_tracking_change_from_nil_to_new_record (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_in_collection (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_get_has_many_through_belongs_to_ids_with_conditions (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_multiple_columns_and_selection_clause (unexpected)
--- FAIL: OptimisticLockingTest#test_lock_destroy (unexpected)
--- FAIL: InverseBelongsToTests#test_child_instance_should_be_shared_with_replaced_via_accessor_parent (unexpected)
--- FAIL: PostGISTest#test_spatial_factory_attrs_parsing (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_counter_with_assigning_new_object (unexpected)
--- FAIL: BelongsToAssociationsTest#test_should_set_foreign_key_on_create_association! (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_inner_join_with_quoted_table_name (unexpected)
--- FAIL: CalculationsTest#test_count_with_column_parameter (unexpected)
--- FAIL: DirtyTest#test_datetime_attribute_can_be_updated_with_fractional_seconds (unexpected)
--- FAIL: HasManyAssociationsTest#test_first_or_create!_adds_the_record_to_the_association (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14385491/activerecord/1710395720490/1710403507688)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14401472?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14401472?buildTab=artifacts#/activerecord) on master @ [ebf92d22a7e4afe2423e681d49c615104b5f0098](https://github.com/cockroachdb/cockroach/commits/ebf92d22a7e4afe2423e681d49c615104b5f0098):


```
--- FAIL: TransactionTest#test_add_to_null_transaction (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_preload_sti_middle_relation (unexpected)
--- FAIL: BelongsToAssociationsTest#test_building_the_belonging_object_with_an_invalid_type (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_has_and_belongs_to_many (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_that_should_not_be_fully_merged (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_of_optional_instance_dependent_associations_is_supported (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_before_save (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_assign_ids (unexpected)
--- FAIL: PersistenceTest#test_becomes_errors_base (unexpected)
--- FAIL: PersistenceTest#test_destroy_with_single_composite_primary_key (unexpected)
--- FAIL: ActiveRecord::Migration::ForeignKeyChangeColumnWithSuffixTest#test_remove_foreign_key_by_column (unexpected)
--- FAIL: HasManyAssociationsTest#test_building_the_associated_object_with_an_invalid_type (unexpected)
--- FAIL: TransactionTest#test_raising_exception_in_nested_transaction_restore_state_in_save (unexpected)
--- FAIL: DefaultScopingTest#test_default_scope_runs_on_select (unexpected)
--- FAIL: FinderTest#test_find_one_message_with_custom_primary_key (unexpected)
--- FAIL: PostgresqlByteaTest#test_read_value (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_an_overridden_original_method_issues_a_deprecation (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependence_for_associations_with_hash_condition (unexpected)
--- FAIL: PersistenceTest#test_update_columns_with_default_scope (unexpected)
--- FAIL: PersistenceTest#test_create_many (unexpected)
--- FAIL: RelationTest#test_select_with_blank_value (unexpected)
--- FAIL: AssociationCallbacksTest#test_has_and_belongs_to_many_callbacks_for_save_on_parent (unexpected)
--- FAIL: RelationTest#test_finding_with_reorder (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_with_touch_option_on_destroy (unexpected)
--- FAIL: HasManyAssociationsTest#test_in_memory_replacements_do_not_execute_callbacks (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_names_with_a_custom_select (unexpected)
--- FAIL: HasManyAssociationsTest#test_regular_create_on_has_many_when_parent_is_new_raises (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_update_counter_caches_on_delete (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_first_or_last_on_loaded_association_should_not_fetch_with_query (unexpected)
--- FAIL: ValidationsTest#test_numericality_validator_wont_be_affected_by_custom_getter (unexpected)
--- FAIL: HasOneThroughAssociationsTest#test_reassigning_has_one_through (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_destroy_on_association_clears_scope (unexpected)
--- FAIL: InheritanceTest#test_becomes_sets_variables_before_initialization_callbacks (unexpected)
--- FAIL: EagerAssociationTest#test_limited_eager_with_order (unexpected)
--- FAIL: AttributeMethodsTest#test_array_content (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_loading_for_new_record (unexpected)
--- FAIL: RelationTest#test_create_with_array (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_for_inspect_with_an_array (unexpected)
--- FAIL: RelationTest#test_preloading_with_associations_default_scopes_and_merges (unexpected)
--- FAIL: OptimisticLockingTest#test_touch_stale_object (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_joining_has_many_through_belongs_to (unexpected)
--- FAIL: InheritanceTest#test_different_namespace_subclass_should_load_correctly_with_store_full_sti_class_option (unexpected)
--- FAIL: PersistenceTest#test_update_columns_returns_boolean (unexpected)
--- FAIL: ReflectionTest#test_reflect_on_association_accepts_symbols (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_delete_all_for_with_dependent_option_delete_all (unexpected)
--- FAIL: RelationTest#test_loaded_all (unexpected)
--- FAIL: CounterCacheTest#test_reset_counter_performs_query_for_correct_counter_with_touch:_true (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14401472/activerecord/1710482036163/1710489647484)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14417897?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14417897?buildTab=artifacts#/activerecord) on master @ [34caf9d2c267f95f6aef2713708806e6f14948d6](https://github.com/cockroachdb/cockroach/commits/34caf9d2c267f95f6aef2713708806e6f14948d6):


```
--- FAIL: FinderTest#test_find_with_nil_inside_set_passed_for_one_attribute (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_with_touch_option_on_update (unexpected)
--- FAIL: CalculationsTest#test_limit_with_offset_is_kept (unexpected)
--- FAIL: CalculationsTest#test_count_with_eager_loading_and_custom_select_and_order (unexpected)
--- FAIL: BelongsToAssociationsTest#test_with_polymorphic_and_condition (unexpected)
--- FAIL: FinderTest#test_exists_with_distinct_association_includes_and_limit (unexpected)
--- FAIL: CalculationsTest#test_grouped_calculation_with_polymorphic_relation (unexpected)
--- FAIL: PersistenceTest#test_becomes_default_sti_subclass (unexpected)
--- FAIL: CalculationsTest#test_should_count_manual_with_count_all (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_get_collection_singular_ids_on_has_many_through_with_conditions_and_include (unexpected)
--- FAIL: HasManyAssociationsTest#test_ids_reader_cache_should_be_cleared_when_collection_is_deleted (unexpected)
--- FAIL: CalculationsTest#test_count_with_column_parameter (unexpected)
--- FAIL: CalculationsTest#test_ids_with_includes_offset (unexpected)
--- FAIL: RelationTest#test_loaded_all (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_grouped (unexpected)
--- FAIL: RelationTest#test_one_with_destroy (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_with_bang_with_purpose (unexpected)
--- FAIL: PersistenceTest#test_create_many_through_factory_with_block (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_invalid_adding_before_save (unexpected)
--- FAIL: HasManyAssociationsTest#test_finder_method_with_dirty_target (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_field_having_condition_from_select (unexpected)
--- FAIL: CalculationsTest#test_maximum_with_not_auto_table_name_prefix_if_column_included (unexpected)
--- FAIL: FinderTest#test_find_by_two_attributes_but_passing_only_one (unexpected)
--- FAIL: BelongsToAssociationsTest#test_composite_primary_key_malformed_association_class (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_preload_with_nested_association (unexpected)
--- FAIL: BelongsToAssociationsTest#test_create! (unexpected)
--- FAIL: FinderTest#test_implicit_order_column_is_configurable (unexpected)
--- FAIL: RelationTest#test_create_bang_with_array (unexpected)
--- FAIL: RelationTest#test_finding_with_reorder_by_aliased_attributes (unexpected)
--- FAIL: FinderTest#test_find_with_ids_with_id_out_of_range (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_alternate_database (unexpected)
--- FAIL: RelationTest#test_relations_limit_the_records_in_#inspect_at_10 (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_two_levels_and_self_table_reference (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_association_through_a_has_many_association_to_self (unexpected)
--- FAIL: FinderTest#test_find_on_multiple_hash_conditions (unexpected)
--- FAIL: RelationTest#test_find_by_returns_nil_if_the_record_is_missing (unexpected)
--- FAIL: RelationTest#test_count (unexpected)
--- FAIL: BasicsTest#test_previously_changed_dup (unexpected)
--- FAIL: BelongsToAssociationsTest#test_multiple_counter_cache_with_after_create_update (unexpected)
--- FAIL: FinderTest#test_select_rows (unexpected)
--- FAIL: RelationTest#test_having_with_binds_for_both_where_and_having (unexpected)
--- FAIL: BasicsTest#test_last (unexpected)
--- FAIL: HasManyAssociationsTest#test_building_the_associated_object_with_explicit_sti_base_class (unexpected)
--- FAIL: PersistenceTest#test_increment_aliased_attribute (unexpected)
--- FAIL: RelationTest#test_unscope_grouped_where (unexpected)
--- FAIL: RelationTest#test_find_by!_requires_at_least_one_argument (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14417897/activerecord/1710568450032/1710576437999)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14429188?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14429188?buildTab=artifacts#/activerecord) on master @ [067e48d29b9093038f6fcf2074cd761ffdcd4fe2](https://github.com/cockroachdb/cockroach/commits/067e48d29b9093038f6fcf2074cd761ffdcd4fe2):


```
--- FAIL: FinderTest#test_#skip_query_cache!_for_#exists?_with_a_limited_eager_load (unexpected)
--- FAIL: FinderTest#test_last_bang_present (unexpected)
--- FAIL: FinderTest#test_find_by_id_with_conditions_with_or (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates_after_parameterize_with_hyphens (unexpected)
--- FAIL: BasicsTest#test_typecasting_aliases (unexpected)
--- FAIL: FinderTest#test_exists_with_scope (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_aggregate_attribute_having_same_name_as_field_and_key_value_being_aggregate (unexpected)
--- FAIL: BasicsTest#test_attributes_on_dummy_time_with_invalid_time (unexpected)
--- FAIL: FinderTest#test_find_by_nil_and_not_nil_attributes (unexpected)
--- FAIL: FinderTest#test_named_bind_variables (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_two_levels_with_two_has_many_associations (unexpected)
--- FAIL: FinderTest#test_second_to_last (unexpected)
--- FAIL: BasicsTest#test_distinct_delegates_to_scoped (unexpected)
--- FAIL: FinderTest#test_find_with_ids_with_id_out_of_range (unexpected)
--- FAIL: BasicsTest#test_group_weirds_by_from (unexpected)
--- FAIL: BasicsTest#test_column_names_are_quoted_when_using_#from_clause_and_model_has_ignored_columns (unexpected)
--- FAIL: BasicsTest#test_reload (unexpected)
--- FAIL: FinderTest#test_exists_with_aggregate_having_three_mappings_with_one_difference (unexpected)
--- FAIL: BasicsTest#test_limit_should_sanitize_sql_injection_for_limit_without_commas (unexpected)
--- FAIL: BasicsTest#test_singular_table_name_guesses_with_prefixes_and_suffixes (unexpected)
--- FAIL: FinderTest#test_find_by_two_attributes_but_passing_only_one (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_grafts_stashed_associations_to_correct_parent (unexpected)
--- FAIL: BasicsTest#test_comparison_with_different_objects (unexpected)
--- FAIL: FinderTest#test_fourth_have_primary_key_order_by_default (unexpected)
--- FAIL: FinderTest#test_find_by_on_a_scope_does_not_perform_statement_caching (unexpected)
--- FAIL: FinderTest#test_last_with_integer_and_order_should_keep_the_order (unexpected)
--- FAIL: FinderTest#test_find_by_bang_on_relation_with_large_number (unexpected)
--- FAIL: FinderTest#test_find_with_nil_inside_set_passed_for_one_attribute (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_multiple_ranges (unexpected)
--- FAIL: FinderTest#test_fourth (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_where_first_level_returns_nil (unexpected)
--- FAIL: BasicsTest#test_create_after_initialize_without_block (unexpected)
--- FAIL: FinderTest#test_model_class_responds_to_fourth_bang (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_cascaded_eager_association_loading_with_join_for_count (unexpected)
--- FAIL: FinderTest#test_second_to_last_have_primary_key_order_by_default (unexpected)
--- FAIL: BasicsTest#test_dont_clear_sequence_name_when_setting_explicitly (unexpected)
--- FAIL: FinderTest#test_find_with_ids_where_and_limit (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_recursive_cascading_four_levels_has_many_through (unexpected)
--- FAIL: BasicsTest#test_scoped_can_take_a_values_hash (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_preloading_across_has_one_constrains_loaded_records (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_three_levels_by_ping_pong (unexpected)
--- FAIL: IntegrationTest#test_cache_version_changes_with_versioning_on (unexpected)
--- FAIL: BasicsTest#test_clear_cache_when_setting_table_name (unexpected)
--- FAIL: FinderTest#test_member_on_unloaded_relation_without_match (unexpected)
--- FAIL: FinderTest#test_find_with_a_multiple_sets_of_composite_primary_key_wrapped_in_an_array (unexpected)
--- FAIL: FinderTest#test_unexisting_record_exception_handling (unexpected)
--- FAIL: IntegrationTest#test_to_param_returns_id_if_not_persisted_but_id_is_set (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14429188/activerecord/1710654074221/1710661337399)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14467090?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14467090?buildTab=artifacts#/activerecord) on master @ [f0116ea373a2b87155e7f0264df4f783ce177360](https://github.com/cockroachdb/cockroach/commits/f0116ea373a2b87155e7f0264df4f783ce177360):


```
--- FAIL: RelationTest#test_finding_with_complex_order_and_limit (unexpected)
--- FAIL: RelationTest#test_select_with_subquery_string_in_from_does_not_use_original_table_name (unexpected)
--- FAIL: RelationTest#test_extracted_association (unexpected)
--- FAIL: FinderTest#test_second_with_offset (unexpected)
--- FAIL: RelationTest#test_joins_with_select (unexpected)
--- FAIL: HasManyAssociationsTest#test_in_memory_replacements_do_not_execute_callbacks (unexpected)
--- FAIL: HasOneThroughDisableJoinsAssociationsTest#test_has_one_through_with_belongs_to_on_disable_joins (unexpected)
--- FAIL: RelationTest#test_first_or_initialize_with_no_parameters (unexpected)
--- FAIL: BasicsTest#test_marshal_inspected_round_trip (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_update_changing_ids_changes_the_counter_cache (unexpected)
--- FAIL: RelationTest#test_finding_with_order_and_take (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_numeric_range_for_string (unexpected)
--- FAIL: FinderTest#test_find_with_nil_inside_set_passed_for_one_attribute (unexpected)
--- FAIL: FinderTest#test_third_with_offset (unexpected)
--- FAIL: FinderTest#test_find_with_ids_with_id_out_of_range (unexpected)
--- FAIL: HasManyAssociationsTest#test_collection_association_with_private_kernel_method (unexpected)
--- FAIL: RelationTest#test_create_or_find_by_with_bang_with_non_unique_attributes (unexpected)
--- FAIL: RelationTest#test_reverse_arel_assoc_order_with_multiargument_function (unexpected)
--- FAIL: FinderTest#test_find_by_array_of_one_id (unexpected)
--- FAIL: FinderTest#test_find_on_combined_explicit_and_hashed_table_names (unexpected)
--- FAIL: PersistenceTest#test_increment_with_no_arg (unexpected)
--- FAIL: InheritanceAttributeMappingTest#test_polymorphic_associations_custom_type (unexpected)
--- FAIL: PersistenceTest#test_save_with_duping_of_destroyed_object (unexpected)
--- FAIL: BelongsToAssociationsTest#test_creating_the_belonging_object_with_primary_key (unexpected)
--- FAIL: PersistenceTest#test_becomes (unexpected)
--- FAIL: CalculationsTest#test_count_selected_arel_attribute_in_joined_table (unexpected)
--- FAIL: BasicsTest#test_quoted_table_name_after_set_table_name (unexpected)
--- FAIL: FinderTest#test_sole_failing_many (unexpected)
--- FAIL: FinderTest#test_find_by_with_non-hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: RelationTest#test_size_with_eager_loading_and_custom_order (unexpected)
--- FAIL: ReflectionTest#test_default_association_validation (unexpected)
--- FAIL: HasManyAssociationsTest#test_counting_with_association_limit (unexpected)
--- FAIL: HasManyAssociationsTest#test_update_all_respects_association_scope (unexpected)
--- FAIL: FinderTest#test_fourth_with_offset (unexpected)
--- FAIL: PersistenceTest#test_update_uses_query_constraints_config (unexpected)
--- FAIL: PersistenceTest#test_update_columns_should_not_leave_the_object_dirty (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_first_after_reset_scope (unexpected)
--- FAIL: CalculationsTest#test_no_queries_for_empty_relation_on_sum (unexpected)
--- FAIL: PersistenceTest#test_decrement_attribute_by (unexpected)
--- FAIL: HasManyAssociationsTest#test_replace_with_new (unexpected)
--- FAIL: RelationTest#test_find_or_create_by! (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_explicit_table_name_and_aggregate (unexpected)
--- FAIL: RelationTest#test_select_takes_an_aliased_attribute (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_qualified_attribute_dot_notation_string (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroy_all (unexpected)
--- FAIL: CalculationsTest#test_should_limit_calculation (unexpected)
--- FAIL: HasManyAssociationsTest#test_building_the_associated_object_with_implicit_sti_base_class (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14467090/activerecord/1710913935380/1710922468631)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14483482?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14483482?buildTab=artifacts#/activerecord) on master @ [3060756a084cc437114d64265e79ace5720a7317](https://github.com/cockroachdb/cockroach/commits/3060756a084cc437114d64265e79ace5720a7317):


```
--- FAIL: BasicsTest#test_find_keeps_multiple_group_values (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all_on_association_is_the_same_as_not_loaded (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_with_rewhere_doesnt_set_inverse_instance_key (unexpected)
--- FAIL: BasicsTest#test_marshal_between_processes (unexpected)
--- FAIL: AttributeMethodsTest#test_read_overridden_attribute_with_predicate_respects_override (unexpected)
--- FAIL: RelationTest#test_select_with_subquery_string_in_from_does_not_use_original_table_name (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_keys_bypass_attribute_protection (unexpected)
--- FAIL: RelationTest#test_select_takes_a_variable_list_of_args (unexpected)
--- FAIL: RelationTest#test_pluck_with_from_includes_quoted_original_table_name (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_store_association_in_two_relations_with_one_save_in_existing_object_with_values (unexpected)
--- FAIL: RelationTest#test_size_with_zero_limit (unexpected)
--- FAIL: BasicsTest#test_preserving_time_objects_with_local_time_conversion_to_default_timezone_utc (unexpected)
--- FAIL: RelationTest#test_find_all_using_where_with_relation_with_no_selects_and_composite_primary_key_raises (unexpected)
--- FAIL: BasicsTest#test_default_values (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_predicates_respect_access_control (unexpected)
--- FAIL: ActiveRecord::CockroachDB::Migration::ColumnsTest#test_change_column_default (unexpected)
--- FAIL: HasManyAssociationsTest#test_clearing_an_exclusively_dependent_association_collection (unexpected)
--- FAIL: RelationTest#test_create_or_find_by_with_bang_should_raise_due_to_validation_errors (unexpected)
--- FAIL: AssociationsTest#test_belongs_to_association_does_not_use_parent_query_constraints_if_not_configured_to (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attribute (unexpected)
--- FAIL: BasicsTest#test_equality (unexpected)
--- FAIL: CallbacksOnMultipleInstancesInATransactionTest#test_updated_callback_called_on_first_to_save_in_transaction_with_old_configuration (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_callbacks_firing_order_on_create (unexpected)
--- FAIL: AttributeMethodsTest#test_hashes_are_not_mangled (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_create_fixtures_resets_sequences_when_not_cached (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_method? (unexpected)
--- FAIL: BasicsTest#test_primary_key_and_references_columns_should_be_identical_type (unexpected)
--- FAIL: HasManyAssociationsTest#test_get_ids_for_unloaded_associations_does_not_load_them (unexpected)
--- FAIL: HasManyAssociationsTest#test_clear_collection_should_not_change_updated_at (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_size_on_an_association_that_has_been_loaded_does_not_perform_query (unexpected)
--- FAIL: HasManyAssociationsTest#test_has_many_preloading_with_duplicate_records (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_with_a_bang (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record (unexpected)
--- FAIL: BasicsTest#test_connection_in_local_time (unexpected)
--- FAIL: HasManyAssociationsTest#test_sti_subselect_count (unexpected)
--- FAIL: RelationTest#test_extracted_association (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::TypesTest#test_attributes_which_are_invalid_for_database_can_still_be_reassigned (unexpected)
--- FAIL: BasicsTest#test_readonly_attributes (unexpected)
--- FAIL: RelationTest#test_find_by_with_delegated_ar_object (unexpected)
--- FAIL: BasicsTest#test_attribute_names_on_table_not_exists (unexpected)
--- FAIL: HasManyAssociationsTest#test_finding_using_primary_key (unexpected)
--- FAIL: HasManyAssociationsTest#test_finding_with_condition (unexpected)
--- FAIL: BasicsTest#test_find_reverse_ordered_last (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_eager_load_raise_errors (unexpected)
--- FAIL: RelationTest#test_create_or_find_by_with_non_unique_attributes (unexpected)
--- FAIL: HasManyAssociationsTest#test_anonymous_has_many (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_one_should_defer_to_collection_if_using_a_block (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14483482/activerecord/1711000422706/1711009099852)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14502091?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14502091?buildTab=artifacts#/activerecord) on master @ [c561383c9d86a52cf63a2c8aaa3ee20270635f11](https://github.com/cockroachdb/cockroach/commits/c561383c9d86a52cf63a2c8aaa3ee20270635f11):


```
--- FAIL: FinderTest#test_find_by_invalid_method_syntax (unexpected)
--- FAIL: BelongsToAssociationsTest#test_failing_create! (unexpected)
--- FAIL: MultiParameterAttributeTest#test_multiparameter_attributes_on_date_with_empty_day (unexpected)
--- FAIL: JsonAttributeTest#test_select (unexpected)
--- FAIL: FinderTest#test_exists_with_includes_limit_and_empty_result (unexpected)
--- FAIL: FinderTest#test_take_failing (unexpected)
--- FAIL: AttributeMethodsTest#test_hashes_are_not_mangled (unexpected)
--- FAIL: BelongsToAssociationsTest#test_type_mismatch (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_association_proxy_transaction_method_starts_transaction_in_association_class (unexpected)
--- FAIL: ReflectionTest#test_scope_chain_does_not_interfere_with_hmt_with_polymorphic_case (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_present (unexpected)
--- FAIL: PersistenceTest#test_increment_attribute_by (unexpected)
--- FAIL: AttributeMethodsTest#test_calling_super_when_the_parent_does_not_define_method_raises_NoMethodError (unexpected)
--- FAIL: ReflectionTest#test_column_null_not_null (unexpected)
--- FAIL: ReflectionTest#test_association_primary_key_uses_explicit_primary_key_option_as_first_priority (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_assignment_with_nil (unexpected)
--- FAIL: EnumTest#test_unicode_characters_for_enum_names (unexpected)
--- FAIL: FinderTest#test_find_by_with_hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: AttributeMethodsTest#test_number_attribute_predicate (unexpected)
--- FAIL: FinderTest#test_implicit_order_column_is_configurable (unexpected)
--- FAIL: JsonAttributeTest#test_changes_in_place (unexpected)
--- FAIL: AttributeMethodsTest#test_write_attribute_does_not_raise_when_the_attribute_isn't_selected (unexpected)
--- FAIL: SignedIdTest#test_cannot_get_a_signed_ID_for_a_new_record (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_destroy_all (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_multiple_stis_and_order (unexpected)
--- FAIL: InheritanceTest#test_alt_destroy_all_within_inheritance (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_aggregate_having_one_mapping_array (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associating_new (unexpected)
--- FAIL: ReflectionTest#test_has_and_belongs_to_many_reflection (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_destroy_on_association_clears_scope (unexpected)
--- FAIL: MultiParameterAttributeTest#test_multiparameter_attributes_setting_date_and_time_attribute (unexpected)
--- FAIL: PersistenceTest#test_update_all (unexpected)
--- FAIL: AttributeMethodsTest#test_time_attributes_are_retrieved_in_the_current_time_zone (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_update_counter_caches_on_delete_with_dependent_destroy (unexpected)
--- FAIL: BelongsToAssociationsTest#test_building_the_belonging_object_with_sti_subclass (unexpected)
--- FAIL: FinderTest#test_last_bang_present (unexpected)
--- FAIL: FinderTest#test_find_with_proc_parameter_and_block (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_a_time_zone-aware_attribute_interprets_time_zone-unaware_string_in_time_zone (unexpected)
--- FAIL: PersistenceTest#test_create_model_with_custom_named_uuid_pk_populates_id (unexpected)
--- FAIL: AttributeMethodsTest#test_#id_value_alias_is_not_defined_if_id_column_doesn't_exist (unexpected)
--- FAIL: BelongsToAssociationsTest#test_create_with_conditions (unexpected)
--- FAIL: InheritanceTest#test_compute_type_on_undefined_method (unexpected)
--- FAIL: BelongsToAssociationsTest#test_association_with_query_constraints_assigns_id_on_replacement (unexpected)
--- FAIL: PersistenceTest#test_decrement_with_touch_updates_timestamps (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_names_on_a_queried_record (unexpected)
--- FAIL: BelongsToAssociationsTest#test_association_assignment_sticks (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14502091/activerecord/1711086450461/1711094849866)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14521690?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14521690?buildTab=artifacts#/activerecord) on master @ [4a9385cacb82e7a8d6d37e5d9a26a6b7c845aab6](https://github.com/cockroachdb/cockroach/commits/4a9385cacb82e7a8d6d37e5d9a26a6b7c845aab6):


```
--- FAIL: HasManyAssociationsTest#test_dependent_association_respects_optional_hash_conditions_on_delete (unexpected)
--- FAIL: HasManyAssociationsTest#test_clearing_updates_counter_cache_when_inverse_counter_cache_is_a_symbol_with_dependent_destroy (unexpected)
--- FAIL: RelationTest#test_select_with_blank_value (unexpected)
--- FAIL: RelationTest#test_where_id_with_delegated_ar_object (unexpected)
--- FAIL: HasManyAssociationsTest#test_in_memory_replacements_sets_inverse_instance (unexpected)
--- FAIL: HasManyAssociationsTest#test_clearing_a_dependent_association_collection (unexpected)
--- FAIL: RelationTest#test_already-loaded_relations_don't_perform_a_new_query_in_#inspect (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all_accepts_limited_parameters (unexpected)
--- FAIL: HasManyAssociationsTest#test_create_from_association_with_nil_values_should_work (unexpected)
--- FAIL: RelationTest#test_finding_with_reversed_assoc_order (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroying (unexpected)
--- FAIL: HasManyAssociationsTest#test_taking_with_inverse_of (unexpected)
--- FAIL: RelationTest#test_finding_with_reversed_arel_assoc_order (unexpected)
--- FAIL: RelationTest#test_where_with_take_memoization (unexpected)
--- FAIL: RelationTest#test_find_all_with_join (unexpected)
--- FAIL: RelationTest#test_empty_with_zero_limit (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependent_association_respects_optional_sanitized_conditions_on_delete (unexpected)
--- FAIL: RelationTest#test_joins_with_order_by_custom_attribute (unexpected)
--- FAIL: HasManyAssociationsTest#test_counter_cache_updates_in_memory_after_create (unexpected)
--- FAIL: RelationTest#test_find_ids (unexpected)
--- FAIL: AssociationsTest#test_should_construct_new_finder_sql_after_create (unexpected)
--- FAIL: RelationTest#test_size_with_zero_limit (unexpected)
--- FAIL: HasManyAssociationsTest#test_first_or_create_adds_the_record_to_the_association (unexpected)
--- FAIL: RelationTest#test_create_or_find_by_within_transaction (unexpected)
--- FAIL: AssociationsTest#test_assign_persisted_composite_foreign_key_belongs_to_association (unexpected)
--- FAIL: RelationTest#test_unscoped_block_style (unexpected)
--- FAIL: HasManyAssociationsTest#test_counter_cache_updates_in_memory_after_update_with_inverse_of_enabled (unexpected)
--- FAIL: RelationTest#test_eager_association_loading_of_stis_with_multiple_references (unexpected)
--- FAIL: HasManyAssociationsTest#test_update_all_respects_association_scope (unexpected)
--- FAIL: RelationTest#test_find_with_list_of_ar (unexpected)
--- FAIL: RelationTest#test_find_all_using_where_with_relation (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependence_with_transaction_support_on_failure (unexpected)
--- FAIL: RelationTest#test_select_with_subquery_string_in_from_does_not_use_original_table_name (unexpected)
--- FAIL: HasManyAssociationsTest#test_has_many_with_out_of_range_value (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_the_association_with_an_array (unexpected)
--- FAIL: RelationTest#test_size_with_limit (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_one_should_count_instead_of_loading_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_get_ids_for_association_on_new_record_does_not_try_to_find_records (unexpected)
--- FAIL: RelationTest#test_one (unexpected)
--- FAIL: HasManyAssociationsTest#test_loading_association_in_validate_callback_doesnt_affect_persistence (unexpected)
--- FAIL: RelationTest#test_find_with_included_associations (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_first_after_reset (unexpected)
--- FAIL: RelationTest#test_find_all_using_where_with_relation_with_joins (unexpected)
--- FAIL: RelationTest#test_two_scopes_with_includes_should_not_drop_any_include (unexpected)
--- FAIL: RelationTest#test_scoped_build (unexpected)
--- FAIL: RelationTest#test_size_with_eager_loading_and_custom_order (unexpected)
--- FAIL: RelationTest#test_find_by_id_with_list_of_ar (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14521690/activerecord/1711172993952/1711183332795)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14549398?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14549398?buildTab=artifacts#/activerecord) on master @ [c994982a8be5af89f594e115e897dd6d62cf99d8](https://github.com/cockroachdb/cockroach/commits/c994982a8be5af89f594e115e897dd6d62cf99d8):


```
(orm_helpers.go:227).summarizeFailed: 
Tests run on Cockroach v24.1.0-alpha.4-dev-c994982a8be5af89f594e115e897dd6d62cf99d8
Tests run against activerecord 7.1.3
8108 Total Tests Run
8079 tests passed
29 tests failed
66 tests skipped
0 tests ignored
0 tests passed unexpectedly
3 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_default_pk_and_sequence (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_create_fixtures_resets_sequences_when_not_cached (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14549398/activerecord/1711346118495/1711354489272)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14559433?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14559433?buildTab=artifacts#/activerecord) on master @ [6df7436b6388eb7823b9f63a42aa75e1fbc89a85](https://github.com/cockroachdb/cockroach/commits/6df7436b6388eb7823b9f63a42aa75e1fbc89a85):


```
--- FAIL: BelongsToAssociationsTest#test_replace_counter_cache (unexpected)
--- FAIL: PostgresqlDateTest#test_bc_date_year_zero (unexpected)
--- FAIL: ActiveRecord::Migration::ForeignKeyChangeColumnTest#test_change_column_of_parent_table (unexpected)
--- FAIL: ActiveRecord::RelationTest#test_selecting_aliased_attribute_quotes_column_name_when_from_is_used (unexpected)
--- FAIL: RelationTest#test_unscope_with_subquery (unexpected)
--- FAIL: SerializationTest#test_read_attribute_for_serialization_with_format_after_find (unexpected)
--- FAIL: BelongsToAssociationsTest#test_with_polymorphic_and_condition (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_has_one_with_cpk (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_for_existing_record_with_updated_at_and_custom_cache_timestamp_format (unexpected)
--- FAIL: ActiveRecord::PostgresqlTransactionTest#test_raises_QueryCanceled_when_statement_timeout_exceeded (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_with_touch_option_on_touch (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_a_regular_association_through_a_polymorphic_association_doesn't_require_the_association_to_exist_on_all_types (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_attributes_with_overridden_types_keep_their_type_when_a_default_value_is_configured_separately (unexpected)
--- FAIL: RelationTest#test_finding_with_assoc_reorder_by_aliased_attributes (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_counter_after_touch (unexpected)
--- FAIL: AttributeMethodsTest#test_write_time_to_date_attribute (unexpected)
--- FAIL: BelongsToAssociationsTest#test_setting_foreign_key_after_nil_target_loaded (unexpected)
--- FAIL: AttributeMethodsTest#test_accessed_fields (unexpected)
--- FAIL: IntegrationTest#test_param_delimiter_changes_delimiter_used_in_to_param (unexpected)
--- FAIL: BasicsTest#test_readonly_attributes_on_a_new_record (unexpected)
--- FAIL: CockroachDB::FixturesTest#test_clean_fixtures (unexpected)
--- FAIL: NamedScopingTest#test_should_build_new_on_top_of_scope (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_preloading_empty_through_with_polymorphic_source_association (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_preloaded_records_are_not_duplicated (unexpected)
--- FAIL: AttributeMethodsTest#test_aliasing_``id``_attribute_allows_reading_the_column_value_for_a_CPK_model (unexpected)
--- FAIL: SerializedAttributeTestWithYamlSafeLoad#test_where_by_serialized_attribute_with_hash_in_array (unexpected)
--- FAIL: PersistenceTest#test_destroy_with_invalid_ids_for_a_model_that_expects_composite_keys (unexpected)
--- FAIL: HasOneAssociationsTest#test_reset_association (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_overloaded_properties_save (unexpected)
--- FAIL: BasicsTest#test_dup_with_aggregate_of_same_name_as_attribute (unexpected)
--- FAIL: PostgresqlVirtualColumnTest#test_stored_column (unexpected)
--- FAIL: ActiveRecord::Migration::ForeignKeyChangeColumnTest#test_remove_reference_column_of_child_table (unexpected)
--- FAIL: RelationTest#test_includes_with_blank_value (unexpected)
--- FAIL: ReflectionTest#test_reflect_on_association_accepts_strings (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_attributes_added_after_subclasses_load_are_inherited (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_with_a_bang_for_single_table_inheritance_(STI_Models) (unexpected)
--- FAIL: HasOneAssociationsTest#test_dependence (unexpected)
--- FAIL: RelationTest#test_unscope_with_triple_dot_where (unexpected)
--- FAIL: DateTest#test_date_with_string_value (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_consider_type (unexpected)
--- FAIL: AttributeMethodsTest#test_string_attribute_predicate (unexpected)
--- FAIL: PersistenceTest#test_update_sti_type (unexpected)
--- FAIL: EagerAssociationTest#test_preload_has_one_using_primary_key (unexpected)
--- FAIL: BasicsTest#test_hashing (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_through_has_one_association_with_conditions_on_association (unexpected)
--- FAIL: YamlSerializationTest#test_yaml_encoding_keeps_false_values (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14559433/activerecord/1711432019478/1711439329767)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14577688?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14577688?buildTab=artifacts#/activerecord) on master @ [ec9444f8287663777c5ebceff61bcd280491ba08](https://github.com/cockroachdb/cockroach/commits/ec9444f8287663777c5ebceff61bcd280491ba08):


```
--- FAIL: CalculationsTest#test_grouped_calculation_with_polymorphic_relation (unexpected)
--- FAIL: BasicsTest#test_equality_of_relation_and_association_relation (unexpected)
--- FAIL: BasicsTest#test_table_exists (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_invalid_adding_with_validate_false (unexpected)
--- FAIL: BasicsTest#test_readonly_attributes_on_a_new_record (unexpected)
--- FAIL: HasManyAssociationsTest#test_get_ids (unexpected)
--- FAIL: BasicsTest#test_quoting_arrays (unexpected)
--- FAIL: BasicsTest#test_hashing (unexpected)
--- FAIL: BasicsTest#test_previously_new_record_on_destroyed_record (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_empty_on_an_association_that_has_not_been_loaded_performs_a_query (unexpected)
--- FAIL: BasicsTest#test_readonly_attributes_in_abstract_class_descendant (unexpected)
--- FAIL: CalculationsTest#test_ids_with_scope (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_field (unexpected)
--- FAIL: BasicsTest#test_previously_changed (unexpected)
--- FAIL: CalculationsTest#test_count_with_block_and_column_name_raises_an_error (unexpected)
--- FAIL: BasicsTest#test_unicode_column_name (unexpected)
--- FAIL: CalculationsTest#test_count_should_shortcut_with_limit_zero (unexpected)
--- FAIL: CalculationsTest#test_group_by_attribute_with_custom_type (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_updates_counter_cache_with_dependent_destroy (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_includes_limit_and_empty_result (unexpected)
--- FAIL: HasManyAssociationsTest#test_cant_save_has_many_readonly_association (unexpected)
--- FAIL: BasicsTest#test_#connected_to_many_sets_prevent_writes_if_role_is_reading (unexpected)
--- FAIL: CalculationsTest#test_pluck_loaded_relation_sql_fragment (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_record_with_complex_joins (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_squishes (unexpected)
--- FAIL: CalculationsTest#test_should_sum_scoped_field_with_conditions (unexpected)
--- FAIL: BasicsTest#test_clone_of_new_object_marks_attributes_as_dirty (unexpected)
--- FAIL: CalculationsTest#test_group_alias_is_properly_quoted (unexpected)
--- FAIL: HasManyAssociationsTest#test_custom_named_counter_cache (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_none_should_defer_to_collection_if_using_a_block (unexpected)
--- FAIL: HasManyAssociationsTest#test_reloading_unloaded_associations_with_query_cache (unexpected)
--- FAIL: HasManyAssociationsTest#test_taking_with_inverse_of (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_build_many_before_save (unexpected)
--- FAIL: HasManyAssociationsTest#test_counter_cache_updates_in_memory_after_create_with_array (unexpected)
--- FAIL: HasManyAssociationsTest#test_dynamic_find_should_respect_association_order (unexpected)
--- FAIL: HasManyAssociationsTest#test_update_all_on_association_accessed_before_save (unexpected)
--- FAIL: BasicsTest#test_table_name_with_2_abstract_subclasses (unexpected)
--- FAIL: HasManyAssociationsTest#test_first_or_initialize_adds_the_record_to_the_association (unexpected)
--- FAIL: CalculationsTest#test_calculation_grouped_by_association_doesnt_error_when_no_records_have_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependent_association_respects_optional_conditions_on_delete (unexpected)
--- FAIL: HasManyAssociationsTest#test_associations_order_should_be_priority_over_throughs_order (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_one_should_return_false_if_zero (unexpected)
--- FAIL: HasManyAssociationsTest#test_clearing_updates_counter_cache (unexpected)
--- FAIL: CalculationsTest#test_limit_should_apply_before_count (unexpected)
--- FAIL: CalculationsTest#test_pick_loaded_relation_multiple_columns (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_field_through_association_and_having (unexpected)
--- FAIL: CalculationsTest#test_pluck_joined_with_polymorphic_relation (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14577688/activerecord/1711518815270/1711526560121)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14594861?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14594861?buildTab=artifacts#/activerecord) on master @ [460707194644edbf4c6258d09b46d9877c8eefa6](https://github.com/cockroachdb/cockroach/commits/460707194644edbf4c6258d09b46d9877c8eefa6):


```
--- FAIL: CalculationsTest#test_count_for_a_composite_primary_key_model (unexpected)
--- FAIL: EagerAssociationTest#test_associations_loaded_for_all_records (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_grouped_by_function_with_table_alias (unexpected)
--- FAIL: CalculationsTest#test_group_by_multiple_same_field (unexpected)
--- FAIL: CalculationsTest#test_ids_with_polymorphic_relation_join (unexpected)
--- FAIL: PersistenceTest#test_update_column_with_default_scope (unexpected)
--- FAIL: EagerAssociationTest#test_preconfigured_includes_with_has_many (unexpected)
--- FAIL: PersistenceTest#test_save_invalid_record (unexpected)
--- FAIL: FinderTest#test_model_class_responds_to_second_bang (unexpected)
--- FAIL: InheritanceTest#test_destroy_all_within_inheritance (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_polymorphic_with_custom_foreign_type (unexpected)
--- FAIL: BasicsTest#test_#connected_to_many_with_a_single_argument_for_classes (unexpected)
--- FAIL: CalculationsTest#test_should_return_decimal_average_if_db_returns_such (unexpected)
--- FAIL: BasicsTest#test_successful_comparison_of_like_class_records (unexpected)
--- FAIL: InheritanceTest#test_scope_inherited_properly (unexpected)
--- FAIL: EagerAssociationTest#test_finding_with_includes_on_belongs_to_association_with_same_include_includes_only_once (unexpected)
--- FAIL: BasicsTest#test_switching_between_table_name (unexpected)
--- FAIL: AssociationsTest#test_loading_the_association_target_should_keep_child_records_marked_for_destruction (unexpected)
--- FAIL: BasicsTest#test_new_record_returns_boolean (unexpected)
--- FAIL: EagerAssociationTest#test_with_ordering (unexpected)
--- FAIL: PersistenceTest#test_update_many_with_active_record_base_object (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_adding_multiple (unexpected)
--- FAIL: BasicsTest#test_clear_cache_when_setting_table_name (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_qualified_attribute_dot_notation_symbol (unexpected)
--- FAIL: PersistenceTest#test_becomes_initializes_missing_attributes (unexpected)
--- FAIL: FinderTest#test_take_bang_present (unexpected)
--- FAIL: PersistenceTest#test_update_many_with_array_of_active_record_base_objects! (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates_after_parameterize (unexpected)
--- FAIL: FinderTest#test_find_with_ids_and_offset (unexpected)
--- FAIL: PersistenceTest#test_delete_record_with_associations (unexpected)
--- FAIL: EagerAssociationTest#test_association_loading_notification (unexpected)
--- FAIL: FinderTest#test_take_and_first_and_last_with_integer_should_use_sql_limit (unexpected)
--- FAIL: FinderTest#test_find_on_array_conditions (unexpected)
--- FAIL: CalculationsTest#test_async_pluck_on_loaded_relation (unexpected)
--- FAIL: BasicsTest#test_equality_of_destroyed_records (unexpected)
--- FAIL: CalculationsTest#test_group_by_with_quoted_count_and_order_by_alias (unexpected)
--- FAIL: EagerAssociationTest#test_with_two_tables_in_from_without_getting_double_quoted (unexpected)
--- FAIL: FinderTest#test_exists_with_distinct_and_offset_and_select (unexpected)
--- FAIL: CalculationsTest#test_count_with_column_parameter (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_for_existing_record_with_updated_at_and_custom_cache_timestamp_format (unexpected)
--- FAIL: FinderTest#test_exists_passing_active_record_object_is_not_permitted (unexpected)
--- FAIL: FinderTest#test_exists_with_nil_arg (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_proxy_should_respond_to_private_methods_via_send (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_singular_ids_are_reloaded_after_collection_concat (unexpected)
--- FAIL: FinderTest#test_find_by_ids_missing_one (unexpected)
--- FAIL: PersistenceTest#test_increment_attribute (unexpected)
--- FAIL: FinderTest#test_find_by_with_associations (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14594861/activerecord/1711604914188/1711612811956)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14610653?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14610653?buildTab=artifacts#/activerecord) on master @ [0dce29441187eda187aad472543997b06c2b01ea](https://github.com/cockroachdb/cockroach/commits/0dce29441187eda187aad472543997b06c2b01ea):


```
--- FAIL: HasOneAssociationsTest#test_create_with_inexistent_foreign_key_failing (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_count (unexpected)
--- FAIL: FinderTest#test_find_with_ids_and_offset (unexpected)
--- FAIL: HasManyAssociationsTest#test_replace_with_new (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_first_or_last_with_integer_on_association_should_not_load_association (unexpected)
--- FAIL: AttributeMethodsTest#test_global_methods_are_overwritten (unexpected)
--- FAIL: ModulesTest#test_module_table_name_prefix_with_global_prefix (unexpected)
--- FAIL: HasManyAssociationsTest#test_adding_a_collection (unexpected)
--- FAIL: FinderTest#test_take_and_first_and_last_with_integer_should_return_an_array (unexpected)
--- FAIL: SerializedAttributeTestWithYamlSafeLoad#test_serialized_attribute_should_raise_exception_on_assignment_with_wrong_type (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_empty_in (unexpected)
--- FAIL: ForbiddenAttributesProtectionTest#test_create_with_works_with_params_values (unexpected)
--- FAIL: HasManyThroughDisableJoinsAssociationsTest#test_first_and_scope_in_double_join_applies_order_in_memory (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- FAIL: PersistenceTest#test_update_raises_record_not_found_exception (unexpected)
--- FAIL: CalculationsTest#test_should_perform_joined_include_when_referencing_included_tables (unexpected)
--- FAIL: FinderTest#test_first_failing (unexpected)
--- FAIL: ActiveRecord::Migration::ForeignKeyChangeColumnWithSuffixTest#test_rename_column_of_child_table (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_push_with_invalid_join_record (unexpected)
--- FAIL: DefaultScopingTest#test_nilable_default_scope_with_all_queries_runs_on_destroy (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_build_for_has_many_through_association (unexpected)
--- FAIL: RelationTest#test_find_by_id_with_list_of_ar (unexpected)
--- FAIL: FinderTest#test_third_to_last_have_primary_key_order_by_default (unexpected)
--- FAIL: RelationTest#test_dynamic_find_by_attributes (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_get_collection_singular_ids_on_has_many_through_with_conditions_and_include (unexpected)
--- FAIL: TestAutosaveAssociationValidationsOnAHasManyAssociation#test_validations_still_fire_on_unchanged_association_with_custom_validation_context (unexpected)
--- FAIL: HasManyAssociationsTest#test_finding_array_compatibility (unexpected)
--- FAIL: FinderTest#test_model_class_responds_to_third_to_last_bang (unexpected)
--- FAIL: FinderTest#test_#last_for_a_model_with_composite_query_constraints (unexpected)
--- FAIL: FinderTest#test_take (unexpected)
--- FAIL: AttributeMethodsTest#test_user-defined_date_attribute_predicate (unexpected)
--- FAIL: CalculationsTest#test_ids_with_includes_and_scope (unexpected)
--- FAIL: CallbacksOnMultipleActionsTest#test_after_commit_on_multiple_actions (unexpected)
--- FAIL: EagerAssociationTest#test_loading_with_multiple_associations (unexpected)
--- FAIL: PostgresqlByteaTest#test_write_value (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_polymorphic_with_custom_foreign_type (unexpected)
--- FAIL: FinderTest#test_bind_variables (unexpected)
--- FAIL: ValidationsTest#test_validates_acceptance_of_as_database_column (unexpected)
--- FAIL: BelongsToAssociationsTest#test_creating_the_belonging_object_with_primary_key (unexpected)
--- FAIL: InheritanceTest#test_alt_inheritance_find_all (unexpected)
--- FAIL: RelationTest#test_loading_query_is_annotated_in_#inspect (unexpected)
--- FAIL: CalculationsTest#test_ids_on_loaded_relation_with_scope (unexpected)
--- FAIL: SerializedAttributeTest#test_serialized_attribute_on_custom_attribute_with_default (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_get_ids (unexpected)
--- FAIL: EagerAssociationTest#test_eager_loading_with_conditions_on_joined_table_preloads (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_marshal_dump (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroy_with_bang_bubbles_errors_from_associations (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14610653/activerecord/1711691519330/1711697499887)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14626026?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14626026?buildTab=artifacts#/activerecord) on master @ [2a5e231716c436781f12452d800651f51c6383b7](https://github.com/cockroachdb/cockroach/commits/2a5e231716c436781f12452d800651f51c6383b7):


```
--- FAIL: RelationTest#test_reverse_arel_assoc_order_with_multiargument_function (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_counter (unexpected)
--- FAIL: PersistenceTest#test_becomes_after_reload_schema_from_cache (unexpected)
--- FAIL: PersistenceTest#test_update_attribute_for_aborted_callback! (unexpected)
--- FAIL: RelationTest#test_finding_with_subquery_with_eager_loading_in_where (unexpected)
--- FAIL: SignedIdTest#test_can_get_a_signed_ID_in_an_after_create (unexpected)
--- FAIL: BelongsToAssociationsTest#test_stale_tracking_doesn't_care_about_the_type (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_for_inspect_with_an_array (unexpected)
--- FAIL: SignedIdTest#test_fail_to_find_signed_record_within_expiration_time (unexpected)
--- FAIL: RelationTest#test_find_by_requires_at_least_one_argument (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_the_same_alias_as_parent_doesn't_issue_a_deprecation (unexpected)
--- FAIL: RelationTest#test_find_all_using_where_with_relation_with_no_selects_and_composite_primary_key_raises (unexpected)
--- FAIL: AttributeMethodsTest#test_respond_to? (unexpected)
--- FAIL: BelongsToAssociationsTest#test_with_select (unexpected)
--- FAIL: DirtyTest#test_partial_insert (unexpected)
--- FAIL: BelongsToAssociationsTest#test_build_with_conditions (unexpected)
--- FAIL: RelationTest#test_build (unexpected)
--- FAIL: TransactionCallbacksTest#test_no_after_commit_on_destroy_after_transaction_commits_for_destroyed_new_record (unexpected)
--- FAIL: SignedIdTest#test_finding_signed_record_that_has_been_destroyed_raises_on_the_bang (unexpected)
--- FAIL: RelationTest#test_select_with_from_includes_original_table_name (unexpected)
--- FAIL: SignedIdTest#test_fail_to_work_without_when_signed_id_verifier_secret_lambda_is_nil (unexpected)
--- FAIL: RelationTest#test_find_all_with_join (unexpected)
--- FAIL: RelationTest#test_group_with_select_and_includes (unexpected)
--- FAIL: RelationTest#test_delete_by (unexpected)
--- FAIL: RelationTest#test_#skip_query_cache! (unexpected)
--- FAIL: RelationTest#test_select_takes_an_aliased_attribute (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_assignment_foreign_type_field_updating (unexpected)
--- FAIL: AttributeMethodsTest#test_overridden_write_attribute (unexpected)
--- FAIL: RelationTest#test_multiple_selects (unexpected)
--- FAIL: RelationTest#test_joins_with_nil_argument (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_touch_option_on_touch_without_updated_at_attributes (unexpected)
--- FAIL: RelationTest#test_locked_should_not_build_arel (unexpected)
--- FAIL: AttributeMethodsTest#test_attributes_for_database (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_marshal_dump_and_load_via_disk (unexpected)
--- FAIL: PersistenceTest#test_update_columns_with_default_scope (unexpected)
--- FAIL: PersistenceTest#test_update_many_with_invalid_id! (unexpected)
--- FAIL: RelationTest#test_joins_with_string_array (unexpected)
--- FAIL: BelongsToAssociationsTest#test_dependent_delete_and_destroy_with_belongs_to (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attributes_before_type_cast_on_a_datetime (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_yaml_loads_5_1_dump_without_indexes_still_queries_for_indexes (unexpected)
--- FAIL: RelationTest#test_where_with_take_memoization (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_default_pk_and_sequence (unexpected)
--- FAIL: PersistenceTest#test_save_for_record_with_only_primary_key (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_regroup_raise_errors (unexpected)
--- FAIL: RelationTest#test_using_a_custom_table_with_joins_affects_the_joins (unexpected)
--- FAIL: ModulesTest#test_module_table_name_suffix (unexpected)
--- FAIL: RelationTest#test_where_with_ar_object (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14626026/activerecord/1711777794655/1711785985803)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14635121?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14635121?buildTab=artifacts#/activerecord) on master @ [7fc4c7bcbbf0c75a62d056da0bf79a5a32714650](https://github.com/cockroachdb/cockroach/commits/7fc4c7bcbbf0c75a62d056da0bf79a5a32714650):


```
--- FAIL: HasManyAssociationsTest#test_associations_replace_in_memory_when_records_have_the_same_id (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_assignment_updates_foreign_id_field_for_new_and_saved_records (unexpected)
--- FAIL: RelationTest#test_scoped_build (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_get_collection_singular_ids_on_has_many_through_with_conditions_and_include (unexpected)
--- FAIL: PersistenceTest#test_update_columns_with_model_having_primary_key_other_than_id (unexpected)
--- FAIL: RelationTest#test_reverse_order_with_function (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_attributes_not_backed_by_database_columns_return_the_default_on_models_loaded_from_database (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_numeric_range_for_string (unexpected)
--- FAIL: HasManyAssociationsTest#test_has_many_through_respects_hash_conditions (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_with_extend_option (unexpected)
--- FAIL: FinderTest#test_implicit_order_column_is_configurable (unexpected)
--- FAIL: CalculationsTest#test_#skip_query_cache!_for_a_grouped_calculation (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_accept_string_and_hash_join (unexpected)
--- FAIL: BelongsToAssociationsTest#test_creating_the_belonging_object_with_primary_key (unexpected)
--- FAIL: FinderTest#test_find_with_entire_select_statement (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_enum_works_properly (unexpected)
--- FAIL: RelationTest#test_first_or_create_bang_with_invalid_block (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_qualified_attribute_dot_notation_string (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_store_association_in_two_relations_with_one_save_in_existing_object_with_values (unexpected)
--- FAIL: RelationTest#test_where_id_with_delegated_ar_object (unexpected)
--- FAIL: FinderTest#test_sole_failing_many (unexpected)
--- FAIL: ModulesTest#test_module_spanning_has_and_belongs_to_many_associations (unexpected)
--- FAIL: FinderTest#test_find_by_with_non-hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_create_with_conditions_hash_on_through_association (unexpected)
--- FAIL: RelationTest#test_already-loaded_relations_don't_perform_a_new_query_in_#pretty_print (unexpected)
--- FAIL: HasManyAssociationsTest#test_replace (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all_on_association_clears_scope (unexpected)
--- FAIL: HasManyAssociationsTest#test_replace_with_less_and_dependent_nullify (unexpected)
--- FAIL: FinderTest#test_hash_condition_utc_time_interpolation_with_default_timezone_local (unexpected)
--- FAIL: PersistenceTest#test_delete_many (unexpected)
--- FAIL: HasManyAssociationsTest#test_reattach_to_new_objects_replaces_inverse_association_and_foreign_key (unexpected)
--- FAIL: EagerAssociationTest#test_eager-loading_readonly_association (unexpected)
--- FAIL: BelongsToAssociationsTest#test_should_set_foreign_key_on_save! (unexpected)
--- FAIL: AttributeMethodsTest#test_inherited_custom_accessors (unexpected)
--- FAIL: OptimisticLockingWithSchemaChangeTest#test_update_counters_updates_custom_lock_version (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_names_with_a_custom_select (unexpected)
--- FAIL: BooleanTest#test_find_by_boolean_string (unexpected)
--- FAIL: DelegatedTypeTest#test_delegated_class_with_custom_foreign_type (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_assignment_before_either_saved (unexpected)
--- FAIL: CalculationsTest#test_calculation_with_polymorphic_relation (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_joining_has_many_through_with_distinct (unexpected)
--- FAIL: BelongsToAssociationsTest#test_default_scope_on_relations_is_not_cached (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_has_many_through_with_distinct (unexpected)
--- FAIL: BasicsTest#test_column_types_typecast (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_build_before_child_saved (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_delete_through_belongs_to_with_dependent_destroy (unexpected)
--- FAIL: BasicsTest#test_has_attribute_with_symbol (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14635121/activerecord/1711864343217/1711873127526)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14643937?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14643937?buildTab=artifacts#/activerecord) on master @ [7fc4c7bcbbf0c75a62d056da0bf79a5a32714650](https://github.com/cockroachdb/cockroach/commits/7fc4c7bcbbf0c75a62d056da0bf79a5a32714650):


```
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_interdependent_one_level_and_two_levels (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
--- FAIL: RelationTest#test_automatically_added_reorder_references (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_with_inexistent_foreign_key_failing (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_preload_with_nested_association (unexpected)
--- FAIL: BelongsToAssociationsTest#test_natural_assignment_with_primary_key (unexpected)
--- FAIL: RelationTest#test_select_with_block (unexpected)
--- FAIL: RelationTest#test_size_with_limit (unexpected)
--- FAIL: MigrationTest#test_rename_table_with_prefix_and_suffix (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_get_collection_singular_ids_on_has_many_through_with_conditions_and_include (unexpected)
--- FAIL: TransactionTest#test_rollback_dirty_changes_even_with_raise_during_rollback_doesnt_commit_transaction (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_autosave_with_primary_key_manually_set (unexpected)
--- FAIL: RelationTest#test_do_not_double_quote_string_id_with_array (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates (unexpected)
--- FAIL: FinderTest#test_find_with_a_multiple_sets_of_composite_primary_key_wrapped_in_an_array_ordered (unexpected)
--- FAIL: ReflectionTest#test_association_primary_key_uses_explicit_primary_key_option_as_first_priority (unexpected)
--- FAIL: RelationTest#test_loaded_all (unexpected)
--- FAIL: InheritanceTest#test_abstract_class (unexpected)
--- FAIL: CalculationsTest#test_should_sum_arel_attribute (unexpected)
--- FAIL: RelationTest#test_reverse_arel_assoc_order_with_function (unexpected)
--- FAIL: InheritanceTest#test_where_new_with_invalid_type (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_select_chosen_fields_only (unexpected)
--- FAIL: RelationTest#test_first_or_initialize (unexpected)
--- FAIL: RelationTest#test_find_with_included_associations (unexpected)
--- FAIL: CalculationsTest#test_distinct_count_with_order_and_offset (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_destroy_association (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_default_pk_and_sequence (unexpected)
--- FAIL: FinderTest#test_first_failing (unexpected)
--- FAIL: RelationTest#test_where_id_with_delegated_ar_object (unexpected)
--- FAIL: RelationTest#test_finding_with_complex_order (unexpected)
--- FAIL: BelongsToAssociationsTest#test_skips_parent_presence_check_if_parent_has_not_changed (unexpected)
--- FAIL: BelongsToAssociationsTest#test_creating_the_belonging_object_from_new_record (unexpected)
--- FAIL: InheritanceTest#test_inheritance_without_mapping (unexpected)
--- FAIL: CalculationsTest#test_async_pluck_none_relation (unexpected)
--- FAIL: TransactionTest#test_rollback_when_saving_a_frozen_record (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_with_false (unexpected)
--- FAIL: RelationTest#test_joins_with_blank_value (unexpected)
--- FAIL: RelationTest#test_size_with_eager_loading_and_manual_distinct_select_and_custom_order (unexpected)
--- FAIL: IntegrationTest#test_to_param_should_return_string (unexpected)
--- FAIL: FinderTest#test_take_bang_present (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_an_overridden_original_method_issues_a_deprecation (unexpected)
--- FAIL: BelongsToAssociationsTest#test_create! (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_replace_association_with_duplicates (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_new_by_building (unexpected)
--- FAIL: TransactionTest#test_rollback_when_thread_killed (unexpected)
--- FAIL: AssociationsTest#test_assign_belongs_to_cpk_model_by_id_attribute (unexpected)
--- FAIL: PersistenceTest#test_destroy_raises_record_not_found_exception (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14643937/activerecord/1711950704333/1711959535587)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14660501?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14660501?buildTab=artifacts#/activerecord) on master @ [c43f54cdde5b7578f4a0ca61de41463f0d690993](https://github.com/cockroachdb/cockroach/commits/c43f54cdde5b7578f4a0ca61de41463f0d690993):


```
--- FAIL: BasicsTest#test_time_zone_aware_attribute_with_default_timezone_utc_on_utc_can_be_created (unexpected)
--- FAIL: RelationTest#test_destroy_by (unexpected)
--- FAIL: BasicsTest#test_find_by_slug (unexpected)
--- FAIL: BelongsToAssociationsTest#test_default (unexpected)
--- FAIL: RelationTest#test_reverse_order_with_reorder_nil_removes_the_order (unexpected)
--- FAIL: EagerAssociationTest#test_eager_loaded_has_many_association_without_primary_key (unexpected)
--- FAIL: RelationTest#test_default_scope_order_with_scope_order (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_delete_all_for_with_dependent_option_nullify (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_regroup_raise_errors (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependent_association_respects_optional_hash_conditions_on_delete (unexpected)
--- FAIL: TestAutosaveAssociationOnAHasOneAssociation#test_mark_for_destruction_is_ignored_without_autosave_true (unexpected)
--- FAIL: TestDestroyAsPartOfAutosaveAssociation#test_should_run_remove_callback_procs_for_habtm (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_create_has_many_through_with_default_scope_on_join_model (unexpected)
--- FAIL: HasManyAssociationsTest#test_pushing_association_updates_counter_cache (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_updates_counter_cache_with_dependent_destroy (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_does_not_use_order_by (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_the_same_alias_as_parent_doesn't_issue_a_deprecation (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_by_string_id (unexpected)
--- FAIL: RelationTest#test_to_yaml (unexpected)
--- FAIL: TestNestedAttributesOnAHasAndBelongsToManyAssociation#test_should_preserve_order_when_not_overwriting_unsaved_updates (unexpected)
--- FAIL: RelationTest#test_find_all_with_join (unexpected)
--- FAIL: BelongsToAssociationsTest#test_creating_the_belonging_object (unexpected)
--- FAIL: ActiveRecord::SelectTest#test_select_with_hash_and_table_alias (unexpected)
--- FAIL: AttributeMethodsTest#test_typecast_attribute_from_select_to_false (unexpected)
--- FAIL: HasOneAssociationsTest#test_association_protect_foreign_key (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_with_polymorphic_has_many_does_not_allow_to_override_type_and_id (unexpected)
--- FAIL: PostgresqlIntervalTest#test_interval_type_cast_from_invalid_string (unexpected)
--- FAIL: UniquenessValidationWithIndexTest#test_changing_non_unique_attribute (unexpected)
--- FAIL: HasManyAssociationsTest#test_ids_reader_memoization (unexpected)
--- FAIL: EagerAssociationTest#test_calculate_with_string_in_from_and_eager_loading (unexpected)
--- FAIL: FinderTest#test_exists (unexpected)
--- FAIL: CalculationsTest#test_minimum_and_maximum_on_non_numeric_type (unexpected)
--- FAIL: RelationTest#test_delete_by (unexpected)
--- FAIL: InheritanceTest#test_should_store_demodulized_class_name_with_store_full_sti_class_option_disabled (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit_and_conditions_array_on_the_eagers (unexpected)
--- FAIL: BelongsToAssociationsTest#test_with_polymorphic_and_condition (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_invalid_dependent_option_raises_exception (unexpected)
--- FAIL: RelationTest#test_unscope_with_triple_dot_where (unexpected)
--- FAIL: RelationTest#test_scoped_build (unexpected)
--- FAIL: CalculationsTest#test_should_return_nil_as_average (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_on_relation (unexpected)
--- FAIL: SignedIdTest#test_fail_to_find_signed_record_within_expiration_duration (unexpected)
--- FAIL: InheritanceTest#test_compute_type_nonexistent_constant (unexpected)
--- FAIL: CockroachDB::PostgresqlIntervalTest#test_interval_type_cast_from_invalid_string (unexpected)
--- FAIL: SchemaTest#test_remove_index_when_schema_specified (unexpected)
--- FAIL: InheritanceAttributeMappingTest#test_sti_with_custom_type (unexpected)
--- FAIL: InverseHasManyTests#test_parent_instance_should_be_shared_with_replaced_via_accessor_children (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14660501/activerecord/1712037054253/1712044523711)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14675255?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14675255?buildTab=artifacts#/activerecord) on master @ [7166724ff02b9b7f0d5787f74d919487ca347bd0](https://github.com/cockroachdb/cockroach/commits/7166724ff02b9b7f0d5787f74d919487ca347bd0):


```
--- FAIL: RelationTest#test_unscope_with_table_name_qualified_hash (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_bang_with_block (unexpected)
--- FAIL: PersistenceTest#test_update_columns_should_raise_exception_if_new_record (unexpected)
--- FAIL: RelationTest#test_loading_with_one_association_with_non_preload (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_build_then_save_with_has_many_inverse (unexpected)
--- FAIL: RelationTest#test_find_by!_with_multi-arg_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: CalculationsTest#test_should_sum_arel_attribute (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_association_replaces_existing_with_dependent_option (unexpected)
--- FAIL: CalculationsTest#test_pluck_type_cast_with_joins_without_table_name_qualified_column (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_through_record_is_built_when_created_with_where (unexpected)
--- FAIL: RelationTest#test_reorder_with_first (unexpected)
--- FAIL: RelationTest#test_first_or_create (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_with_create_and_valid_options (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_preload_sti_middle_relation (unexpected)
--- FAIL: CalculationsTest#test_ids_on_relation (unexpected)
--- FAIL: HasOneAssociationsTest#test_restrict_with_error (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_assignment_before_either_saved (unexpected)
--- FAIL: PersistenceTest#test_save_destroyed_object (unexpected)
--- FAIL: CalculationsTest#test_pluck_and_distinct (unexpected)
--- FAIL: PersistenceTest#test_save_for_record_with_only_primary_key (unexpected)
--- FAIL: CalculationsTest#test_should_sum_field (unexpected)
--- FAIL: CalculationsTest#test_should_sum_field_with_conditions (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_before_save (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_grouped_with_invalid_field (unexpected)
--- FAIL: CalculationsTest#test_calculation_with_query_cache (unexpected)
--- FAIL: RelationTest#test_raising_exception_on_invalid_hash_params (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_delete_all_for_with_dependent_option_destroy (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_arel_attribute (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_dynamic_find_should_respect_association_include (unexpected)
--- FAIL: RelationTest#test_preloading_with_associations_default_scopes_and_merges (unexpected)
--- FAIL: CalculationsTest#test_count_with_reverse_order (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_polymorphic_with_primary_key_option (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_push_with_invalid_record (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_unscope_default_scope (unexpected)
--- FAIL: CalculationsTest#test_pluck_loaded_relation (unexpected)
--- FAIL: HasOneAssociationsTest#test_type_mismatch (unexpected)
--- FAIL: PersistenceTest#test_destroy_record_with_associations (unexpected)
--- FAIL: HasOneAssociationsTest#test_restrict_with_exception (unexpected)
--- FAIL: CalculationsTest#test_#skip_query_cache!_for_a_simple_calculation (unexpected)
--- FAIL: RelationTest#test_unscope_with_subquery (unexpected)
--- FAIL: PersistenceTest#test_save_touch_false (unexpected)
--- FAIL: PersistenceTest#test_create_through_factory_with_block (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_get_has_many_through_belongs_to_ids_with_conditions (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_get_ids (unexpected)
--- FAIL: CalculationsTest#test_should_count_field_in_joined_table (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_insert_records_via_has_many_through_association_with_scope_and_association_name_different_from_the_joining_table_name (unexpected)
--- FAIL: HasOneAssociationsTest#test_save_of_record_with_loaded_has_one (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14675255/activerecord/1712123641319/1712131766602)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14691306?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14691306?buildTab=artifacts#/activerecord) on master @ [0d3c5b71415de867fedde97de01486d424209848](https://github.com/cockroachdb/cockroach/commits/0d3c5b71415de867fedde97de01486d424209848):


```
--- FAIL: FinderTest#test_last_with_integer_and_reorder_should_use_sql_limit (unexpected)
--- FAIL: FinderTest#test_fourth (unexpected)
--- FAIL: FinderTest#test_condition_local_time_interpolation_with_default_timezone_utc (unexpected)
--- FAIL: FinderTest#test_last_on_relation_with_limit_and_offset (unexpected)
--- FAIL: FinderTest#test_find_by_with_multi-arg_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: FinderTest#test_count_by_sql (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_dynamic_find_should_respect_association_order (unexpected)
--- FAIL: FinderTest#test_find_with_ids_where_and_limit (unexpected)
--- FAIL: FinderTest#test_named_bind_variables_with_quotes (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_find_grouped (unexpected)
--- FAIL: FinderTest#test_exists_with_order_and_distinct (unexpected)
--- FAIL: FinderTest#test_find_by_empty_in_condition (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_count (unexpected)
--- FAIL: FinderTest#test_take_and_first_and_last_with_integer_should_use_sql_limit (unexpected)
--- FAIL: FinderTest#test_#skip_query_cache!_for_#exists? (unexpected)
--- FAIL: FinderTest#test_find_on_association_proxy_conditions (unexpected)
--- FAIL: FinderTest#test_find_by_one_missing_attribute (unexpected)
--- FAIL: FinderTest#test_exists (unexpected)
--- FAIL: FinderTest#test_last_bang_present (unexpected)
--- FAIL: FinderTest#test_second_to_last (unexpected)
--- FAIL: FinderTest#test_fifth (unexpected)
--- FAIL: FinderTest#test_member_on_unloaded_relation_with_offset (unexpected)
--- FAIL: FinderTest#test_exists_with_joins (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_selects_all_columns_by_default (unexpected)
--- FAIL: FinderTest#test_find_with_ids_and_offset (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_for_existing_record_with_updated_at (unexpected)
--- FAIL: FinderTest#test_first_have_determined_order_by_default (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_has_and_belongs_to_many_while_partial_inserts_false (unexpected)
--- FAIL: IntegrationTest#test_to_param_should_return_string (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_build_before_child_saved (unexpected)
--- FAIL: FinderTest#test_third_with_offset (unexpected)
--- FAIL: FinderTest#test_exists_with_string (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_destroy_associations_destroys_multiple_associations (unexpected)
--- FAIL: FinderTest#test_find_with_hash_conditions_on_joined_table_and_with_range (unexpected)
--- FAIL: FinderTest#test_eager_load_for_no_has_many_with_limit_and_joins_for_has_many (unexpected)
--- FAIL: FinderTest#test_find_with_ids_with_limit_and_order_clause (unexpected)
--- FAIL: FinderTest#test_first_bang_present (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_collection_size_from_params (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_proxy_transaction_method_starts_transaction_in_association_class (unexpected)
--- FAIL: FinderTest#test_find_one_message_with_custom_primary_key (unexpected)
--- FAIL: FinderTest#test_find_with_large_number (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_distinct_order_preserved (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_is_precise_enough (unexpected)
--- FAIL: FinderTest#test_find (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_adding_from_the_project (unexpected)
--- FAIL: FinderTest#test_find_by_ids_with_limit_and_offset (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14691306/activerecord/1712209824364/1712218818180)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14705936?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14705936?buildTab=artifacts#/activerecord) on master @ [04e1faeb6b674729e989e001d37429786b584c6e](https://github.com/cockroachdb/cockroach/commits/04e1faeb6b674729e989e001d37429786b584c6e):


```
--- FAIL: AttributeMethodsTest#test_declared_suffixed_attribute_method_affects_respond_to?_and_method_missing (unexpected)
--- FAIL: HasManyAssociationsTest#test_taking_with_inverse_of (unexpected)
--- FAIL: PersistenceTest#test_save_invalid_record (unexpected)
--- FAIL: SerializedAttributeTest#test_serialized_attribute (unexpected)
--- FAIL: RelationTest#test_first_or_create_bang_with_invalid_array (unexpected)
--- FAIL: SerializedAttributeTest#test_serialized_column_should_unserialize_after_update_column (unexpected)
--- FAIL: InheritanceTest#test_alt_becomes_works_with_sti (unexpected)
--- FAIL: ActiveRecord::Migration::ForeignKeyChangeColumnWithSuffixTest#test_remove_foreign_key_by_column (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_merge_join_association_with_has_many_through_association_proxy (unexpected)
--- FAIL: HasOneThroughDisableJoinsAssociationsTest#test_preload_on_disable_joins_through (unexpected)
--- FAIL: HasManyAssociationsTest#test_abstract_class_with_polymorphic_has_many (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_includes_raise_errors (unexpected)
--- FAIL: PersistenceTest#test_delete_new_record (unexpected)
--- FAIL: CalculationsTest#test_no_queries_for_empty_relation_on_average (unexpected)
--- FAIL: BelongsToAssociationsTest#test_skips_parent_presence_check_if_parent_has_not_changed (unexpected)
--- FAIL: InheritanceTest#test_inheritance_save (unexpected)
--- FAIL: I18nValidationTest#test_validates_associated_on_generated_message_given_unless_condition (unexpected)
--- FAIL: DDLTest#test_column_defaults (unexpected)
--- FAIL: FinderTest#test_last_bang_present (unexpected)
--- FAIL: JsonAttributeTest#test_rewrite_array_json_value (unexpected)
--- FAIL: FinderTest#test_find_by_with_hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_custom_join_table (unexpected)
--- FAIL: HasManyAssociationsTest#test_counting (unexpected)
--- FAIL: RelationTest#test_finding_with_reorder (unexpected)
--- FAIL: NormalizedAttributeTest#test_normalizes_nil_if_apply_to_nil (unexpected)
--- FAIL: RelationTest#test_relations_don't_load_all_records_in_#pretty_print (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_through_scope_with_joins (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_from_association_should_respect_scope (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_first_nth_or_last_on_association_should_not_load_association (unexpected)
--- FAIL: FinderTest#test_model_class_responds_to_second_to_last_bang (unexpected)
--- FAIL: RelationTest#test_size_with_eager_loading_and_custom_order_and_distinct (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_update_counter_caches_on_destroy (unexpected)
--- FAIL: AssociationsTest#test_include_with_order_works (unexpected)
--- FAIL: InheritanceTest#test_where_create_bang_with_subclass (unexpected)
--- FAIL: DirtyTest#test_nullable_datetime_not_marked_as_changed_if_new_value_is_blank (unexpected)
--- FAIL: RelationTest#test_unscope_with_table_name_qualified_column (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_none_should_return_false_if_any (unexpected)
--- FAIL: RelationTest#test_find_with_included_associations (unexpected)
--- FAIL: PersistenceTest#test_save_valid_record (unexpected)
--- FAIL: DirtyTest#test_partial_insert (unexpected)
--- FAIL: FinderTest#test_find_by_empty_ids (unexpected)
--- FAIL: CounterCacheTest#test_reset_multiple_counters_with_touch:_:written_on (unexpected)
--- FAIL: ModulesTest#test_nested_models_should_not_raise_exception_when_using_nullify_dependency_on_association (unexpected)
--- FAIL: TestAutosaveAssociationOnAHasManyAssociationWithInverse#test_after_save_callback_with_autosave (unexpected)
--- FAIL: RelationTest#test_size_with_limit (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_present_with_booleans (unexpected)
--- FAIL: ActiveRecord::Migration::ColumnAttributesTest#test_add_column_newline_default (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14705936/activerecord/1712297475468/1712303891576)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14720307?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14720307?buildTab=artifacts#/activerecord) on master @ [eb495cac325786ea3e148ee64d63bd73ed74701b](https://github.com/cockroachdb/cockroach/commits/eb495cac325786ea3e148ee64d63bd73ed74701b):


```
--- FAIL: NamedScopingTest#test_scope_with_object (unexpected)
--- FAIL: RelationTest#test_default_scope_order_with_scope_order (unexpected)
--- FAIL: BasicsTest#test_attribute_names (unexpected)
--- FAIL: AssociationsTest#test_preloads_model_with_query_constraints_by_explicitly_configured_fk_and_pk (unexpected)
--- FAIL: PersistenceTest#test_create_prefetched_pk (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_with_a_polymorphic_association_and_using_the_existential_predicate_but_also_using_a_select (unexpected)
--- FAIL: PostgresqlByteaTest#test_write_value (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_for_existing_record_with_updated_at_and_custom_cache_timestamp_format (unexpected)
--- FAIL: RelationTest#test_doesnt_add_having_values_if_options_are_blank (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_deleting_all (unexpected)
--- FAIL: EagerAssociationTest#test_join_eager_with_empty_order_should_generate_valid_sql (unexpected)
--- FAIL: BasicsTest#test_marshalling_new_record_round_trip_with_associations (unexpected)
--- FAIL: EagerAssociationTest#test_eager_has_many_through_with_order (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_existing_record_twice_should_add_to_target_twice (unexpected)
--- FAIL: HasManyAssociationsTest#test_updates_counter_cache_when_default_scope_is_given (unexpected)
--- FAIL: FinderTest#test_find_by_nil_and_not_nil_attributes (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_raises_UnknownPrimaryKey_when_a_model_has_no_primary_key (unexpected)
--- FAIL: ReflectionTest#test_has_many_reflection_with_array_fk_raises (unexpected)
--- FAIL: CockroachDB::PostgresqlIntervalTest#test_simple_interval (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_reserved_words (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_respects_select (unexpected)
--- FAIL: PersistenceTest#test_update_does_not_run_sql_if_record_has_not_changed (unexpected)
--- FAIL: AssociationsTest#test_loading_the_association_target_should_load_most_recent_attributes_for_child_records_marked_for_destruction (unexpected)
--- FAIL: EagerAssociationTest#test_type_cast_in_where_references_association_name (unexpected)
--- FAIL: CalculationsTest#test_no_queries_for_empty_relation_on_sum (unexpected)
--- FAIL: BasicsTest#test_benchmark_with_log_level (unexpected)
--- FAIL: CalculationsTest#test_ids_with_contradicting_scope (unexpected)
--- FAIL: BasicsTest#test_cannot_call_connected_to_on_the_abstract_class_that_did_not_establish_the_connection (unexpected)
--- FAIL: RelationTest#test_anonymous_extension (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_reassignment_of_associated_type_updates_the_object (unexpected)
--- FAIL: TouchTest#test_update (unexpected)
--- FAIL: BasicsTest#test_connection_in_utc_time (unexpected)
--- FAIL: BasicsTest#test_columns_should_obey_set_primary_key (unexpected)
--- FAIL: CalculationsTest#test_sum_with_from_option (unexpected)
--- FAIL: CalculationsTest#test_should_sum_expression (unexpected)
--- FAIL: TransactionCallbacksTest#test_no_after_commit_on_destroy_after_transaction_commits_for_destroyed_new_record (unexpected)
--- FAIL: PersistenceTest#test_preserve_original_sti_type (unexpected)
--- FAIL: BasicsTest#test_preserving_time_objects_with_time_with_zone_conversion_to_default_timezone_local (unexpected)
--- FAIL: AttributeMethodsTest#test_custom_field_attribute_predicate (unexpected)
--- FAIL: FinderTest#test_exists_with_scope (unexpected)
--- FAIL: FullStiClassNamesTest#test_class_names_with_find_by (unexpected)
--- FAIL: ActiveRecord::CacheKeyTest#test_cache_key_with_version_always_has_both_key_and_version (unexpected)
--- FAIL: PostgresqlIntervalTest#test_interval_type_cast_from_invalid_string (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_has_one_reflection (unexpected)
--- FAIL: AttributeMethodsTest#test_#define_attribute_methods_brings_back_undefined_aliases (unexpected)
--- FAIL: EagerAssociationTest#test_eager_load_has_one_quotes_table_and_column_names (unexpected)
--- FAIL: EagerAssociationTest#test_including_duplicate_objects_from_has_many (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/cpu_arch=arm64/run_1
```

Parameters:
 - <code>ROACHTEST_arch=arm64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14720307/activerecord/1712383827599/1712395371694)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14729833?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14729833?buildTab=artifacts#/activerecord) on master @ [ff52bbf0c5fcc807828c6fa5aeeb1d19c24270fc](https://github.com/cockroachdb/cockroach/commits/ff52bbf0c5fcc807828c6fa5aeeb1d19c24270fc):


```
--- FAIL: ModulesTest#test_module_table_name_suffix_with_global_suffix (unexpected)
--- FAIL: RelationTest#test_unscope_with_subquery (unexpected)
--- FAIL: BelongsToAssociationsTest#test_reloading_the_belonging_object (unexpected)
--- FAIL: BasicsTest#test_utc_as_time_zone (unexpected)
--- FAIL: BelongsToAssociationsTest#test_building_the_belonging_object_with_an_unrelated_type (unexpected)
--- FAIL: HasOneAssociationsTest#test_successful_build_association (unexpected)
--- FAIL: BasicsTest#test_dup_for_a_composite_primary_key_model (unexpected)
--- FAIL: FinderTest#test_find_with_group_and_sanitized_having_method (unexpected)
--- FAIL: FinderTest#test_exists_does_not_select_columns_without_alias (unexpected)
--- FAIL: FinderTest#test_include_when_non_AR_object_passed_on_loaded_relation (unexpected)
--- FAIL: RelationTest#test_references_with_blank_value (unexpected)
--- FAIL: RelationTest#test_relations_show_the_records_in_#inspect (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_existing_record_twice_should_add_to_target_twice (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_with_touch_option_on_update (unexpected)
--- FAIL: FinderTest#test_condition_interpolation (unexpected)
--- FAIL: RelationTest#test_raising_exception_on_invalid_hash_params (unexpected)
--- FAIL: RelationTest#test_find_or_create_by_race_condition (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_with_touch_option_on_destroy (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_nil_with_aggregate_having_multiple_mappings (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_a_manually_defined_method_issues_a_deprecation_warning (unexpected)
--- FAIL: InheritanceTest#test_alt_inheritance_find_all (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_line_endings (unexpected)
--- FAIL: NumericalityValidationTest#test_virtual_attribute_with_precision_and_scale (unexpected)
--- FAIL: CalculationsTest#test_count_with_arel_attribute (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_with_conditions_join_model_with_include (unexpected)
--- FAIL: CalculationsTest#test_should_count_manual_with_count_all (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_a_polymorphic_association_with_references_to_the_associated_table (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_with_bang_with_purpose_raises (unexpected)
--- FAIL: FinderTest#test_find_with_ids_with_limit_and_order_clause (unexpected)
--- FAIL: EagerAssociationTest#test_preconfigured_includes_with_has_many (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_duplicated_has_many_through_with_through_scope_with_joins (unexpected)
--- FAIL: CalculationsTest#test_pluck_in_relation (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_with_false (unexpected)
--- FAIL: FinderTest#test_third (unexpected)
--- FAIL: BasicsTest#test_preserving_time_objects_with_time_with_zone_conversion_to_default_timezone_utc (unexpected)
--- FAIL: AssociationsTest#test_polymorphic_belongs_to_uses_parent_query_constraints (unexpected)
--- FAIL: BelongsToAssociationsTest#test_build_with_block (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_through_has_one_association_with_order_on_association (unexpected)
--- FAIL: FinderTest#test_last_with_integer_and_order_should_keep_the_order (unexpected)
--- FAIL: InheritanceTest#test_new_with_complex_inheritance (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_end_exclusive_range (unexpected)
--- FAIL: BasicsTest#test_dup_does_not_copy_associations (unexpected)
--- FAIL: BasicsTest#test_default_values (unexpected)
--- FAIL: FinderTest#test_find_one_message_with_custom_primary_key (unexpected)
--- FAIL: TestAutosaveAssociationValidationsOnABelongsToAssociation#test_validations_still_fire_on_unchanged_association_with_custom_validation_context (unexpected)
--- FAIL: HasOneAssociationsTest#test_restrict_with_exception (unexpected)
--- FAIL: CalculationsTest#test_async_pluck_none_relation (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14729833/activerecord/1712473322395/1712481175591)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14738640?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14738640?buildTab=artifacts#/activerecord) on master @ [909e91dc1ea45e8223658c1144067332df7ed1ec](https://github.com/cockroachdb/cockroach/commits/909e91dc1ea45e8223658c1144067332df7ed1ec):


```
--- FAIL: FinderTest#test_exists_with_eager_load (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_collection_build_with_nonstandard_primary_key_on_belongs_to (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: FinderTest#test_include_on_unloaded_relation_with_limit (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_counter_cache (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_get_collection_singular_ids_on_has_many_through_with_conditions_and_include (unexpected)
--- FAIL: BelongsToAssociationsTest#test_cant_save_readonly_association (unexpected)
--- FAIL: ModulesTest#test_compute_type_can_infer_class_name_of_sibling_inside_module (unexpected)
--- FAIL: FinderTest#test_find_with_group_and_sanitized_having_method (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_association_through_a_has_many_association_with_nonstandard_primary_keys (unexpected)
--- FAIL: ModulesTest#test_nested_models_should_not_raise_exception_when_using_delete_all_dependency_on_association (unexpected)
--- FAIL: FinderTest#test_second_to_last (unexpected)
--- FAIL: AssociationsTest#test_querying_by_relation_with_composite_key (unexpected)
--- FAIL: FinderTest#test_exists_with_distinct_and_offset_and_joins (unexpected)
--- FAIL: FinderTest#test_count_by_sql (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_with_touch_option_on_nonpersisted_built_associations_doesnt_update_parent (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_proxy_should_respond_to_private_methods_via_send (unexpected)
--- FAIL: HasOneAssociationsTest#test_building_the_associated_object_with_sti_subclass (unexpected)
--- FAIL: FinderTest#test_take_and_first_and_last_with_integer_should_use_sql_limit (unexpected)
--- FAIL: FinderTest#test_hash_condition_utc_time_interpolation_with_default_timezone_local (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_through_scope_with_includes (unexpected)
--- FAIL: FinderTest#test_find_by_with_non-hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: HasOneAssociationsTest#test_natural_assignment (unexpected)
--- FAIL: HasOneAssociationsTest#test_association_attributes_are_available_to_after_initialize (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_escaped_characters (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_empty_array_with_aggregate_having_multiple_mappings (unexpected)
--- FAIL: BelongsToAssociationsTest#test_type_mismatch (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_get_ids_for_has_many_through_with_conditions_should_not_preload (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_primary_key_joins_on_correct_column (unexpected)
--- FAIL: HasOneAssociationsTest#test_build_with_block (unexpected)
--- FAIL: AssociationsTest#test_belongs_to_a_model_with_composite_primary_key_uses_composite_pk_in_sql (unexpected)
--- FAIL: BelongsToAssociationsTest#test_building_the_belonging_object_with_an_unrelated_type (unexpected)
--- FAIL: FinderTest#test_model_class_responds_to_third_to_last_bang (unexpected)
--- FAIL: FinderTest#test_find_with_a_multiple_sets_of_composite_primary_key_wrapped_in_an_array (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_default_scope_on_join_model (unexpected)
--- FAIL: FinderTest#test_find_with_large_number (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_default_scope_on_the_target (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_both_parent_ids_set_when_saving_new (unexpected)
--- FAIL: FinderTest#test_find_by_with_associations (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_add_with_sti_middle_relation (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_end_exclusive_range (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_with_create_with_through_having_conditions (unexpected)
--- FAIL: BelongsToAssociationsTest#test_should_set_composite_foreign_key_on_association_when_key_changes_on_associated_record (unexpected)
--- FAIL: FinderTest#test_find_on_association_proxy_conditions (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_polymorphic_with_primary_key_option (unexpected)
--- FAIL: FinderTest#test_find_by_one_attribute_with_several_options (unexpected)
--- FAIL: FinderTest#test_exists_with_order (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14738640/activerecord/1712556731483/1712565549362)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14754736?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14754736?buildTab=artifacts#/activerecord) on master @ [5c0cad2ff83656d15f54f11272c0f710bee41728](https://github.com/cockroachdb/cockroach/commits/5c0cad2ff83656d15f54f11272c0f710bee41728):


```
--- FAIL: RelationTest#test_find_all_using_where_with_relation_with_joins (unexpected)
--- FAIL: RelationTest#test_first_or_create_bang_with_valid_options (unexpected)
--- FAIL: BelongsToAssociationsTest#test_reloading_association_with_key_change (unexpected)
--- FAIL: BelongsToAssociationsTest#test_destroying_child_with_unloaded_parent_and_foreign_key_and_touch_is_possible_with_has_many_inversing (unexpected)
--- FAIL: FinderTest#test_find_by_two_attributes_that_are_both_aggregates (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_through_record_is_built_when_created_with_where (unexpected)
--- FAIL: FinderTest#test_find_an_empty_array (unexpected)
--- FAIL: BelongsToAssociationsTest#test_default_scope_on_relations_is_not_cached (unexpected)
--- FAIL: ReflectionTest#test_has_many_through_reflection (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_do_not_cache_association_reader_if_the_though_method_has_default_scopes (unexpected)
--- FAIL: RelationTest#test_count_with_distinct (unexpected)
--- FAIL: RelationTest#test_grouping_by_column_with_reserved_name (unexpected)
--- FAIL: FinderTest#test_find_ignores_previously_inserted_record (unexpected)
--- FAIL: FinderTest#test_implicit_order_for_model_without_primary_key (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_for_existing_record_with_updated_at (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_assignment_with_primary_key_updates_foreign_id_field_for_new_and_saved_records (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: RelationTest#test_reorder_deduplication (unexpected)
--- FAIL: RelationTest#test_unscope_with_aliased_column (unexpected)
--- FAIL: ReflectionTest#test_name_error_from_incidental_code_is_not_converted_to_name_error_for_association (unexpected)
--- FAIL: RelationTest#test_find_all_using_where_with_relation_with_bound_values (unexpected)
--- FAIL: RelationTest#test_finding_with_reversed_assoc_order (unexpected)
--- FAIL: AssociationsTest#test_belongs_to_with_foreign_key_as_an_array_raises (unexpected)
--- FAIL: CockroachDB::FixturesTest#test_multiple_clean_fixtures (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_with_create_and_no_options (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_replace_association_with_duplicates (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_marshal_dump (unexpected)
--- FAIL: RelationTest#test_last (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_numeric_range_for_string (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_add_with_sti_middle_relation (unexpected)
--- FAIL: RelationTest#test_order_using_scoping (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_qualified_attribute_dot_notation_string (unexpected)
--- FAIL: BelongsToAssociationsTest#test_failing_create! (unexpected)
--- FAIL: RelationTest#test_finding_with_order_limit_and_offset (unexpected)
--- FAIL: RelationTest#test_only (unexpected)
--- FAIL: BelongsToAssociationsTest#test_destroying_polymorphic_child_with_unloaded_parent_and_touch_is_possible_with_has_many_inversing (unexpected)
--- FAIL: RelationTest#test_first_get_more_than_available (unexpected)
--- FAIL: RelationTest#test_count_explicit_columns (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_create_record (unexpected)
--- FAIL: FinderTest#test_model_class_responds_to_second_bang (unexpected)
--- FAIL: RelationTest#test_unscope_with_merge (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_count_has_many_through_with_named_scope (unexpected)
--- FAIL: FinderTest#test_exists_does_not_select_columns_without_alias (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_concat (unexpected)
--- FAIL: RelationTest#test_finding_with_reversed_arel_assoc_order (unexpected)
--- FAIL: BelongsToAssociationsTest#test_building_the_belonging_object_with_an_invalid_type (unexpected)
--- FAIL: CockroachDB::InheritanceComputeTypeTest#test_inheritance_new_with_subclass_as_default (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14754736/activerecord/1712644291193/1712653458765)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
- #118823 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14772140?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14772140?buildTab=artifacts#/activerecord) on master @ [911f1ce389342459592d89fe7b78bcba3a2d265a](https://github.com/cockroachdb/cockroach/commits/911f1ce389342459592d89fe7b78bcba3a2d265a):


```
--- FAIL: HasManyAssociationsTest#test_transaction_when_deleting_new_record (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_update_counter_caches_on_delete_with_dependent_destroy (unexpected)
--- FAIL: PersistenceTest#test_destroy! (unexpected)
--- FAIL: PersistenceTest#test_build_many_through_factory_with_block (unexpected)
--- FAIL: AssociationsJoinModelTest#test_has_many_with_piggyback (unexpected)
--- FAIL: CalculationsTest#test_distinct_joins_count_with_order_and_offset (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::ConnectionHandlersShardingDbTest#test_sharding_separation (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_destroy_on_association_clears_scope (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_open_ended_range (unexpected)
--- FAIL: AttributeMethodsTest#test_hash_content (unexpected)
--- FAIL: TransactionTest#test_successful (unexpected)
--- FAIL: AttributeMethodsTest#test_#id_value_alias_returns_id_column_only_for_composite_primary_key_models (unexpected)
--- FAIL: AttributeMethodsTest#test_declared_suffixed_attribute_method_affects_respond_to?_and_method_missing (unexpected)
--- FAIL: FinderTest#test_include_on_unloaded_relation_with_match (unexpected)
--- FAIL: TransactionTest#test_restore_active_record_state_for_all_records_in_a_transaction (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_a_time_zone_conversion_for_attributes_should_write_the_value_on_a_class_variable (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_grouped_with_invalid_field (unexpected)
--- FAIL: InheritanceTest#test_new_with_unrelated_namespaced_type (unexpected)
--- FAIL: PersistenceTest#test_create_model_with_uuid_pk_populates_id (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_through_scope_isnt_affected_by_scoping (unexpected)
--- FAIL: HasManyAssociationsTest#test_collection_size_with_dirty_target (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_field (unexpected)
--- FAIL: HasManyAssociationsTest#test_custom_named_counter_cache (unexpected)
--- FAIL: DelegatedTypeTest#test_delegated_class (unexpected)
--- FAIL: PostgresqlEnumTest#test_enum_defaults (unexpected)
--- FAIL: FinderTest#test_third_to_last (unexpected)
--- FAIL: FinderTest#test_find (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_via_block (unexpected)
--- FAIL: AttributeMethodsTest#test_boolean_attributes (unexpected)
--- FAIL: PersistenceTest#test_save_invalid_record (unexpected)
--- FAIL: CalculationsTest#test_count_selected_arel_attribute_in_joined_table (unexpected)
--- FAIL: CalculationsTest#test_sum_takes_attribute_type_precedence_over_database_type (unexpected)
--- FAIL: FinderTest#test_find_by_empty_ids (unexpected)
--- FAIL: HasManyAssociationsTest#test_include_returns_false_for_non_matching_record_to_verify_scoping (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_many_on_loaded_association_should_not_use_query (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_readers_respect_access_control (unexpected)
--- FAIL: FinderTest#test_exists_with_left_joins (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_update_on_id_changes_the_counter_cache (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_build_and_then_save_parent_should_not_reload_target (unexpected)
--- FAIL: SerializedAttributeTestWithYamlSafeLoad#test_serialized_attribute_on_custom_attribute_with_default (unexpected)
--- FAIL: HasManyAssociationsTest#test_adding_array_and_collection (unexpected)
--- FAIL: CalculationsTest#test_select_avg_with_joins_and_group_by_as_virtual_attribute_with_sql (unexpected)
--- FAIL: HasManyAssociationsTest#test_reattach_to_new_objects_replaces_inverse_association_and_foreign_key (unexpected)
--- FAIL: PersistenceTest#test_becomes_default_sti_subclass (unexpected)
--- FAIL: FinderTest#test_find_with_ids_and_order_clause (unexpected)
--- FAIL: PersistenceTest#test_update_many_with_invalid_id (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_save_returns_falsy_when_join_record_has_errors (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14772140/activerecord/1712729566899/1712737195456)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14790158?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14790158?buildTab=artifacts#/activerecord) on master @ [cead8be66c30a7377dacbf1c678911d2cb17194d](https://github.com/cockroachdb/cockroach/commits/cead8be66c30a7377dacbf1c678911d2cb17194d):


```
--- FAIL: HasManyAssociationsTest#test_destroying_by_string_id (unexpected)
--- FAIL: CalculationsTest#test_should_sum_field_with_conditions (unexpected)
--- FAIL: CalculationsTest#test_should_not_perform_joined_include_by_default (unexpected)
--- FAIL: EagerAssociationTest#test_eager_has_many_through_with_order (unexpected)
--- FAIL: CalculationsTest#test_should_generate_valid_sql_with_joins_and_group (unexpected)
--- FAIL: CalculationsTest#test_pluck_columns_with_same_name (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_limit_and_offset (unexpected)
--- FAIL: HasManyAssociationsTest#test_defining_has_many_association_with_delete_all_dependency_lazily_evaluates_target_class (unexpected)
--- FAIL: CalculationsTest#test_count_with_distinct (unexpected)
--- FAIL: EagerAssociationTest#test_including_duplicate_objects_from_has_many (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_adding_unsavable_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_transactions_when_replacing_on_new_record (unexpected)
--- FAIL: CalculationsTest#test_minimum_and_maximum_on_time_attributes (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_and_belongs_to_many_and_limit (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_conditions_hash (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_has_many_through_association_associated_by_a_composite_query_constraints (unexpected)
--- FAIL: PersistenceTest#test_build_many_through_factory_with_block (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_dynamic_find_should_respect_association_include (unexpected)
--- FAIL: CalculationsTest#test_distinct_joins_count_with_order_and_offset (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_many (unexpected)
--- FAIL: PersistenceTest#test_destroy_many_with_invalid_id (unexpected)
--- FAIL: ReflectionTest#test_content_columns (unexpected)
--- FAIL: CalculationsTest#test_ids_with_includes (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_first_sanitized (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_include_uses_array_include_after_loaded (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_and_create_from_association_should_respect_passed_attributes_over_default_scope (unexpected)
--- FAIL: EagerAssociationTest#test_eager-loading_readonly_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_updates_counter_cache_without_dependent_option (unexpected)
--- FAIL: PersistenceTest#test_destroy_new_record (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_does_not_raise_exception_when_association_does_not_exist (unexpected)
--- FAIL: PersistenceTest#test_save_for_record_with_only_primary_key (unexpected)
--- FAIL: CalculationsTest#test_should_limit_calculation (unexpected)
--- FAIL: InheritanceAttributeTest#test_inheritance_new_with_subclass_as_default (unexpected)
--- FAIL: CalculationsTest#test_average_with_from_option (unexpected)
--- FAIL: HasManyAssociationsTest#test_has_many_associations_on_new_records_use_null_relations (unexpected)
--- FAIL: HasManyAssociationsTest#test_anonymous_has_many (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_destroying_many (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_line_endings (unexpected)
--- FAIL: HasManyAssociationsTest#test_assign_ids_ignoring_blanks (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_explicit_join (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependent_association_respects_optional_hash_conditions_on_delete (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_find_in_association_with_options (unexpected)
--- FAIL: CalculationsTest#test_group_by_with_quoted_count_and_order_by_alias (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_does_not_cache_has_many_association_subset_when_preloaded_with_a_through_association (unexpected)
--- FAIL: PersistenceTest#test_delete_record_with_associations (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_index_sort_order (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_symbols_as_keys (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14790158/activerecord/1712816060185/1712824010693)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14807897?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14807897?buildTab=artifacts#/activerecord) on master @ [7dfff9430b0aedaac5bb57e06d704a527336863e](https://github.com/cockroachdb/cockroach/commits/7dfff9430b0aedaac5bb57e06d704a527336863e):


```
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_recursive_cascading_four_levels_has_many_through (unexpected)
--- FAIL: CalculationsTest#test_pick_one (unexpected)
--- FAIL: EagerAssociationTest#test_eager_has_many_through_multiple_with_order (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_through_has_one_association_with_conditions_on_association (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_against_given_relation (unexpected)
--- FAIL: PostgresqlVirtualColumnTest#test_virtual_column_with_full_inserts (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_conditions_hash (unexpected)
--- FAIL: CalculationsTest#test_apply_distinct_in_count (unexpected)
--- FAIL: AttributeMethodsTest#test_read_overridden_attribute (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_a_time_zone-aware_datetime_in_the_current_time_zone (unexpected)
--- FAIL: BasicsTest#test_sequence_name_with_abstract_class (unexpected)
--- FAIL: BasicsTest#test_generated_association_methods_module_name (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_empty_belongs_to_polymorphic (unexpected)
--- FAIL: BasicsTest#test_switching_between_table_name (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_present_with_booleans (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnNewRecord#test_autosave_new_record_on_belongs_to_can_be_disabled_per_relationship (unexpected)
--- FAIL: CalculationsTest#test_ids_on_loaded_relation_with_includes_and_table_scope (unexpected)
--- FAIL: CalculationsTest#test_pluck (unexpected)
--- FAIL: BasicsTest#test_find_multiple_ordered_last (unexpected)
--- FAIL: ActiveRecord::CollectionCacheKeyTest#test_cache_key_with_custom_timestamp_column (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_of_instance_dependent_associations_is_supported (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_one_dependent_does_not_destroy_dependent (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_a_time_zone-aware_attribute_interprets_time_zone-unaware_string_in_time_zone (unexpected)
--- FAIL: CalculationsTest#test_pluck_type_cast_with_eager_load_without_table_name_qualified_column (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_associations_with_string_joins_and_order_references (unexpected)
--- FAIL: EagerAssociationTest#test_count_with_include (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_with_invalid_field (unexpected)
--- FAIL: EagerAssociationTest#test_including_duplicate_objects_from_belongs_to (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attribute_for_database (unexpected)
--- FAIL: CalculationsTest#test_select_avg_with_joins_and_group_by_as_virtual_attribute_with_ar (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_order_string_with_quoted_table_name (unexpected)
--- FAIL: BasicsTest#test_comparison_with_different_objects_in_array (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_a_belongs_to_association (unexpected)
--- FAIL: BasicsTest#test_default_values (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_includes_limit_and_empty_result (unexpected)
--- FAIL: AttributeMethodsTest#test_hash_content (unexpected)
--- FAIL: AttributeMethodsTest#test_#define_attribute_methods_brings_back_undefined_aliases (unexpected)
--- FAIL: AttributeMethodsTest#test_aliasing_``id``_attribute_allows_reading_the_column_value_for_a_CPK_model (unexpected)
--- FAIL: ActiveRecord::Migration::ColumnsTest#test_change_column_default_to_null_with_not_null (unexpected)
--- FAIL: BasicsTest#test_ignored_columns_are_stored_as_an_array_of_string (unexpected)
--- FAIL: BasicsTest#test_dup_does_not_copy_associations (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_through_has_one_association_with_conditions_on_nested_association (unexpected)
--- FAIL: AttributeMethodsTest#test_define_attribute_method_works_with_both_symbol_and_string (unexpected)
--- FAIL: BasicsTest#test_benchmark_with_use_silence (unexpected)
--- FAIL: BasicsTest#test_table_exists (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_includes_offset (unexpected)
--- FAIL: ActiveRecord::StringTypeTest#test_string_mutations_are_detected (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14807897/activerecord/1712902263059/1712910518444)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14824676?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14824676?buildTab=artifacts#/activerecord) on master @ [f117eea22dd7be380c7141cdf6cd7aba92dd9c70](https://github.com/cockroachdb/cockroach/commits/f117eea22dd7be380c7141cdf6cd7aba92dd9c70):


```
--- FAIL: RelationTest#test_pluck_with_from_includes_original_table_name (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_create_with_conditions_hash_on_through_association (unexpected)
--- FAIL: PersistenceTest#test_reload_removes_custom_selects (unexpected)
--- FAIL: AssociationsTest#test_include_with_order_works (unexpected)
--- FAIL: FinderTest#test_joins_with_string_array (unexpected)
--- FAIL: CalculationsTest#test_select_avg_with_group_by_as_virtual_attribute_with_sql (unexpected)
--- FAIL: RelationTest#test_order_by_relation_attribute (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_empty_array_with_aggregate_having_multiple_mappings (unexpected)
--- FAIL: PersistenceTest#test_update (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_and_belongs_to_many_and_limit (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_adding_from_the_project (unexpected)
--- FAIL: BelongsToAssociationsTest#test_reloading_the_belonging_object (unexpected)
--- FAIL: InheritanceTest#test_scope_inherited_properly (unexpected)
--- FAIL: RelationTest#test_where_with_ar_object (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_updates_counter_cache_with_dependent_destroy (unexpected)
--- FAIL: BasicsTest#test_protected_environments_by_default_is_an_array_with_production (unexpected)
--- FAIL: CalculationsTest#test_should_return_zero_if_sum_conditions_return_nothing (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_reorder_raise_errors (unexpected)
--- FAIL: PersistenceTest#test_destroy_new_record (unexpected)
--- FAIL: RelationTest#test_finding_with_order_concatenated (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_reassignment_of_associated_type_updates_the_object (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_deleting (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_collection_delete_with_nonstandard_primary_key_on_belongs_to (unexpected)
--- FAIL: RelationTest#test_intersection_with_array (unexpected)
--- FAIL: FinderTest#test_find_by_nil_and_not_nil_attributes (unexpected)
--- FAIL: ActiveRecord::Migration::ForeignKeyChangeColumnTest#test_rename_column_of_child_table (unexpected)
--- FAIL: EagerAssociationTest#test_eager_has_many_through_multiple_with_order (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_squishes (unexpected)
--- FAIL: PersistenceTest#test_update_columns_not_equal_attributes (unexpected)
--- FAIL: InheritanceTest#test_company_descends_from_active_record (unexpected)
--- FAIL: RelationTest#test_find_id (unexpected)
--- FAIL: RelationTest#test_order_with_blank_value (unexpected)
--- FAIL: BasicsTest#test_limit_should_sanitize_sql_injection_for_limit_with_commas (unexpected)
--- FAIL: RelationTest#test_unscope_specific_where_value (unexpected)
--- FAIL: BasicsTest#test_equality_of_collection_proxy_and_association_relation (unexpected)
--- FAIL: HasOneAssociationsTest#test_nullification_on_cpk_association (unexpected)
--- FAIL: AttributeMethodsTest#test_caching_a_nil_primary_key (unexpected)
--- FAIL: HasManyAssociationsTest#test_update_all_respects_association_scope (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_updates_counter_cache_with_dependent_delete_all (unexpected)
--- FAIL: AttributeMethodsTest#test_number_attribute_predicate (unexpected)
--- FAIL: UniquenessValidationTest#test_validate_uniqueness_when_integer_out_of_range_show_order_does_not_matter (unexpected)
--- FAIL: BelongsToAssociationsTest#test_cant_save_readonly_association (unexpected)
--- FAIL: BelongsToAssociationsTest#test_building_the_belonging_object_with_explicit_sti_base_class (unexpected)
--- FAIL: FinderTest#test_find_by_two_attributes_but_passing_only_one (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_callbacks_firing_order_on_update (unexpected)
--- FAIL: RelationTest#test_ordering_with_extra_spaces (unexpected)
--- FAIL: BasicsTest#test_#connecting_to_with_prevent_writes (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14824676/activerecord/1712988889693/1712995725979)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14833400?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14833400?buildTab=artifacts#/activerecord) on master @ [67547d7724f8a52646e2e8ecb3ca48b923957d90](https://github.com/cockroachdb/cockroach/commits/67547d7724f8a52646e2e8ecb3ca48b923957d90):


```
--- FAIL: FinderTest#test_find_by_returns_nil_if_the_record_is_missing (unexpected)
--- FAIL: RelationTest#test_find_by_returns_nil_if_the_record_is_missing (unexpected)
--- FAIL: BasicsTest#test_find_by_slug_with_array (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_a_time_zone_conversion_for_attributes_should_write_the_value_on_a_class_variable (unexpected)
--- FAIL: BasicsTest#test_#connected_to_many_with_a_single_argument_for_classes (unexpected)
--- FAIL: BasicsTest#test_reload (unexpected)
--- FAIL: AssociationsTest#test_querying_by_single_associated_record_works_using_query_constraints (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_distinct_order_preserved (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_grouped_with_longer_field (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_update_columns_after_push_without_duplicate_join_table_rows (unexpected)
--- FAIL: RelationTest#test_order_using_scoping (unexpected)
--- FAIL: InheritanceTest#test_compute_type_argument_error (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_proxy_should_respond_to_private_methods_via_send (unexpected)
--- FAIL: BelongsToAssociationsTest#test_create! (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_with_validate_false_does_not_run_associated_validation_callbacks_on_create (unexpected)
--- FAIL: BelongsToAssociationsTest#test_clearing_an_association_clears_the_associations_inverse (unexpected)
--- FAIL: EagerAssociationTest#test_eager_habtm_with_association_inheritance (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_invalid_build (unexpected)
--- FAIL: InheritanceTest#test_new_without_storing_full_sti_class (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_child_is_visible_to_join_model_in_add_association_callbacks (unexpected)
--- FAIL: BelongsToAssociationsTest#test_building_the_belonging_object_with_an_invalid_type (unexpected)
--- FAIL: FinderTest#test_find_with_ids_with_id_out_of_range (unexpected)
--- FAIL: CalculationsTest#test_pluck_type_cast_with_eager_load_without_table_name_qualified_column (unexpected)
--- FAIL: OptimisticLockingTest#test_lock_new (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_adding_before_save (unexpected)
--- FAIL: FinderTest#test_find_an_empty_array (unexpected)
--- FAIL: AttributeMethodsTest#test_user-defined_text_attribute_predicate (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_push_with_invalid_record (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attribute_with_nil_should_not_asplode (unexpected)
--- FAIL: InheritanceTest#test_eager_load_belongs_to_primary_key_quoting (unexpected)
--- FAIL: CockroachDB::FixturesTest#test_multiple_clean_fixtures (unexpected)
--- FAIL: EagerAssociationTest#test_eager_load_belongs_to_quotes_table_and_column_names (unexpected)
--- FAIL: FinderTest#test_exists_with_empty_table_and_no_args_given (unexpected)
--- FAIL: EagerAssociationTest#test_belongs_to_association_ignores_the_scoping (unexpected)
--- FAIL: InheritanceTest#test_update_all_within_inheritance (unexpected)
--- FAIL: EagerAssociationTest#test_duplicate_middle_objects (unexpected)
--- FAIL: RelationTest#test_count_with_distinct (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_hashed_table_name (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_index_sort_order (unexpected)
--- FAIL: CalculationsTest#test_#skip_query_cache!_for_a_grouped_calculation (unexpected)
--- FAIL: FinderTest#test_find_with_nil_inside_set_passed_for_attribute (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_has_many_with_cpk (unexpected)
--- FAIL: FinderTest#test_first_bang_missing (unexpected)
--- FAIL: FinderTest#test_sole (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_destroy_on_association_clears_scope (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_respects_select (unexpected)
--- FAIL: BelongsToAssociationsTest#test_should_set_foreign_key_on_save! (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14833400/activerecord/1713074987049/1713083154146)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14842113?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14842113?buildTab=artifacts#/activerecord) on master @ [67547d7724f8a52646e2e8ecb3ca48b923957d90](https://github.com/cockroachdb/cockroach/commits/67547d7724f8a52646e2e8ecb3ca48b923957d90):


```
--- FAIL: SerializedAttributeTestWithYamlSafeLoad#test_serialized_attribute_on_alias_attribute (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_polymorphic_with_custom_foreign_type (unexpected)
--- FAIL: AttributeMethodsTest#test_declared_suffixed_attribute_method_affects_respond_to?_and_method_missing (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_touch_option_on_touch_without_updated_at_attributes (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_through_has_one_association_with_conditions_on_association (unexpected)
--- FAIL: RelationTest#test_create_or_find_by_with_bang (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_delete_on_association_clears_scope (unexpected)
--- FAIL: AttributeMethodsTest#test_case-sensitive_attributes_hash (unexpected)
--- FAIL: InheritanceTest#test_class_with_blank_sti_name (unexpected)
--- FAIL: PersistenceTest#test_delete_all (unexpected)
--- FAIL: CalculationsTest#test_count_selected_arel_attribute_in_joined_table (unexpected)
--- FAIL: BasicsTest#test_bignum_pk (unexpected)
--- FAIL: FinderTest#test_implicit_order_column_reorders_query_constraints (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_an_overridden_original_method_from_a_module_issues_a_deprecation (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_size_calculation_works_with_default_scoped_selects_when_not_previously_fetched (unexpected)
--- FAIL: UniquenessValidationTest#test_validate_uniqueness_when_integer_out_of_range (unexpected)
--- FAIL: RelationTest#test_relations_don't_load_all_records_in_#inspect (unexpected)
--- FAIL: HasManyAssociationsTest#test_depends_and_nullify_on_polymorphic_assoc (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_a_manually_defined_method_issues_a_deprecation_warning (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_aggregate_attribute_having_same_name_as_field_and_key_value_being_aggregate (unexpected)
--- FAIL: AttributeMethodsTest#test_undeclared_attribute_method_does_not_affect_respond_to?_and_method_missing (unexpected)
--- FAIL: HasManyAssociationsTest#test_finding_array_compatibility (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_delete_all_on_association_clears_scope (unexpected)
--- FAIL: BelongsToAssociationsTest#test_should_set_foreign_key_on_create_association_with_unpersisted_owner (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_with_has_one_through_an_sti_with_after_initialize (unexpected)
--- FAIL: PersistenceTest#test_becomes_initializes_missing_attributes (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_adding_before_save (unexpected)
--- FAIL: AttributeMethodsTest#test_declared_affixed_attribute_method_affects_respond_to?_and_method_missing (unexpected)
--- FAIL: ActiveRecord::Type::IntegerTest#test_casting_ActiveRecord_models (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_a_belongs_to_association (unexpected)
--- FAIL: BasicsTest#test_previously_changed (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_line_endings (unexpected)
--- FAIL: RelationTest#test_multi_where_ands_queries (unexpected)
--- FAIL: CalculationsTest#test_ids_on_loaded_relation_with_includes_and_table_scope (unexpected)
--- FAIL: FinderTest#test_include_on_unloaded_relation_with_offset (unexpected)
--- FAIL: FinderTest#test_find_by_and_where_consistency_with_active_record_instance (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_caching_of_columns (unexpected)
--- FAIL: BasicsTest#test_comparison_with_different_objects_in_array (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_touch_option_on_touch_and_removed_parent (unexpected)
--- FAIL: RelationTest#test_relations_don't_load_all_records_in_#pretty_print (unexpected)
--- FAIL: BasicsTest#test_scoped_can_take_a_values_hash (unexpected)
--- FAIL: FinderTest#test_#last_for_a_model_with_composite_query_constraints (unexpected)
--- FAIL: UniquenessValidationTest#test_validates_uniqueness_inside_scoping (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_has_many_through_with_distinct (unexpected)
--- FAIL: PersistenceTest#test_update_attribute_for_readonly_attribute! (unexpected)
--- FAIL: RelationTest#test_relations_show_the_records_in_#inspect (unexpected)
--- FAIL: BasicsTest#test_ignored_columns_are_stored_as_an_array_of_string (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14842113/activerecord/1713161409960/1713169546243)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14856478?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14856478?buildTab=artifacts#/activerecord) on master @ [3c643b003890560b16c4fee1d1c18bea1871803b](https://github.com/cockroachdb/cockroach/commits/3c643b003890560b16c4fee1d1c18bea1871803b):


```
--- FAIL: RelationTest#test_no_arguments_to_order_raise_errors (unexpected)
--- FAIL: BasicsTest#test_has_attribute_with_symbol (unexpected)
--- FAIL: HasManyAssociationsTest#test_has_many_through_respects_hash_conditions (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_one_should_return_false_if_zero (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependence (unexpected)
--- FAIL: RelationTest#test_find_by_with_non-hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: FinderTest#test_find_by_association_subquery (unexpected)
--- FAIL: HasManyAssociationsTest#test_exists_respects_association_scope (unexpected)
--- FAIL: HasManyAssociationsTest#test_assign_ids_ignoring_blanks (unexpected)
--- FAIL: AttributeMethodsTest#test_accessed_fields (unexpected)
--- FAIL: RelationTest#test_size_with_limit (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroy_on_association_clears_scope (unexpected)
--- FAIL: BasicsTest#test_#connecting_to_with_role_and_shard (unexpected)
--- FAIL: EagerAssociationTest#test_eager_load_has_many_through_quotes_table_and_column_names (unexpected)
--- FAIL: RelationTest#test_find_on_hash_conditions (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_keys_on_a_new_instance (unexpected)
--- FAIL: CockroachDB::DefaultNumbersTest#test_default_decimal_zero_with_large_scale (unexpected)
--- FAIL: BelongsToAssociationsTest#test_tracking_change_from_persisted_record_to_nil (unexpected)
--- FAIL: RelationTest#test_build_with_array (unexpected)
--- FAIL: ReflectionTest#test_collection_association (unexpected)
--- FAIL: FinderTest#test_fifth_have_primary_key_order_by_default (unexpected)
--- FAIL: RelationTest#test_find_by_returns_nil_if_the_record_is_missing (unexpected)
--- FAIL: HasManyAssociationsTest#test_counting_with_single_hash (unexpected)
--- FAIL: RelationTest#test_find_by!_with_hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: HasOneAssociationsTest#test_dependence_with_missing_association (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_has_many_through_with_custom_scope (unexpected)
--- FAIL: EagerAssociationTest#test_including_duplicate_objects_from_has_many (unexpected)
--- FAIL: HasOneAssociationsTest#test_dependence (unexpected)
--- FAIL: RelationTest#test_intersection_with_array (unexpected)
--- FAIL: TransactionCallbacksTest#test_only_call_after_commit_on_create_after_transaction_commits_for_new_record_if_create_succeeds_creating_through_association (unexpected)
--- FAIL: TransactionCallbacksTest#test_only_call_after_commit_on_save_after_transaction_commits_for_saving_record (unexpected)
--- FAIL: TouchTest#test_create_turned_off (unexpected)
--- FAIL: HasManyAssociationsTest#test_collection_size_with_dirty_target (unexpected)
--- FAIL: StrictLoadingTest#test_strict_loading_n_plus_one_only_mode (unexpected)
--- FAIL: RelationTest#test_size_with_eager_loading_and_manual_distinct_select_and_custom_order (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_left_outer_joins_raise_errors (unexpected)
--- FAIL: SerializedAttributeTest#test_nil_is_not_changed_when_serialized_with_a_class (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_custom_join_table (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_join_model_with_conditions (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_order_string_with_quoted_table_name (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_touch_option_on_destroy_with_destroyed_parent (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_invalid_adding (unexpected)
--- FAIL: AttributeMethodsTest#test_on_the_fly_super_invokable_generated_attribute_methods_via_method_missing (unexpected)
--- FAIL: HasOneAssociationsTest#test_restrict_with_error_with_locale (unexpected)
--- FAIL: TestAutosaveAssociationOnAHasManyAssociationWithInverse#test_after_save_callback_with_autosave (unexpected)
--- FAIL: BasicsTest#test_find_on_abstract_base_class_doesnt_use_type_condition (unexpected)
--- FAIL: ReflectionTest#test_automatic_inverse_does_not_suppress_name_error_from_incidental_code (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14856478/activerecord/1713247885576/1713256801965)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14871820?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14871820?buildTab=artifacts#/activerecord) on master @ [934684d8134fd8bb34eae1a37f3aa83a4ac066b7](https://github.com/cockroachdb/cockroach/commits/934684d8134fd8bb34eae1a37f3aa83a4ac066b7):


```
--- FAIL: RelationTest#test_find_by_with_delegated_ar_object (unexpected)
--- FAIL: DateTimeTest#test_assign_empty_date (unexpected)
--- FAIL: RelationTest#test_find_ids (unexpected)
--- FAIL: HasManyAssociationsTest#test_new_the_association_with_an_array (unexpected)
--- FAIL: CalculationsTest#test_async_pluck_on_loaded_relation (unexpected)
--- FAIL: CalculationsTest#test_ids_on_loaded_relation_with_scope (unexpected)
--- FAIL: FinderTest#test_find_with_ids_returning_ordered (unexpected)
--- FAIL: RelationTest#test_#skip_query_cache! (unexpected)
--- FAIL: AttributeMethodsTest#test_string_attribute_predicate (unexpected)
--- FAIL: RelationTest#test_size_with_eager_loading_and_custom_select_and_order (unexpected)
--- FAIL: BelongsToAssociationsTest#test_building_the_belonging_object_with_explicit_sti_base_class (unexpected)
--- FAIL: TransactionTest#test_no_savepoint_in_nested_transaction_without_force (unexpected)
--- FAIL: FinderTest#test_sole (unexpected)
--- FAIL: CalculationsTest#test_count_with_arel_star (unexpected)
--- FAIL: RelationTest#test_create_bang (unexpected)
--- FAIL: EagerAssociationTest#test_should_work_inverse_of_with_eager_load (unexpected)
--- FAIL: ReflectionTest#test_has_many_reflection_with_array_fk_raises (unexpected)
--- FAIL: CalculationsTest#test_should_count_manual_select_with_group_with_count_all (unexpected)
--- FAIL: MultiParameterAttributeTest#test_multiparameter_attributes_on_date_with_empty_year_and_month (unexpected)
--- FAIL: TransactionTest#test_rollback_dirty_changes_then_retry_save_on_new_record (unexpected)
--- FAIL: EnumTest#test_capital_characters_for_enum_names (unexpected)
--- FAIL: RelationTest#test_create_or_find_by (unexpected)
--- FAIL: EagerAssociationTest#test_eager_count_performed_on_a_has_many_through_association_with_multi_table_conditional (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_first_after_reload (unexpected)
--- FAIL: ModulesTest#test_table_name (unexpected)
--- FAIL: RelationTest#test_reverse_order_with_reorder_nil_removes_the_order (unexpected)
--- FAIL: RelationTest#test_#where_with_empty_set (unexpected)
--- FAIL: InverseBelongsToTests#test_recursive_model_has_many_inversing (unexpected)
--- FAIL: AssociationsTest#test_cpk_model_has_many_records_by_id_attribute (unexpected)
--- FAIL: RelationTest#test_first_or_initialize (unexpected)
--- FAIL: RelationTest#test_first_get_more_than_available (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroy_on_association_clears_scope (unexpected)
--- FAIL: FinderTest#test_third (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_yaml_dump_and_load (unexpected)
--- FAIL: RelationTest#test_typecasting_where_with_array (unexpected)
--- FAIL: HasManyAssociationsTest#test_build (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_invalid_string_to_a_zone-aware_time_attribute (unexpected)
--- FAIL: IntegrationTest#test_cache_key_for_newer_updated_at (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_with_rewhere_doesnt_set_inverse_instance_key (unexpected)
--- FAIL: CalculationsTest#test_limit_should_apply_before_count_arel_attribute (unexpected)
--- FAIL: PessimisticLockingTest#test_with_lock_rolls_back_transaction (unexpected)
--- FAIL: FinderTest#test_count_by_sql (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_aggregate_having_three_mappings_array (unexpected)
--- FAIL: EagerAssociationTest#test_three_level_nested_preloading_does_not_raise_exception_when_association_does_not_exist (unexpected)
--- FAIL: FinderTest#test_second_with_offset (unexpected)
--- FAIL: AttributeMethodsTest#test_boolean_attributes (unexpected)
--- FAIL: FinderTest#test_include_on_loaded_relation_without_match (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14871820/activerecord/1713334347664/1713341333626)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14886544?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14886544?buildTab=artifacts#/activerecord) on master @ [2d67111f0db7bec9bfed537542c60f37f8340f69](https://github.com/cockroachdb/cockroach/commits/2d67111f0db7bec9bfed537542c60f37f8340f69):


```
--- FAIL: CalculationsTest#test_ids_with_join (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_with_custom_primary_key (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_with_purpose (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_include_method_in_has_and_belongs_to_many_association_should_return_true_for_instance_added_with_build (unexpected)
--- FAIL: CalculationsTest#test_should_return_decimal_average_of_integer_field (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_create_record (unexpected)
--- FAIL: RelationTest#test_intersection_with_array (unexpected)
--- FAIL: CalculationsTest#test_should_not_use_alias_for_grouped_field (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_build_then_remove_then_save (unexpected)
--- FAIL: CalculationsTest#test_no_limit_no_offset (unexpected)
--- FAIL: FinderTest#test_find_by_one_attribute_that_is_an_aggregate (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_collection_create_with_nonstandard_primary_key_on_belongs_to (unexpected)
--- FAIL: CalculationsTest#test_should_not_perform_joined_include_by_default (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_existing_with_nonstandard_primary_key_on_belongs_to (unexpected)
--- FAIL: CalculationsTest#test_pluck_if_table_included (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_has_and_belongs_to_many_associations_on_new_records_use_null_relations (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_marshal_dump (unexpected)
--- FAIL: FinderTest#test_find_by_id_with_hash (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_add_two_instance_and_then_deleting (unexpected)
--- FAIL: FinderTest#test_include_on_loaded_relation_with_composite_primary_key (unexpected)
--- FAIL: RelationTest#test_unscoped_block_style (unexpected)
--- FAIL: RelationTest#test_unscope_with_table_name_qualified_hash (unexpected)
--- FAIL: RelationTest#test_relations_with_cached_arel_can't_be_mutated_[internal_API] (unexpected)
--- FAIL: RelationTest#test_first_or_create (unexpected)
--- FAIL: IntegrationTest#test_to_param_for_a_composite_primary_key_model (unexpected)
--- FAIL: FinderTest#test_with_limiting_with_custom_select (unexpected)
--- FAIL: HasOneAssociationsTest#test_restrict_with_error (unexpected)
--- FAIL: CalculationsTest#test_pluck_and_distinct (unexpected)
--- FAIL: HasOneAssociationsTest#test_cant_save_readonly_association (unexpected)
--- FAIL: HasOneAssociationsTest#test_build_and_create_should_not_happen_within_scope (unexpected)
--- FAIL: CalculationsTest#test_distinct_joins_count_with_order_and_limit_and_offset (unexpected)
--- FAIL: RelationTest#test_create_or_find_by_within_transaction (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_field_with_conditions (unexpected)
--- FAIL: RelationTest#test_find_by_with_hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: CalculationsTest#test_ids_with_includes (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_with_touch_option_on_nonpersisted_built_associations_doesnt_update_parent (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_association (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_has_many_through_polymorphic_has_manys_works (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_marshal_dump (unexpected)
--- FAIL: RelationTest#test_#where_with_empty_set (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_replace_association (unexpected)
--- FAIL: FinderTest#test_exists_with_left_joins (unexpected)
--- FAIL: RelationTest#test_finding_with_subquery (unexpected)
--- FAIL: RelationTest#test_joins_with_nil_argument (unexpected)
--- FAIL: RelationTest#test_first_or_create_with_block (unexpected)
--- FAIL: CalculationsTest#test_minimum_with_from_option (unexpected)
--- FAIL: CalculationsTest#test_pluck_type_cast_with_eager_load_without_table_name_qualified_column (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14886544/activerecord/1713420957812/1713428331519)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14886572?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14886572?buildTab=artifacts#/activerecord) on master @ [2d67111f0db7bec9bfed537542c60f37f8340f69](https://github.com/cockroachdb/cockroach/commits/2d67111f0db7bec9bfed537542c60f37f8340f69):


```
--- FAIL: HasManyThroughAssociationsTest#test_include_method_in_association_through_should_return_true_for_instance_added_with_build (unexpected)
--- FAIL: BelongsToAssociationsTest#test_runs_parent_presence_check_if_parent_has_not_changed_and_belongs_to_required_validates_foreign_key_is_set (unexpected)
--- FAIL: RelationTest#test_first_or_initialize_with_no_parameters (unexpected)
--- FAIL: BasicsTest#test_previously_changed_dup (unexpected)
--- FAIL: RelationTest#test_find_or_create_by! (unexpected)
--- FAIL: RelationTest#test_relations_with_cached_arel_can't_be_mutated_[internal_API] (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_collection_create_with_nonstandard_primary_key_on_belongs_to (unexpected)
--- FAIL: BasicsTest#test_find_reverse_ordered_last (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_with_create (unexpected)
--- FAIL: RelationTest#test_eager_load_with_blank_value (unexpected)
--- FAIL: ReflectionTest#test_has_one_reflection (unexpected)
--- FAIL: ReflectionTest#test_association_primary_key (unexpected)
--- FAIL: BelongsToAssociationsTest#test_reflect_the_most_recent_change (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_association_through_a_has_many_association_to_self (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_polymorphic_with_primary_key_option (unexpected)
--- FAIL: CalculationsTest#test_ids_on_loaded_relation_with_includes_and_table_scope (unexpected)
--- FAIL: InheritanceTest#test_class_without_store_full_sti_class_returns_demodulized_name (unexpected)
--- FAIL: RelationTest#test_preload_applies_to_all_chained_preloaded_scopes (unexpected)
--- FAIL: RelationTest#test_find_all_with_multiple_should_use_and (unexpected)
--- FAIL: BasicsTest#test_connection_in_utc_time (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_field_with_conditions_and_having (unexpected)
--- FAIL: ReflectionTest#test_collection_association (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_eager_load_raise_errors (unexpected)
--- FAIL: RelationTest#test_order_with_hash_and_symbol_generates_the_same_sql (unexpected)
--- FAIL: CockroachDB::FixturesTest#test_multiple_clean_fixtures (unexpected)
--- FAIL: CalculationsTest#test_pick_loaded_relation_aliased_attribute (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_build_then_save_with_has_many_inverse (unexpected)
--- FAIL: BasicsTest#test_distinct_delegates_to_scoped (unexpected)
--- FAIL: CalculationsTest#test_ids_with_includes_and_non_primary_key_order (unexpected)
--- FAIL: BelongsToAssociationsTest#test_building_the_belonging_object_with_sti_subclass (unexpected)
--- FAIL: CalculationsTest#test_pluck (unexpected)
--- FAIL: CalculationsTest#test_should_count_manual_select_with_group_with_count_all (unexpected)
--- FAIL: RelationTest#test_to_xml (unexpected)
--- FAIL: BelongsToAssociationsTest#test_dangerous_association_name_raises_ArgumentError (unexpected)
--- FAIL: InheritanceTest#test_where_create_with_unrelated_type (unexpected)
--- FAIL: CalculationsTest#test_distinct_joins_count_with_order_and_limit (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_unscope_raise_errors (unexpected)
--- FAIL: CalculationsTest#test_grouped_calculation_with_polymorphic_relation (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_size_of_through_association_should_increase_correctly_when_has_many_association_is_added (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_existing_with_nonstandard_primary_key_on_belongs_to (unexpected)
--- FAIL: RelationTest#test_scoped_all (unexpected)
--- FAIL: InheritanceTest#test_new_with_unrelated_namespaced_type (unexpected)
--- FAIL: CalculationsTest#test_should_return_decimal_average_of_integer_field (unexpected)
--- FAIL: InheritanceTest#test_alt_inheritance_save (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_belongs_to_with_dependent_destroy (unexpected)
--- FAIL: BelongsToAssociationsTest#test_create_bang_with_block (unexpected)
--- FAIL: RelationTest#test_unscope_with_double_dot_where (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14886572/activerecord/1713420787855/1713428799424)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14886600?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14886600?buildTab=artifacts#/activerecord) on master @ [2d67111f0db7bec9bfed537542c60f37f8340f69](https://github.com/cockroachdb/cockroach/commits/2d67111f0db7bec9bfed537542c60f37f8340f69):


```
--- FAIL: FinderTest#test_implicit_order_for_model_without_primary_key (unexpected)
--- FAIL: HasOneAssociationsTest#test_attributes_are_being_set_when_initialized_from_has_one_association_with_where_clause (unexpected)
--- FAIL: ReflectionTest#test_foreign_key_is_inferred_from_model_name (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_updates_counter_cache_with_dependent_delete_all (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_one_dependent_does_not_destroy_dependent (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_aggregate_having_three_mappings (unexpected)
--- FAIL: PersistenceTest#test_preserve_original_sti_type (unexpected)
--- FAIL: PersistenceTest#test_update_columns_should_not_use_setter_method (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_and_create_should_not_happen_within_scope (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_ids_and_inverse_of (unexpected)
--- FAIL: PersistenceTest#test_save_touch_false (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_followed_by_save_does_not_load_target (unexpected)
--- FAIL: EagerAssociationTest#test_eager_count_performed_on_a_has_many_through_association_with_multi_table_conditional (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_many_should_return_false_if_none_or_one (unexpected)
--- FAIL: EagerAssociationTest#test_finding_with_includes_on_null_belongs_to_polymorphic_association (unexpected)
--- FAIL: FinderTest#test_exists_with_scope (unexpected)
--- FAIL: ReflectionTest#test_column_string_type_and_limit (unexpected)
--- FAIL: FinderTest#test_#last_for_a_model_with_composite_query_constraints (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit_with_no_results (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_with_block (unexpected)
--- FAIL: HasOneAssociationsTest#test_association_enum_works_properly (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method (unexpected)
--- FAIL: EagerAssociationTest#test_loading_with_multiple_associations (unexpected)
--- FAIL: PersistenceTest#test_becomes_includes_changed_attributes (unexpected)
--- FAIL: FinderTest#test_find_by!_with_non-hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: HasManyAssociationsTest#test_associations_replace_in_memory_when_records_have_the_same_id (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroy_all (unexpected)
--- FAIL: HasManyAssociationsTest#test_adding_a_collection (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_one_should_return_false_if_more_than_one (unexpected)
--- FAIL: IntegrationTest#test_param_delimiter_changes_delimiter_used_in_to_param (unexpected)
--- FAIL: FinderTest#test_exists_with_distinct_association_includes_limit_and_order (unexpected)
--- FAIL: PersistenceTest#test_create_with_custom_timestamps (unexpected)
--- FAIL: FinderTest#test_hash_condition_local_time_interpolation_with_default_timezone_utc (unexpected)
--- FAIL: ReflectionTest#test_reflection_klass_not_found_with_no_class_name_option (unexpected)
--- FAIL: PersistenceTest#test_update_attribute_for_updated_at_on! (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_association_with_bang (unexpected)
--- FAIL: FinderTest#test_member_when_non_AR_object_passed_on_unloaded_relation (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_cache_nils (unexpected)
--- FAIL: FinderTest#test_find_some_message_with_custom_primary_key (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_updates_counter_cache (unexpected)
--- FAIL: FinderTest#test_select_rows (unexpected)
--- FAIL: HasManyAssociationsTest#test_update_all_on_association_accessed_before_save (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependence_on_account (unexpected)
--- FAIL: HasOneAssociationsTest#test_reload_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_scoped_grouped (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_first_or_last_on_loaded_association_should_not_fetch_with_query (unexpected)
--- FAIL: FinderTest#test_member_on_loaded_relation_with_match (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14886600/activerecord/1713420979426/1713429060742)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14898257?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14898257?buildTab=artifacts#/activerecord) on master @ [737df7fe75c5698d9ba384ad322f30963582cc42](https://github.com/cockroachdb/cockroach/commits/737df7fe75c5698d9ba384ad322f30963582cc42):


```
--- FAIL: FinderTest#test_fifth_have_primary_key_order_by_default (unexpected)
--- FAIL: PostgresqlBitStringTest#test_roundtrip (unexpected)
--- FAIL: PostgresqlEnumTest#test_enum_mapping (unexpected)
--- FAIL: FinderTest#test_find_by_nil_attribute (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_one_dependent_does_not_destroy_dependent (unexpected)
--- FAIL: TransactionTest#test_successful_with_return_outside_inner_transaction (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_with_validate_false_does_not_run_associated_validation_callbacks_on_create (unexpected)
--- FAIL: DirtyTest#test_attribute_will_change!_doesn't_try_to_save_non-persistable_attributes (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_for_inspect_with_a_non-primary_key_id_attribute (unexpected)
--- FAIL: AttributeMethodsTest#test_#undefine_attribute_methods_undefines_alias_attribute_methods (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_a_item_which_is_not_in_the_collection (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_through_has_one_association_with_conditions_on_association (unexpected)
--- FAIL: EagerAssociationTest#test_eager-loading_readonly_association (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_for_inspect_with_an_array (unexpected)
--- FAIL: FinderTest#test_fifth (unexpected)
--- FAIL: BasicsTest#test_readonly_attributes_in_abstract_class_descendant (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_field_with_conditions (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_belongs_to_association_associated_by_a_composite_query_constraints (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_replace_on_new_object (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_association_with_bang (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_names_with_a_custom_select (unexpected)
--- FAIL: HasOneThroughAssociationsTest#test_building_works_with_has_one_through_belongs_to (unexpected)
--- FAIL: BasicsTest#test_attribute_names_on_abstract_class (unexpected)
--- FAIL: JsonAttributeTest#test_select_nil_json_after_create (unexpected)
--- FAIL: CalculationsTest#test_should_return_nil_as_average (unexpected)
--- FAIL: BelongsToAssociationsTest#test_assignment_before_child_saved_with_primary_key (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_obeys_order_on_through_association (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_overloading_types (unexpected)
--- FAIL: CalculationsTest#test_count_should_shortcut_with_limit_zero (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_adding_multiple (unexpected)
--- FAIL: ReflectionTest#test_join_table_with_common_prefix (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_find_scoped_grouped (unexpected)
--- FAIL: BelongsToAssociationsTest#test_assigning_an_association_doesn't_result_in_duplicate_objects (unexpected)
--- FAIL: BasicsTest#test_equality_of_collection_proxy_and_association_relation (unexpected)
--- FAIL: ReflectionTest#test_has_many_reflection (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_default_scope (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_update_columns_after_push_without_duplicate_join_table_rows (unexpected)
--- FAIL: FinderTest#test_exists_returns_false_with_false_arg (unexpected)
--- FAIL: BelongsToAssociationsTest#test_custom_attribute_with_select (unexpected)
--- FAIL: HasManyAssociationsTest#test_included_in_collection (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_find_scoped_grouped_having (unexpected)
--- FAIL: BasicsTest#test_clear_cache! (unexpected)
--- FAIL: CalculationsTest#test_should_return_float_average_if_db_returns_such (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_a_time_zone-aware_datetime_in_the_current_time_zone (unexpected)
--- FAIL: BasicsTest#test_readonly_attributes (unexpected)
--- FAIL: SerializedAttributeTest#test_serialize_with_coder (unexpected)
--- FAIL: EagerAssociationTest#test_finding_with_includes_on_has_many_association_with_same_include_includes_only_once (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14898257/activerecord/1713507063996/1713514536687)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14910161?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14910161?buildTab=artifacts#/activerecord) on master @ [c4ab095c4f65b9140661ed57adddc690b1e3ce3f](https://github.com/cockroachdb/cockroach/commits/c4ab095c4f65b9140661ed57adddc690b1e3ce3f):


```
--- FAIL: RelationTest#test_default_scope_order_with_scope_order (unexpected)
--- FAIL: RelationTest#test_create_with_block (unexpected)
--- FAIL: RelationTest#test_automatically_added_where_references (unexpected)
--- FAIL: RelationTest#test_unscope_with_double_dot_where (unexpected)
--- FAIL: HasOneAssociationsTest#test_polymorphic_has_one_with_touch_option_on_update_will_touch_record_by_fetching_from_database_if_needed (unexpected)
--- FAIL: RelationTest#test_reverse_arel_order_with_function (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroy_all_on_desynced_counter_cache_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_to_a_should_dup_target (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_build_before_child_saved (unexpected)
--- FAIL: RelationTest#test_unscope_with_triple_dot_where (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_via_block (unexpected)
--- FAIL: HasOneAssociationsTest#test_successful_build_association (unexpected)
--- FAIL: RelationTest#test_presence (unexpected)
--- FAIL: RelationTest#test_already-loaded_relations_don't_perform_a_new_query_in_#pretty_print (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_save_fails_for_invalid_has_one (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_with_extend_option (unexpected)
--- FAIL: InheritanceTest#test_inheritance_condition (unexpected)
--- FAIL: HasManyAssociationsTest#test_sending_new_to_association_proxy_should_have_same_effect_as_calling_new (unexpected)
--- FAIL: RelationTest#test_relations_with_cached_arel_can't_be_mutated_[internal_API] (unexpected)
--- FAIL: RelationTest#test_anonymous_extension (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_first_nth_or_last_on_new_record_should_not_run_queries (unexpected)
--- FAIL: HasManyAssociationsTest#test_replace_with_less (unexpected)
--- FAIL: RelationTest#test_default_reverse_order_on_table_without_primary_key (unexpected)
--- FAIL: HasManyAssociationsTest#test_do_not_call_callbacks_for_delete_all (unexpected)
--- FAIL: HasOneAssociationsTest#test_association_enum_works_properly_with_nested_join (unexpected)
--- FAIL: AssociationsTest#test_assign_composite_foreign_key_belongs_to_association_with_autosave (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_by_string_id (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_all_with_include_and_conditions (unexpected)
--- FAIL: RelationTest#test_finding_with_reorder_by_aliased_attributes (unexpected)
--- FAIL: HasManyAssociationsTest#test_create_without_loading_association (unexpected)
--- FAIL: ModulesTest#test_nested_models_should_not_raise_exception_when_using_delete_all_dependency_on_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_restrict_with_error_with_locale (unexpected)
--- FAIL: RelationTest#test_to_a_should_dup_target (unexpected)
--- FAIL: HasManyAssociationsTest#test_raises_RecordNotDestroyed_when_replaced_child_can't_be_destroyed (unexpected)
--- FAIL: RelationTest#test_find_by!_with_multi-arg_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: AssociationsTest#test_belongs_to_a_cpk_model_by_id_attribute (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_none_should_count_instead_of_loading_association (unexpected)
--- FAIL: InheritanceTest#test_new_without_storing_full_sti_class (unexpected)
--- FAIL: RelationTest#test_two_scopes_with_includes_should_not_drop_any_include (unexpected)
--- FAIL: RelationTest#test_find_with_list_of_ar (unexpected)
--- FAIL: RelationTest#test_reverse_order_with_function (unexpected)
--- FAIL: HasOneAssociationsTest#test_association_change_calls_delete (unexpected)
--- FAIL: RelationTest#test_first_or_create_bang_with_valid_block (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_one_on_loaded_association_should_not_use_query (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_updates_counter_cache (unexpected)
--- FAIL: HasManyAssociationsTest#test_associations_replace_in_memory_when_records_have_the_same_id (unexpected)
--- FAIL: RelationTest#test_find_by_with_hash_conditions_returns_the_first_matching_record (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14910161/activerecord/1713593486193/1713601977530)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14913779?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14913779?buildTab=artifacts#/activerecord) on master @ [c4ab095c4f65b9140661ed57adddc690b1e3ce3f](https://github.com/cockroachdb/cockroach/commits/c4ab095c4f65b9140661ed57adddc690b1e3ce3f):


```
--- FAIL: BasicsTest#test_select_symbol (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_an_sti_join_model_with_conditions_on_both (unexpected)
--- FAIL: CalculationsTest#test_async_pluck_none_relation (unexpected)
--- FAIL: AttributeMethodsTest#test_bulk_updates_respect_access_control (unexpected)
--- FAIL: RelationTest#test_preload_applies_to_all_chained_preloaded_scopes (unexpected)
--- FAIL: RelationTest#test_reload (unexpected)
--- FAIL: AttributeMethodsTest#test_global_methods_are_overwritten_when_subclassing (unexpected)
--- FAIL: PersistenceTest#test_delete_all (unexpected)
--- FAIL: PersistenceTest#test_update_columns_with_one_readonly_attribute (unexpected)
--- FAIL: FinderTest#test_member_on_unloaded_relation_with_match (unexpected)
--- FAIL: AssociationsTest#test_assign_composite_foreign_key_belongs_to_association_with_autosave (unexpected)
--- FAIL: CalculationsTest#test_pluck_on_aliased_attribute (unexpected)
--- FAIL: BasicsTest#test_equality_of_relation_and_array (unexpected)
--- FAIL: FinderTest#test_find_on_multiple_hash_conditions (unexpected)
--- FAIL: FinderTest#test_nth_to_last_with_order_uses_limit (unexpected)
--- FAIL: RelationTest#test_unscope_with_table_name_qualified_hash (unexpected)
--- FAIL: CalculationsTest#test_pluck_for_a_composite_primary_key (unexpected)
--- FAIL: RelationTest#test_first_or_create_with_block (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_joins_raise_errors (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_assignment_before_either_saved (unexpected)
--- FAIL: PersistenceTest#test_build_many (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_the_same_association_twice_works (unexpected)
--- FAIL: RelationTest#test_alias_tracker_respects_a_custom_table (unexpected)
--- FAIL: RelationTest#test_find_by_with_take_memoization (unexpected)
--- FAIL: CalculationsTest#test_ids_async_on_loaded_relation (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_has_one_using_reorder (unexpected)
--- FAIL: FinderTest#test_find_by_with_range_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: FinderTest#test_find_with_ids_and_offset (unexpected)
--- FAIL: PersistenceTest#test_becomes_default_sti_subclass (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_distinct_before_the_fact (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: RelationTest#test_finding_last_with_arel_order (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_a_time_zone-aware_datetime_in_the_current_time_zone (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_custom_join_table (unexpected)
--- FAIL: BasicsTest#test_limit_should_take_value_from_latest_limit (unexpected)
--- FAIL: RelationTest#test_relations_limit_the_records_in_#pretty_print_at_10 (unexpected)
--- FAIL: AttributeMethodsTest#test_typecast_attribute_from_select_to_true (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_find_scoped_grouped_having (unexpected)
--- FAIL: CalculationsTest#test_should_get_maximum_of_field_with_include (unexpected)
--- FAIL: RelationTest#test_ordering_with_extra_spaces (unexpected)
--- FAIL: RelationTest#test_find_on_hash_conditions (unexpected)
--- FAIL: CalculationsTest#test_select_avg_with_group_by_as_virtual_attribute_with_ar (unexpected)
--- FAIL: FinderTest#test_first_have_primary_key_order_by_default (unexpected)
--- FAIL: BasicsTest#test_equality_of_relation_and_association_relation (unexpected)
--- FAIL: CalculationsTest#test_no_limit_no_offset (unexpected)
--- FAIL: RelationTest#test_find_by_id_with_list_of_ar (unexpected)
--- FAIL: EagerAssociationTest#test_join_eager_with_empty_order_should_generate_valid_sql (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14913779/activerecord/1713613628990/1713621647181)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14918392?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14918392?buildTab=artifacts#/activerecord) on master @ [e0068814dfcb4f975a53b79b5546b5fb85c0f927](https://github.com/cockroachdb/cockroach/commits/e0068814dfcb4f975a53b79b5546b5fb85c0f927):


```
--- FAIL: EagerAssociationTest#test_join_eager_with_empty_order_should_generate_valid_sql (unexpected)
--- FAIL: FinderTest#test_find_with_bad_sql (unexpected)
--- FAIL: BasicsTest#test_marshal_inspected_round_trip (unexpected)
--- FAIL: CalculationsTest#test_should_sum_arel_attribute (unexpected)
--- FAIL: CalculationsTest#test_should_average_arel_attribute (unexpected)
--- FAIL: RelationTest#test_size_with_eager_loading_and_custom_order (unexpected)
--- FAIL: ActiveRecord::SelectTest#test_select_with_hash_argument (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_preload_sti_middle_relation (unexpected)
--- FAIL: RelationTest#test_create_with_block (unexpected)
--- FAIL: EagerAssociationTest#test_conditions_on_join_table_with_include_and_limit (unexpected)
--- FAIL: CalculationsTest#test_count_arel_attribute_in_joined_table_with (unexpected)
--- FAIL: EagerAssociationTest#test_eager-loading_with_a_polymorphic_association_won't_work_consistently (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_has_many_through_with_implicit_source (unexpected)
--- FAIL: FinderTest#test_find_by!_with_non-hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: FinderTest#test_exists_with_large_number (unexpected)
--- FAIL: BasicsTest#test_typecasting_aliases (unexpected)
--- FAIL: AttributeMethodsTest#test_came_from_user? (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_has_one_with_cpk (unexpected)
--- FAIL: BasicsTest#test_marshal_round_trip (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_aggregate_having_one_mapping_array (unexpected)
--- FAIL: BasicsTest#test_find_symbol_ordered_last (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_has_many_through_with_custom_scope (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_should_raise_exception_for_destroying_mismatching_records (unexpected)
--- FAIL: PersistenceTest#test_becomes_includes_changed_attributes (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_dynamic_find_should_respect_association_include (unexpected)
--- FAIL: RelationTest#test_create_or_find_by_with_non_unique_attributes (unexpected)
--- FAIL: InheritanceTest#test_base_class_activerecord_error (unexpected)
--- FAIL: EagerAssociationTest#test_polymorphic_type_condition (unexpected)
--- FAIL: EagerAssociationTest#test_loading_association_with_string_joins (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_proxy_should_respond_to_private_methods_via_send (unexpected)
--- FAIL: PersistenceTest#test_becomes_default_sti_subclass (unexpected)
--- FAIL: CalculationsTest#test_count_on_invalid_columns_raises (unexpected)
--- FAIL: BasicsTest#test_ignored_columns_don't_prevent_explicit_declaration_of_attribute_methods (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_destroy_all_on_composite_primary_key_model (unexpected)
--- FAIL: EagerAssociationTest#test_loading_polymorphic_association_with_mixed_table_conditions (unexpected)
--- FAIL: TransactionInstrumentationTest#test_transaction_instrumentation_with_restart_savepoint_parent_transactions_on_commit (unexpected)
--- FAIL: CalculationsTest#test_ids_with_scope (unexpected)
--- FAIL: FinderTest#test_exists_with_string (unexpected)
--- FAIL: AssociationsTest#test_append_composite_foreign_key_has_many_association_with_autosave (unexpected)
--- FAIL: MultiParameterAttributeTest#test_multiparameter_attributes_setting_date_and_time_attribute (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_an_overridden_original_method_along_with_an_overridden_alias_method_doesn't_issue_a_deprecation (unexpected)
--- FAIL: BelongsToAssociationsTest#test_building_the_belonging_object_with_an_unrelated_type (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_circular_autosave_association_correctly_saves_multiple_records (unexpected)
--- FAIL: FinderTest#test_third_have_primary_key_order_by_default (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_adding_from_the_project_fixed_timestamp (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_uses_default_if_blank (unexpected)
--- FAIL: CalculationsTest#test_limit_should_apply_before_count (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14918392/activerecord/1713680133985/1713688080495)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14976785?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14976785?buildTab=artifacts#/activerecord) on master @ [ad622067f1c18c1c7e983c6e95758f34a90153b3](https://github.com/cockroachdb/cockroach/commits/ad622067f1c18c1c7e983c6e95758f34a90153b3):


```
--- FAIL: AttributeMethodsTest#test_caching_a_nil_primary_key (unexpected)
--- FAIL: CalculationsTest#test_ids_for_a_composite_primary_key_on_loaded_relation (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_marshal_dump (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_multiple_columns_and_includes (unexpected)
--- FAIL: ReflectionTest#test_columns (unexpected)
--- FAIL: RelationTest#test_find_by!_with_non-hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_method?_returns_false_if_the_table_does_not_exist (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_association_through_a_has_many_association_with_nonstandard_primary_keys (unexpected)
--- FAIL: CallbacksOnMultipleInstancesInATransactionTest#test_updated_callback_called_on_first_to_save_in_transaction_with_old_configuration (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_size_of_through_association_should_increase_correctly_when_has_many_association_is_added (unexpected)
--- FAIL: RelationTest#test_finding_with_order_and_take (unexpected)
--- FAIL: PersistenceTest#test_becomes (unexpected)
--- FAIL: RelationTest#test_left_outer_joins_with_blank_value (unexpected)
--- FAIL: AttributeMethodsTest#test_undeclared_attribute_method_does_not_affect_respond_to?_and_method_missing (unexpected)
--- FAIL: TransactionsWithTransactionalFixturesTest#test_automatic_savepoint_in_outer_transaction (unexpected)
--- FAIL: TransactionTest#test_cancellation_from_before_filters_rollbacks_in_validation (unexpected)
--- FAIL: TransactionTest#test_nested_explicit_transactions (unexpected)
--- FAIL: PersistenceTest#test_save_uses_query_constraints_config (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_grouped_association_with_foreign_key_option (unexpected)
--- FAIL: RelationTest#test_size (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_collection_singular_ids_setter_raises_exception_when_invalid_ids_set (unexpected)
--- FAIL: RelationTest#test_unscope_grouped_where (unexpected)
--- FAIL: CalculationsTest#test_count_with_reverse_order (unexpected)
--- FAIL: ReflectionTest#test_association_primary_key_raises_when_missing_primary_key (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_collection_singular_ids_setter_with_string_primary_keys (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_left_joins_raise_errors (unexpected)
--- FAIL: RelationTest#test_select_with_from_includes_quoted_original_table_name (unexpected)
--- FAIL: RelationTest#test_default_reverse_order_on_table_without_primary_key (unexpected)
--- FAIL: PersistenceTest#test_returns_object_even_if_validations_failed (unexpected)
--- FAIL: StrictLoadingTest#test_strict_loading_n_plus_one_only_mode (unexpected)
--- FAIL: ReflectionTest#test_integer_columns (unexpected)
--- FAIL: TestAutosaveAssociationValidationsOnAHABTMAssociation#test_should_not_automatically_validate_associations_without_:validate_=>_true (unexpected)
--- FAIL: AttributeMethodsTest#test_respond_to? (unexpected)
--- FAIL: TransactionTest#test_write_attribute_after_rollback (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_for_inspect_with_a_long_array (unexpected)
--- FAIL: RelationTest#test_reverse_order_with_function_other_predicates (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_update_counter_caches_on_destroy (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_create_on_new_record (unexpected)
--- FAIL: CalculationsTest#test_should_count_scoped_select_with_options (unexpected)
--- FAIL: ReflectionTest#test_reflect_on_association_accepts_symbols (unexpected)
--- FAIL: ReflectionTest#test_foreign_key_is_inferred_from_model_name (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_join (unexpected)
--- FAIL: ReflectionTest#test_includes_accepts_strings (unexpected)
--- FAIL: RelationTest#test_reverse_arel_assoc_order_with_multiargument_function (unexpected)
--- FAIL: TransactionTest#test_raising_exception_in_nested_transaction_restore_state_in_save (unexpected)
--- FAIL: CalculationsTest#test_pick_loaded_relation (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_association (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14976785/activerecord/1714025432876/1714033502270)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14992679?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14992679?buildTab=artifacts#/activerecord) on master @ [7c8bf4cd88e2065d5f2d900a9880838062265cf5](https://github.com/cockroachdb/cockroach/commits/7c8bf4cd88e2065d5f2d900a9880838062265cf5):


```
--- FAIL: AssociationsTest#test_model_with_composite_query_constraints_has_many_association_sql (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_adding_before_save (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all_with_option_nullify (unexpected)
--- FAIL: BelongsToAssociationsTest#test_optional_relation_can_be_set_per_model (unexpected)
--- FAIL: CalculationsTest#test_ids_with_includes_and_scope (unexpected)
--- FAIL: AttributeMethodsTest#test_hashes_are_not_mangled (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_for_inspect_with_a_long_array (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_against_given_relation (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_associations_with_string_joins_and_order_references (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_method_on_an_abstract_class_is_available_on_subclasses (unexpected)
--- FAIL: CalculationsTest#test_ids_on_relation (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_default_scope_as_class_method_using_find_by_method (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_with_or_doesnt_set_inverse_instance_key (unexpected)
--- FAIL: HasManyAssociationsTest#test_has_many_without_counter_cache_option (unexpected)
--- FAIL: AttributeMethodsTest#test_user-defined_text_attribute_predicate (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_grouped_by_function (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_each (unexpected)
--- FAIL: AttributeMethodsTest#test_attributes_without_primary_key (unexpected)
--- FAIL: CalculationsTest#test_should_return_integer_average_if_db_returns_such (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_scoped_find_on_through_association_doesnt_return_read_only_records (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_with_extend_option (unexpected)
--- FAIL: HasOneAssociationsTest#test_dependence (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_grouped_by_function_with_table_alias (unexpected)
--- FAIL: HasManyAssociationsTest#test_create_with_bang_on_has_many_when_parent_is_new_raises (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_include_checks_if_record_exists_if_target_not_loaded (unexpected)
--- FAIL: CalculationsTest#test_limit_should_apply_before_count (unexpected)
--- FAIL: HasManyAssociationsTest#test_transactions_when_replacing_on_persisted (unexpected)
--- FAIL: HasOneAssociationsTest#test_natural_assignment_to_already_associated_record (unexpected)
--- FAIL: CalculationsTest#test_ids_for_a_composite_primary_key (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_association_replaces_existing_with_dependent_option (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_an_overridden_original_method_issues_a_deprecation (unexpected)
--- FAIL: HasManyAssociationsTest#test_finding_array_compatibility (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit (unexpected)
--- FAIL: EagerAssociationTest#test_loading_association_with_same_table_joins (unexpected)
--- FAIL: HasOneAssociationsTest#test_build_respects_hash_condition (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_line_endings (unexpected)
--- FAIL: EagerAssociationTest#test_joins_with_includes_should_preload_via_joins (unexpected)
--- FAIL: AssociationsTest#test_nullify_composite_has_many_through_association (unexpected)
--- FAIL: PersistenceTest#test_delete_isnt_affected_by_scoping (unexpected)
--- FAIL: AttributeMethodsTest#test_bulk_update_raises_ActiveRecord::UnknownAttributeError (unexpected)
--- FAIL: HasOneAssociationsTest#test_reset_association (unexpected)
--- FAIL: EagerAssociationTest#test_exceptions_have_suggestions_for_fix (unexpected)
--- FAIL: PersistenceTest#test_update_columns_should_not_modify_updated_at (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_many_via_block (unexpected)
--- FAIL: BelongsToAssociationsTest#test_self_referential_belongs_to_with_counter_cache_assigning_nil (unexpected)
--- FAIL: AssociationsTest#test_querying_by_single_associated_record_works_using_query_constraints (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_first_or_last_on_loaded_association_should_not_fetch_with_query (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14992679/activerecord/1714111844080/1714121099832)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15008018?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15008018?buildTab=artifacts#/activerecord) on master @ [5d02bd9ff6b2bccecf6d43fc6cd647167b91f782](https://github.com/cockroachdb/cockroach/commits/5d02bd9ff6b2bccecf6d43fc6cd647167b91f782):


```
--- FAIL: RelationTest#test_pluck_with_from_includes_original_table_name (unexpected)
--- FAIL: CalculationsTest#test_pluck_loaded_relation_sql_fragment (unexpected)
--- FAIL: TransactionTest#test_cancellation_from_before_filters_rollbacks_in_validation! (unexpected)
--- FAIL: FinderTest#test_first_bang_present (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_dynamic_find_all_should_respect_readonly_access (unexpected)
--- FAIL: RelationTest#test_unscope_with_arel_sql (unexpected)
--- FAIL: CalculationsTest#test_count_selected_arel_attribute (unexpected)
--- FAIL: AttributeMethodsTest#test_came_from_user? (unexpected)
--- FAIL: EagerAssociationTest#test_preload_belongs_to_uses_exclusive_scope (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_empty_on_an_association_that_has_not_been_loaded_performs_a_query (unexpected)
--- FAIL: CalculationsTest#test_ids_with_includes (unexpected)
--- FAIL: InheritanceTest#test_new_with_unrelated_type (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_has_many_through_polymorphic_has_manys_works (unexpected)
--- FAIL: TransactionTest#test_raising_exception_in_nested_transaction_restore_state_in_save (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_has_and_belongs_to_many_in_a_namespaced_model_pointing_to_a_non_namespaced_model (unexpected)
--- FAIL: BelongsToAssociationsTest#test_not_optional_relation (unexpected)
--- FAIL: RelationTest#test_unscope_grouped_where (unexpected)
--- FAIL: BasicsTest#test_singular_table_name_guesses_for_individual_table (unexpected)
--- FAIL: NumericalityValidationTest#test_virtual_attribute (unexpected)
--- FAIL: InheritanceTest#test_a_bad_type_column (unexpected)
--- FAIL: TransactionCallbacksTest#test_no_after_commit_on_destroy_after_transaction_commits_for_destroyed_new_record (unexpected)
--- FAIL: HasOneAssociationsTest#test_assignment_before_child_saved (unexpected)
--- FAIL: BelongsToAssociationsTest#test_should_set_foreign_key_on_create_association_with_unpersisted_owner (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_multi_table_conditional_properly_counts_the_records_when_using_size (unexpected)
--- FAIL: RelationTest#test_already-loaded_relations_don't_perform_a_new_query_in_#inspect (unexpected)
--- FAIL: RelationTest#test_loading_query_is_annotated_in_#pretty_print (unexpected)
--- FAIL: AttributeMethodsTest#test_string_attribute_predicate (unexpected)
--- FAIL: SerializedAttributeTest#test_serialized_string_attribute (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_assignment_with_nil (unexpected)
--- FAIL: ReflectionTest#test_aggregation_reflection (unexpected)
--- FAIL: FinderTest#test_implicit_order_column_is_configurable (unexpected)
--- FAIL: CalculationsTest#test_should_not_perform_joined_include_by_default (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_each (unexpected)
--- FAIL: CalculationsTest#test_should_get_minimum_of_field (unexpected)
--- FAIL: BasicsTest#test_create_after_initialize_without_block (unexpected)
--- FAIL: TransactionCallbacksTest#test_only_call_after_rollback_on_create_after_transaction_rollsback_for_new_record (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_default_pk_and_sequence (unexpected)
--- FAIL: RelationTest#test_create_or_find_by_with_bang_with_non_unique_attributes (unexpected)
--- FAIL: ModulesTest#test_compute_type_can_infer_class_name_of_sibling_inside_module (unexpected)
--- FAIL: RelationTest#test_eager_load_with_blank_value (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_autosave_with_primary_key_manually_set (unexpected)
--- FAIL: RelationTest#test_count_with_distinct (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_get_ids (unexpected)
--- FAIL: RelationTest#test_dynamic_find_by_attributes_bang (unexpected)
--- FAIL: CalculationsTest#test_pluck_loaded_relation (unexpected)
--- FAIL: DefaultNumbersTest#test_default_negative_integer (unexpected)
--- FAIL: BelongsToAssociationsTest#test_clearing_an_association_clears_the_associations_inverse (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15008018/activerecord/1714198385338/1714206429216)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15016504?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15016504?buildTab=artifacts#/activerecord) on master @ [5d02bd9ff6b2bccecf6d43fc6cd647167b91f782](https://github.com/cockroachdb/cockroach/commits/5d02bd9ff6b2bccecf6d43fc6cd647167b91f782):


```
--- FAIL: FinderTest#test_find_by_one_attribute_with_conditions (unexpected)
--- FAIL: RelationTest#test_select_with_subquery_in_from_does_not_use_original_table_name (unexpected)
--- FAIL: RelationTest#test_finding_with_order_and_take (unexpected)
--- PASS: CockroachDB::ConnectionAdapters::TypeTest#test_type_can_be_used_with_various_db - unknown (unexpected)
--- FAIL: RelationTest#test_count_with_distinct (unexpected)
--- FAIL: FinderTest#test_find_with_proc_parameter_and_block (unexpected)
--- FAIL: RelationTest#test_first_or_initialize_with_block (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_valid_adding_with_validate_false (unexpected)
--- FAIL: FinderTest#test_#last_for_a_model_with_composite_query_constraints (unexpected)
--- FAIL: FinderTest#test_exists_with_scope (unexpected)
--- FAIL: FinderTest#test_fifth_have_primary_key_order_by_default (unexpected)
--- FAIL: PersistenceTest#test_raises_error_when_validations_failed (unexpected)
--- FAIL: FinderTest#test_member_when_non_AR_object_passed_on_loaded_relation (unexpected)
--- FAIL: RelationTest#test_includes_with_select (unexpected)
--- FAIL: RelationTest#test_find_all_using_where_with_relation_with_select_to_build_subquery (unexpected)
--- FAIL: PersistenceTest#test_save_for_record_with_only_primary_key (unexpected)
--- FAIL: FinderTest#test_find_by_id_with_large_number (unexpected)
--- FAIL: RelationTest#test_to_sql_on_scoped_proxy (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_save_succeeds_for_invalid_belongs_to_with_validate_false (unexpected)
--- FAIL: FinderTest#test_condition_array_interpolation (unexpected)
--- FAIL: RelationTest#test_grouping_by_column_with_reserved_name (unexpected)
--- FAIL: RelationTest#test_find_with_preloaded_associations (unexpected)
--- FAIL: PersistenceTest#test_decrement_attribute (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_aggregate_having_one_mapping_array (unexpected)
--- FAIL: PersistenceTest#test_save_nil_string_attributes (unexpected)
--- FAIL: FinderTest#test_bind_variables_with_quotes (unexpected)
--- FAIL: RelationTest#test_find_or_create_by_with_create_with (unexpected)
--- FAIL: RelationTest#test_joins_with_select (unexpected)
--- FAIL: FinderTest#test_find_some_message_with_custom_primary_key (unexpected)
--- FAIL: RelationTest#test_unscoped_block_style (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_numeric_range_for_string (unexpected)
--- FAIL: RelationTest#test_automatically_added_where_references (unexpected)
--- FAIL: FinderTest#test_take_and_first_and_last_with_integer_should_use_sql_limit (unexpected)
--- FAIL: RelationTest#test_loaded_first_with_limit (unexpected)
--- FAIL: PersistenceTest#test_update! (unexpected)
--- FAIL: FinderTest#test_with_limiting_with_custom_select (unexpected)
--- FAIL: RelationTest#test_doesnt_add_having_values_if_options_are_blank (unexpected)
--- FAIL: FinderTest#test_select_values (unexpected)
--- FAIL: FinderTest#test_find_by_with_alias (unexpected)
--- FAIL: FinderTest#test_find_on_relation_with_large_number (unexpected)
--- FAIL: RelationTest#test_first_or_initialize_with_no_parameters (unexpected)
--- FAIL: FinderTest#test_find_passing_active_record_object_is_not_permitted (unexpected)
--- FAIL: FinderTest#test_named_bind_variables_with_quotes (unexpected)
--- FAIL: FinderTest#test_bind_variables (unexpected)
--- FAIL: InheritanceTest#test_new_with_abstract_class (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: PersistenceTest#test_becomes_includes_errors (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15016504/activerecord/1714284465390/1714293327052)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15024765?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15024765?buildTab=artifacts#/activerecord) on master @ [2b5baff23f9b53da26f0fc1a763ca43384b464b0](https://github.com/cockroachdb/cockroach/commits/2b5baff23f9b53da26f0fc1a763ca43384b464b0):


```
--- FAIL: CalculationsTest#test_ids_with_polymorphic_relation_join (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: HasManyAssociationsTest#test_to_a_should_dup_target (unexpected)
--- FAIL: FinderTest#test_second (unexpected)
--- FAIL: CalculationsTest#test_count_with_eager_loading_and_custom_order (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_models_with_composite_keys (unexpected)
--- FAIL: FinderTest#test_select_rows (unexpected)
--- FAIL: PersistenceTest#test_destroy_record_with_associations (unexpected)
--- FAIL: HasManyAssociationsTest#test_prevent_double_firing_the_before_save_callback_of_new_object_when_the_parent_association_saved_in_the_callback (unexpected)
--- FAIL: FinderTest#test_exists_with_joins (unexpected)
--- FAIL: HasManyAssociationsTest#test_clearing_an_association_collection (unexpected)
--- FAIL: FinderTest#test_condition_array_interpolation (unexpected)
--- FAIL: FinderTest#test_sole_failing_many (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_aggregate_attribute_having_same_name_as_field_and_key_value_being_attribute_value (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_join_with_group (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependence_on_account (unexpected)
--- FAIL: SignedIdTest#test_fail_to_find_record_from_that_has_since_been_destroyed (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all_on_association_is_the_same_as_not_loaded (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_with_instance_dependent_scope (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependent_association_respects_optional_hash_conditions_on_delete (unexpected)
--- FAIL: PersistenceTest#test_create_with_custom_timestamps (unexpected)
--- FAIL: CalculationsTest#test_should_not_perform_joined_include_by_default (unexpected)
--- FAIL: HasOneAssociationsTest#test_can_marshal_has_one_association_with_nil_target (unexpected)
--- FAIL: FinderTest#test_member_when_non_AR_object_passed_on_unloaded_relation (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_should_property_quote_string_primary_keys (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_all (unexpected)
--- FAIL: CalculationsTest#test_minimum_and_maximum_on_time_attributes (unexpected)
--- FAIL: FinderTest#test_find_with_a_single_composite_primary_key_wrapped_in_an_array (unexpected)
--- FAIL: PersistenceTest#test_delete (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_multiple_fields_when_table_name_is_too_long (unexpected)
--- FAIL: HasManyAssociationsTest#test_reload_with_query_cache (unexpected)
--- FAIL: HasManyAssociationsTest#test_create_with_bang_on_habtm_when_parent_is_new_raises (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_custom_join_table (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_reserved_words (unexpected)
--- FAIL: HasManyAssociationsTest#test_create_with_bang_on_has_many_when_parent_is_new_raises (unexpected)
--- FAIL: HasManyAssociationsTest#test_key_ensuring_owner_was_is_not_valid_without_dependent_option (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_association_replaces_existing_without_dependent_option (unexpected)
--- FAIL: HasManyAssociationsTest#test_counter_cache_on_unloaded_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_a_item_which_is_not_in_the_collection (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_valid_adding_with_validate_false (unexpected)
--- FAIL: FinderTest#test_custom_select_takes_precedence_over_original_value (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_does_not_use_order_by (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_with_block (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_marshal_dump (unexpected)
--- FAIL: HasManyAssociationsTest#test_select_with_block (unexpected)
--- FAIL: PersistenceTest#test_reload_removes_custom_selects (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15024765/activerecord/1714371025652/1714379002745)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15041953?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15041953?buildTab=artifacts#/activerecord) on master @ [086f930e2b94913737db9ef9877d10976355874c](https://github.com/cockroachdb/cockroach/commits/086f930e2b94913737db9ef9877d10976355874c):


```
--- FAIL: BelongsToAssociationsTest#test_assignment_before_child_saved (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_yaml_loads_5_1_dump_without_indexes_still_queries_for_indexes (unexpected)
--- FAIL: DirtyTest#test_integer_zero_to_integer_zero_not_marked_as_changed (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_all_sanitized (unexpected)
--- FAIL: CalculationsTest#test_count_with_block (unexpected)
--- FAIL: HabtmDestroyOrderTest#test_may_not_delete_a_lesson_with_students (unexpected)
--- FAIL: HasManyAssociationsTest#test_updates_counter_cache_when_default_scope_is_given (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_with_rewhere_doesnt_set_inverse_instance_key (unexpected)
--- FAIL: CalculationsTest#test_should_sum_field (unexpected)
--- FAIL: DirtyTest#test_nullable_datetime_not_marked_as_changed_if_new_value_is_blank (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_proxy_should_not_respond_to_private_methods (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_grouped_association_with_foreign_key_option (unexpected)
--- FAIL: EnumTest#test_overloaded_default_by_:default (unexpected)
--- FAIL: RelationTest#test_size_with_limit (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_reassignment_of_associated_id_updates_the_object (unexpected)
--- FAIL: BelongsToAssociationsTest#test_destroying_polymorphic_child_with_unloaded_parent_and_touch_is_possible_with_has_many_inversing (unexpected)
--- FAIL: AssociationCallbacksTest#test_has_and_belongs_to_many_after_add_called_after_save (unexpected)
--- FAIL: RelationTest#test_build (unexpected)
--- FAIL: EnumTest#test_enum_on_custom_attribute_with_default (unexpected)
--- FAIL: InheritanceTest#test_finding_incorrect_type_data (unexpected)
--- FAIL: PostgresqlIntervalTest#test_interval_type_cast_from_numeric (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_counter_cache (unexpected)
--- FAIL: AttributeMethodsTest#test_typecast_attribute_from_select_to_false (unexpected)
--- FAIL: HasManyAssociationsTest#test_clear_collection_should_not_change_updated_at (unexpected)
--- FAIL: HasManyAssociationsTest#test_dynamic_find_should_respect_association_order (unexpected)
--- FAIL: EagerAssociationTest#test_preconfigured_includes_with_has_many (unexpected)
--- FAIL: RelationTest#test_find_by!_requires_at_least_one_argument (unexpected)
--- FAIL: HasManyAssociationsTest#test_associations_order_should_be_priority_over_throughs_order (unexpected)
--- FAIL: BelongsToAssociationsTest#test_build_with_conditions (unexpected)
--- FAIL: EagerAssociationTest#test_has_and_belongs_to_many_should_not_instantiate_same_records_multiple_times (unexpected)
--- FAIL: StrictLoadingTest#test_strict_loading_with_has_one_through_does_not_prevent_creation_of_association (unexpected)
--- FAIL: BelongsToAssociationsTest#test_create! (unexpected)
--- FAIL: RelationTest#test_find_or_create_by! (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: RelationTest#test_select_with_from_includes_quoted_original_table_name (unexpected)
--- FAIL: HasManyAssociationsTest#test_collection_size_after_building (unexpected)
--- FAIL: InheritanceTest#test_base_class_activerecord_error (unexpected)
--- FAIL: BasicsTest#test_limit_without_comma (unexpected)
--- FAIL: EagerAssociationTest#test_with_ordering (unexpected)
--- FAIL: RelationTest#test_order_doesnt_trigger_eager_loading_when_ordering_using_owner_table_and_symbols (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all_on_association_with_nil_dependency_is_the_same_as_not_loaded (unexpected)
--- FAIL: RelationTest#test_create_or_find_by_with_bang_with_non_unique_attributes (unexpected)
--- FAIL: BasicsTest#test_custom_mutator (unexpected)
--- FAIL: RelationTest#test_finding_with_arel_assoc_order (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_counter (unexpected)
--- FAIL: EnumTest#test_deserialize_enum_value_to_original_hash_key (unexpected)
--- FAIL: BasicsTest#test_attributes (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15041953/activerecord/1714457466533/1714464764764)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15057518?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15057518?buildTab=artifacts#/activerecord) on master @ [af3173a899e16af30d2ca9489d8c2c926c410121](https://github.com/cockroachdb/cockroach/commits/af3173a899e16af30d2ca9489d8c2c926c410121):


```
(orm_helpers.go:227).summarizeFailed: 
Tests run on Cockroach v24.1.0-alpha.5-dev-af3173a899e16af30d2ca9489d8c2c926c410121
Tests run against activerecord 7.1.3
8108 Total Tests Run
8059 tests passed
49 tests failed
66 tests skipped
0 tests ignored
1 test passed unexpectedly
24 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: ModulesTest#test_module_table_name_suffix_with_global_suffix (unexpected)
--- FAIL: TestAutosaveAssociationValidationsOnABelongsToAssociation#test_validations_still_fire_on_unchanged_association_with_custom_validation_context (unexpected)
--- FAIL: ModulesTest#test_nested_models_should_not_raise_exception_when_using_delete_all_dependency_on_association (unexpected)
--- FAIL: HasOneThroughDisableJoinsAssociationsTest#test_preload_on_disable_joins_through (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_create_fixtures_resets_sequences_when_not_cached (unexpected)
--- FAIL: ModulesTest#test_eager_loading_in_modules (unexpected)
--- PASS: CockroachDB::ConnectionAdapters::TypeTest#test_type_can_be_used_with_various_db - unknown (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: ModulesTest#test_module_spanning_associations (unexpected)
--- FAIL: HasOneThroughDisableJoinsAssociationsTest#test_disable_joins_through_with_enum_type (unexpected)
--- FAIL: ModulesTest#test_module_table_name_prefix (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_default_pk_and_sequence (unexpected)
--- FAIL: HasOneThroughDisableJoinsAssociationsTest#test_has_one_through_with_belongs_to_on_disable_joins (unexpected)
--- FAIL: ModulesTest#test_assign_ids (unexpected)
--- FAIL: ModulesTest#test_find_account_and_include_company (unexpected)
--- FAIL: ModulesTest#test_nested_models_should_not_raise_exception_when_using_nullify_dependency_on_association (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: ModulesTest#test_module_table_name_suffix (unexpected)
--- FAIL: HasOneThroughDisableJoinsAssociationsTest#test_counting_on_disable_joins_through (unexpected)
--- FAIL: ModulesTest#test_module_spanning_has_and_belongs_to_many_associations (unexpected)
--- FAIL: ModulesTest#test_module_table_name_prefix_with_global_prefix (unexpected)
--- FAIL: ModulesTest#test_compute_type_can_infer_class_name_of_sibling_inside_module (unexpected)
--- FAIL: ActiveRecord::Type::IntegerTest#test_casting_ActiveRecord_models (unexpected)
--- FAIL: ModulesTest#test_associations_spanning_cross_modules (unexpected)
--- FAIL: HasOneThroughDisableJoinsAssociationsTest#test_nil_on_disable_joins_through (unexpected)
--- FAIL: ModulesTest#test_table_name (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-1714542508/activerecord/1714544391721/1714552727521)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15074804?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15074804?buildTab=artifacts#/activerecord) on master @ [f2459fb5e6edb110109d5b5ef24ef529a3a85a48](https://github.com/cockroachdb/cockroach/commits/f2459fb5e6edb110109d5b5ef24ef529a3a85a48):


```
--- FAIL: DirtyTest#test_clear_attribute_change (unexpected)
--- FAIL: PersistenceTest#test_increment_nil_attribute (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_inner_join_with_quoted_table_name (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_grouped_association_with_invalid_field (unexpected)
--- FAIL: RelationTest#test_reload (unexpected)
--- FAIL: DirtyTest#test_attribute_changed?_properly_type_casts_enum_values (unexpected)
--- FAIL: RelationTest#test_finding_with_reorder_by_aliased_attributes (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependence (unexpected)
--- FAIL: HasManyAssociationsTest#test_associations_order_should_be_priority_over_throughs_order (unexpected)
--- FAIL: InverseBelongsToTests#test_recursive_inverse_on_recursive_model_has_many_inversing (unexpected)
--- FAIL: CalculationsTest#test_should_return_float_average_if_db_returns_such (unexpected)
--- FAIL: ActiveRecord::SelectTest#test_type_casted_extra_select_with_eager_loading (unexpected)
--- FAIL: FinderTest#test_find_by_doesn't_have_implicit_ordering (unexpected)
--- FAIL: BasicsTest#test_reload (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_one_message_on_primary_key (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_accept_string_and_hash_join (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_include_method_in_association_through_should_return_true_for_instance_added_with_build (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_update_on_id_changes_the_counter_cache (unexpected)
--- FAIL: DirtyTest#test_datetime_attribute_doesnt_change_if_zone_is_modified_in_string (unexpected)
--- FAIL: CalculationsTest#test_count_with_distinct (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_for_inspect_with_a_non-primary_key_id_attribute (unexpected)
--- FAIL: ReflectionTest#test_irregular_reflection_class_name (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_adding_before_save (unexpected)
--- FAIL: AssociationsTest#test_querying_by_single_associated_record_works_using_query_constraints (unexpected)
--- FAIL: DefaultScopingTest#test_default_scope (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_both_parent_ids_set_when_saving_new (unexpected)
--- FAIL: ActiveRecord::Migration::ColumnAttributesTest#test_add_column_newline_default (unexpected)
--- FAIL: DirtyTest#test_save_always_should_update_timestamps_when_serialized_attributes_are_present (unexpected)
--- FAIL: FinderTest#test_find_by!_doesn't_have_implicit_ordering (unexpected)
--- FAIL: CalculationsTest#test_ids_with_includes_and_non_primary_key_order (unexpected)
--- FAIL: FinderTest#test_find_by_with_associations (unexpected)
--- FAIL: AttributeMethodsTest#test_read_overridden_attribute (unexpected)
--- FAIL: FinderTest#test_find_by_on_attribute_that_is_a_reserved_word (unexpected)
--- FAIL: PostgresqlInfinityTest#test_type_casting_infinity_on_a_date_column (unexpected)
--- FAIL: RelationTest#test_find_by!_raises_RecordNotFound_if_the_record_is_missing (unexpected)
--- FAIL: PersistenceTest#test_update_attribute_with_one_updated (unexpected)
--- FAIL: ReflectionTest#test_nested? (unexpected)
--- FAIL: CounterCacheTest#test_reset_counters_with_modularized_and_camelized_classnames (unexpected)
--- FAIL: DirtyTest#test_dup_objects_should_not_copy_dirty_flag_from_creator (unexpected)
--- FAIL: RelationTest#test_scoped_first (unexpected)
--- FAIL: RelationTest#test_loading_query_is_annotated_in_#inspect (unexpected)
--- FAIL: IntegrationTest#test_cache_key_changes_when_child_touched (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_child_is_visible_to_join_model_in_add_association_callbacks (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_method_on_an_abstract_class_is_available_on_subclasses (unexpected)
--- FAIL: RelationTest#test_count_complex_chained_relations (unexpected)
--- FAIL: BasicsTest#test_distinct_delegates_to_scoped (unexpected)
--- FAIL: HasManyAssociationsTest#test_add_record_to_collection_should_change_its_updated_at (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15074804/activerecord/1714631063474/1714638152744)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15094229?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15094229?buildTab=artifacts#/activerecord) on master @ [6ff7ecaa829de0ff01d9ede3a72a3be7541a04df](https://github.com/cockroachdb/cockroach/commits/6ff7ecaa829de0ff01d9ede3a72a3be7541a04df):


```
--- FAIL: FinderTest#test_last_with_integer_and_order_should_use_sql_limit (unexpected)
--- FAIL: FinderTest#test_select_value (unexpected)
--- FAIL: ReflectionTest#test_join_table_with_common_prefix (unexpected)
--- FAIL: FinderTest#test_find_with_large_number (unexpected)
--- FAIL: FinderTest#test_find_with_proc_parameter_and_block (unexpected)
--- FAIL: FinderTest#test_find_by_ids_with_limit_and_offset (unexpected)
--- FAIL: AssociationsTest#test_has_many_association_with_composite_foreign_key_loads_records (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_partial_indices (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: ReflectionTest#test_foreign_key_is_inferred_from_model_name (unexpected)
--- FAIL: FinderTest#test_find_on_array_conditions (unexpected)
--- FAIL: FinderTest#test_include_on_unloaded_relation_with_match (unexpected)
--- FAIL: ReflectionTest#test_reflect_on_association_accepts_strings (unexpected)
--- FAIL: FinderTest#test_find_by!_raises_RecordNotFound_if_the_record_is_missing (unexpected)
--- FAIL: FinderTest#test_last_bang_missing (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_index_length (unexpected)
--- FAIL: FinderTest#test_include_on_unloaded_relation_with_offset (unexpected)
--- FAIL: AssociationsTest#test_nullify_composite_foreign_key_belongs_to_association (unexpected)
--- FAIL: FinderTest#test_find_by_with_alias (unexpected)
--- FAIL: FinderTest#test_select_rows (unexpected)
--- FAIL: FinderTest#test_exists_with_distinct_and_offset_and_joins (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_open_ended_range (unexpected)
--- FAIL: FinderTest#test_find_ignores_previously_inserted_record (unexpected)
--- FAIL: FinderTest#test_exists_should_reference_correct_aliases_while_joining_tables_of_has_many_through_association (unexpected)
--- FAIL: FinderTest#test_find_by_association_subquery (unexpected)
--- FAIL: FinderTest#test_second_to_last_have_primary_key_order_by_default (unexpected)
--- FAIL: FinderTest#test_first_bang_present (unexpected)
--- FAIL: FinderTest#test_implicit_order_for_model_without_primary_key (unexpected)
--- FAIL: FinderTest#test_take_and_first_and_last_with_integer_should_use_sql_limit (unexpected)
--- FAIL: FinderTest#test_exists_with_distinct_and_offset_and_select (unexpected)
--- FAIL: FinderTest#test_find_by_id_with_conditions_with_or (unexpected)
--- FAIL: FinderTest#test_unexisting_record_exception_handling (unexpected)
--- FAIL: FinderTest#test_first_on_relation_with_limit_and_offset (unexpected)
--- FAIL: ReflectionTest#test_non_existent_types_are_identity_types (unexpected)
--- FAIL: FinderTest#test_find_by_ids_missing_one (unexpected)
--- FAIL: FinderTest#test_take_and_first_and_last_with_integer_should_return_an_array (unexpected)
--- FAIL: FinderTest#test_find_by_id_with_large_number (unexpected)
--- FAIL: FinderTest#test_#find_by_with_composite_primary_key_and_query_caching (unexpected)
--- FAIL: FinderTest#test_exists_with_eager_load (unexpected)
--- FAIL: FinderTest#test_exists_with_strong_parameters (unexpected)
--- FAIL: ReflectionTest#test_column_null_not_null (unexpected)
--- FAIL: FinderTest#test_last_on_loaded_relation_should_not_use_sql (unexpected)
--- FAIL: FinderTest#test_model_class_responds_to_fourth_bang (unexpected)
--- FAIL: FinderTest#test_find_with_a_single_composite_primary_key_wrapped_in_an_array (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_index_sort_order (unexpected)
--- FAIL: FinderTest#test_find_an_empty_array (unexpected)
--- FAIL: AssociationsTest#test_querying_by_whole_associated_records_using_query_constraints (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15094229/activerecord/1714716899075/1714725570233)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.0-rc release-blocker]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15109710?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15109710?buildTab=artifacts#/activerecord) on master @ [6e23d32dd0d36181f5c886c9a0eda9c756005bd6](https://github.com/cockroachdb/cockroach/commits/6e23d32dd0d36181f5c886c9a0eda9c756005bd6):


```
--- FAIL: RelationTest#test_relations_show_the_records_in_#inspect (unexpected)
--- FAIL: AttributeMethodsTest#test_boolean_attributes (unexpected)
--- FAIL: FinderTest#test_third_have_primary_key_order_by_default (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_preload_sti_middle_relation (unexpected)
--- FAIL: EagerAssociationTest#test_joins_with_includes_should_preload_via_joins (unexpected)
--- FAIL: InsertAllTest#test_insert_all_and_upsert_all_with_sti (unexpected)
--- FAIL: FinderTest#test_count_by_sql (unexpected)
--- FAIL: EagerAssociationTest#test_eager_has_one_with_association_inheritance (unexpected)
--- FAIL: HasManyAssociationsTest#test_finding_with_condition_hash (unexpected)
--- FAIL: FinderTest#test_find_by_bang_on_relation_with_large_number (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_build_before_save (unexpected)
--- FAIL: AssociationsJoinModelTest#test_has_many_with_pluralize_table_names_false (unexpected)
--- FAIL: HasManyAssociationsTest#test_reload_with_query_cache (unexpected)
--- FAIL: TransactionTest#test_raising_exception_in_callback_rollbacks_in_save (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_respects_hash_condition (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_array (unexpected)
--- FAIL: HasOneAssociationsTest#test_with_select (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all_accepts_limited_parameters (unexpected)
--- FAIL: FinderTest#test_nth_to_last_with_order_uses_limit (unexpected)
--- FAIL: StoreTest#test_updating_the_store_will_mark_it_as_changed (unexpected)
--- FAIL: BelongsToAssociationsTest#test_should_set_foreign_key_on_create_association_with_unpersisted_owner (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_aggregate_having_one_mapping_and_key_value_being_attribute_value (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_with_bang_with_purpose (unexpected)
--- FAIL: SerializedAttributeTest#test_serialized_attribute_on_custom_attribute_with_default (unexpected)
--- FAIL: OptimisticLockingTest#test_quote_table_name (unexpected)
--- FAIL: FinderTest#test_first_have_primary_key_order_by_default (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_works_with_model_called_Record (unexpected)
--- FAIL: ActiveRecord::Type::TimeTest#test_default_year_is_correct (unexpected)
--- FAIL: CalculationsTest#test_should_average_field (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_assignment_triggers_save_on_change_on_replacing_object (unexpected)
--- FAIL: HasManyAssociationsTest#test_three_levels_of_dependence (unexpected)
--- FAIL: AttributeMethodsTest#test_hash_content (unexpected)
--- FAIL: RelationTest#test_first_or_initialize (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_time_zone_aware_attribute (unexpected)
--- FAIL: TransactionTest#test_persisted_in_a_model_with_custom_primary_key_after_failed_save (unexpected)
--- FAIL: AssociationsTest#test_append_composite_foreign_key_has_many_association_with_autosave (unexpected)
--- FAIL: HasManyAssociationsTest#test_finding_default_orders (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_build (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_distinct_before_the_fact (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_first_after_reset_scope (unexpected)
--- FAIL: RelationTest#test_automatically_added_order_references (unexpected)
--- FAIL: HasManyAssociationsTest#test_sending_new_to_association_proxy_should_have_same_effect_as_calling_new (unexpected)
--- FAIL: FinderTest#test_exists_with_distinct_and_offset_and_joins (unexpected)
--- FAIL: OptimisticLockingTest#test_lock_with_custom_column_without_default_should_work_with_null_in_the_database (unexpected)
--- FAIL: FinderTest#test_fourth (unexpected)
--- FAIL: SerializationTest#test_read_attribute_for_serialization_with_format_after_init (unexpected)
--- FAIL: AssociationsTest#test_querying_by_whole_associated_records_using_query_constraints (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15109710/activerecord/1714803060403/1714811240954)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15118434?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15118434?buildTab=artifacts#/activerecord) on master @ [6e23d32dd0d36181f5c886c9a0eda9c756005bd6](https://github.com/cockroachdb/cockroach/commits/6e23d32dd0d36181f5c886c9a0eda9c756005bd6):


```
--- FAIL: BelongsToAssociationsTest#test_polymorphic_setting_foreign_key_after_nil_target_loaded (unexpected)
--- FAIL: CalculationsTest#test_should_get_minimum_of_arel_attribute (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_many_with_merged_options (unexpected)
--- FAIL: FinderTest#test_hash_condition_local_time_interpolation_with_default_timezone_utc (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_a_item_which_is_not_in_the_collection (unexpected)
--- FAIL: BelongsToAssociationsTest#test_reloading_association_with_key_change (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_save_succeeds_for_invalid_belongs_to_with_validate_false (unexpected)
--- FAIL: SerializationTest#test_find_records_by_serialized_attributes_through_join (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_array_of_ranges (unexpected)
--- FAIL: AttributeMethodsTest#test_bulk_update_raises_ActiveRecord::UnknownAttributeError (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_through_has_one_association_with_order_on_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_all (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_build (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_explicit_join (unexpected)
--- FAIL: CalculationsTest#test_should_count_field_in_joined_table_with_group_by_when_tables_share_column_names (unexpected)
--- FAIL: HasManyAssociationsTest#test_has_many_through_respects_hash_conditions (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_count (unexpected)
--- FAIL: FinderTest#test_fifth_have_primary_key_order_by_default (unexpected)
--- FAIL: NamedScopingTest#test_should_create_on_top_of_scope (unexpected)
--- FAIL: EagerAssociationTest#test_eager_loading_of_instance_dependent_associations_is_not_supported (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_attributes_are_available_to_after_initialize (unexpected)
--- FAIL: RelationTest#test_select_with_from_includes_quoted_original_table_name (unexpected)
--- FAIL: CockroachDB::PostgresqlIntervalTest#test_interval_type (unexpected)
--- FAIL: UniquenessValidationTest#test_validate_case_insensitive_uniqueness_with_special_sql_like_chars (unexpected)
--- FAIL: DefaultScopingTest#test_unscoped_with_named_scope_should_not_have_default_scope (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependence_with_transaction_support_on_failure (unexpected)
--- FAIL: CalculationsTest#test_limit_is_kept (unexpected)
--- FAIL: HasManyAssociationsTest#test_attributes_are_set_when_initialized_from_has_many_null_relationship (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_multiple_fields (unexpected)
--- FAIL: FinderTest#test_find_with_large_number (unexpected)
--- FAIL: AttributeMethodsTest#test_came_from_user? (unexpected)
--- FAIL: FinderTest#test_find_with_ids_returning_ordered (unexpected)
--- FAIL: HasManyAssociationsTest#test_in_memory_replacement_executes_no_queries (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_aggregate_having_one_mapping_and_key_value_being_attribute_value (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_delete_all_for_with_dependent_option_delete_all (unexpected)
--- FAIL: ReflectionTest#test_foreign_type (unexpected)
--- FAIL: DirtyTest#test_reverted_changes_are_not_dirty_after_multiple_changes (unexpected)
--- FAIL: HasManyAssociationsTest#test_counter_cache_updates_in_memory_after_concat (unexpected)
--- FAIL: BelongsToAssociationsTest#test_id_assignment (unexpected)
--- FAIL: RelationTest#test_to_yaml (unexpected)
--- FAIL: EagerAssociationTest#test_string_id_column_joins (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_a_time_zone-aware_attribute_to_a_blank_string_returns_nil (unexpected)
--- FAIL: RelationTest#test_unscope_with_aliased_column (unexpected)
--- FAIL: FinderTest#test_find_ignores_previously_inserted_record (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_followed_by_save_does_not_load_target (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_with_create_and_no_options (unexpected)
--- FAIL: FinderTest#test_exists_with_distinct_association_includes_limit_and_order (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15118434/activerecord/1714889321543/1714897474374)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15127361?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15127361?buildTab=artifacts#/activerecord) on master @ [379d332c9a4ce49cda4a3565a852e7ddb850ffb5](https://github.com/cockroachdb/cockroach/commits/379d332c9a4ce49cda4a3565a852e7ddb850ffb5):


```
--- FAIL: InheritanceTest#test_inheritance_base_class (unexpected)
--- FAIL: BasicsTest#test_readonly_attributes_when_configured_to_not_raise (unexpected)
--- FAIL: InheritanceTest#test_should_store_full_class_name_with_store_full_sti_class_option_enabled (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_indexes_for_existent_table (unexpected)
--- FAIL: AssociationsTest#test_polymorphic_belongs_to_uses_parent_query_constraints (unexpected)
--- FAIL: FinderTest#test_find_by!_with_hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: StoreTest#test_updating_the_store_won't_mark_accessor_as_changed_if_the_whole_store_was_updated (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_update_counter_caches_on_delete_with_dependent_nullify (unexpected)
--- FAIL: PersistenceTest#test_update_column_with_default_scope (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_build_before_either_saved (unexpected)
--- FAIL: BelongsToAssociationsTest#test_stale_tracking_doesn't_care_about_the_type (unexpected)
--- FAIL: BasicsTest#test_attributes_on_dummy_time_with_invalid_time (unexpected)
--- FAIL: BasicsTest#test_find_multiple_ordered_last (unexpected)
--- FAIL: AttributeMethodsTest#test_#undefine_attribute_methods_undefines_alias_attribute_methods (unexpected)
--- FAIL: TransactionTest#test_write_attribute_after_rollback (unexpected)
--- FAIL: RelationTest#test_size_with_distinct (unexpected)
--- FAIL: RelationTest#test_find_by_id_with_list_of_ar (unexpected)
--- FAIL: PersistenceTest#test_update_columns_should_not_use_setter_method (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_preloaded_associations_size (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_present_with_booleans (unexpected)
--- FAIL: RelationTest#test_already-loaded_relations_don't_perform_a_new_query_in_#inspect (unexpected)
--- FAIL: ActiveRecord::SelectTest#test_select_with_one_level_hash_argument (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_assign_ids (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit_and_high_offset_and_multiple_hash_conditions (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_with_scope_with_joins (unexpected)
--- FAIL: RelationTest#test_alias_tracker_respects_a_custom_table (unexpected)
--- FAIL: FilterAttributesTest#test_proc_filter_attributes_don't_prevent_marshal_dump (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
--- FAIL: ActiveRecord::SelectTest#test_select_with_hash_and_table_alias (unexpected)
--- FAIL: StoreTest#test_convert_store_attributes_from_any_format_other_than_Hash_or_HashWithIndifferentAccess_losing_the_data (unexpected)
--- FAIL: PersistenceTest#test_update_attribute_does_not_run_sql_if_attribute_is_not_changed (unexpected)
--- FAIL: BasicsTest#test_#connected_to_many_cannot_be_called_with_classes_that_include_ActiveRecord::Base (unexpected)
--- FAIL: RelationTest#test_count_on_association_relation (unexpected)
--- FAIL: PersistenceTest#test_update_columns_should_not_modify_updated_at (unexpected)
--- FAIL: HasOneAssociationsTest#test_dependence_with_missing_association (unexpected)
--- FAIL: BasicsTest#test_create_without_prepared_statement (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_join_model_ignores_default_includes (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_accept_string_and_hash_join (unexpected)
--- FAIL: BasicsTest#test_equality_of_new_records (unexpected)
--- FAIL: StoreTest#test_prefix/suffix_do_not_affect_stored_attributes (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_ids_and_inverse_of (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_time_zone-aware_read_attribute (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_aggregate_having_one_mapping_array (unexpected)
--- FAIL: HasManyAssociationsTest#test_get_ids_for_ordered_association (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_open_ended_range (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15127361/activerecord/1714975560280/1714983118562)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15141255?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15141255?buildTab=artifacts#/activerecord) on master @ [e90d15c9807775a1a1d89f75cac4c9383c325709](https://github.com/cockroachdb/cockroach/commits/e90d15c9807775a1a1d89f75cac4c9383c325709):


```
0 tests expected failed but not run
---
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_has_many_sti (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_three_levels_by_ping_pong (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_where_first_level_returns_nil (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_validation_does_not_validate_non_dirty_association_target (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_join_for_count (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_assignment_before_either_saved (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_cascaded_eager_association_loading_with_twice_includes_edge_cases (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_belongs_to_sti (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_preload_through_missing_records (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_store_association_in_two_relations_with_one_save (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_interdependent_one_level_and_two_levels (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_composite_primary_key_autosave (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_store_association_in_two_relations_with_one_save_in_existing_object (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_preloading_across_has_one_through_constrains_loaded_records (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_save_fails_for_invalid_belongs_to (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_validation_does_not_validate_stale_association_target (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_grafts_stashed_associations_to_correct_parent (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_save_succeeds_for_invalid_belongs_to_with_validate_false (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_two_levels_and_self_table_reference (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_build_and_then_save_parent_should_not_reload_target (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_preloaded_records_are_not_duplicated (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_store_two_association_with_one_save (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_assignment_before_parent_saved (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_default_pk_and_sequence (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_two_levels_with_condition (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_has_many_sti_and_subclasses (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_recursive_cascading_four_levels_has_many_through (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_cascaded_eager_association_loading_with_duplicated_includes (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_multiple_stis_and_order (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_two_levels_with_two_has_many_associations (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_cascaded_eager_association_loading_with_join_for_count (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_missing_first_record (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_store_association_with_a_polymorphic_relationship (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_two_levels_and_one_level (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_should_save_parent_but_not_invalid_child (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_recursive_cascading_four_levels_has_and_belongs_to_many (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_of_stis_with_multiple_references (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_create_fixtures_resets_sequences_when_not_cached (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_preloading_across_has_one_constrains_loaded_records (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_store_association_in_two_relations_with_one_save_in_existing_object_with_values (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_two_levels (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_hmt_does_not_table_name_collide_when_joining_associations (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15141255/activerecord/1715062264960/1715069986629)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15157859?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15157859?buildTab=artifacts#/activerecord) on master @ [2f498d0abcf4130e5c872b73200086d6a3af9f02](https://github.com/cockroachdb/cockroach/commits/2f498d0abcf4130e5c872b73200086d6a3af9f02):


```
--- FAIL: RelationTest#test_select_with_from_includes_quoted_original_table_name (unexpected)
--- FAIL: RelationTest#test_typecasting_where_with_array (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_create_fixtures_resets_sequences_when_not_cached (unexpected)
--- FAIL: RelationTest#test_first_or_initialize (unexpected)
--- FAIL: RelationTest#test_first_or_create_bang_with_valid_array (unexpected)
--- FAIL: RelationTest#test_references_with_blank_value (unexpected)
--- FAIL: ReflectionTest#test_never_validate_association_if_explicit (unexpected)
--- FAIL: RelationTest#test_unscope_with_aliased_column (unexpected)
--- FAIL: EagerAssociationTest#test_preload_with_invalid_argument (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit_with_no_results (unexpected)
--- FAIL: BasicsTest#test_ignored_columns_don't_prevent_explicit_declaration_of_attribute_methods (unexpected)
--- FAIL: RelationTest#test_create_with_polymorphic_association (unexpected)
--- FAIL: BasicsTest#test_#connected_to_many_sets_prevent_writes_if_role_is_reading (unexpected)
--- FAIL: RelationTest#test_loading_with_one_association_with_non_preload (unexpected)
--- FAIL: BasicsTest#test_dup (unexpected)
--- FAIL: BasicsTest#test_arel_attribute_normalization (unexpected)
--- FAIL: BasicsTest#test_readonly_attributes (unexpected)
--- FAIL: RelationTest#test_find_by_with_take_memoization (unexpected)
--- FAIL: EagerAssociationTest#test_preconfigured_includes_with_has_many (unexpected)
--- FAIL: RelationTest#test_first_or_initialize_with_no_parameters (unexpected)
--- FAIL: RelationTest#test_find_all_using_where_with_relation_with_joins (unexpected)
--- FAIL: EagerAssociationTest#test_loading_from_an_association (unexpected)
--- FAIL: RelationTest#test_find_by_with_non-hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_of_stis_with_multiple_references (unexpected)
--- FAIL: RelationTest#test_anonymous_extension (unexpected)
--- FAIL: EagerAssociationTest#test_preconfigured_includes_with_belongs_to (unexpected)
--- FAIL: RelationTest#test_find_with_list_of_ar (unexpected)
--- FAIL: RelationTest#test_relations_don't_load_all_records_in_#pretty_print (unexpected)
--- FAIL: RelationTest#test_to_yaml (unexpected)
--- FAIL: RelationTest#test_count_complex_chained_relations (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit (unexpected)
--- FAIL: BasicsTest#test_attribute_names_on_table_not_exists (unexpected)
--- FAIL: EagerAssociationTest#test_loading_polymorphic_association_with_mixed_table_conditions (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_through_has_one_association_with_order (unexpected)
--- FAIL: SchemaIndexIncludeColumnsTest#test_schema_dumps_index_included_columns (unexpected)
--- FAIL: ReflectionTest#test_content_columns (unexpected)
--- FAIL: InheritanceTest#test_inherits_custom_primary_key (unexpected)
--- FAIL: CockroachDB::FixturesTest#test_multiple_clean_fixtures (unexpected)
--- FAIL: BasicsTest#test_records_of_different_classes_have_different_hashes (unexpected)
--- FAIL: RelationTest#test_select_with_subquery_string_in_from_does_not_use_original_table_name (unexpected)
--- FAIL: BasicsTest#test_utc_as_time_zone (unexpected)
--- FAIL: BasicsTest#test_find_by_slug_with_array (unexpected)
--- FAIL: RelationTest#test_empty_with_zero_limit (unexpected)
--- FAIL: EagerAssociationTest#test_preload_has_many_with_association_condition_and_default_scope (unexpected)
--- FAIL: BasicsTest#test_clone_preserves_subtype (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_join_model_with_conditions (unexpected)
--- FAIL: BasicsTest#test_scoped_can_take_a_values_hash (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15157859/activerecord/1715148762752/1715156820512)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15175579?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15175579?buildTab=artifacts#/activerecord) on master @ [1ee6c19177041c25481e465186ce8f7294c82547](https://github.com/cockroachdb/cockroach/commits/1ee6c19177041c25481e465186ce8f7294c82547):


```
--- FAIL: AttributeMethodsTest#test_attribute_for_inspect_with_an_array (unexpected)
--- FAIL: RelationTest#test_count_on_association_relation (unexpected)
--- FAIL: FinderTest#test_second_to_last_have_primary_key_order_by_default (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_post_has_many_tags_through_association_with_composite_query_constraints (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_multiple_associations_with_same_table_has_one (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_reorder_raise_errors (unexpected)
--- FAIL: RelationTest#test_where_with_take_memoization (unexpected)
--- FAIL: AssociationsTest#test_loading_the_association_target_should_load_most_recent_attributes_for_child_records_marked_for_destruction (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_has_and_belongs_to_many_is_usable_with_belongs_to_required_by_default (unexpected)
--- FAIL: InheritanceTest#test_should_store_full_class_name_with_store_full_sti_class_option_enabled (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_join_model_ignores_default_includes (unexpected)
--- FAIL: BelongsToAssociationsTest#test_tracking_polymorphic_changes (unexpected)
--- FAIL: RelationTest#test_find_or_create_by_with_block (unexpected)
--- FAIL: RelationTest#test_unscope_with_double_dot_where (unexpected)
--- FAIL: OptimisticLockingTest#test_lock_new_when_explicitly_passing_value (unexpected)
--- FAIL: FinderTest#test_find_by_empty_in_condition (unexpected)
--- FAIL: SignedIdTest#test_use_a_custom_verifier (unexpected)
--- FAIL: BelongsToAssociationsTest#test_dependency_should_halt_parent_destruction (unexpected)
--- FAIL: PersistenceTest#test_update_columns_should_not_use_setter_method (unexpected)
--- FAIL: CalculationsTest#test_should_return_nil_as_average (unexpected)
--- FAIL: CalculationsTest#test_count_with_distinct (unexpected)
--- FAIL: FinderTest#test_eager_load_for_no_has_many_with_limit_and_left_joins_for_has_many (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_nulls_not_distinct (unexpected)
--- FAIL: AttributeMethodsTest#test_boolean_attributes (unexpected)
--- FAIL: PersistenceTest#test_find_raises_record_not_found_exception (unexpected)
--- FAIL: CalculationsTest#test_minimum_and_maximum_on_time_attributes (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_association_through_a_has_many_association_to_self (unexpected)
--- FAIL: HasOneAssociationsTest#test_build_with_block (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_grouped_association_with_foreign_key_option (unexpected)
--- FAIL: FinderTest#test_second (unexpected)
--- FAIL: AttributeMethodsTest#test_write_attribute_raises_ActiveModel::MissingAttributeError_when_the_attribute_does_not_exist (unexpected)
--- FAIL: RelationTest#test_find_or_create_by! (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_composite_primary_key_join_table (unexpected)
--- FAIL: BasicsTest#test_many_mutations (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_build_for_has_many_through_association (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_loading_for_new_record (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_default_scope_as_callable (unexpected)
--- FAIL: BasicsTest#test_default_in_utc (unexpected)
--- FAIL: InheritanceTest#test_alt_inheritance_find (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_assign_ids (unexpected)
--- FAIL: ReflectionTest#test_columns (unexpected)
--- FAIL: BelongsToAssociationsTest#test_attributes_are_set_without_error_when_initialized_from_belongs_to_association_with_array_in_where_clause (unexpected)
--- FAIL: FinderTest#test_find_with_hash_conditions_on_joined_table (unexpected)
--- FAIL: ModulesTest#test_assign_ids (unexpected)
--- FAIL: AttributeMethodsTest#test_set_attributes (unexpected)
--- FAIL: RelationTest#test_eager_load_with_blank_value (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_that_should_not_be_fully_merged (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15175579/activerecord/1715235408087/1715244211113)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15213439?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15213439?buildTab=artifacts#/activerecord) on master @ [59b261a579cbe2c032a5dd3e182ff67aeee900b9](https://github.com/cockroachdb/cockroach/commits/59b261a579cbe2c032a5dd3e182ff67aeee900b9):


```
--- FAIL: CalculationsTest#test_maximum_with_from_option (unexpected)
--- FAIL: CalculationsTest#test_pluck_not_auto_table_name_prefix_if_column_included (unexpected)
--- FAIL: ReflectionTest#test_reflection_klass_for_nested_class_name (unexpected)
--- FAIL: HasManyAssociationsTest#test_associations_order_should_be_priority_over_throughs_order (unexpected)
--- FAIL: HasManyAssociationsTest#test_replace_with_same_content (unexpected)
--- FAIL: HasManyAssociationsTest#test_pushing_association_updates_counter_cache (unexpected)
--- FAIL: ReflectionTest#test_belongs_to_reflection_with_array_fk_raises (unexpected)
--- FAIL: HasManyAssociationsTest#test_associations_replace_in_memory_when_records_have_the_same_id (unexpected)
--- FAIL: HasManyAssociationsTest#test_inverse_on_before_validate (unexpected)
--- FAIL: SignedIdTest#test_fail_to_find_signed_record_within_expiration_duration (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_field_through_association_and_having (unexpected)
--- FAIL: HasManyAssociationsTest#test_include_checks_if_record_exists_if_target_not_loaded (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_proxy_should_not_respond_to_private_methods (unexpected)
--- FAIL: CalculationsTest#test_async_pluck_on_loaded_relation (unexpected)
--- FAIL: HasManyAssociationsTest#test_to_a_should_dup_target (unexpected)
--- FAIL: HasManyAssociationsTest#test_anonymous_has_many (unexpected)
--- FAIL: SignedIdTest#test_fail_to_find_signed_record_within_expiration_time (unexpected)
--- FAIL: ReflectionTest#test_active_record_primary_key (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_in_batches (unexpected)
--- FAIL: HasManyAssociationsTest#test_in_memory_replacement_executes_no_queries (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_one_should_return_false_if_more_than_one (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_via_block (unexpected)
--- FAIL: AttributeMethodsTest#test_inherited_custom_accessors (unexpected)
--- FAIL: HasManyAssociationsTest#test_build (unexpected)
--- FAIL: HasManyAssociationsTest#test_key_ensuring_owner_was_is_valid_when_dependent_option_is_destroy_async (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_updates_counter_cache_without_dependent_option (unexpected)
--- FAIL: ReflectionTest#test_integer_columns (unexpected)
--- FAIL: ModulesTest#test_table_name (unexpected)
--- FAIL: HasManyAssociationsTest#test_create (unexpected)
--- FAIL: CalculationsTest#test_should_sum_scoped_field (unexpected)
--- FAIL: ReflectionTest#test_automatic_inverse_does_not_suppress_name_error_from_incidental_code (unexpected)
--- FAIL: HasManyAssociationsTest#test_merging_with_custom_attribute_writer (unexpected)
--- FAIL: CalculationsTest#test_pluck_without_column_names (unexpected)
--- FAIL: HasManyAssociationsTest#test_get_ids_for_loaded_associations (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroying_by_string_id (unexpected)
--- FAIL: AttributeMethodsTest#test_custom_field_attribute_predicate (unexpected)
--- FAIL: HasOneAssociationsTest#test_natural_assignment (unexpected)
--- FAIL: HasManyAssociationsTest#test_modifying_a_through_a_has_many_should_raise (unexpected)
--- FAIL: AttributeMethodsTest#test_user-defined_JSON_attribute_predicate (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_for_inspect_with_a_non-primary_key_id_attribute (unexpected)
--- FAIL: CalculationsTest#test_pluck_loaded_relation_multiple_columns (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_association (unexpected)
--- FAIL: ModulesTest#test_eager_loading_in_modules (unexpected)
--- FAIL: HasManyAssociationsTest#test_transaction_when_deleting_persisted (unexpected)
--- FAIL: HasManyAssociationsTest#test_adding_a_mismatch_class (unexpected)
--- FAIL: HasManyAssociationsTest#test_counter_cache_updates_in_memory_after_update_with_inverse_of_enabled (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_none_on_loaded_association_should_not_use_query (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15213439/activerecord/1715407872690/1715417115442)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15224239?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15224239?buildTab=artifacts#/activerecord) on master @ [4c2e7761acd050aaee565443932b6b0eca55620b](https://github.com/cockroachdb/cockroach/commits/4c2e7761acd050aaee565443932b6b0eca55620b):


```
--- FAIL: HasManyThroughAssociationsTest#test_primary_key_option_on_source (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_primary_key_option (unexpected)
--- FAIL: MultiParameterAttributeTest#test_create_with_multiparameter_attributes_setting_date_and_time_attribute (unexpected)
--- FAIL: EnumTest#test_validation_with_'validate:_hash'_option (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_callbacks_firing_order_on_create (unexpected)
--- FAIL: EagerAssociationTest#test_conditions_on_join_table_with_include_and_limit (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_empty_on_an_association_that_has_not_been_loaded_performs_a_query (unexpected)
--- FAIL: HasManyAssociationsTest#test_finding_with_foreign_key (unexpected)
--- FAIL: EagerAssociationTest#test_preload_has_many_uses_exclusive_scope (unexpected)
--- FAIL: CalculationsTest#test_count_with_eager_loading_and_custom_select_and_order (unexpected)
--- FAIL: FinderTest#test_find_with_eager_loading_collection_and_ordering_by_collection_primary_key (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_through_has_one_association_with_conditions (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_aggregate_having_one_mapping_array (unexpected)
--- FAIL: HasManyAssociationsTest#test_clear_collection_should_not_change_updated_at (unexpected)
--- FAIL: BasicsTest#test_singular_table_name_guesses_with_prefixes_and_suffixes (unexpected)
--- FAIL: BelongsToAssociationsTest#test_dont_find_target_when_foreign_key_is_null (unexpected)
--- FAIL: TimestampTest#test_touching_a_record_with_a_belongs_to_that_uses_a_counter_cache_should_update_the_parent (unexpected)
--- FAIL: MigrationTest#test_rename_table_with_prefix_and_suffix (unexpected)
--- FAIL: HasManyAssociationsTest#test_counting_with_counter_sql (unexpected)
--- FAIL: EagerAssociationTest#test_preconfigured_includes_with_habtm (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_saving_multiple_relationships (unexpected)
--- FAIL: NumericalityValidationTest#test_on_abstract_class (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_none_should_defer_to_collection_if_using_a_block (unexpected)
--- FAIL: AttributeMethodsTest#test_inherited_custom_accessors_with_reserved_names (unexpected)
--- FAIL: CalculationsTest#test_should_count_scoped_select (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_destroy_on_association_clears_scope (unexpected)
--- FAIL: FinderTest#test_exists_with_distinct_and_offset_and_select (unexpected)
--- FAIL: HasManyAssociationsTest#test_raises_RecordNotDestroyed_when_replaced_child_can't_be_destroyed (unexpected)
--- FAIL: EnumTest#test_validate_inclusion_of_value_in_array (unexpected)
--- FAIL: BasicsTest#test_cannot_call_connected_to_on_the_abstract_class_that_did_not_establish_the_connection (unexpected)
--- FAIL: PersistenceTest#test_update_columns_with_default_scope (unexpected)
--- FAIL: FinderTest#test_find_by_one_attribute (unexpected)
--- FAIL: PersistenceTest#test_becomes_errors_base (unexpected)
--- FAIL: FinderTest#test_exists_with_distinct_and_offset_and_eagerload_and_order (unexpected)
--- FAIL: DelegatedTypeTest#test_delegated_class (unexpected)
--- FAIL: CalculationsTest#test_distinct_count_with_order_and_limit (unexpected)
--- FAIL: FinderTest#test_exists_with_distinct_association_includes_and_limit (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_empty_array_with_aggregate_having_multiple_mappings (unexpected)
--- FAIL: InheritanceAttributeMappingTest#test_polymorphic_associations_custom_type (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_limit (unexpected)
--- FAIL: ReflectionTest#test_foreign_key_is_inferred_from_model_name (unexpected)
--- FAIL: HasOneAssociationsTest#test_dependency_should_halt_parent_destruction (unexpected)
--- FAIL: PersistenceTest#test_save_with_duping_of_destroyed_object (unexpected)
--- FAIL: InheritanceTest#test_becomes_and_change_tracking_for_inheritance_columns (unexpected)
--- FAIL: StoreTest#test_updating_the_store_populates_the_changed_array_correctly (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_relationship_cannot_have_a_counter_cache (unexpected)
--- FAIL: PostgresqlInfinityTest#test_where_clause_with_infinite_range_on_a_datetime_column (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15224239/activerecord/1715494366142/1715501643867)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15235275?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15235275?buildTab=artifacts#/activerecord) on master @ [4cc0bfcc14771331fea57de01e1ea78b07393f3d](https://github.com/cockroachdb/cockroach/commits/4cc0bfcc14771331fea57de01e1ea78b07393f3d):


```
--- FAIL: HasAndBelongsToManyAssociationsTest#test_join_with_group (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_find_grouped (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_assignment_before_parent_saved (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_store_association_in_two_relations_with_one_save_in_existing_object (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_create (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_new_with_values_in_collection (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_destroying (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_distinct_order_preserved (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_store_two_association_with_one_save (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_adding_type_mismatch (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_scoped_find_on_through_association_doesnt_return_read_only_records (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_creation_respects_hash_condition (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_dynamic_find_should_respect_association_order (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_has_and_belongs_to_many (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_has_many_through_polymorphic_has_manys_works (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_distinct_when_association_already_loaded (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_attributes_are_being_set_when_initialized_from_habtm_association_with_multiple_where_clauses (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_proxy_transaction_method_starts_transaction_in_association_class (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_include_checks_if_record_exists_if_target_not_loaded (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_count (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_singular_ids_are_reloaded_after_collection_concat (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_find_in_association_with_options (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_distinct_after_the_fact (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_join_table_alias (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_include_uses_array_include_after_loaded (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_validation_does_not_validate_non_dirty_association_target (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_validation_does_not_validate_stale_association_target (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_get_ids_for_unloaded_associations_does_not_load_them (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_symbols_as_keys (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_has_and_belongs_to_many_in_a_namespaced_model_pointing_to_a_namespaced_model (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_adding_from_the_project_fixed_timestamp (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_with_extend_option (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_distinct_before_the_fact (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_save_fails_for_invalid_belongs_to (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_has_and_belongs_to_many_in_a_namespaced_model_pointing_to_a_non_namespaced_model (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_has_and_belongs_to_many_associations_on_new_records_use_null_relations (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_include_returns_false_for_non_matching_record_to_verify_scoping (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_alternate_database (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_build_by_new_record (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_proper_usage_of_primary_keys_and_join_table (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_find_with_merged_options (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_has_and_belongs_to_many_is_usable_with_belongs_to_required_by_default (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_get_ids (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_distinct_option_prevents_duplicate_push (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_destroy_all (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15235275/activerecord/1715580865250/1715589910740)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15254568?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15254568?buildTab=artifacts#/activerecord) on master @ [b185402d34768c898d63a61de3e585e4723fbe81](https://github.com/cockroachdb/cockroach/commits/b185402d34768c898d63a61de3e585e4723fbe81):


```
--- FAIL: HasManyAssociationsTest#test_first_or_create_adds_the_record_to_the_association (unexpected)
--- FAIL: BasicsTest#test_custom_mutator (unexpected)
--- FAIL: HasManyAssociationsTest#test_has_many_association_with_same_foreign_key_name (unexpected)
--- FAIL: PersistenceTest#test_update (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_store_association_with_a_polymorphic_relationship (unexpected)
--- FAIL: HasManyAssociationsTest#test_include_returns_false_for_non_matching_record_to_verify_scoping (unexpected)
--- FAIL: PersistenceTest#test_update_attribute_with_one_updated (unexpected)
--- FAIL: PersistenceTest#test_class_level_update_without_ids (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: BasicsTest#test_create_after_initialize_without_block (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_with_extend_option (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all_on_association_clears_scope (unexpected)
--- FAIL: HasManyAssociationsTest#test_create_with_bang_on_habtm_when_parent_is_new_raises (unexpected)
--- FAIL: PersistenceTest#test_save_destroyed_object (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependent_association_respects_optional_sanitized_conditions_on_delete (unexpected)
--- FAIL: BasicsTest#test_previously_changed (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_scoped_grouped_having (unexpected)
--- FAIL: HasManyAssociationsTest#test_defining_has_many_association_with_nullify_dependency_lazily_evaluates_target_class (unexpected)
--- FAIL: HasManyAssociationsTest#test_finding_array_compatibility (unexpected)
--- FAIL: PersistenceTest#test_update_columns (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_save_succeeds_for_invalid_belongs_to_with_validate_false (unexpected)
--- FAIL: HasManyAssociationsTest#test_get_ids_for_through (unexpected)
--- FAIL: PersistenceTest#test_update_many_with_array_of_active_record_base_objects! (unexpected)
--- FAIL: HasManyAssociationsTest#test_replace_returns_target (unexpected)
--- FAIL: HasManyAssociationsTest#test_get_ids_ignores_include_option (unexpected)
--- FAIL: BasicsTest#test_many_mutations (unexpected)
--- FAIL: BasicsTest#test_all_with_conditions (unexpected)
--- FAIL: HasManyAssociationsTest#test_sharded_deleting_models (unexpected)
--- FAIL: PersistenceTest#test_update_column (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_grouped (unexpected)
--- FAIL: BasicsTest#test_preserving_date_objects (unexpected)
--- FAIL: HasManyAssociationsTest#test_belongs_to_with_new_object (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_the_association_with_an_array (unexpected)
--- FAIL: BasicsTest#test_limit_should_sanitize_sql_injection_for_limit_without_commas (unexpected)
--- FAIL: BasicsTest#test_connection_handler_can_be_overridden (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_by_string_id (unexpected)
--- FAIL: PersistenceTest#test_update_column_should_not_use_setter_method (unexpected)
--- FAIL: PersistenceTest#test_create (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_each_with_conditions (unexpected)
--- FAIL: PersistenceTest#test_save_touch_false (unexpected)
--- FAIL: BasicsTest#test_clear_cache_when_setting_table_name (unexpected)
--- FAIL: PersistenceTest#test_increment_with_touch_updates_timestamps (unexpected)
--- FAIL: PersistenceTest#test_destroy_many_with_invalid_id (unexpected)
--- FAIL: HasManyAssociationsTest#test_counter_cache_updates_in_memory_after_create_with_overlapping_counter_cache_columns (unexpected)
--- FAIL: PersistenceTest#test_delete_all (unexpected)
--- FAIL: BasicsTest#test_table_name_guesses (unexpected)
--- FAIL: BasicsTest#test_default_values_on_empty_strings (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15254568/activerecord/1715667103926/1715675532596)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15273548?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15273548?buildTab=artifacts#/activerecord) on master @ [6300c3c3367ad46ac48bf24915cf0d73cae446a0](https://github.com/cockroachdb/cockroach/commits/6300c3c3367ad46ac48bf24915cf0d73cae446a0):


```
--- FAIL: HasManyAssociationsTest#test_delete_all_on_association_is_the_same_as_not_loaded (unexpected)
--- FAIL: HasManyAssociationsTest#test_associations_autosaves_when_object_is_already_persisted (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_reselect_raise_errors (unexpected)
--- FAIL: CalculationsTest#test_having_with_strong_parameters (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_fields_with_table_alias (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_through_scope_with_includes (unexpected)
--- FAIL: HasManyAssociationsTest#test_transactions_when_adding_to_new_record (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_updating_attributes_on_non_rich_associations (unexpected)
--- FAIL: PersistenceTest#test_update_many (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_marshal_dump (unexpected)
--- FAIL: CalculationsTest#test_should_count_field_in_joined_table (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_with_a_bang (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_parent_should_save_children_record_with_foreign_key_validation_set_in_before_save_callback (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_build_then_save_with_has_many_inverse (unexpected)
--- FAIL: PersistenceTest#test_update_attribute_uses_query_constraints_config (unexpected)
--- FAIL: HasManyAssociationsTest#test_restrict_with_error_with_locale (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroy_all_on_desynced_counter_cache_association (unexpected)
--- FAIL: CalculationsTest#test_select_avg_with_group_by_as_virtual_attribute_with_sql (unexpected)
--- FAIL: HasManyAssociationsTest#test_transactions_when_adding_to_persisted (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_collection_singular_ids_setter_raises_exception_when_invalid_ids_set (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_singleton_has_many_through (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all_with_option_delete_all (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_should_property_quote_string_primary_keys (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_one_should_return_false_if_zero (unexpected)
--- FAIL: CalculationsTest#test_should_not_use_alias_for_grouped_field (unexpected)
--- FAIL: RelationTest#test_build (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_create (unexpected)
--- FAIL: HasManyAssociationsTest#test_sending_new_to_association_proxy_should_have_same_effect_as_calling_new (unexpected)
--- FAIL: RelationTest#test_count_on_association_relation (unexpected)
--- FAIL: IntegrationTest#test_cache_key_for_newer_updated_at (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_through_record_is_built_when_created_with_where (unexpected)
--- FAIL: RelationTest#test_optimizer_hints_with_blank_value (unexpected)
--- FAIL: HasManyAssociationsTest#test_attributes_are_being_set_when_initialized_from_has_many_association_with_where_clause (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_updates_counter_cache_with_dependent_destroy (unexpected)
--- FAIL: RelationTest#test_create_or_find_by_with_bang_should_raise_due_to_validation_errors (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_that_has_joined_same_table_with_parent_relation (unexpected)
--- FAIL: HasManyAssociationsTest#test_building_the_associated_object_with_implicit_sti_base_class (unexpected)
--- FAIL: CalculationsTest#test_pluck_functions_with_alias (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_find_should_append_to_association_order (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_preload_sti_rhs_class (unexpected)
--- FAIL: RelationTest#test_find_with_list_of_ar (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_many_on_loaded_association_should_not_use_query (unexpected)
--- FAIL: PersistenceTest#test_destroy_with_invalid_ids_for_a_model_that_expects_composite_keys (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_joining_has_many_through_belongs_to (unexpected)
--- FAIL: CalculationsTest#test_group_by_attribute_with_custom_type (unexpected)
--- FAIL: AssociationsTest#test_belongs_to_a_model_with_composite_foreign_key_finds_associated_record (unexpected)
--- FAIL: CalculationsTest#test_should_get_minimum_of_arel_attribute (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15273548/activerecord/1715753460412/1715761901010)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15288601?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15288601?buildTab=artifacts#/activerecord) on master @ [992d4573fb8b71e28717b4d14c4c7ebccb38c412](https://github.com/cockroachdb/cockroach/commits/992d4573fb8b71e28717b4d14c4c7ebccb38c412):


```
--- FAIL: EagerAssociationTest#test_eager_loading_of_instance_dependent_associations_is_not_supported (unexpected)
--- FAIL: RelationTest#test_where_id_with_delegated_ar_object (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_multiple_associations_with_same_table_has_one (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_multiple_fields_having_functions (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_a_time_zone-aware_attribute_to_a_blank_string_returns_nil (unexpected)
--- FAIL: StoreTest#test_updating_the_store_won't_mark_accessor_as_changed_if_the_whole_store_was_updated (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_destroy_on_association_clears_scope (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_should_raise_exception_for_destroying_mismatching_records (unexpected)
--- FAIL: PersistenceTest#test_delete_many (unexpected)
--- FAIL: RelationTest#test_default_scope_order_with_scope_order (unexpected)
--- FAIL: RelationTest#test_reverse_order_with_function_other_predicates (unexpected)
--- FAIL: PersistenceTest#test_save_invalid_record (unexpected)
--- FAIL: CalculationsTest#test_pluck_columns_with_same_name (unexpected)
--- FAIL: HasManyThroughDisableJoinsAssociationsTest#test_first_and_scope_in_double_join_applies_order_in_memory (unexpected)
--- FAIL: HasManyAssociationsTest#test_with_polymorphic_has_many_with_custom_columns_name (unexpected)
--- FAIL: SerializedAttributeTestWithYamlSafeLoad#test_nil_not_serialized_without_class_constraint (unexpected)
--- FAIL: BasicsTest#test_column_types_typecast (unexpected)
--- FAIL: FinderTest#test_exists_with_nil_arg (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_destroy_all_on_association_clears_scope (unexpected)
--- FAIL: HasManyAssociationsTest#test_counter_cache_on_unloaded_association (unexpected)
--- FAIL: AttributeMethodsTest#test_set_attributes (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnNewRecord#test_autosave_new_record_with_after_create_callback (unexpected)
--- FAIL: FinderTest#test_find_doesnt_have_implicit_ordering (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all,_when_not_loaded,_doesn't_load_the_records (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_none_on_loaded_association_should_not_use_query (unexpected)
--- FAIL: RelationTest#test_finding_with_desc_order_with_string (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_count_with_include_should_alias_join_table (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_source_scope (unexpected)
--- FAIL: FinderTest#test_model_class_responds_to_last_bang (unexpected)
--- FAIL: PersistenceTest#test_update_columns_should_not_modify_updated_at (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_consider_type (unexpected)
--- FAIL: CalculationsTest#test_should_return_zero_if_sum_conditions_return_nothing (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_creation_respects_hash_condition (unexpected)
--- FAIL: FinderTest#test_fourth_with_offset (unexpected)
--- FAIL: PersistenceTest#test_create_through_factory_with_block (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_find_in_association (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_store_two_association_with_one_save (unexpected)
--- FAIL: StoreTest#test_writing_with_not_nullable_column_encoded_with_JSON (unexpected)
--- FAIL: FinderTest#test_find_with_ids_and_limit (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_left_outer_joins_raise_errors (unexpected)
--- FAIL: BasicsTest#test_find_keeps_multiple_order_values (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_left_joined_same_table_with_through_table (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_belongs_to_association_SQL (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_with_conditions_join_model_with_include (unexpected)
--- FAIL: PersistenceTest#test_find_raises_record_not_found_exception (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_grouped_with_invalid_field (unexpected)
--- FAIL: RelationTest#test_dynamic_finder (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15288601/activerecord/1715840131963/1715847443408)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15301919?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15301919?buildTab=artifacts#/activerecord) on master @ [855b9cc97afa3df4f7e17f928c04ab0834b2630c](https://github.com/cockroachdb/cockroach/commits/855b9cc97afa3df4f7e17f928c04ab0834b2630c):


```
--- FAIL: CalculationsTest#test_ids (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_includes_limit_and_empty_result (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_association_with_non_numeric_foreign_key (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_field_through_association_and_having (unexpected)
--- FAIL: CalculationsTest#test_pick_loaded_relation_multiple_columns (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_field (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_empty_in (unexpected)
--- FAIL: CalculationsTest#test_pluck_on_aliased_attribute (unexpected)
--- FAIL: CalculationsTest#test_count_with_aliased_attribute (unexpected)
--- FAIL: CalculationsTest#test_should_not_use_alias_for_grouped_field (unexpected)
--- FAIL: CalculationsTest#test_pluck_functions_without_alias (unexpected)
--- FAIL: CalculationsTest#test_should_reference_correct_aliases_while_joining_tables_of_has_many_through_association (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_line_endings (unexpected)
--- FAIL: CalculationsTest#test_should_count_manual_select_with_count_all (unexpected)
--- FAIL: CalculationsTest#test_pluck_if_table_included (unexpected)
--- FAIL: CalculationsTest#test_limit_should_apply_before_count_arel_attribute (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_assignment_before_either_saved (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_build_and_then_save_parent_should_not_reload_target (unexpected)
--- FAIL: CalculationsTest#test_should_get_maximum_of_field_with_include (unexpected)
--- FAIL: CalculationsTest#test_count_selected_arel_attribute_in_joined_table (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_field_with_conditions_and_having (unexpected)
--- FAIL: CalculationsTest#test_#skip_query_cache!_for_a_simple_calculation (unexpected)
--- FAIL: CalculationsTest#test_limit_with_offset_is_kept (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_should_save_parent_but_not_invalid_child (unexpected)
--- FAIL: CalculationsTest#test_pick_loaded_relation_aliased_attribute (unexpected)
--- FAIL: CalculationsTest#test_should_perform_joined_include_when_referencing_included_tables (unexpected)
--- FAIL: CalculationsTest#test_pick_one (unexpected)
--- FAIL: CalculationsTest#test_count_selected_arel_attribute (unexpected)
--- FAIL: CalculationsTest#test_should_sum_arel_attribute (unexpected)
--- FAIL: CalculationsTest#test_count_with_block_and_column_name_raises_an_error (unexpected)
--- FAIL: CalculationsTest#test_no_queries_for_empty_relation_on_minimum (unexpected)
--- FAIL: CalculationsTest#test_calculation_with_query_cache (unexpected)
--- FAIL: CalculationsTest#test_sum_uses_enumerable_version_when_block_is_given (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_create_fixtures_resets_sequences_when_not_cached (unexpected)
--- FAIL: CalculationsTest#test_pluck_type_cast_with_left_joins_without_table_name_qualified_column (unexpected)
--- FAIL: CalculationsTest#test_count_with_from_option (unexpected)
--- FAIL: CalculationsTest#test_sum_with_not_auto_table_name_prefix_if_column_included (unexpected)
--- FAIL: CalculationsTest#test_count_with_arel_attribute (unexpected)
--- FAIL: CalculationsTest#test_pluck_loaded_relation_sql_fragment (unexpected)
--- FAIL: CalculationsTest#test_minimum_and_maximum_on_time_attributes (unexpected)
--- FAIL: CalculationsTest#test_count_with_column_and_options_parameter (unexpected)
--- FAIL: CalculationsTest#test_sum_with_from_option (unexpected)
--- FAIL: CalculationsTest#test_limit_is_kept (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_grouped_with_longer_field (unexpected)
--- FAIL: CalculationsTest#test_ids_on_loaded_relation_with_includes_and_table_scope (unexpected)
--- FAIL: CalculationsTest#test_should_return_integer_average_if_db_returns_such (unexpected)
--- FAIL: CalculationsTest#test_should_order_by_grouped_field (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15301919/activerecord/1715926270240/1715934507703)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15327658?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15327658?buildTab=artifacts#/activerecord) on master @ [5d013285fa696f53df2abb39f44ffc777125fe1c](https://github.com/cockroachdb/cockroach/commits/5d013285fa696f53df2abb39f44ffc777125fe1c):


```
--- FAIL: BasicsTest#test_equality_of_relation_and_collection_proxy (unexpected)
--- FAIL: FinderTest#test_count_by_sql (unexpected)
--- FAIL: BasicsTest#test_create_without_prepared_statement (unexpected)
--- FAIL: AttributeMethodsTest#test_aliases_to_the_same_attribute_name_do_not_conflict_with_each_other (unexpected)
--- FAIL: PersistenceTest#test_save_touch_false (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_through_scope_is_affected_by_unscoping (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_field_having_condition (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_join_model_with_conditions_on_top_level (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_multiple_associations_with_same_table_has_many_and_habtm (unexpected)
--- FAIL: BooleanTest#test_boolean_cast_from_string (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_relationship_cannot_have_a_counter_cache (unexpected)
--- FAIL: FinderTest#test_find_by_empty_in_condition (unexpected)
--- FAIL: ReflectionTest#test_belongs_to_reflection_with_array_fk_raises (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_build_before_child_saved (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_for_existing_record_with_updated_at_and_custom_cache_timestamp_format (unexpected)
--- FAIL: BasicsTest#test_find_keeps_multiple_order_values (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_with_touch_option_on_destroy (unexpected)
--- FAIL: EagerAssociationTest#test_joins_with_includes_should_preload_via_joins (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_enum_method_issues_a_deprecation_warning (unexpected)
--- FAIL: HasOneAssociationsTest#test_build_and_create_should_not_happen_within_scope (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_create_bang_should_raise_exception_when_join_record_has_errors (unexpected)
--- FAIL: ActiveRecord::Migration::ForeignKeyChangeColumnWithSuffixTest#test_remove_reference_column_of_child_table (unexpected)
--- FAIL: BasicsTest#test_preserving_time_objects_with_time_with_zone_conversion_to_default_timezone_utc (unexpected)
--- FAIL: BasicsTest#test_touch_should_raise_error_on_a_new_object (unexpected)
--- FAIL: BasicsTest#test_assert_queries (unexpected)
--- FAIL: InheritanceTest#test_where_create_bang_with_invalid_type (unexpected)
--- FAIL: EagerAssociationTest#test_dont_create_temporary_active_record_instances (unexpected)
--- FAIL: CalculationsTest#test_ids_on_loaded_relation_with_scope (unexpected)
--- FAIL: FinderTest#test_third_to_last_have_primary_key_order_by_default (unexpected)
--- FAIL: PostgresqlVirtualColumnTest#test_stored_column (unexpected)
--- FAIL: FinderTest#test_find_by_with_range_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: FinderTest#test_exists_with_left_joins (unexpected)
--- FAIL: AssociationsTest#test_assign_belongs_to_cpk_model_by_id_attribute (unexpected)
--- FAIL: CalculationsTest#test_should_get_maximum_of_field (unexpected)
--- FAIL: FinderTest#test_find_by_one_attribute_bang (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_validation_does_not_validate_stale_association_target (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_index_length (unexpected)
--- FAIL: BasicsTest#test_when_ignored_attribute_is_loaded,_cast_type_should_be_preferred_over_DB_type (unexpected)
--- FAIL: CalculationsTest#test_ids_for_a_composite_primary_key_with_scope (unexpected)
--- FAIL: BasicsTest#test_marshal_new_record_round_trip (unexpected)
--- FAIL: FinderTest#test_first_failing (unexpected)
--- FAIL: BasicsTest#test_all (unexpected)
--- FAIL: RelationTest#test_multivalue_where (unexpected)
--- FAIL: EagerAssociationTest#test_eager_loaded_has_many_association_without_primary_key (unexpected)
--- FAIL: DefaultStringsTest#test_default_strings (unexpected)
--- FAIL: RelationTest#test_references_triggers_eager_loading (unexpected)
--- FAIL: FinderTest#test_exists_should_reference_correct_aliases_while_joining_tables_of_has_many_through_association (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15327658/activerecord/1716185526264/1716193340737)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15340204?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15340204?buildTab=artifacts#/activerecord) on master @ [d146ecff6f687e438706cf63591cafca60cc116d](https://github.com/cockroachdb/cockroach/commits/d146ecff6f687e438706cf63591cafca60cc116d):


```
--- FAIL: InheritanceTest#test_alt_becomes_works_with_sti (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_association_through_a_belongs_to_association_where_the_association_doesnt_exist (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_attributes_with_overridden_types_keep_their_type_when_a_default_value_is_configured_separately (unexpected)
--- FAIL: AttributeMethodsTest#test_inherited_custom_accessors_with_reserved_names (unexpected)
--- FAIL: CalculationsTest#test_should_count_scoped_select_with_options (unexpected)
--- FAIL: InheritanceTest#test_alt_update_all_within_inheritance (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_through_association_with_joins (unexpected)
--- FAIL: RelationTest#test_find_or_initialize_by (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_one_through_join_model_with_conditions_on_the_through (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_through_scope_with_includes (unexpected)
--- FAIL: PreloaderTest#test_some_already_loaded_associations (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_works_with_model_called_Record (unexpected)
--- FAIL: CalculationsTest#test_should_average_arel_attribute (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_field_having_condition_from_select (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_one_dependent_does_not_destroy_dependent (unexpected)
--- FAIL: EagerAssociationTest#test_eager_has_many_with_association_inheritance (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_composite_primary_key_join_table (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnNewRecord#test_autosave_new_record_on_has_one_can_be_disabled_per_relationship (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_loading_cpk_association_with_unpersisted_owner (unexpected)
--- FAIL: HasOneAssociationsTest#test_association_keys_bypass_attribute_protection (unexpected)
--- FAIL: AttributeMethodsTest#test_converted_values_are_returned_after_assignment (unexpected)
--- FAIL: AttributeMethodsTest#test_write_nil_to_time_attribute (unexpected)
--- FAIL: HasManyAssociationsTest#test_counting_with_counter_sql (unexpected)
--- FAIL: CalculationsTest#test_should_get_maximum_of_field (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_multiple_fields (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_against_given_relation (unexpected)
--- FAIL: CalculationsTest#test_minimum_and_maximum_on_non_numeric_type (unexpected)
--- FAIL: RelationTest#test_select_with_from_includes_quoted_original_table_name (unexpected)
--- FAIL: IntegrationTest#test_cache_version_changes_with_versioning_on (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_build_then_save_with_has_one_inverse (unexpected)
--- FAIL: EagerAssociationTest#test_count_eager_with_has_many_and_limit_and_high_offset (unexpected)
--- FAIL: CalculationsTest#test_distinct_is_honored_when_used_with_count_operation_after_group (unexpected)
--- FAIL: HasOneAssociationsTest#test_destroyed_by_association_set_in_child_destroy_callback_on_parent_destroy (unexpected)
--- FAIL: BelongsToAssociationsTest#test_composite_primary_key_malformed_association_class (unexpected)
--- FAIL: RelationTest#test_finding_with_cross_table_order_and_limit (unexpected)
--- FAIL: RelationTest#test_automatically_added_where_references (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependent_association_respects_optional_hash_conditions_on_delete (unexpected)
--- FAIL: RelationTest#test_find_by_doesn't_have_implicit_ordering (unexpected)
--- FAIL: HasManyAssociationsTest#test_finding_with_different_class_name_and_order (unexpected)
--- FAIL: PostgresqlEnumTest#test_enum_mapping (unexpected)
--- FAIL: DefaultBinaryTest#test_default_varbinary_string (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_association_callback_ordering (unexpected)
--- FAIL: RelationTest#test_create_or_find_by_should_not_raise_due_to_validation_errors (unexpected)
--- FAIL: HasManyAssociationsTest#test_key_ensuring_owner_was_is_not_valid_without_dependent_option (unexpected)
--- FAIL: HasManyAssociationsTest#test_counter_cache_updates_in_memory_after_update_with_inverse_of_enabled (unexpected)
--- FAIL: BelongsToAssociationsTest#test_build_with_block (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroy_all_on_association_clears_scope (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15340204/activerecord/1716272111148/1716278984174)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15356582?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15356582?buildTab=artifacts#/activerecord) on master @ [8156d095961d7271df61fe2e2f716ef072bc5f96](https://github.com/cockroachdb/cockroach/commits/8156d095961d7271df61fe2e2f716ef072bc5f96):


```
--- FAIL: ReflectionTest#test_class_for_source_type (unexpected)
--- FAIL: AttributeMethodsTest#test_allocated_objects_can_be_inspected (unexpected)
--- FAIL: AttributeMethodsTest#test_declared_affixed_attribute_method_affects_respond_to?_and_method_missing (unexpected)
--- FAIL: HasOneAssociationsTest#test_can_marshal_has_one_association_with_nil_target (unexpected)
--- FAIL: BasicsTest#test_successful_comparison_of_like_class_records (unexpected)
--- FAIL: AttributeMethodsTest#test_accessed_fields (unexpected)
--- FAIL: AttributeMethodsTest#test_write_attribute_raises_ActiveModel::MissingAttributeError_when_the_attribute_does_not_exist (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_collection_size_from_params (unexpected)
--- FAIL: AttributeMethodsTest#test_respond_to?_with_a_custom_primary_key (unexpected)
--- FAIL: InheritanceAttributeTest#test_inheritance_new_with_subclass_as_default (unexpected)
--- FAIL: ReflectionTest#test_validate_association_if_autosave (unexpected)
--- FAIL: ModulesTest#test_associations_spanning_cross_modules (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_association_replaces_existing_without_dependent_option (unexpected)
--- FAIL: CockroachDB::LeftOuterJoinAssociationTest#test_does_not_override_select (unexpected)
--- FAIL: BelongsToAssociationsTest#test_assignment_before_child_saved_with_primary_key (unexpected)
--- FAIL: EnumTest#test_deserialize_enum_value_to_original_hash_key (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_caching_of_columns (unexpected)
--- FAIL: BasicsTest#test_table_name_guesses_with_prefixes_and_suffixes (unexpected)
--- FAIL: PostgresqlIntervalTest#test_interval_type_cast_string_and_numeric_from_user (unexpected)
--- FAIL: BasicsTest#test_#connecting_to_with_role (unexpected)
--- FAIL: BasicsTest#test_using_table_name_qualified_column_names_unless_having_SELECT_list_explicitly (unexpected)
--- FAIL: PersistenceTest#test_update_many_with_array_of_active_record_base_objects (unexpected)
--- FAIL: BasicsTest#test_.columns_hash_raises_an_error_if_the_record_has_an_empty_table_name (unexpected)
--- FAIL: ReflectionTest#test_never_validate_association_if_explicit (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_touch_option_on_update (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_touch_option_on_empty_update (unexpected)
--- FAIL: AttributeMethodsTest#test_declared_suffixed_attribute_method_affects_respond_to?_and_method_missing (unexpected)
--- FAIL: BelongsToAssociationsTest#test_create! (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_an_overridden_original_method_from_a_module_issues_a_deprecation (unexpected)
--- FAIL: ValidationsTest#test_numericality_validation_with_mutation (unexpected)
--- FAIL: BelongsToAssociationsTest#test_skips_parent_presence_check_if_parent_has_not_changed (unexpected)
--- FAIL: ModulesTest#test_module_spanning_associations (unexpected)
--- FAIL: PersistenceTest#test_build_through_factory_with_block (unexpected)
--- FAIL: PersistenceTest#test_update_columns_should_not_use_setter_method (unexpected)
--- FAIL: PersistenceTest#test_it_is_possible_to_update_parts_of_the_query_constraints_config (unexpected)
--- FAIL: DelegatedTypeTest#test_delegated_type_predicates_with_custom_foreign_type (unexpected)
--- FAIL: AttributeMethodsTest#test_on-the-fly_super-invokable_generated_attribute_predicates_via_method_missing (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_touch_with_reassigning (unexpected)
--- FAIL: PersistenceTest#test_update_attribute_uses_query_constraints_config (unexpected)
--- FAIL: EnumTest#test_validation_with_'validate:_true'_option (unexpected)
--- FAIL: ActiveRecord::Migration::ForeignKeyChangeColumnTest#test_rename_reference_column_of_child_table (unexpected)
--- FAIL: AttributeMethodsTest#test_boolean_attributes_writing_and_reading (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_include_uses_array_include_after_loaded (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_collection_size_from_build (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_a_time_zone-aware_time_in_the_current_time_zone (unexpected)
--- FAIL: AttributeMethodsTest#test_YAML_dumping_a_record_with_time_zone-aware_attribute (unexpected)
--- FAIL: BelongsToAssociationsTest#test_assigning_belongs_to_on_destroyed_object (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15356582/activerecord/1716358315629/1716368453750)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15371548?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15371548?buildTab=artifacts#/activerecord) on master @ [c580e634736b2d2b6da544eecf16664d3caca740](https://github.com/cockroachdb/cockroach/commits/c580e634736b2d2b6da544eecf16664d3caca740):


```
--- FAIL: RelationTest#test_reload (unexpected)
--- FAIL: RelationTest#test_find_all_with_multiple_should_use_and (unexpected)
--- FAIL: AttributeMethodsTest#test_time_zone-aware_attribute_saved (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_through_association_readonly_should_be_false (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_an_overridden_original_method_from_a_module_issues_a_deprecation (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_the_same_alias_as_parent_doesn't_issue_a_deprecation (unexpected)
--- FAIL: RelationTest#test_order_doesnt_trigger_eager_loading_when_ordering_using_the_owner_table_and_hash_syntax (unexpected)
--- FAIL: ModulesTest#test_module_table_name_prefix_with_global_prefix (unexpected)
--- FAIL: RelationTest#test_loaded_relations_cannot_be_mutated_by_extending! (unexpected)
--- FAIL: AttributeMethodsTest#test_method_overrides_in_multi-level_subclasses (unexpected)
--- FAIL: RelationTest#test_find_id (unexpected)
--- FAIL: RelationTest#test_references_triggers_eager_loading (unexpected)
--- FAIL: AttributeMethodsTest#test_time_zone-aware_attributes_do_not_recurse_infinitely_on_invalid_values (unexpected)
--- FAIL: RelationTest#test_reorder_deduplication (unexpected)
--- FAIL: RelationTest#test_create (unexpected)
--- FAIL: AttributeMethodsTest#test_string_attribute_predicate (unexpected)
--- FAIL: RelationTest#test_loading_with_one_association_with_non_preload (unexpected)
--- FAIL: RelationTest#test_left_outer_joins_with_blank_value (unexpected)
--- FAIL: RelationTest#test_delegations_do_not_leak_to_other_classes (unexpected)
--- FAIL: RelationTest#test_where_relation_with_delegated_ar_object (unexpected)
--- FAIL: RelationTest#test_loaded_relations_cannot_be_mutated_by_single_value_methods (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_marshal_dump (unexpected)
--- FAIL: RelationTest#test_find_all_using_where_twice_should_or_the_relation (unexpected)
--- FAIL: RelationTest#test_reverse_order_with_multiargument_function (unexpected)
--- FAIL: BasicsTest#test_table_name_for_base_class (unexpected)
--- FAIL: AttributeMethodsTest#test_write_attribute_can_write_aliased_attributes_as_well (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_belongs_to_with_dependent_destroy (unexpected)
--- FAIL: CallbacksOnDestroyUpdateActionRaceTest#test_trigger_on_update_where_row_was_deleted (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_time_zone-aware_attribute_in_other_time_zone (unexpected)
--- FAIL: BasicsTest#test_generated_association_methods_module_name (unexpected)
--- FAIL: BasicsTest#test_previously_persisted_returns_boolean (unexpected)
--- FAIL: BasicsTest#test_connection_in_local_time (unexpected)
--- FAIL: BasicsTest#test_all (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_collection_singular_ids_setter (unexpected)
--- FAIL: BasicsTest#test_initialize_with_attributes (unexpected)
--- FAIL: AttributeMethodsTest#test_inherited_custom_accessors_with_reserved_names (unexpected)
--- FAIL: BasicsTest#test_singular_table_name_guesses_with_prefixes_and_suffixes (unexpected)
--- FAIL: BasicsTest#test_dup_does_not_copy_associations (unexpected)
--- FAIL: RelationTest#test_finding_with_subquery_with_eager_loading_in_where (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_that_should_not_be_fully_merged (unexpected)
--- FAIL: BasicsTest#test_touch_should_raise_error_on_a_new_object (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_callbacks_firing_order_on_create (unexpected)
--- FAIL: BasicsTest#test_clone_of_new_object_marks_attributes_as_dirty (unexpected)
--- FAIL: BasicsTest#test_benchmark_with_use_silence (unexpected)
--- FAIL: BasicsTest#test_preserving_time_objects (unexpected)
--- FAIL: RelationTest#test_find_all_using_where_with_relation (unexpected)
--- FAIL: RelationTest#test_only (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15371548/activerecord/1716444618094/1716452956334)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15386474?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15386474?buildTab=artifacts#/activerecord) on master @ [c7f16c96d4db6ef97085f972f094c7e4e7c1518f](https://github.com/cockroachdb/cockroach/commits/c7f16c96d4db6ef97085f972f094c7e4e7c1518f):


```
--- FAIL: ReflectionTest#test_reflection_should_not_raise_error_when_compared_to_other_object (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_no_pk_join_model_callbacks (unexpected)
--- FAIL: AssociationsTest#test_using_limitable_reflections_helper (unexpected)
--- FAIL: RelationTest#test_finding_with_conditions (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_find_in_association_with_options (unexpected)
--- FAIL: NumericalityValidationTest#test_virtual_attribute_without_precision (unexpected)
--- FAIL: AssociationCallbacksTest#test_has_and_belongs_to_many_add_callback (unexpected)
--- FAIL: RelationTest#test_unscope_specific_where_value (unexpected)
--- FAIL: InheritanceTest#test_new_with_complex_inheritance (unexpected)
--- FAIL: PersistenceTest#test_build_through_factory_with_block (unexpected)
--- FAIL: RelationTest#test_create_or_find_by_with_block (unexpected)
--- FAIL: PersistenceTest#test_update_columns_should_raise_exception_if_new_record (unexpected)
--- FAIL: RelationTest#test_automatically_added_where_references (unexpected)
--- FAIL: ReflectionTest#test_join_table_can_be_overridden (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_creation_respects_hash_condition (unexpected)
--- FAIL: RelationTest#test_size_with_eager_loading_and_custom_select_and_order (unexpected)
--- FAIL: AssociationsTest#test_include_with_order_works (unexpected)
--- FAIL: ReflectionTest#test_class_for_class_name (unexpected)
--- FAIL: UniquenessValidationTest#test_validate_uniqueness_with_columns_which_are_sql_keywords (unexpected)
--- FAIL: MultiParameterAttributeTest#test_multiparameter_attributes_on_time_with_seconds_will_ignore_date_if_empty (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_of_stis_with_multiple_references (unexpected)
--- FAIL: PersistenceTest#test_instantiate_creates_a_new_instance (unexpected)
--- FAIL: PersistenceTest#test_persisted_returns_boolean (unexpected)
--- FAIL: RelationTest#test_find_by_with_multi-arg_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: RelationTest#test_select_with_subquery_in_from_does_not_use_original_table_name (unexpected)
--- FAIL: RelationTest#test_#where_with_set (unexpected)
--- FAIL: PersistenceTest#test_duped_becomes_persists_changes_from_the_original (unexpected)
--- FAIL: PersistenceTest#test_update_attribute_for_aborted_callback! (unexpected)
--- FAIL: DefaultScopingTest#test_create_with_empty_hash_will_not_reset (unexpected)
--- FAIL: AssociationsTest#test_nullify_composite_has_many_through_association (unexpected)
--- FAIL: PersistenceTest#test_update_attribute_with_one_updated (unexpected)
--- FAIL: DefaultScopingTest#test_default_scope_with_references_works_with_find_by (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_nulls_not_distinct (unexpected)
--- FAIL: ReflectionTest#test_columns_are_returned_in_the_order_they_were_declared (unexpected)
--- FAIL: AttributeMethodsTest#test_time_zone-aware_attribute_saved (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_has_many_sti (unexpected)
--- FAIL: AttributeMethodsTest#test_string_attribute_predicate (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_scope_can_unscope (unexpected)
--- FAIL: TestAutosaveAssociationValidationsOnABelongsToAssociation#test_validations_still_fire_on_unchanged_association_with_custom_validation_context (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_has_one_reflection (unexpected)
--- FAIL: AssociationsTest#test_query_constraints_that_dont_include_the_primary_key_raise (unexpected)
--- FAIL: PersistenceTest#test_becomes_keeps_extra_attributes (unexpected)
--- FAIL: PostgresqlByteaTest#test_write_value (unexpected)
--- FAIL: DDLTest#test_column_defaults (unexpected)
--- FAIL: ReflectionTest#test_includes_accepts_symbols (unexpected)
--- FAIL: RelationTest#test_unscope_with_blank_value (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_raises_UnknownPrimaryKey_when_a_model_has_no_primary_key (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15386474/activerecord/1716531050213/1716540658315)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15399674?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15399674?buildTab=artifacts#/activerecord) on master @ [b5081e997adaf5950c9b85dfb6655f226bf16d29](https://github.com/cockroachdb/cockroach/commits/b5081e997adaf5950c9b85dfb6655f226bf16d29):


```
--- FAIL: AttributeMethodsTest#test_attribute_predicates_respect_access_control (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_assign_ids_ignoring_blanks (unexpected)
--- FAIL: AttributeMethodsTest#test_typecast_attribute_from_select_to_true (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_scoped_field (unexpected)
--- FAIL: CalculationsTest#test_#skip_query_cache!_for_a_grouped_calculation (unexpected)
--- FAIL: AssociationsTest#test_eager_loading_should_not_change_count_of_children (unexpected)
--- FAIL: ReflectionTest#test_reflect_on_association_accepts_symbols (unexpected)
--- FAIL: ReflectionTest#test_aggregation_reflection (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_caching_of_columns (unexpected)
--- FAIL: YamlSerializationTest#test_cast_types_are_not_changed_on_round_trip (unexpected)
--- FAIL: CalculationsTest#test_select_avg_with_group_by_as_virtual_attribute_with_sql (unexpected)
--- FAIL: CalculationsTest#test_select_avg_with_joins_and_group_by_as_virtual_attribute_with_sql (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_destroying_many (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_multiple_columns_and_selection_clause (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_selects_all_columns_by_default (unexpected)
--- FAIL: CalculationsTest#test_should_sum_scoped_field_with_conditions (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_multiple_stis_and_order (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_include_returns_false_for_non_matching_record_to_verify_scoping (unexpected)
--- FAIL: AttributeMethodsTest#test_hash_content (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_qualified_column_name (unexpected)
--- FAIL: ReflectionTest#test_association_reflection_in_modules (unexpected)
--- FAIL: ReflectionTest#test_reflection_klass_for_nested_class_name (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_time_zone-aware_attribute_with_a_string (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_marshal_dump (unexpected)
--- FAIL: AttributeMethodsTest#test_respond_to? (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_get_ids (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_distinct_order_preserved (unexpected)
--- FAIL: AssociationsTest#test_query_constraints_that_dont_include_the_primary_key_raise (unexpected)
--- FAIL: AttributeMethodsTest#test_declared_affixed_attribute_method_affects_respond_to?_and_method_missing (unexpected)
--- FAIL: CalculationsTest#test_should_limit_calculation_with_offset (unexpected)
--- FAIL: ReflectionTest#test_belongs_to_inferred_foreign_key_from_assoc_name (unexpected)
--- FAIL: CalculationsTest#test_should_not_use_alias_for_grouped_field (unexpected)
--- FAIL: AssociationsTest#test_cpk_model_has_many_records_by_id_attribute (unexpected)
--- FAIL: CalculationsTest#test_sum_expression_returns_zero_when_no_records_to_sum (unexpected)
--- FAIL: ReflectionTest#test_join_table_can_be_overridden (unexpected)
--- FAIL: AttributeMethodsTest#test_respond_to?_with_a_custom_primary_key (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_a_time_zone-aware_attribute_interprets_time_zone-unaware_string_in_time_zone (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_preloading_across_has_one_constrains_loaded_records (unexpected)
--- FAIL: CalculationsTest#test_pluck_in_relation (unexpected)
--- FAIL: AttributeMethodsTest#test_#id_value_alias_returns_id_column_only_for_composite_primary_key_models (unexpected)
--- FAIL: AttributeMethodsTest#test_caching_a_nil_primary_key (unexpected)
--- FAIL: ReflectionTest#test_scope_chain_does_not_interfere_with_hmt_with_polymorphic_case (unexpected)
--- FAIL: AttributeMethodsTest#test_global_methods_are_overwritten_when_subclassing (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_time_zone-aware_read_attribute (unexpected)
--- FAIL: CalculationsTest#test_distinct_joins_count_with_order_and_limit_and_offset (unexpected)
--- FAIL: CalculationsTest#test_count_takes_attribute_type_precedence_over_database_type (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_preloading_across_has_one_through_constrains_loaded_records (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15399674/activerecord/1716618015400/1716620885729)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15409076?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15409076?buildTab=artifacts#/activerecord) on master @ [e999ede1ed2bc87b34e01eedba1da58e09b8c4d3](https://github.com/cockroachdb/cockroach/commits/e999ede1ed2bc87b34e01eedba1da58e09b8c4d3):


```
--- FAIL: RelationTest#test_create_or_find_by_with_non_unique_attributes (unexpected)
--- FAIL: RelationTest#test_first_or_create_bang_with_invalid_block (unexpected)
--- FAIL: AttributeMethodsTest#test_inherited_custom_accessors_with_reserved_names (unexpected)
--- FAIL: AssociationsTest#test_force_reload (unexpected)
--- FAIL: HasOneThroughDisableJoinsAssociationsTest#test_nil_on_disable_joins_through (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_grouped_association_with_foreign_key_option (unexpected)
--- FAIL: HasOneAssociationsTest#test_restrict_with_error_with_locale (unexpected)
--- FAIL: BasicsTest#test_bignum_pk (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_first_after_reset (unexpected)
--- FAIL: BasicsTest#test_has_attribute (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_with_has_one_through_an_sti_with_after_initialize (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_nonexistent_attribute (unexpected)
--- FAIL: PersistenceTest#test_save_for_record_with_only_primary_key_that_is_provided (unexpected)
--- FAIL: BelongsToAssociationsTest#test_default_scope_on_relations_is_not_cached (unexpected)
--- FAIL: HasOneAssociationsTest#test_dangerous_association_name_raises_ArgumentError (unexpected)
--- FAIL: AttributeMethodsTest#test_user-defined_text_attribute_predicate (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_with_create (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_with_conditions_join_model_with_include (unexpected)
--- FAIL: BasicsTest#test_preserving_date_objects (unexpected)
--- FAIL: CalculationsTest#test_should_count_manual_select_with_count_all (unexpected)
--- FAIL: NumericalityValidationTest#test_virtual_attribute_with_precision_round_half_even (unexpected)
--- FAIL: EagerAssociationTest#test_loading_association_with_same_table_joins (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit_and_high_offset (unexpected)
--- FAIL: BasicsTest#test_create_after_initialize_with_block (unexpected)
--- FAIL: BasicsTest#test_ignored_columns_have_no_attribute_methods (unexpected)
--- FAIL: AttributeMethodsTest#test_caching_a_nil_primary_key (unexpected)
--- FAIL: BelongsToAssociationsTest#test_with_select (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_escaped_characters (unexpected)
--- FAIL: RelationTest#test_loaded_relations_cannot_be_mutated_by_extending! (unexpected)
--- FAIL: BelongsToAssociationsTest#test_assignment_before_child_saved (unexpected)
--- FAIL: FinderTest#test_model_class_responds_to_third_to_last_bang (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_no_pk_join_table_delete (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_saving_multiple_relationships (unexpected)
--- FAIL: HasOneAssociationsTest#test_creation_failure_due_to_new_record_should_raise_error (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_self_type_mismatch (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_belongs_to_association_associated_by_a_composite_query_constraints (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_many_should_return_true_if_more_than_one (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_find_in_association (unexpected)
--- FAIL: CalculationsTest#test_count_with_arel_star (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_join_model_ignores_default_includes (unexpected)
--- FAIL: PersistenceTest#test_persisted_returns_boolean (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_keys_bypass_attribute_protection (unexpected)
--- FAIL: PersistenceTest#test_update (unexpected)
--- FAIL: HasManyAssociationsTest#test_select_with_block_and_dirty_target (unexpected)
--- FAIL: HasManyAssociationsTest#test_include_returns_false_for_non_matching_record_to_verify_scoping (unexpected)
--- FAIL: BelongsToAssociationsTest#test_building_the_belonging_object_with_explicit_sti_base_class (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_get_has_many_through_belongs_to_ids_with_conditions (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15409076/activerecord/1716704023502/1716711435665)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15417787?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15417787?buildTab=artifacts#/activerecord) on master @ [0ea8b622c5354a891cfb867d8d872a0746ba77b3](https://github.com/cockroachdb/cockroach/commits/0ea8b622c5354a891cfb867d8d872a0746ba77b3):


```
--- FAIL: RelationTest#test_relation_join_method (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_method? (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_a_time_zone-aware_attribute_interprets_time_zone-unaware_string_in_time_zone (unexpected)
--- FAIL: PersistenceTest#test_update_many_with_duplicated_ids (unexpected)
--- FAIL: InheritanceTest#test_new_without_storing_full_sti_class (unexpected)
--- FAIL: HasManyAssociationsTest#test_collection_not_empty_after_building (unexpected)
--- FAIL: AttributeMethodsTest#test_converted_values_are_returned_after_assignment (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_one_should_return_false_if_more_than_one (unexpected)
--- FAIL: CalculationsTest#test_should_get_maximum_of_arel_attribute (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_enum_works_properly (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_many_on_loaded_association_should_not_use_query (unexpected)
--- FAIL: CalculationsTest#test_count_with_aliased_attribute (unexpected)
--- FAIL: IntegrationTest#test_to_param_should_return_string (unexpected)
--- FAIL: CockroachDB::AdapterTestWithoutTransaction#test_reset_empty_table_with_custom_pk_sequence (unexpected)
--- FAIL: RelationTest#test_create_or_find_by_should_not_raise_due_to_validation_errors (unexpected)
--- FAIL: PersistenceTest#test_update_attribute_for_updated_at_on (unexpected)
--- FAIL: RelationTest#test_finding_with_order_by_aliased_attributes (unexpected)
--- FAIL: BasicsTest#test_initialize_with_attributes (unexpected)
--- FAIL: RelationTest#test_typecasting_where_with_array (unexpected)
--- FAIL: CalculationsTest#test_distinct_joins_count_with_order_and_limit (unexpected)
--- FAIL: InheritanceTest#test_class_with_blank_sti_name (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_many_via_block (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates_edge_case (unexpected)
--- FAIL: RelationTest#test_order_with_hash_and_symbol_generates_the_same_sql (unexpected)
--- FAIL: PersistenceTest#test_save_with_duping_of_destroyed_object (unexpected)
--- FAIL: BelongsToAssociationsTest#test_natural_assignment_to_nil_with_primary_key (unexpected)
--- FAIL: CalculationsTest#test_distinct_count_with_order_and_offset (unexpected)
--- FAIL: CalculationsTest#test_pluck_auto_table_name_prefix (unexpected)
--- FAIL: BelongsToAssociationsTest#test_assigning_belongs_to_on_destroyed_object (unexpected)
--- FAIL: RelationTest#test_size_with_eager_loading_and_custom_select_and_order (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_present_with_booleans (unexpected)
--- FAIL: AssociationsTest#test_querying_by_whole_associated_records_using_query_constraints (unexpected)
--- FAIL: IntegrationTest#test_cache_version_format_is_not_too_precise (unexpected)
--- FAIL: DirtyTest#test_virtual_attribute_will_change (unexpected)
--- FAIL: RelationTest#test_destroy_by (unexpected)
--- FAIL: RelationTest#test_reverse_order_with_function_other_predicates (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attribute_when_true (unexpected)
--- FAIL: RelationTest#test_delegations_do_not_leak_to_other_classes (unexpected)
--- FAIL: HasManyAssociationsTest#test_adding_a_mismatch_class (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_keys_bypass_attribute_protection (unexpected)
--- FAIL: PersistenceTest#test_populates_non_primary_key_autoincremented_column_for_a_cpk_model (unexpected)
--- FAIL: InheritanceTest#test_alt_update_all_within_inheritance (unexpected)
--- FAIL: BasicsTest#test_has_attribute_with_symbol (unexpected)
--- FAIL: CalculationsTest#test_ids_with_eager_load (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_counter_after_touch (unexpected)
--- FAIL: RelationTest#test_unscope_with_table_name_qualified_hash (unexpected)
--- FAIL: RelationTest#test_find_all_with_join (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15417787/activerecord/1716789923118/1716797837271)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15427543?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15427543?buildTab=artifacts#/activerecord) on master @ [0ea8b622c5354a891cfb867d8d872a0746ba77b3](https://github.com/cockroachdb/cockroach/commits/0ea8b622c5354a891cfb867d8d872a0746ba77b3):


```
--- FAIL: HasManyAssociationsTest#test_create_the_association_with_an_array (unexpected)
--- FAIL: PersistenceTest#test_destroy_new_record (unexpected)
--- FAIL: AttributeMethodsTest#test_integers_as_nil (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit (unexpected)
--- FAIL: BasicsTest#test_hashing (unexpected)
--- FAIL: HasManyAssociationsTest#test_get_ids (unexpected)
--- FAIL: HasOneAssociationsTest#test_nullification_on_association_change (unexpected)
--- FAIL: DDLTest#test_column_defaults (unexpected)
--- FAIL: HasManyAssociationsTest#test_set_ids_for_association_on_new_record_applies_association_correctly (unexpected)
--- FAIL: ActiveRecord::DupTest#test_dup_record_not_persisted_after_rollback_transaction (unexpected)
--- FAIL: ActiveRecord::DupTest#test_dup_with_default_scope (unexpected)
--- FAIL: PostGISTest#test_spatial_factory_retrieval (unexpected)
--- FAIL: ModulesTest#test_compute_type_can_infer_class_name_of_sibling_inside_module (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_indexes_for_existent_table (unexpected)
--- FAIL: EagerAssociationTest#test_include_has_one_using_primary_key (unexpected)
--- FAIL: HasOneAssociationsTest#test_proxy_assignment (unexpected)
--- FAIL: RelationTest#test_find_by!_with_hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: BasicsTest#test_preserving_time_objects_with_utc_time_conversion_to_default_timezone_local (unexpected)
--- FAIL: RelationTest#test_reorder_deduplication (unexpected)
--- FAIL: EagerAssociationTest#test_loading_with_no_associations (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_order_string_with_unquoted_table_name (unexpected)
--- FAIL: EagerAssociationTest#test_calculate_with_string_in_from_and_eager_loading (unexpected)
--- FAIL: PersistenceTest#test_update_attribute_uses_query_constraints_config (unexpected)
--- FAIL: HasManyAssociationsTest#test_include_returns_false_for_non_matching_record_to_verify_scoping (unexpected)
--- FAIL: BasicsTest#test_quote (unexpected)
--- FAIL: BasicsTest#test_using_table_name_qualified_column_names_unless_having_SELECT_list_explicitly (unexpected)
--- FAIL: DefaultScopingTest#test_default_scope_as_block_referencing_scope (unexpected)
--- FAIL: ValidationsTest#test_exception_on_create_bang_with_block (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_index_columns_in_right_order (unexpected)
--- FAIL: RelationTest#test_unscope_with_blank_value (unexpected)
--- FAIL: TouchLaterTest#test_touch_touches_immediately (unexpected)
--- FAIL: HasOneAssociationsTest#test_building_the_associated_object_with_an_unrelated_type (unexpected)
--- FAIL: RelationTest#test_extracted_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_first_nth_or_last_on_association_should_not_load_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_composite_primary_key_malformed_association_class (unexpected)
--- FAIL: BasicsTest#test_find_by_slug_with_range (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_with_instance_dependent_scope (unexpected)
--- FAIL: NamedScopingTest#test_scope_with_kwargs (unexpected)
--- FAIL: BasicsTest#test_dont_clear_inheritance_column_when_setting_explicitly (unexpected)
--- FAIL: DefaultScopingTest#test_default_scope_with_all_queries_runs_on_reload (unexpected)
--- FAIL: SerializedAttributeTestWithYamlSafeLoad#test_serialized_column_should_unserialize_after_update_attribute (unexpected)
--- FAIL: BasicsTest#test_preserving_time_objects (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attributes_after_type_cast_on_a_date (unexpected)
--- FAIL: BasicsTest#test_find_reverse_ordered_last (unexpected)
--- FAIL: BasicsTest#test_default_values_are_deeply_dupped (unexpected)
--- FAIL: SerializationTest#test_serialize_should_allow_attribute_except_filtering (unexpected)
--- FAIL: PersistenceTest#test_increment_with_touch_updates_timestamps (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15427543/activerecord/1716876593652/1716884948778)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15444703?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15444703?buildTab=artifacts#/activerecord) on master @ [d5f825ce04362a75781b75e19cf429f2ed5ed61b](https://github.com/cockroachdb/cockroach/commits/d5f825ce04362a75781b75e19cf429f2ed5ed61b):


```
--- FAIL: FinderTest#test_find_by_one_attribute_that_is_an_aggregate (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_explicit_table_name_and_aggregate (unexpected)
--- FAIL: AttributeMethodsTest#test_set_attributes_with_a_block (unexpected)
--- FAIL: AttributeMethodsTest#test_accessed_fields (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_method? (unexpected)
--- FAIL: FinderTest#test_find_with_a_single_composite_primary_key_wrapped_in_an_array (unexpected)
--- FAIL: FinderTest#test_hash_condition_local_time_interpolation_with_default_timezone_utc (unexpected)
--- FAIL: AttributeMethodsTest#test_declared_affixed_attribute_method_affects_respond_to?_and_method_missing (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates_case_shown_in_doc (unexpected)
--- FAIL: FinderTest#test_find_with_hash_parameter (unexpected)
--- FAIL: StrictLoadingTest#test_strict_loading_with_has_one_through_does_not_prevent_creation_of_association (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_does_not_affect_reading_from_the_column (unexpected)
--- FAIL: AssociationsTest#test_include_with_order_works (unexpected)
--- FAIL: FinderTest#test_fifth_have_primary_key_order_by_default (unexpected)
--- FAIL: I18nGenerateMessageValidationTest#test_generate_message_invalid_with_custom_message (unexpected)
--- FAIL: FinderTest#test_find_by_with_alias (unexpected)
--- FAIL: PrimaryKeysTest#test_id_column_that_is_not_primary_key (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnNewRecord#test_autosave_new_record_on_belongs_to_can_be_disabled_per_relationship (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_for_inspect_with_a_date (unexpected)
--- FAIL: FinderTest#test_last_with_integer_and_order_should_use_sql_limit (unexpected)
--- FAIL: FinderTest#test_find_by_on_attribute_that_is_a_reserved_word (unexpected)
--- FAIL: FinderTest#test_exists_with_polymorphic_relation (unexpected)
--- FAIL: AssociationsTest#test_assign_composite_foreign_key_belongs_to_association_with_autosave (unexpected)
--- FAIL: FinderTest#test_#skip_query_cache!_for_#exists?_with_a_limited_eager_load (unexpected)
--- FAIL: FinderTest#test_exists_passing_active_record_object_is_not_permitted (unexpected)
--- FAIL: AssociationsTest#test_has_many_association_with_composite_foreign_key_loads_records (unexpected)
--- FAIL: FinderTest#test_find_with_string (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_is_precise_enough (unexpected)
--- FAIL: AssociationsTest#test_belongs_to_a_model_with_composite_primary_key_uses_composite_pk_in_sql (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_a_time_zone-aware_attribute_to_a_blank_string_returns_nil (unexpected)
--- FAIL: AttributeMethodsTest#test_instance_methods_should_be_defined_on_the_base_class (unexpected)
--- FAIL: FinderTest#test_find_with_ids_with_no_id_passed (unexpected)
--- FAIL: AttributeMethodsTest#test_converted_values_are_returned_after_assignment (unexpected)
--- FAIL: EnumTest#test_serializable?_with_large_number_label (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_overloaded_default_but_keeping_its_own_type (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_for_existing_record_with_updated_at (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_open_ended_range (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_an_overridden_original_method_along_with_an_overridden_alias_method_in_a_parent_class_doesn't_issue_a_deprecation (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_names_with_a_custom_select (unexpected)
--- FAIL: EnumTest#test_option_names_can_be_used_as_label (unexpected)
--- FAIL: TransactionTest#test_rolling_back_in_a_callback_rollbacks_before_save (unexpected)
--- FAIL: EnumTest#test_overloaded_default_by_:_default (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_names_on_a_new_record (unexpected)
--- FAIL: FinderTest#test_find_by_id_with_large_number (unexpected)
--- FAIL: FinderTest#test_model_class_responds_to_first_bang (unexpected)
--- FAIL: AssociationsTest#test_preloads_model_with_query_constraints_by_explicitly_configured_fk_and_pk (unexpected)
--- FAIL: FinderTest#test_bind_variables (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15444703/activerecord/1716963298586/1716971529610)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15449929?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15449929?buildTab=artifacts#/activerecord) on master @ [d5f825ce04362a75781b75e19cf429f2ed5ed61b](https://github.com/cockroachdb/cockroach/commits/d5f825ce04362a75781b75e19cf429f2ed5ed61b):


```
--- FAIL: HasManyThroughAssociationsTest#test_find_on_has_many_association_collection_with_include_and_conditions (unexpected)
--- FAIL: EagerAssociationTest#test_belongs_to_association_ignores_the_scoping (unexpected)
--- FAIL: ReflectionTest#test_foreign_key_is_inferred_from_model_name (unexpected)
--- FAIL: CockroachDB::LeftOuterJoinAssociationTest#test_does_not_override_select (unexpected)
--- FAIL: FinderTest#test_model_class_responds_to_second_bang (unexpected)
--- FAIL: CalculationsTest#test_should_get_maximum_of_arel_attribute (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_limit_and_offset (unexpected)
--- FAIL: HasOneAssociationsTest#test_building_the_associated_object_with_an_invalid_type (unexpected)
--- FAIL: CalculationsTest#test_should_sum_expression (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all_accepts_limited_parameters (unexpected)
--- FAIL: EagerAssociationTest#test_loading_association_with_same_table_joins (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_loading_cpk_association_with_unpersisted_owner (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependence_on_account (unexpected)
--- FAIL: FinderTest#test_find_by_records (unexpected)
--- FAIL: CalculationsTest#test_should_sum_arel_attribute (unexpected)
--- FAIL: HasManyAssociationsTest#test_depends_and_nullify_on_polymorphic_assoc (unexpected)
--- FAIL: HasManyAssociationsTest#test_dynamic_find_should_respect_association_order_for_through (unexpected)
--- FAIL: FinderTest#test_find_by_with_alias (unexpected)
--- FAIL: RelationTest#test_joins_with_string_array (unexpected)
--- FAIL: CockroachDB::MarshalSerializationTest#test_deserializing_rails_6_1_marshal_with_loaded_association_cache (unexpected)
--- FAIL: RelationTest#test_except (unexpected)
--- FAIL: HasManyAssociationsTest#test_transactions_when_adding_to_new_record (unexpected)
--- FAIL: RelationTest#test_joins_with_blank_value (unexpected)
--- FAIL: BasicsTest#test_preserving_time_objects_with_time_with_zone_conversion_to_default_timezone_local (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one (unexpected)
--- FAIL: CalculationsTest#test_ids_on_loaded_relation_with_scope (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_with_reflection_using_class_name_and_fixtures (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through (unexpected)
--- FAIL: TransactionInstrumentationTest#test_transaction_instrumentation_with_savepoints (unexpected)
--- FAIL: CalculationsTest#test_count_selected_arel_attribute_in_joined_table (unexpected)
--- FAIL: BasicsTest#test_dont_clear_inheritance_column_when_setting_explicitly (unexpected)
--- FAIL: BelongsToAssociationsTest#test_should_set_foreign_key_on_save! (unexpected)
--- FAIL: HasManyAssociationsTest#test_regular_create_on_has_many_when_parent_is_new_raises (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_source_scope (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnNewRecord#test_autosave_new_record_on_has_one_can_be_disabled_per_relationship (unexpected)
--- FAIL: IntegrationTest#test_cache_version_changes_with_versioning_on (unexpected)
--- FAIL: PersistenceTest#test_destroy_uses_query_constraints_config (unexpected)
--- FAIL: EagerAssociationTest#test_three_level_nested_preloading_does_not_raise_exception_when_association_does_not_exist (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_one_message_on_primary_key (unexpected)
--- FAIL: PersistenceTest#test_build_through_factory_with_block (unexpected)
--- FAIL: ReflectionTest#test_association_primary_key_uses_explicit_primary_key_option_as_first_priority (unexpected)
--- FAIL: ReflectionTest#test_scope_chain_does_not_interfere_with_hmt_with_polymorphic_case (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_is_precise_enough (unexpected)
--- FAIL: BasicsTest#test_when_ignored_attribute_is_loaded,_cast_type_should_be_preferred_over_DB_type (unexpected)
--- FAIL: BelongsToAssociationsTest#test_destroying_child_with_unloaded_parent_and_foreign_key_and_touch_is_possible_with_has_many_inversing (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_ordered_has_many_through (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_through_association_with_left_joins (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15449929/activerecord/1716994100586/1717001347196)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15454923?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15454923?buildTab=artifacts#/activerecord) on master @ [e5f5597b1c1570fff03ee6dfd4a1e8be65c0e008](https://github.com/cockroachdb/cockroach/commits/e5f5597b1c1570fff03ee6dfd4a1e8be65c0e008):


```
--- FAIL: BelongsToAssociationsTest#test_where_with_custom_primary_key (unexpected)
--- FAIL: AttributeMethodsTest#test_boolean_attributes_writing_and_reading (unexpected)
--- FAIL: FinderTest#test_last_on_loaded_relation_should_not_use_sql (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_counter_with_reassigning (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_associations_sum_on_columns (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_new_with_values_in_collection (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_grouped_by_function_with_table_alias (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_aggregate_having_one_mapping_and_key_value_being_attribute_value (unexpected)
--- FAIL: HasManyAssociationsTest#test_collection_empty_with_dirty_target (unexpected)
--- FAIL: FinderTest#test_fourth (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_removing_associations_on_destroy (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_before_save (unexpected)
--- FAIL: InheritanceTest#test_compute_type_no_method_error (unexpected)
--- FAIL: FinderTest#test_condition_hash_interpolation (unexpected)
--- FAIL: InheritanceTest#test_inheritance_find (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_cascaded_eager_association_loading_with_duplicated_includes (unexpected)
--- FAIL: HasManyAssociationsTest#test_has_many_associations_on_new_records_use_null_relations (unexpected)
--- FAIL: AttributeMethodsTest#test_#id_value_alias_is_not_defined_if_id_column_doesn't_exist (unexpected)
--- FAIL: EagerAssociationTest#test_exceptions_have_suggestions_for_fix (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_does_not_cache_has_many_association_subset_when_preloaded_with_a_through_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_has_many_without_counter_cache_option (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_belongs_to_with_dependent_delete_all (unexpected)
--- FAIL: InheritanceTest#test_abstract_class (unexpected)
--- FAIL: BelongsToAssociationsTest#test_creating_the_belonging_object_with_primary_key (unexpected)
--- FAIL: FinderTest#test_exists_with_includes_limit_and_empty_result (unexpected)
--- FAIL: HasManyAssociationsTest#test_counter_cache_updates_in_memory_after_create_with_overlapping_counter_cache_columns (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_adding_type_mismatch (unexpected)
--- FAIL: ActiveRecord::RelationTest#test_update_all_goes_through_normal_type_casting (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependence_on_account (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_include_method_in_has_and_belongs_to_many_association_should_return_true_for_instance_added_with_build (unexpected)
--- FAIL: CalculationsTest#test_should_count_manual_select_with_group_with_count_all (unexpected)
--- FAIL: AttributeMethodsTest#test_define_attribute_method_works_with_both_symbol_and_string (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_many_should_defer_to_collection_if_using_a_block (unexpected)
--- FAIL: HasManyAssociationsTest#test_dynamic_find_should_respect_association_order_for_through (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_loading_cpk_association_with_unpersisted_owner (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_preloading_empty_through_with_polymorphic_source_association (unexpected)
--- FAIL: FinderTest#test_exists_with_nil_arg (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_has_and_belongs_to_many_in_a_namespaced_model_pointing_to_a_non_namespaced_model (unexpected)
--- FAIL: BelongsToAssociationsTest#test_cant_save_readonly_association (unexpected)
--- FAIL: CalculationsTest#test_average_with_from_option (unexpected)
--- FAIL: FinderTest#test_find_by_empty_in_condition (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_both_parent_ids_set_when_saving_new (unexpected)
--- FAIL: CalculationsTest#test_group_alias_is_properly_quoted (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_association_with_non_numeric_foreign_key (unexpected)
--- FAIL: InheritanceTest#test_becomes_sets_variables_before_initialization_callbacks (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_a_item_which_is_not_in_the_collection (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_keys_bypass_attribute_protection (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15454923/activerecord/1717049737241/1717058941460)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15465774?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15465774?buildTab=artifacts#/activerecord) on master @ [64bd4bbac8f8d27a2fe4968404cc3d048480375f](https://github.com/cockroachdb/cockroach/commits/64bd4bbac8f8d27a2fe4968404cc3d048480375f):


```
--- FAIL: InheritanceTest#test_alt_becomes_works_with_sti (unexpected)
--- FAIL: InheritanceTest#test_compute_type_no_method_error (unexpected)
--- FAIL: InheritanceTest#test_compute_type_success (unexpected)
--- FAIL: BasicsTest#test_clear_cache_when_setting_table_name (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_adding_before_save (unexpected)
--- FAIL: InheritanceTest#test_compute_type_on_undefined_method (unexpected)
--- FAIL: ModulesTest#test_module_table_name_suffix (unexpected)
--- FAIL: BasicsTest#test_new_threads_get_default_the_default_connection_handler (unexpected)
--- FAIL: InheritanceTest#test_where_new_with_invalid_type (unexpected)
--- FAIL: BasicsTest#test_#connecting_to_with_prevent_writes (unexpected)
--- FAIL: InheritanceTest#test_descends_from_active_record (unexpected)
--- FAIL: ModulesTest#test_module_table_name_prefix_with_global_prefix (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_replace_on_duplicated_object (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_build_many_via_block_before_save (unexpected)
--- FAIL: InheritanceTest#test_eager_load_belongs_to_something_inherited (unexpected)
--- FAIL: BasicsTest#test_comparison_with_different_objects (unexpected)
--- FAIL: ModulesTest#test_module_spanning_associations (unexpected)
--- FAIL: BasicsTest#test_readonly_attributes_in_abstract_class_descendant (unexpected)
--- FAIL: InheritanceTest#test_inherits_custom_primary_key (unexpected)
--- FAIL: BasicsTest#test_set_table_name_with_inheritance (unexpected)
--- FAIL: BasicsTest#test_clone_of_new_object_marks_attributes_as_dirty (unexpected)
--- FAIL: BasicsTest#test_initialize_with_attributes (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_has_one_cpk_has_one_autosave_with_id (unexpected)
--- FAIL: BasicsTest#test_ignored_columns_have_no_attribute_methods (unexpected)
--- FAIL: BasicsTest#test_previously_persisted_returns_boolean (unexpected)
--- FAIL: BasicsTest#test_destroy_without_prepared_statement (unexpected)
--- FAIL: BasicsTest#test_table_exists (unexpected)
--- FAIL: BasicsTest#test_using_table_name_qualified_column_names_unless_having_SELECT_list_explicitly (unexpected)
--- FAIL: BasicsTest#test_attributes (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_adding_unsavable_association (unexpected)
--- FAIL: BasicsTest#test_#connected_to_many_cannot_be_called_on_anything_but_ActiveRecord::Base (unexpected)
--- FAIL: BasicsTest#test_dup_of_saved_object_marks_as_dirty_only_changed_attributes (unexpected)
--- FAIL: InheritanceTest#test_new_with_abstract_class (unexpected)
--- FAIL: BasicsTest#test_preserving_time_objects_with_time_with_zone_conversion_to_default_timezone_local (unexpected)
--- FAIL: BasicsTest#test_protected_environments_by_default_is_an_array_with_production (unexpected)
--- FAIL: BasicsTest#test_primary_key_with_no_id (unexpected)
--- FAIL: BasicsTest#test_cannot_call_connected_to_on_the_abstract_class_that_did_not_establish_the_connection (unexpected)
--- FAIL: InheritanceTest#test_class_with_blank_sti_name (unexpected)
--- FAIL: BasicsTest#test_auto_id (unexpected)
--- FAIL: BasicsTest#test_dup (unexpected)
--- FAIL: BasicsTest#test_bignum (unexpected)
--- FAIL: BasicsTest#test_limit_without_comma (unexpected)
--- FAIL: BasicsTest#test_marshal_round_trip (unexpected)
--- FAIL: InheritanceTest#test_destroy_all_within_inheritance (unexpected)
--- FAIL: BasicsTest#test_preserving_time_objects_with_utc_time_conversion_to_default_timezone_local (unexpected)
--- FAIL: ModulesTest#test_module_table_name_suffix_with_global_suffix (unexpected)
--- FAIL: BasicsTest#test_benchmark_with_use_silence (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15465774/activerecord/1717135460757/1717142599786)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15478988?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15478988?buildTab=artifacts#/activerecord) on master @ [d77b6e709e7d90625ff31bc0016eed8c7a6dc65a](https://github.com/cockroachdb/cockroach/commits/d77b6e709e7d90625ff31bc0016eed8c7a6dc65a):


```
--- FAIL: AttributeMethodsTest#test_read_attributes_before_type_cast (unexpected)
--- FAIL: StoreTest#test_convert_store_attributes_from_Hash_to_HashWithIndifferentAccess_saving_the_data_and_access_attributes_indifferently (unexpected)
--- FAIL: HasOneAssociationsTest#test_dependence_with_missing_association (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_through_has_one_association_with_conditions (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_a_regular_association_with_a_typo_through_a_polymorphic_association_still_raises (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_through_scope_with_joins (unexpected)
--- FAIL: ForbiddenAttributesProtectionTest#test_where_works_with_params_values (unexpected)
--- FAIL: CalculationsTest#test_count_with_block_and_column_name_raises_an_error (unexpected)
--- FAIL: PersistenceTest#test_update_columns_should_raise_exception_if_new_record (unexpected)
--- FAIL: StrictLoadingTest#test_strict_loading_with_reflection_is_ignored_in_validation_context (unexpected)
--- FAIL: DelegatedTypeTest#test_scope_with_custom_foreign_type (unexpected)
--- FAIL: CalculationsTest#test_count_with_where_and_order (unexpected)
--- FAIL: ForbiddenAttributesProtectionTest#test_create_with_works_with_permitted_params (unexpected)
--- FAIL: StoreTest#test_new_record_and_no_accessors_changes (unexpected)
--- FAIL: BasicsTest#test_find_by_slug (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_accept_string_and_hash_join (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_association_replaces_existing_without_dependent_option (unexpected)
--- FAIL: CalculationsTest#test_distinct_is_honored_when_used_with_count_operation_after_group (unexpected)
--- FAIL: BasicsTest#test_table_name_guesses_with_prefixes_and_suffixes (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_with_a_polymorphic_association_and_using_the_existential_predicate_but_also_using_a_select (unexpected)
--- FAIL: HasOneAssociationsTest#test_restrict_with_exception (unexpected)
--- FAIL: StoreTest#test_write_accessor_without_pre/suffix_in_the_same_store_as_other_pre/suffixed_accessors_still_works (unexpected)
--- FAIL: BasicsTest#test_default_in_utc (unexpected)
--- FAIL: BasicsTest#test_cannot_call_connected_to_on_the_abstract_class_that_did_not_establish_the_connection (unexpected)
--- FAIL: ReflectionTest#test_collection_association (unexpected)
--- FAIL: HasOneAssociationsTest#test_association_enum_works_properly_with_nested_join (unexpected)
--- FAIL: CalculationsTest#test_count_with_eager_loading_and_custom_order_and_distinct (unexpected)
--- FAIL: BasicsTest#test_#connecting_to_with_role (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_not_resaved_when_unchanged (unexpected)
--- FAIL: BasicsTest#test_set_table_name_with_inheritance (unexpected)
--- FAIL: CalculationsTest#test_pluck_loaded_relation_aliased_attribute (unexpected)
--- FAIL: BasicsTest#test_sequence_name_with_abstract_class (unexpected)
--- FAIL: CalculationsTest#test_group_by_with_quoted_count_and_order_by_alias (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit_and_conditions_array_on_the_eagers (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_join_model_ignores_default_includes (unexpected)
--- FAIL: DirtyTest#test_reverted_changes_are_not_dirty (unexpected)
--- FAIL: CalculationsTest#test_count_with_too_many_parameters_raises (unexpected)
--- FAIL: OptimisticLockingTest#test_lock_existing (unexpected)
--- FAIL: CalculationsTest#test_async_pluck_none_relation (unexpected)
--- FAIL: BasicsTest#test_connection_in_local_time (unexpected)
--- FAIL: PersistenceTest#test_create (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_type_cast_does_not_corrupt_the_query_cache (unexpected)
--- FAIL: EagerAssociationTest#test_has_many_through_with_order (unexpected)
--- FAIL: MultiParameterAttributeTest#test_multiparameter_attributes_on_time_with_empty_seconds (unexpected)
--- FAIL: ReflectionTest#test_reflection_klass_for_nested_class_name (unexpected)
--- FAIL: CalculationsTest#test_should_count_selected_field_with_include (unexpected)
--- FAIL: StoreTest#test_dirty_methods_for_suffixed_accessors (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15478988/activerecord/1717222176923/1717231127984)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15484123?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15484123?buildTab=artifacts#/activerecord) on master @ [a21a5fc2c955415a5e1b44cf0d995e1048f667c9](https://github.com/cockroachdb/cockroach/commits/a21a5fc2c955415a5e1b44cf0d995e1048f667c9):


```
--- FAIL: BasicsTest#test_clone_preserves_subtype (unexpected)
--- FAIL: CalculationsTest#test_count_with_column_and_options_parameter (unexpected)
--- FAIL: SerializedAttributeTestWithYamlSafeLoad#test_values_cast_from_nil_are_persisted_as_nil (unexpected)
--- FAIL: FoxyFixturesTest#test_supports_timestamps_in_join_tables (unexpected)
--- FAIL: InheritanceTest#test_abstract_class (unexpected)
--- FAIL: HasManyAssociationsTest#test_can_rewhere_the_default_scope_of_the_associated_model (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_existing (unexpected)
--- FAIL: EagerAssociationTest#test_loading_association_with_string_joins (unexpected)
--- FAIL: PostgresqlUUIDTest#test_uuid_change_case_does_not_mark_dirty (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_association_proxy_transaction_method_starts_transaction_in_association_class (unexpected)
--- FAIL: AssociationsTest#test_query_constraints_that_dont_include_the_primary_key_raise (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_collection_build_with_nonstandard_primary_key_on_belongs_to (unexpected)
--- FAIL: RelationTest#test_finding_with_subquery_without_select_does_not_change_the_select (unexpected)
--- FAIL: BasicsTest#test_clear_cache! (unexpected)
--- FAIL: SerializedAttributeTest#test_serialized_attributes_from_database_on_subclass (unexpected)
--- FAIL: CalculationsTest#test_should_get_minimum_of_field (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_no_pk_join_table_delete (unexpected)
--- FAIL: AttributeMethodsTest#test_overridden_read_attribute (unexpected)
--- FAIL: UniquenessValidationTest#test_validate_uniqueness_ignores_itself_when_primary_key_changed (unexpected)
--- FAIL: RelationTest#test_finding_with_arel_order (unexpected)
--- FAIL: FinderTest#test_exists_with_large_number (unexpected)
--- FAIL: TransactionIsolationTest#test_repeatable_read (unexpected)
--- FAIL: StoreTest#test_reading_store_attributes_through_accessors_encoded_with_JSON (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_deleting (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_should_save_parent_but_not_invalid_child (unexpected)
--- FAIL: BasicsTest#test_default_values (unexpected)
--- FAIL: FinderTest#test_find_by_one_attribute_with_several_options (unexpected)
--- FAIL: PersistenceTest#test_reload_via_querycache (unexpected)
--- FAIL: EagerAssociationTest#test_include_has_many_using_primary_key (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_updates_counter_cache_without_dependent_option (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_a_polymorphic_association_with_references_to_the_associated_table (unexpected)
--- FAIL: EnumTest#test_declare_multiple_enums_at_a_time (unexpected)
--- FAIL: FinderTest#test_#find_by_with_composite_primary_key (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_through_record_is_built_when_created_with_where (unexpected)
--- FAIL: RelationTest#test_loaded_relations_cannot_be_mutated_by_multi_value_methods (unexpected)
--- FAIL: PersistenceTest#test_update_raises_record_not_found_exception (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_self_type_mismatch (unexpected)
--- FAIL: FinderTest#test_select_value (unexpected)
--- FAIL: CalculationsTest#test_should_count_selected_field_with_include (unexpected)
--- FAIL: RelationTest#test_size_with_distinct (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit_and_conditions_on_the_eagers (unexpected)
--- FAIL: AttributeMethodsTest#test_undeclared_attribute_method_does_not_affect_respond_to?_and_method_missing (unexpected)
--- FAIL: CalculationsTest#test_select_avg_with_joins_and_group_by_as_virtual_attribute_with_ar (unexpected)
--- FAIL: AttributeMethodsTest#test_inherited_custom_accessors_with_reserved_names (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_through_scope_with_includes (unexpected)
--- FAIL: RelationTest#test_reverse_arel_assoc_order_with_function (unexpected)
--- FAIL: DirtyTest#test_association_assignment_changes_foreign_key (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15484123/activerecord/1717308547274/1717316071042)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15491068?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15491068?buildTab=artifacts#/activerecord) on master @ [c84a89569fab4cc4896a0234bec1e1822dd420d8](https://github.com/cockroachdb/cockroach/commits/c84a89569fab4cc4896a0234bec1e1822dd420d8):


```
--- FAIL: BasicsTest#test_unicode_column_name (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_limit_and_offset (unexpected)
--- FAIL: EagerAssociationTest#test_loading_with_scope_including_joins (unexpected)
--- FAIL: BasicsTest#test_default_values (unexpected)
--- FAIL: ReflectionTest#test_type (unexpected)
--- FAIL: EagerAssociationTest#test_eager_loading_of_instance_dependent_associations_is_not_supported (unexpected)
--- FAIL: ReflectionTest#test_content_columns (unexpected)
--- FAIL: BasicsTest#test_attributes_on_dummy_time (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_cascaded_eager_association_loading_with_join_for_count (unexpected)
--- FAIL: ReflectionTest#test_name_error_from_incidental_code_is_not_converted_to_name_error_for_association (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit_and_high_offset_and_multiple_hash_conditions (unexpected)
--- FAIL: PersistenceTest#test_reload_via_querycache (unexpected)
--- FAIL: PersistenceTest#test_save_with_duping_of_destroyed_object (unexpected)
--- FAIL: PersistenceTest#test_update! (unexpected)
--- FAIL: BasicsTest#test_when_#reload_called,_ignored_columns'_attribute_methods_are_not_defined (unexpected)
--- FAIL: BasicsTest#test_marshal_inspected_round_trip (unexpected)
--- FAIL: BasicsTest#test_time_zone_aware_attribute_with_default_timezone_utc_on_utc_can_be_created (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit_and_conditions_array (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_limit_and_conditions (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_enum_method_issues_a_deprecation_warning (unexpected)
--- FAIL: PersistenceTest#test_delete (unexpected)
--- FAIL: BasicsTest#test_destroy_without_prepared_statement (unexpected)
--- FAIL: PersistenceTest#test_update_many_with_invalid_id (unexpected)
--- FAIL: ReflectionTest#test_association_primary_key_raises_when_missing_primary_key (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_present_with_booleans (unexpected)
--- FAIL: PersistenceTest#test_update_many_with_duplicated_ids (unexpected)
--- FAIL: PersistenceTest#test_decrement_attribute_by (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_through_has_one_association (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_an_sti_join_model (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: AttributeMethodsTest#test_number_attribute_predicate (unexpected)
--- FAIL: ReflectionTest#test_read_attribute_names (unexpected)
--- FAIL: BasicsTest#test_bignum_pk (unexpected)
--- FAIL: BasicsTest#test_column_names_are_quoted_when_using_#from_clause_and_model_has_ignored_columns (unexpected)
--- FAIL: EagerAssociationTest#test_loading_from_an_association (unexpected)
--- FAIL: AttributeMethodsTest#test_user-defined_datetime_attribute_predicate (unexpected)
--- FAIL: EagerAssociationTest#test_eager-loading_non-readonly_association (unexpected)
--- FAIL: BasicsTest#test_singular_table_name_guesses_with_prefixes_and_suffixes (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_habtm (unexpected)
--- FAIL: PersistenceTest#test_class_level_update_without_ids (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_has_many_sti (unexpected)
--- FAIL: EagerAssociationTest#test_finding_with_includes_on_has_many_association_with_same_include_includes_only_once (unexpected)
--- FAIL: AttributeMethodsTest#test_set_attributes_without_a_hash (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_a_belongs_to_association (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_two_levels (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_yaml_loads_5_1_dump_without_indexes_still_queries_for_indexes (unexpected)
--- FAIL: PersistenceTest#test_update_attribute_with_one_updated (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15491068/activerecord/1717395077841/1717404391679)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15504714?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15504714?buildTab=artifacts#/activerecord) on master @ [3d9eaa7939e4fb5aa0518ba7e19750a55c93cf11](https://github.com/cockroachdb/cockroach/commits/3d9eaa7939e4fb5aa0518ba7e19750a55c93cf11):


```
--- FAIL: RelationTest#test_create_or_find_by_should_not_raise_due_to_validation_errors (unexpected)
--- FAIL: HasOneAssociationsTest#test_with_select (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_has_many_through_with_implicit_source (unexpected)
--- FAIL: DirtyTest#test_attribute_will_change!_doesn't_try_to_save_non-persistable_attributes (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_valid_adding_with_validate_false (unexpected)
--- FAIL: ValidationsTest#test_validates_acceptance_of_as_database_column (unexpected)
--- FAIL: PersistenceTest#test_update_many (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_cpk_stale_target (unexpected)
--- FAIL: TransactionTest#test_rollback_dirty_changes_even_with_raise_during_rollback_removes_from_pool (unexpected)
--- FAIL: BelongsToAssociationsTest#test_creating_the_belonging_object_from_new_record (unexpected)
--- FAIL: PersistenceTest#test_build (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_serialization (unexpected)
--- FAIL: BasicsTest#test_typecasting_aliases (unexpected)
--- FAIL: EnumTest#test_enum_value_after_write_symbol (unexpected)
--- FAIL: InsertAllTest#test_upsert_all_does_not_implicitly_set_timestamps_on_update_when_model_record_timestamps_is_true_but_overridden (unexpected)
--- FAIL: CallbacksOnMultipleActionsTest#test_after_commit_on_multiple_actions (unexpected)
--- FAIL: EagerAssociationTest#test_deep_preload (unexpected)
--- FAIL: PostGISTest#test_custom_factory (unexpected)
--- FAIL: EagerAssociationTest#test_eager_loading_with_order_on_joined_table_preloads (unexpected)
--- FAIL: TestAutosaveAssociationOnABelongsToAssociationDefinedAsRecord#test_should_not_raise_error (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_nested_has_many_through_association_with_unpersisted_parent_instance (unexpected)
--- FAIL: TransactionTest#test_rollback_of_frozen_records (unexpected)
--- FAIL: TestHasManyAutosaveAssociationWhichItselfHasAutosaveAssociations#test_when_grandchild_changed_in_memory,_saving_parent_should_save_grandchild (unexpected)
--- FAIL: StoreTest#test_reading_store_attributes_through_accessors (unexpected)
--- FAIL: InheritanceTest#test_inheritance_condition (unexpected)
--- FAIL: ForbiddenAttributesProtectionTest#test_regular_hash_should_still_be_used_for_mass_assignment (unexpected)
--- FAIL: CalculationsTest#test_pluck_on_aliased_attribute (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_recursive_cascading_four_levels_has_many_through (unexpected)
--- FAIL: FinderTest#test_model_class_responds_to_first_bang (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_save_fails_for_invalid_belongs_to (unexpected)
--- FAIL: HasManyAssociationsTest#test_create_with_bang_on_has_many_raises_when_record_not_saved (unexpected)
--- FAIL: RelationTest#test_multivalue_where (unexpected)
--- FAIL: RelationTest#test_unscoped_block_style (unexpected)
--- FAIL: FixturesTest#test_serialized_fixtures (unexpected)
--- FAIL: PersistenceTest#test_becomes_initializes_missing_attributes (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_with_custom_primary_key (unexpected)
--- FAIL: ActiveRecord::Encryption::EncryptableRecordApiTest#test_encrypted_attribute?_returns_false_for_encrypted_attributes_which_content_is_not_encrypted (unexpected)
--- FAIL: FinderTest#test_find_with_a_multiple_sets_of_composite_primary_key_wrapped_in_an_array_ordered (unexpected)
--- FAIL: PersistenceTest#test_save_with_duping_of_destroyed_object (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_has_one_reflection (unexpected)
--- FAIL: FinderTest#test_find_by!_with_hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: RelationTest#test_find_by_doesn't_have_implicit_ordering (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_counter_after_touch (unexpected)
--- FAIL: AssociationsTest#test_assign_persisted_composite_foreign_key_belongs_to_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_with_instance_dependent_scope (unexpected)
--- FAIL: HasManyAssociationsTest#test_building_the_associated_object_with_an_unrelated_type (unexpected)
--- FAIL: DefaultScopingTest#test_default_scope_with_all_queries_runs_on_destroy (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15504714/activerecord/1717481914361/1717490113697)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15535942?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15535942?buildTab=artifacts#/activerecord) on master @ [d0582983521533ffc15a8b8525a188d80da2a0b7](https://github.com/cockroachdb/cockroach/commits/d0582983521533ffc15a8b8525a188d80da2a0b7):


```
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_two_levels_with_condition (unexpected)
--- FAIL: RelationTest#test_first_or_create_with_block (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_preloaded_associations_size (unexpected)
--- FAIL: RelationTest#test_joins_with_select_custom_attribute (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_deleting (unexpected)
--- FAIL: HasOneThroughDisableJoinsAssociationsTest#test_has_one_through_with_belongs_to_on_disable_joins (unexpected)
--- FAIL: FinderTest#test_last_on_loaded_relation_should_not_use_sql (unexpected)
--- FAIL: ModulesTest#test_module_table_name_prefix (unexpected)
--- FAIL: FinderTest#test_find_by_ids_with_limit_and_offset (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_caching_of_columns (unexpected)
--- FAIL: FinderTest#test_include_on_unloaded_relation_with_mismatched_class (unexpected)
--- FAIL: RelationTest#test_finding_with_complex_order_and_limit (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_dynamic_find_should_respect_association_include (unexpected)
--- FAIL: PersistenceTest#test_reload_via_querycache (unexpected)
--- FAIL: FinderTest#test_eager_load_for_no_has_many_with_limit_and_joins_for_has_many (unexpected)
--- FAIL: PersistenceTest#test_save_destroyed_object (unexpected)
--- FAIL: RelationTest#test_first_get_more_than_available (unexpected)
--- FAIL: PersistenceTest#test_create_many (unexpected)
--- FAIL: RelationTest#test_first_or_create_bang_with_invalid_array (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_validation_does_not_validate_stale_association_target (unexpected)
--- FAIL: FinderTest#test_#skip_query_cache!_for_#exists? (unexpected)
--- FAIL: RelationTest#test_order_using_scoping (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_attributes_are_being_set_when_initialized_from_habtm_association_with_multiple_where_clauses (unexpected)
--- FAIL: RelationTest#test_ordering_with_extra_spaces (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_marshal_dump (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_clear_associations (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_find_should_append_to_association_order (unexpected)
--- FAIL: PersistenceTest#test_class_level_update_without_ids! (unexpected)
--- FAIL: FinderTest#test_select_value (unexpected)
--- FAIL: PersistenceTest#test_delete (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_numeric_range_for_string (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_push_with_invalid_join_record (unexpected)
--- FAIL: FinderTest#test_find_with_bad_sql (unexpected)
--- FAIL: FinderTest#test_exists_with_large_number (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_accept_string_and_hash_join (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_belongs_to_should_update_when_the_through_foreign_key_changes (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_distinct_option_prevents_duplicate_push (unexpected)
--- FAIL: ModulesTest#test_module_table_name_suffix_with_global_suffix (unexpected)
--- FAIL: PersistenceTest#test_class_level_destroy_is_affected_by_scoping (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_reorder_raise_errors (unexpected)
--- FAIL: PersistenceTest#test_update_sti_type (unexpected)
--- FAIL: RelationTest#test_reverse_order_with_nulls_first_or_last (unexpected)
--- FAIL: RelationTest#test_select_with_from_includes_quoted_original_table_name (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_three_levels_by_ping_pong (unexpected)
--- FAIL: CockroachDB::FixturesTest#test_multiple_clean_fixtures (unexpected)
--- FAIL: RelationTest#test_typecasting_where_with_array (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_polymorphic_with_primary_key_option (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15535942/activerecord/1717654429484/1717663099228)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15551305?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15551305?buildTab=artifacts#/activerecord) on master @ [802554ac59763c31aee034651efcd28f7bdcdd5d](https://github.com/cockroachdb/cockroach/commits/802554ac59763c31aee034651efcd28f7bdcdd5d):


```
--- FAIL: HasAndBelongsToManyAssociationsTest#test_replace_with_new (unexpected)
--- FAIL: RelationTest#test_to_a_should_dup_target (unexpected)
--- FAIL: RelationTest#test_already-loaded_relations_don't_perform_a_new_query_in_#pretty_print (unexpected)
--- FAIL: FinderTest#test_exists_with_distinct_and_offset_and_select (unexpected)
--- FAIL: BasicsTest#test_table_name_for_base_class (unexpected)
--- FAIL: RelationTest#test_find_all_using_where_twice_should_or_the_relation (unexpected)
--- FAIL: EagerAssociationTest#test_eager_load_multiple_associations_with_references (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_for_single_table_inheritance_(STI_Models) (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_preload_sti_rhs_class (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_destroying_many (unexpected)
--- FAIL: FinderTest#test_member_on_unloaded_relation_with_mismatched_class (unexpected)
--- FAIL: RelationTest#test_select_takes_a_variable_list_of_args (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_through_record_is_built_when_created_with_where (unexpected)
--- FAIL: FinderTest#test_find_with_ids_with_id_out_of_range (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_include_uses_array_include_after_loaded (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_build_then_save_with_has_one_inverse (unexpected)
--- FAIL: RelationTest#test_find_all_with_join (unexpected)
--- FAIL: EagerAssociationTest#test_eager_loaded_has_many_association_without_primary_key (unexpected)
--- FAIL: RelationTest#test_finding_with_arel_order (unexpected)
--- FAIL: AssociationsTest#test_has_many_with_foreign_key_as_an_array_raises (unexpected)
--- FAIL: AssociationsTest#test_subselect (unexpected)
--- FAIL: FinderTest#test_first_failing (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates_words_properly (unexpected)
--- FAIL: BasicsTest#test_limit_should_sanitize_sql_injection_for_limit_with_commas (unexpected)
--- FAIL: BasicsTest#test_equality_of_collection_proxy_and_association_relation (unexpected)
--- FAIL: RelationTest#test_finding_with_order_concatenated (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_through_has_one_association_with_order_on_nested_association (unexpected)
--- FAIL: BasicsTest#test_toggle_attribute (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_distinct_before_the_fact (unexpected)
--- FAIL: RelationTest#test_size_with_zero_limit (unexpected)
--- FAIL: RelationTest#test_finding_with_hash_conditions_on_joined_table (unexpected)
--- FAIL: RelationTest#test_first_or_initialize (unexpected)
--- FAIL: RelationTest#test_finding_with_assoc_reorder_by_aliased_attributes (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_one_dependent_does_not_destroy_dependent (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_conditions_hash (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_one_condition_being_aggregate_and_another_not (unexpected)
--- FAIL: RelationTest#test_find_by_with_hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_includes_in_through_association_scope (unexpected)
--- FAIL: EagerAssociationTest#test_eager_loaded_has_one_association_without_primary_key (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_touch_option_on_touch_without_updated_at_attributes (unexpected)
--- FAIL: BelongsToAssociationsTest#test_building_the_belonging_object_with_an_unrelated_type (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_valid_association_as_string_not_symbol (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_out_of_range_value_assigning (unexpected)
--- FAIL: RelationTest#test_none? (unexpected)
--- FAIL: RelationTest#test_find_by_with_delegated_ar_object (unexpected)
--- FAIL: FinderTest#test_symbols_table_ref (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_find_with_merged_options (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15551305/activerecord/1717740959685/1717749928482)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15565196?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15565196?buildTab=artifacts#/activerecord) on master @ [c7f37e813dbd1e6d2afdb88c5e9cd986dc061299](https://github.com/cockroachdb/cockroach/commits/c7f37e813dbd1e6d2afdb88c5e9cd986dc061299):


```
41 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: IntegrationTest#test_to_param_class_method_truncates_words_properly (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_for_existing_record_with_updated_at_and_custom_cache_timestamp_format (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_default_pk_and_sequence (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates_case_shown_in_doc (unexpected)
--- FAIL: IntegrationTest#test_cache_version_changes_with_versioning_on (unexpected)
--- FAIL: IntegrationTest#test_cache_key_for_newer_updated_at (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_is_not_too_precise (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_for_existing_record_with_nil_updated_timestamps (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates_after_parameterize (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_uses_default_if_blank (unexpected)
--- FAIL: IntegrationTest#test_to_param_with_no_arguments (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_squishes (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_multibyte_character (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates_edge_case (unexpected)
--- FAIL: IntegrationTest#test_cache_version_format_is_precise_enough (unexpected)
--- FAIL: IntegrationTest#test_cache_key_for_newer_updated_on (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: IntegrationTest#test_param_delimiter_changes_delimiter_used_in_to_param (unexpected)
--- FAIL: IntegrationTest#test_cache_key_for_existing_record_is_not_timezone_dependent (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates_after_parameterize_with_hyphens (unexpected)
--- FAIL: IntegrationTest#test_cache_key_retains_version_when_custom_timestamp_is_used (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_uses_default_if_not_persisted (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_partial_indices (unexpected)
--- FAIL: IntegrationTest#test_to_param_returns_id_if_not_persisted_but_id_is_set (unexpected)
--- FAIL: IntegrationTest#test_cache_key_changes_when_child_touched (unexpected)
--- FAIL: IntegrationTest#test_cache_key_is_stable_with_versioning_on (unexpected)
--- FAIL: IntegrationTest#test_to_param_returns_nil_if_not_persisted (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: SchemaDumperTest#test_schema_dumps_index_columns_in_right_order (unexpected)
--- FAIL: IntegrationTest#test_param_delimiter_is_defined_per_class (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_index_length (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_index_sort_order (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_for_existing_record_with_updated_at (unexpected)
--- FAIL: IntegrationTest#test_cache_key_for_updated_on (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_nulls_not_distinct (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_create_fixtures_resets_sequences_when_not_cached (unexpected)
--- FAIL: IntegrationTest#test_to_param_for_a_composite_primary_key_model (unexpected)
--- FAIL: IntegrationTest#test_to_param_should_return_string (unexpected)
--- FAIL: IntegrationTest#test_cache_version_format_is_not_too_precise (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_is_precise_enough (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15565196/activerecord/1717827258796/1717836784401)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15574156?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15574156?buildTab=artifacts#/activerecord) on master @ [c7f37e813dbd1e6d2afdb88c5e9cd986dc061299](https://github.com/cockroachdb/cockroach/commits/c7f37e813dbd1e6d2afdb88c5e9cd986dc061299):


```
--- FAIL: AttributeMethodsTest#test_hash_content (unexpected)
--- FAIL: CalculationsTest#test_should_count_field_in_joined_table (unexpected)
--- FAIL: CalculationsTest#test_aggregate_attribute_on_enum_type (unexpected)
--- FAIL: AttributeMethodsTest#test_update_array_content (unexpected)
--- FAIL: DirtyTest#test_changed?_in_after_callbacks_returns_false (unexpected)
--- FAIL: ActiveRecord::Migration::ForeignKeyChangeColumnWithPrefixTest#test_rename_reference_column_of_child_table (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_method_on_an_abstract_class_is_available_on_subclasses (unexpected)
--- FAIL: PostGISTest#test_spatial_factory_attrs_parsing (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_replace_order_is_preserved (unexpected)
--- FAIL: IntegrationTest#test_cache_key_for_updated_on (unexpected)
--- FAIL: CalculationsTest#test_group_by_attribute_with_custom_type (unexpected)
--- FAIL: AttributeMethodsTest#test_boolean_attribute_predicate (unexpected)
--- FAIL: BelongsToAssociationsTest#test_reloading_association_with_key_change (unexpected)
--- FAIL: EagerAssociationTest#test_include_has_one_using_primary_key (unexpected)
--- FAIL: CalculationsTest#test_should_sum_scoped_field (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_new_aliased_to_build (unexpected)
--- FAIL: EagerAssociationTest#test_preconfigured_includes_with_has_many (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_limit_and_offset_and_conditions_array (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_circular_autosave_association_correctly_saves_multiple_records (unexpected)
--- FAIL: BelongsToAssociationsTest#test_create_with_block (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_attributes_not_backed_by_database_columns_are_not_dirty_when_unchanged (unexpected)
--- FAIL: EagerAssociationTest#test_conditions_on_join_table_with_include_and_limit (unexpected)
--- FAIL: AttributeMethodsTest#test_boolean_attributes_writing_and_reading (unexpected)
--- FAIL: AssociationsJoinModelTest#test_has_many_through_goes_through_all_sti_classes (unexpected)
--- FAIL: EagerAssociationTest#test_finding_with_includes_on_has_many_association_with_same_include_includes_only_once (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_polymorphic_with_primary_key_option (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_unscope_should_affect_to_through_scope (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_with_a_bang_for_single_table_inheritance_(STI_Models) (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_multiple_stis_and_order (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_join_model_ignores_default_includes (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_children_inherit_custom_properties (unexpected)
--- FAIL: ActiveRecord::Migration::ColumnsTest#test_change_column_with_nil_default (unexpected)
--- FAIL: CounterCacheTest#test_increment_counters_with_touch:_true (unexpected)
--- FAIL: InheritanceAttributeTest#test_inheritance_new_with_subclass_as_default (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_joining_has_many_through_belongs_to (unexpected)
--- FAIL: TransactionTest#test_cancellation_from_before_filters_rollbacks_in_save (unexpected)
--- FAIL: CalculationsTest#test_count_with_block (unexpected)
--- FAIL: EagerAssociationTest#test_eager_load_multiple_associations_with_references (unexpected)
--- FAIL: PostGISTest#test_custom_factory (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_a_polymorphic_association_with_references_to_the_associated_table (unexpected)
--- FAIL: DirtyTest#test_attribute_will_change!_doesn't_try_to_save_non-persistable_attributes (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_with_reflection_using_class_name_and_fixtures (unexpected)
--- FAIL: ActiveRecord::PostgresqlTransactionTest#test_raises_QueryCanceled_when_statement_timeout_exceeded (unexpected)
--- FAIL: AssociationsJoinModelTest#test_add_to_join_table_with_no_id (unexpected)
--- FAIL: IntegrationTest#test_param_delimiter_is_defined_per_class (unexpected)
--- FAIL: IntegrationTest#test_cache_key_changes_when_child_touched (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_raises_UnknownPrimaryKey_when_a_model_has_no_primary_key (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15574156/activerecord/1717913429600/1717920564065)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15601083?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15601083?buildTab=artifacts#/activerecord) on master @ [2ce93809d799bacc078f8a1d8e24a710dbff4d66](https://github.com/cockroachdb/cockroach/commits/2ce93809d799bacc078f8a1d8e24a710dbff4d66):


```
--- FAIL: HasOneAssociationsTest#test_clearing_an_association_clears_the_associations_inverse (unexpected)
--- FAIL: HasOneAssociationsTest#test_dependence (unexpected)
--- FAIL: CalculationsTest#test_count_selected_arel_attribute (unexpected)
--- FAIL: CalculationsTest#test_should_perform_joined_include_when_referencing_included_tables (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_with_create_and_valid_options (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_an_overridden_original_method_along_with_an_overridden_alias_method_in_a_parent_class_doesn't_issue_a_deprecation (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_add_with_sti_middle_relation (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_assignment_before_either_saved (unexpected)
--- FAIL: RelationTest#test_find_all_using_where_with_relation_with_select_to_build_subquery (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_collection_create_with_nonstandard_primary_key_on_belongs_to (unexpected)
--- FAIL: RelationTest#test_first_or_create_bang_with_invalid_options (unexpected)
--- FAIL: RelationTest#test_references_with_blank_value (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_count_with_include_should_alias_join_table (unexpected)
--- FAIL: CalculationsTest#test_distinct_count_with_order_and_offset (unexpected)
--- FAIL: RelationTest#test_select_with_from_includes_original_table_name (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_with_a_bang_for_single_table_inheritance_(STI_Models) (unexpected)
--- FAIL: RelationTest#test_group_with_blank_value (unexpected)
--- FAIL: RelationTest#test_size_with_eager_loading_and_custom_order (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_validation_does_not_validate_stale_association_target (unexpected)
--- FAIL: RelationTest#test_#skip_query_cache!_with_a_preload (unexpected)
--- FAIL: RelationTest#test_find_or_create_by! (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_callbacks_firing_order_on_update (unexpected)
--- FAIL: CalculationsTest#test_pick_two (unexpected)
--- FAIL: ReflectionTest#test_foreign_key_is_inferred_from_model_name (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_with_touch_option_on_empty_update (unexpected)
--- FAIL: RelationTest#test_unscope_with_triple_dot_where (unexpected)
--- FAIL: RelationTest#test_finding_with_subquery_with_binds (unexpected)
--- FAIL: InheritanceTest#test_company_descends_from_active_record (unexpected)
--- FAIL: AttributeMethodsTest#test_aliasing_``id``_attribute_allows_reading_the_column_value (unexpected)
--- FAIL: CalculationsTest#test_pluck_for_a_composite_primary_key (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_save_bang_should_raise_exception_when_join_record_has_errors (unexpected)
--- FAIL: RelationTest#test_reverse_order_with_function (unexpected)
--- FAIL: CalculationsTest#test_count_with_from_option (unexpected)
--- FAIL: InheritanceTest#test_inheritance_with_default_scope (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_existing_with_nonstandard_primary_key_on_belongs_to (unexpected)
--- FAIL: RelationTest#test_finding_with_complex_order_and_limit (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_get_ids (unexpected)
--- FAIL: CalculationsTest#test_should_sum_field_with_conditions (unexpected)
--- FAIL: CalculationsTest#test_distinct_is_honored_when_used_with_count_operation_after_group (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_update_counter_caches_on_replace_association (unexpected)
--- FAIL: RelationTest#test_alias_tracker_respects_a_custom_table (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_belongs_to_sti (unexpected)
--- FAIL: CalculationsTest#test_ids_on_loaded_relation_with_scope (unexpected)
--- FAIL: RelationTest#test_loaded_first (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_field_through_association_and_having (unexpected)
--- FAIL: InheritanceTest#test_scope_inherited_properly (unexpected)
--- FAIL: RelationTest#test_create_or_find_by_with_bang_within_transaction (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15601083/activerecord/1718086359476/1718093301999)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2 no-test-failure-activity]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15618383?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15618383?buildTab=artifacts#/activerecord) on master @ [c88a1cedb064f554f38bb5b310e83ac3259ee02e](https://github.com/cockroachdb/cockroach/commits/c88a1cedb064f554f38bb5b310e83ac3259ee02e):


```
--- FAIL: IntegrationTest#test_to_param_class_method_truncates_edge_case (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: IntegrationTest#test_to_param_class_method_uses_default_if_blank (unexpected)
--- FAIL: IntegrationTest#test_cache_key_for_newer_updated_at (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_assignment_before_parent_saved (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_default_pk_and_sequence (unexpected)
--- FAIL: IntegrationTest#test_to_param_for_a_composite_primary_key_model (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_for_existing_record_with_nil_updated_timestamps (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_validation_does_not_validate_stale_association_target (unexpected)
--- FAIL: IntegrationTest#test_cache_key_is_stable_with_versioning_on (unexpected)
--- FAIL: IntegrationTest#test_cache_key_for_updated_on (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_save_fails_for_invalid_belongs_to (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_is_precise_enough (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_for_existing_record_with_updated_at_and_custom_cache_timestamp_format (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_build_and_then_save_parent_should_not_reload_target (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates_case_shown_in_doc (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_squishes (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates_after_parameterize (unexpected)
--- FAIL: IntegrationTest#test_to_param_returns_id_if_not_persisted_but_id_is_set (unexpected)
--- FAIL: IntegrationTest#test_param_delimiter_changes_delimiter_used_in_to_param (unexpected)
--- FAIL: IntegrationTest#test_to_param_with_no_arguments (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_create_fixtures_resets_sequences_when_not_cached (unexpected)
--- FAIL: IntegrationTest#test_cache_key_for_newer_updated_on (unexpected)
--- FAIL: IntegrationTest#test_to_param_should_return_string (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_for_existing_record_with_updated_at (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_store_association_with_a_polymorphic_relationship (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_store_association_in_two_relations_with_one_save_in_existing_object_with_values (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_store_two_association_with_one_save (unexpected)
--- FAIL: ActiveRecord::Type::IntegerTest#test_casting_ActiveRecord_models (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_save_succeeds_for_invalid_belongs_to_with_validate_false (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: IntegrationTest#test_cache_version_changes_with_versioning_on (unexpected)
--- FAIL: IntegrationTest#test_to_param_returns_nil_if_not_persisted (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_should_save_parent_but_not_invalid_child (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_validation_does_not_validate_non_dirty_association_target (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates_after_parameterize_with_hyphens (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_store_association_in_two_relations_with_one_save (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_store_association_in_two_relations_with_one_save_in_existing_object (unexpected)
--- FAIL: IntegrationTest#test_cache_version_format_is_precise_enough (unexpected)
--- FAIL: IntegrationTest#test_cache_key_retains_version_when_custom_timestamp_is_used (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates (unexpected)
--- FAIL: IntegrationTest#test_cache_key_changes_when_child_touched (unexpected)
--- FAIL: IntegrationTest#test_cache_key_for_existing_record_is_not_timezone_dependent (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_assignment_before_either_saved (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_multibyte_character (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15618383/activerecord/1718172755073/1718181575104)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2 no-test-failure-activity]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15637170?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15637170?buildTab=artifacts#/activerecord) on master @ [9d3dd0530818b9c5c0f7dac04e8a8de4acf1bba4](https://github.com/cockroachdb/cockroach/commits/9d3dd0530818b9c5c0f7dac04e8a8de4acf1bba4):


```
--- FAIL: InheritanceTest#test_inherits_custom_primary_key (unexpected)
--- FAIL: BasicsTest#test_arel_attribute_normalization (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_default_scope_as_class_method (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_through_association_with_through_scope_and_nested_where (unexpected)
--- FAIL: HasOneThroughDisableJoinsAssociationsTest#test_disable_joins_through_with_enum_type (unexpected)
--- FAIL: EagerAssociationTest#test_eager_load_has_many_with_string_keys (unexpected)
--- FAIL: PersistenceTest#test_destroy (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_attributes_are_available_to_after_initialize (unexpected)
--- FAIL: BasicsTest#test_select_symbol (unexpected)
--- FAIL: CalculationsTest#test_should_limit_calculation (unexpected)
--- FAIL: CalculationsTest#test_count_with_block_and_column_name_raises_an_error (unexpected)
--- FAIL: PersistenceTest#test_update_attribute_does_not_run_sql_if_attribute_is_not_changed (unexpected)
--- FAIL: EagerAssociationTest#test_eager_load_has_many_through_quotes_table_and_column_names (unexpected)
--- FAIL: BelongsToAssociationsTest#test_create_bang_with_conditions (unexpected)
--- FAIL: FinderTest#test_second_have_primary_key_order_by_default (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_through_has_one_association_with_conditions (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroy_does_not_raise_when_association_errors_on_destroy (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_touch_option_on_empty_update (unexpected)
--- FAIL: CalculationsTest#test_limit_should_apply_before_count_arel_attribute (unexpected)
--- FAIL: AssociationsTest#test_polymorphic_belongs_to_uses_parent_query_constraints (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_new_by_building (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_belongs_to_with_cpk (unexpected)
--- FAIL: FinderTest#test_find_on_array_conditions (unexpected)
--- FAIL: FinderTest#test_condition_utc_time_interpolation_with_default_timezone_local (unexpected)
--- FAIL: PersistenceTest#test_increment_with_no_arg (unexpected)
--- FAIL: BasicsTest#test_records_of_different_classes_have_different_hashes (unexpected)
--- FAIL: FinderTest#test_last_with_integer_and_order_should_keep_the_order (unexpected)
--- FAIL: InheritanceTest#test_compute_type_nonexistent_constant (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_explicitly_joining_join_table (unexpected)
--- FAIL: BasicsTest#test_toggle_attribute (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_attributes_are_being_set_when_initialized_from_hm_through_association_with_multiple_where_clauses (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_create_fixtures_resets_sequences_when_not_cached (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_with_bang_with_purpose_raises (unexpected)
--- FAIL: BelongsToAssociationsTest#test_create_bang_with_block (unexpected)
--- FAIL: PersistenceTest#test_increment_with_touch_an_attribute_updates_timestamps (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_size_on_an_association_that_has_not_been_loaded_performs_a_query (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit_and_high_offset_and_multiple_hash_conditions (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit_and_conditions_on_the_eagers (unexpected)
--- FAIL: HasManyAssociationsTest#test_select_with_block (unexpected)
--- FAIL: HasManyAssociationsTest#test_creating_using_primary_key (unexpected)
--- FAIL: BasicsTest#test_connection_in_utc_time (unexpected)
--- FAIL: PersistenceTest#test_update (unexpected)
--- FAIL: EagerAssociationTest#test_including_duplicate_objects_from_has_many (unexpected)
--- FAIL: HasManyAssociationsTest#test_load_target_respects_protected_attributes (unexpected)
--- FAIL: FinderTest#test_include_on_unloaded_relation_without_match (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_association_replaces_existing_with_dependent_option (unexpected)
--- FAIL: HasManyAssociationsTest#test_loading_association_in_validate_callback_doesnt_affect_persistence (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15637170/activerecord/1718259141555/1718267867131)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2 no-test-failure-activity]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15653277?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15653277?buildTab=artifacts#/activerecord) on master @ [9d3dd0530818b9c5c0f7dac04e8a8de4acf1bba4](https://github.com/cockroachdb/cockroach/commits/9d3dd0530818b9c5c0f7dac04e8a8de4acf1bba4):


```
--- FAIL: HasAndBelongsToManyAssociationsTest#test_destroying (unexpected)
--- FAIL: FinderTest#test_last_with_integer_and_order_should_keep_the_order (unexpected)
--- FAIL: PersistenceTest#test_update_attribute_for_aborted_callback! (unexpected)
--- FAIL: InheritanceTest#test_alt_finding_incorrect_type_data (unexpected)
--- FAIL: SignedIdTest#test_can_get_a_signed_ID_in_an_after_create (unexpected)
--- FAIL: FinderTest#test_find_by_one_attribute_with_conditions (unexpected)
--- FAIL: PersistenceTest#test_update_sti_subclass_type (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_destroy_association (unexpected)
--- FAIL: ReflectionTest#test_belongs_to_inferred_foreign_key_from_assoc_name (unexpected)
--- FAIL: HasManyAssociationsTest#test_building_the_associated_object_with_an_unrelated_type (unexpected)
--- FAIL: FinderTest#test_find_by_array_of_one_id (unexpected)
--- FAIL: HasManyAssociationsTest#test_joining_through_a_polymorphic_association_with_a_where_clause (unexpected)
--- FAIL: ReflectionTest#test_aggregation_reflection (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_tags_has_manu_posts_through_association_with_composite_query_constraints (unexpected)
--- FAIL: ModulesTest#test_nested_models_should_not_raise_exception_when_using_nullify_dependency_on_association (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_proper_usage_of_primary_keys_and_join_table (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_update_counter_caches_on_destroy_with_indestructible_through_record (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroying_by_string_id (unexpected)
--- FAIL: HasManyAssociationsTest#test_counting_with_counter_sql (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_many_with_merged_options (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_with_rewhere_doesnt_set_inverse_instance_key (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_validation_does_not_validate_non_dirty_association_target (unexpected)
--- FAIL: PersistenceTest#test_update_uses_query_constraints_config (unexpected)
--- FAIL: HasManyAssociationsTest#test_can_unscope_and_where_the_default_scope_of_the_associated_model (unexpected)
--- FAIL: FinderTest#test_find_passing_active_record_object_is_not_permitted (unexpected)
--- FAIL: AssociationsTest#test_assign_belongs_to_cpk_model_by_id_attribute (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_delete_association (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_find_grouped (unexpected)
--- FAIL: HasManyAssociationsTest#test_default_scope_on_relations_is_not_cached (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_distinct_before_the_fact (unexpected)
--- FAIL: InheritanceTest#test_where_create_bang_with_unrelated_type (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_index_columns_in_right_order (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all_on_association_clears_scope (unexpected)
--- FAIL: HasManyAssociationsTest#test_counting_with_single_hash (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_many_should_return_false_if_none_or_one (unexpected)
--- FAIL: PersistenceTest#test_save_touch_false (unexpected)
--- FAIL: InheritanceAttributeMappingTest#test_sti_with_custom_type (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_assign_ids (unexpected)
--- FAIL: HasManyAssociationsTest#test_set_ids_for_association_on_new_record_applies_association_correctly (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_belongs_to_should_update_when_the_through_foreign_key_changes (unexpected)
--- FAIL: InheritanceTest#test_becomes_sets_variables_before_initialization_callbacks (unexpected)
--- FAIL: AssociationsTest#test_cpk_model_has_many_records_by_id_attribute (unexpected)
--- FAIL: ModulesTest#test_compute_type_can_infer_class_name_of_sibling_inside_module (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associating_new (unexpected)
--- FAIL: PersistenceTest#test_returns_object_even_if_validations_failed (unexpected)
--- FAIL: FinderTest#test_find_by_with_associations (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_in_batches (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15653277/activerecord/1718327740789/1718336204625)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2 no-test-failure-activity]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15653977?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15653977?buildTab=artifacts#/activerecord) on master @ [62e90edbe21d0afe9dfb26c216ba91730e74ee43](https://github.com/cockroachdb/cockroach/commits/62e90edbe21d0afe9dfb26c216ba91730e74ee43):


```
--- FAIL: BelongsToAssociationsTest#test_polymorphic_reassignment_of_associated_id_updates_the_object (unexpected)
--- FAIL: RelationTest#test_first_or_create_bang_with_valid_array (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_build_many_before_save (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_post_has_many_tags_through_association_with_composite_query_constraints (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_merge_join_association_with_has_many_through_association_proxy (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependence_on_account (unexpected)
--- FAIL: FinderTest#test_find_with_group_and_sanitized_having_method (unexpected)
--- FAIL: RelationTest#test_where_id_with_delegated_ar_object (unexpected)
--- FAIL: RelationTest#test_select_with_subquery_in_from_does_not_use_original_table_name (unexpected)
--- FAIL: BelongsToAssociationsTest#test_build_with_block (unexpected)
--- FAIL: FinderTest#test_#last_for_a_model_with_composite_query_constraints (unexpected)
--- FAIL: CalculationsTest#test_should_average_arel_attribute (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_build_via_block_before_save (unexpected)
--- FAIL: ReflectionTest#test_has_one_reflection (unexpected)
--- FAIL: HasOneAssociationsTest#test_exclusive_dependence (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_a_item_which_is_not_in_the_collection (unexpected)
--- FAIL: CalculationsTest#test_should_not_overshadow_enumerable_sum (unexpected)
--- FAIL: ReflectionTest#test_name_error_from_incidental_code_is_not_converted_to_name_error_for_association (unexpected)
--- FAIL: BelongsToAssociationsTest#test_assigning_nil_on_an_association_clears_the_associations_inverse (unexpected)
--- FAIL: CalculationsTest#test_pluck_type_cast_with_conflict_column_names (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_with_a_bang_within_expiration_duration (unexpected)
--- FAIL: HasManyAssociationsTest#test_finder_method_with_dirty_target (unexpected)
--- FAIL: HotCompatibilityTest#test_cleans_up_after_prepared_statement_failure_in_a_transaction (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_time_zone_aware_attribute (unexpected)
--- FAIL: ActiveRecord::SelectTest#test_select_with_hash_argument (unexpected)
--- FAIL: InverseBelongsToTests#test_recursive_inverse_on_recursive_model_has_many_inversing (unexpected)
--- FAIL: PersistenceTest#test_save_destroyed_object (unexpected)
--- FAIL: HasManyAssociationsTest#test_collection_size_twice_for_regressions (unexpected)
--- FAIL: HasManyAssociationsTest#test_create_from_association_with_nil_values_should_work (unexpected)
--- FAIL: FinderTest#test_exists_with_order (unexpected)
--- FAIL: CalculationsTest#test_should_count_manual_with_count_all (unexpected)
--- FAIL: CounterCacheTest#test_update_counters_with_touch:_:written_on (unexpected)
--- FAIL: OptimisticLockingTest#test_explicit_update_lock_column_raise_error (unexpected)
--- FAIL: AttributeMethodsTest#test_time_attributes_are_retrieved_in_the_current_time_zone (unexpected)
--- FAIL: RelationTest#test_find_by!_with_non-hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: ModulesTest#test_compute_type_can_infer_class_name_of_sibling_inside_module (unexpected)
--- FAIL: BasicsTest#test_find_symbol_ordered_last (unexpected)
--- FAIL: CalculationsTest#test_minimum_and_maximum_on_time_attributes (unexpected)
--- FAIL: HasOneAssociationsTest#test_create (unexpected)
--- FAIL: AttributeMethodsTest#test_bulk_update_raises_ActiveRecord::UnknownAttributeError (unexpected)
--- FAIL: HasManyAssociationsTest#test_depends_and_nullify (unexpected)
--- FAIL: BasicsTest#test_benchmark_with_log_level (unexpected)
--- FAIL: I18nValidationTest#test_validates_uniqueness_of_on_generated_message_given_if_condition (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_a_manually_defined_method_issues_a_deprecation_warning (unexpected)
--- FAIL: I18nGenerateMessageValidationTest#test_translation_for_'taken'_can_be_overridden_in_activerecord_scope (unexpected)
--- FAIL: MultiParameterAttributeTest#test_multiparameter_attributes_on_time_with_time_zone_aware_attributes (unexpected)
--- FAIL: SignedIdTest#test_finding_record_from_broken_signed_id_raises_on_the_bang (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15653977/activerecord/1718345583528/1718353032043)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2 no-test-failure-activity]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15670966?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15670966?buildTab=artifacts#/activerecord) on master @ [3aa5bdf40527e2f60b179094403b4302a1c2cbe1](https://github.com/cockroachdb/cockroach/commits/3aa5bdf40527e2f60b179094403b4302a1c2cbe1):


```
--- FAIL: EagerAssociationTest#test_preload_has_many_uses_exclusive_scope (unexpected)
--- FAIL: FinderTest#test_first_bang_missing (unexpected)
--- FAIL: RelationTest#test_reorder_with_first (unexpected)
--- FAIL: BelongsToAssociationsTest#test_with_different_class_name (unexpected)
--- FAIL: FinderTest#test_find_by_empty_ids (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_join_for_count (unexpected)
--- FAIL: AssociationsTest#test_eager_loading_should_not_change_count_of_children (unexpected)
--- FAIL: PersistenceTest#test_create_through_factory_with_block (unexpected)
--- FAIL: CalculationsTest#test_minimum_with_not_auto_table_name_prefix_if_column_included (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_default_scope (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_touch_option_on_update (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_multiple_columns_and_includes (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_delete_all_for_with_dependent_option_nullify (unexpected)
--- FAIL: CalculationsTest#test_should_perform_joined_include_when_referencing_included_tables (unexpected)
--- FAIL: BasicsTest#test_dup_for_a_composite_primary_key_model (unexpected)
--- FAIL: FinderTest#test_first_failing (unexpected)
--- FAIL: RelationTest#test_first_or_create_bang_with_invalid_block (unexpected)
--- FAIL: RelationTest#test_scoped (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_should_save_parent_but_not_invalid_child (unexpected)
--- FAIL: FinderTest#test_find_with_eager_loading_collection_and_ordering_by_collection_primary_key (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: EagerAssociationTest#test_preloading_has_one_with_cpk (unexpected)
--- FAIL: BelongsToAssociationsTest#test_counter_cache_double_destroy (unexpected)
--- FAIL: InheritanceTest#test_alt_inheritance_find_all (unexpected)
--- FAIL: CalculationsTest#test_ids_with_join (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_has_many_through_association_associated_by_a_composite_query_constraints (unexpected)
--- FAIL: RelationTest#test_finding_last_with_arel_order (unexpected)
--- FAIL: ReflectionTest#test_join_table (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_callbacks_firing_order_on_update (unexpected)
--- FAIL: EagerAssociationTest#test_loading_with_no_associations (unexpected)
--- FAIL: AssociationsTest#test_loading_cpk_association_when_persisted_and_in_memory_differ (unexpected)
--- FAIL: PersistenceTest#test_delete_all (unexpected)
--- FAIL: CalculationsTest#test_pluck_in_relation (unexpected)
--- FAIL: FinderTest#test_last_on_loaded_relation_should_not_use_sql (unexpected)
--- FAIL: FinderTest#test_include_on_unloaded_relation_with_having_referencing_aliased_select (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_two_levels_and_self_table_reference (unexpected)
--- FAIL: FinderTest#test_find_by_array_of_one_id (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_build_before_either_saved (unexpected)
--- FAIL: AssociationsTest#test_should_construct_new_finder_sql_after_create (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_for_existing_record_with_nil_updated_timestamps (unexpected)
--- FAIL: BelongsToAssociationsTest#test_should_set_foreign_key_on_create_association_with_unpersisted_owner (unexpected)
--- FAIL: BasicsTest#test_generated_association_methods_module_name (unexpected)
--- FAIL: BasicsTest#test_previously_changed_dup (unexpected)
--- FAIL: RelationTest#test_delete_by (unexpected)
--- FAIL: RelationTest#test_order_with_reorder_nil_removes_the_order (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_escaped_characters (unexpected)
--- FAIL: HasOneAssociationsTest#test_destroyed_by_association_set_in_child_destroy_callback_on_parent_destroy (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15670966/activerecord/1718432107388/1718440114665)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.2-rc release-blocker]
- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2 no-test-failure-activity]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15681888?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15681888?buildTab=artifacts#/activerecord) on master @ [3aa5bdf40527e2f60b179094403b4302a1c2cbe1](https://github.com/cockroachdb/cockroach/commits/3aa5bdf40527e2f60b179094403b4302a1c2cbe1):


```
--- FAIL: BasicsTest#test_attribute_names (unexpected)
--- FAIL: BasicsTest#test_equality_of_relation_and_association_relation (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_field_having_condition (unexpected)
--- FAIL: AttributeMethodsTest#test_boolean_attributes_writing_and_reading (unexpected)
--- FAIL: CalculationsTest#test_count_with_column_and_options_parameter (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_with_create_and_no_options (unexpected)
--- FAIL: CalculationsTest#test_count_on_invalid_columns_raises (unexpected)
--- FAIL: PostGISTest#test_save_and_load_geographic_point (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_callbacks_firing_order_on_create (unexpected)
--- FAIL: CalculationsTest#test_count_with_eager_loading_and_custom_select_and_order (unexpected)
--- FAIL: AttributeMethodsTest#test_set_attributes_with_a_block (unexpected)
--- FAIL: HasOneAssociationsTest#test_reload_association (unexpected)
--- FAIL: HasOneAssociationsTest#test_association_enum_works_properly (unexpected)
--- FAIL: PersistenceTest#test_becomes_includes_errors (unexpected)
--- FAIL: AttributeMethodsTest#test_typecast_attribute_from_select_to_true (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_find_should_append_to_association_order (unexpected)
--- FAIL: CalculationsTest#test_should_get_maximum_of_arel_attribute_with_include (unexpected)
--- FAIL: BasicsTest#test_clear_cache_when_setting_table_name (unexpected)
--- FAIL: PostgresqlTimestampTest#test_timestamp_with_zone_values_without_rails_time_zone_support (unexpected)
--- FAIL: PostgresqlByteaTest#test_read_nil_value (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_proxy_should_not_respond_to_private_methods (unexpected)
--- FAIL: PostgresqlByteaTest#test_serialize (unexpected)
--- FAIL: CalculationsTest#test_should_sum_field_with_conditions (unexpected)
--- FAIL: PersistenceTest#test_save_with_duping_of_destroyed_object (unexpected)
--- FAIL: HasOneAssociationsTest#test_with_select (unexpected)
--- FAIL: PostgresqlByteaTest#test_write_nil (unexpected)
--- FAIL: PersistenceTest#test_increment_nil_attribute (unexpected)
--- FAIL: SpatialQueriesTest#test_query_point (unexpected)
--- FAIL: SerializedAttributeTest#test_serialized_class_attribute (unexpected)
--- FAIL: BasicsTest#test_equality_of_collection_proxy_and_association_relation (unexpected)
--- FAIL: DirtyTest#test_reverted_changes_are_not_dirty (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_with_inexistent_foreign_key_failing (unexpected)
--- FAIL: DirtyTest#test_attribute_should_be_compared_with_type_cast (unexpected)
--- FAIL: InheritanceTest#test_complex_inheritance (unexpected)
--- FAIL: BasicsTest#test_preserving_time_objects_with_time_with_zone_conversion_to_default_timezone_utc (unexpected)
--- FAIL: PostGISTest#test_save_and_load_point (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_multiple_fields_having_functions (unexpected)
--- FAIL: CalculationsTest#test_should_count_manual_select_with_count_all (unexpected)
--- FAIL: CalculationsTest#test_limit_should_apply_before_count_arel_attribute (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_add_two_instance_and_then_deleting (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_has_and_belongs_to_many_while_partial_inserts_false (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_collection_singular_ids_getter_with_string_primary_keys (unexpected)
--- FAIL: BasicsTest#test_generated_relation_methods_module_name (unexpected)
--- FAIL: BasicsTest#test_last (unexpected)
--- FAIL: NumericalityValidationTest#test_virtual_attribute_with_precision_round_down (unexpected)
--- FAIL: ActiveRecord::Migration::ColumnsTest#test_rename_column_using_symbol_arguments (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_multiple_columns_and_selection_clause (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15681888/activerecord/1718518403481/1718525957849)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.2-rc release-blocker]
- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2 no-test-failure-activity]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15693113?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15693113?buildTab=artifacts#/activerecord) on master @ [efc894bfc167fe48af2ab99d1e1ecbedf8110a10](https://github.com/cockroachdb/cockroach/commits/efc894bfc167fe48af2ab99d1e1ecbedf8110a10):


```
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_saving_multiple_relationships (unexpected)
--- FAIL: HasOneAssociationsTest#test_destroyed_by_association_set_in_child_destroy_callback_on_replace (unexpected)
--- FAIL: BelongsToAssociationsTest#test_cant_save_readonly_association (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_collection_size_from_params (unexpected)
--- FAIL: HasOneAssociationsTest#test_association_keys_bypass_attribute_protection (unexpected)
--- FAIL: BelongsToAssociationsTest#test_building_the_belonging_object_with_explicit_sti_base_class (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_with_touch_option_on_create (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_adding_a_collection (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_assign_ids_with_cpk_for_two_models (unexpected)
--- FAIL: HasOneAssociationsTest#test_building_the_associated_object_with_explicit_sti_base_class (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_with_touch_option_on_empty_update (unexpected)
--- FAIL: BelongsToAssociationsTest#test_create! (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_should_save_parent_but_not_invalid_child (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_counter_with_reassigning (unexpected)
--- FAIL: BelongsToAssociationsTest#test_with_different_class_name (unexpected)
--- FAIL: AssociationsTest#test_belongs_to_a_cpk_model_by_id_attribute (unexpected)
--- FAIL: BelongsToAssociationsTest#test_reloading_association_with_key_change (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_distinct_when_association_already_loaded (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_deleting_array (unexpected)
--- FAIL: BelongsToAssociationsTest#test_id_assignment (unexpected)
--- FAIL: AssociationsTest#test_bad_collection_keys (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_touch_option_on_touch_and_reassigned_parent (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_collection_size_from_build (unexpected)
--- FAIL: HasOneAssociationsTest#test_dangerous_association_name_raises_ArgumentError (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_validation_does_not_validate_non_dirty_association_target (unexpected)
--- FAIL: AssociationsTest#test_nullify_composite_foreign_key_has_many_association (unexpected)
--- FAIL: BelongsToAssociationsTest#test_should_set_foreign_key_on_save (unexpected)
--- FAIL: BelongsToAssociationsTest#test_assignment_before_child_saved_with_primary_key (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_valid_adding_with_validate_false (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_association_with_bang (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_redefine_habtm (unexpected)
--- FAIL: BelongsToAssociationsTest#test_skips_parent_presence_check_if_parent_has_not_changed (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_should_save_parent_but_not_invalid_child (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_touch_option_on_destroy_with_destroyed_parent (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_with_custom_primary_key (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_has_and_belongs_to_many_while_partial_inserts_false (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_include_method_in_has_and_belongs_to_many_association_should_return_true_for_instance_added_with_build (unexpected)
--- FAIL: BelongsToAssociationsTest#test_runs_parent_presence_check_if_parent_changed_or_nil (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_should_property_quote_string_primary_keys (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_respects_hash_condition (unexpected)
--- FAIL: HasOneAssociationsTest#test_dependence_with_missing_association (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_with_touch_option_on_nonpersisted_built_associations_doesnt_update_parent (unexpected)
--- FAIL: BelongsToAssociationsTest#test_tracking_change_from_persisted_record_to_nil (unexpected)
--- FAIL: ActiveRecord::Type::IntegerTest#test_casting_ActiveRecord_models (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_build_many_via_block_before_save (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_save_succeeds_for_invalid_belongs_to_with_validate_false (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_association_class (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15693113/activerecord/1718604717288/1718612862272)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.2-rc release-blocker]
- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123928 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-22.2 no-test-failure-activity]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15727827?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15727827?buildTab=artifacts#/activerecord) on master @ [67b4af76ba20ed2f6935da31eda7dfe8fc0b63e2](https://github.com/cockroachdb/cockroach/commits/67b4af76ba20ed2f6935da31eda7dfe8fc0b63e2):


```
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: BasicsTest#test_sql_injection_via_find (unexpected)
--- FAIL: BasicsTest#test_distinct_delegates_to_scoped (unexpected)
--- FAIL: BasicsTest#test_column_types_typecast (unexpected)
--- FAIL: BasicsTest#test_failed_comparison_of_unlike_class_records (unexpected)
--- FAIL: InheritanceTest#test_class_with_blank_sti_name (unexpected)
--- FAIL: BasicsTest#test_incomplete_schema_loading (unexpected)
--- FAIL: CalculationsTest#test_pluck_type_cast (unexpected)
--- FAIL: CalculationsTest#test_having_with_strong_parameters (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_multiple_fields_having_functions (unexpected)
--- FAIL: BasicsTest#test_find_reverse_ordered_last (unexpected)
--- FAIL: InheritanceTest#test_where_create_with_unrelated_type (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_recursive_cascading_four_levels_has_and_belongs_to_many (unexpected)
--- FAIL: BasicsTest#test_dup_for_a_composite_primary_key_model (unexpected)
--- FAIL: BasicsTest#test_utc_as_time_zone_and_new (unexpected)
--- FAIL: CalculationsTest#test_should_return_decimal_average_of_integer_field (unexpected)
--- FAIL: CalculationsTest#test_count_with_aliased_attribute (unexpected)
--- FAIL: BasicsTest#test_all_with_conditions (unexpected)
--- FAIL: BasicsTest#test_#connected_to_many_with_a_single_argument_for_classes (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_multiple_fields_when_table_name_is_too_long (unexpected)
--- FAIL: CalculationsTest#test_pluck_loaded_relation (unexpected)
--- FAIL: CalculationsTest#test_pluck_loaded_relation_aliased_attribute (unexpected)
--- FAIL: BasicsTest#test_preserving_time_objects_with_utc_time_conversion_to_default_timezone_local (unexpected)
--- FAIL: BasicsTest#test_initialize_with_attributes (unexpected)
--- FAIL: CalculationsTest#test_async_pluck_none_relation (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_with_invalid_field (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_grouped_association_with_invalid_field (unexpected)
--- FAIL: InheritanceTest#test_class_with_store_full_sti_class_returns_full_name (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_grouped_with_longer_field (unexpected)
--- FAIL: CalculationsTest#test_sum_takes_attribute_type_precedence_over_database_type (unexpected)
--- FAIL: CalculationsTest#test_limit_is_kept (unexpected)
--- FAIL: CalculationsTest#test_count_with_too_many_parameters_raises (unexpected)
--- FAIL: BasicsTest#test_singular_table_name_guesses_with_prefixes_and_suffixes (unexpected)
--- FAIL: BasicsTest#test_default_values_on_empty_strings (unexpected)
--- FAIL: BasicsTest#test_scoped_can_take_a_values_hash (unexpected)
--- FAIL: BasicsTest#test_bignum (unexpected)
--- FAIL: CalculationsTest#test_minimum_and_maximum_on_time_attributes (unexpected)
--- FAIL: CalculationsTest#test_should_count_field_in_joined_table_with_group_by_when_tables_share_column_names (unexpected)
--- FAIL: BasicsTest#test_primary_key_and_references_columns_should_be_identical_type (unexpected)
--- FAIL: BasicsTest#test_attributes (unexpected)
--- FAIL: CalculationsTest#test_ids_on_loaded_relation_with_includes_and_table_scope (unexpected)
--- FAIL: CalculationsTest#test_from_option_with_table_different_than_class (unexpected)
--- FAIL: CalculationsTest#test_should_sum_expression (unexpected)
--- FAIL: CalculationsTest#test_count_for_a_composite_primary_key_model (unexpected)
--- FAIL: InheritanceTest#test_alt_inheritance_save (unexpected)
--- FAIL: CalculationsTest#test_ids_with_includes (unexpected)
--- FAIL: CalculationsTest#test_should_count_field_of_root_table_with_conflicting_group_by_column (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15727827/activerecord/1718777524198/1718785247791)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15741028?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15741028?buildTab=artifacts#/activerecord) on master @ [2a6193bba6b7919532a29e0370a860028576cebd](https://github.com/cockroachdb/cockroach/commits/2a6193bba6b7919532a29e0370a860028576cebd):


```
--- FAIL: HasOneAssociationsTest#test_with_polymorphic_has_one_with_custom_columns_name (unexpected)
--- FAIL: ModulesTest#test_assign_ids (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_relationship_cannot_have_a_counter_cache (unexpected)
--- FAIL: HasOneAssociationsTest#test_attributes_are_being_set_when_initialized_from_has_one_association_with_where_clause (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_does_not_use_order_by (unexpected)
--- FAIL: ModulesTest#test_compute_type_can_infer_class_name_of_sibling_inside_module (unexpected)
--- FAIL: ModulesTest#test_find_account_and_include_company (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_with_inexistent_foreign_key_failing (unexpected)
--- FAIL: HasOneAssociationsTest#test_creation_failure_due_to_new_record_should_raise_error (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_with_block (unexpected)
--- FAIL: HasOneAssociationsTest#test_polymorphic_has_one_with_touch_option_on_create_wont_cache_association_so_fetching_after_transaction_commit_works (unexpected)
--- FAIL: HasOneAssociationsTest#test_dependence_with_missing_association (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_proxy_should_not_respond_to_private_methods (unexpected)
--- FAIL: HasOneAssociationsTest#test_replacement_failure_due_to_existing_record_should_raise_error (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_respects_hash_condition (unexpected)
--- FAIL: HasOneAssociationsTest#test_build_and_create_should_not_happen_within_scope (unexpected)
--- FAIL: HasOneAssociationsTest#test_update_with_foreign_and_primary_keys (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_with_touch_option_on_nonpersisted_built_associations_doesnt_update_parent (unexpected)
--- FAIL: HasOneAssociationsTest#test_reload_association (unexpected)
--- FAIL: HasOneAssociationsTest#test_association_enum_works_properly_with_nested_join (unexpected)
--- FAIL: HasOneAssociationsTest#test_building_the_associated_object_with_implicit_sti_base_class (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnNewRecord#test_autosave_new_record_on_has_one_can_be_disabled_per_relationship (unexpected)
--- FAIL: HasOneAssociationsTest#test_destroyed_by_association_set_in_child_destroy_callback_on_parent_destroy (unexpected)
--- FAIL: HasOneAssociationsTest#test_nullification_on_destroyed_association (unexpected)
--- FAIL: HasOneAssociationsTest#test_assignment_before_child_saved (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_association_with_bang_failing (unexpected)
--- FAIL: HasOneAssociationsTest#test_with_select (unexpected)
--- FAIL: HasOneAssociationsTest#test_nullification_on_cpk_association (unexpected)
--- FAIL: ModulesTest#test_module_table_name_prefix (unexpected)
--- FAIL: HasOneAssociationsTest#test_natural_assignment_to_nil (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: HasOneAssociationsTest#test_type_mismatch (unexpected)
--- FAIL: ModulesTest#test_module_table_name_suffix_with_global_suffix (unexpected)
--- FAIL: HasOneAssociationsTest#test_dependency_should_halt_parent_destruction (unexpected)
--- FAIL: ModulesTest#test_module_spanning_associations (unexpected)
--- FAIL: HasOneAssociationsTest#test_dangerous_association_name_raises_ArgumentError (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_with_touch_option_on_create (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_default_pk_and_sequence (unexpected)
--- FAIL: HasOneAssociationsTest#test_destroyed_by_association_set_in_child_destroy_callback_on_replace (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_with_touch_option_on_update (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_create_fixtures_resets_sequences_when_not_cached (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_proxy_should_respond_to_private_methods_via_send (unexpected)
--- FAIL: HasOneAssociationsTest#test_exclusive_dependence (unexpected)
--- FAIL: HasOneAssociationsTest#test_association_keys_bypass_attribute_protection (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_loading_for_new_record (unexpected)
--- FAIL: HasOneAssociationsTest#test_finding_with_interpolated_condition (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15741028/activerecord/1718863747442/1718871588621)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15757215?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15757215?buildTab=artifacts#/activerecord) on master @ [065b94022a26d223b0a19809e510a6dca337f155](https://github.com/cockroachdb/cockroach/commits/065b94022a26d223b0a19809e510a6dca337f155):


```
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_touch_option_on_touch_and_reassigned_parent (unexpected)
--- FAIL: ReflectionTest#test_class_for_source_type (unexpected)
--- FAIL: InheritanceTest#test_company_descends_from_active_record (unexpected)
--- FAIL: ModulesTest#test_module_spanning_associations (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_should_save_parent_but_not_invalid_child (unexpected)
--- FAIL: BelongsToAssociationsTest#test_runs_parent_presence_check_if_parent_has_not_changed_and_belongs_to_required_validates_foreign_key_is_set (unexpected)
--- FAIL: BelongsToAssociationsTest#test_creating_the_belonging_object_with_primary_key (unexpected)
--- FAIL: BelongsToAssociationsTest#test_default_with_lambda (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_with_custom_foreign_type (unexpected)
--- FAIL: BelongsToAssociationsTest#test_where_on_polymorphic_association_with_empty_array (unexpected)
--- FAIL: ReflectionTest#test_always_validate_association_if_explicit (unexpected)
--- FAIL: BelongsToAssociationsTest#test_creating_the_belonging_object (unexpected)
--- FAIL: BelongsToAssociationsTest#test_dependent_delete_and_destroy_with_belongs_to (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_reassign_with_namespaced_models_and_counters (unexpected)
--- FAIL: BelongsToAssociationsTest#test_building_the_belonging_object (unexpected)
--- FAIL: ReflectionTest#test_has_many_reflection_with_array_fk_raises (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_counter_after_update (unexpected)
--- FAIL: BelongsToAssociationsTest#test_should_set_foreign_key_on_create_association (unexpected)
--- FAIL: BelongsToAssociationsTest#test_should_set_composite_foreign_key_on_association_when_key_changes_on_associated_record (unexpected)
--- FAIL: InheritanceTest#test_compute_type_argument_error (unexpected)
--- FAIL: InheritanceTest#test_finding_incorrect_type_data (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_assignment_updates_foreign_id_field_for_new_and_saved_records (unexpected)
--- FAIL: InheritanceTest#test_where_new_with_unrelated_type (unexpected)
--- FAIL: InheritanceTest#test_compute_type_no_method_error (unexpected)
--- FAIL: BelongsToAssociationsTest#test_self_referential_belongs_to_with_counter_cache_assigning_nil (unexpected)
--- FAIL: BelongsToAssociationsTest#test_assigning_belongs_to_on_destroyed_object (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_with_custom_primary_key (unexpected)
--- FAIL: InheritanceTest#test_alt_inheritance_find_all (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_counter_when_update_columns (unexpected)
--- FAIL: InheritanceTest#test_where_create_bang_with_unrelated_type (unexpected)
--- FAIL: ReflectionTest#test_columns (unexpected)
--- FAIL: ReflectionTest#test_join_table (unexpected)
--- FAIL: BelongsToAssociationsTest#test_reassigning_the_parent_id_updates_the_object (unexpected)
--- FAIL: InheritanceTest#test_where_new_with_subclass (unexpected)
--- FAIL: ReflectionTest#test_content_columns (unexpected)
--- FAIL: ReflectionTest#test_validate_association_if_autosave (unexpected)
--- FAIL: BelongsToAssociationsTest#test_stale_tracking_doesn't_care_about_the_type (unexpected)
--- FAIL: InheritanceTest#test_alt_inheritance_condition (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_with_false (unexpected)
--- FAIL: InheritanceTest#test_where_create_with_unrelated_type (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_reassignment_of_associated_id_updates_the_object (unexpected)
--- FAIL: BelongsToAssociationsTest#test_assignment_before_child_saved_with_primary_key (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_primary_key (unexpected)
--- FAIL: BelongsToAssociationsTest#test_association_assignment_sticks (unexpected)
--- FAIL: BelongsToAssociationsTest#test_tracking_change_from_nil_to_new_record (unexpected)
--- FAIL: BelongsToAssociationsTest#test_create_bang_with_block (unexpected)
--- FAIL: InheritanceTest#test_new_with_ar_base (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15757215/activerecord/1718949138445/1718956396985)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15773127?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15773127?buildTab=artifacts#/activerecord) on master @ [8a97a5edd98336e2dd04ef12f08628fba84b17dd](https://github.com/cockroachdb/cockroach/commits/8a97a5edd98336e2dd04ef12f08628fba84b17dd):


```
--- FAIL: EagerAssociationTest#test_attribute_alias_in_where_references_association_name (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_with_create_and_valid_options (unexpected)
--- FAIL: BasicsTest#test_readonly_attributes_on_belongs_to_association (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit_with_no_results (unexpected)
--- FAIL: EagerAssociationTest#test_eager-loading_readonly_association (unexpected)
--- FAIL: EagerAssociationTest#test_order_on_join_table_with_include_and_limit (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_update_counter_caches_on_delete_with_dependent_destroy (unexpected)
--- FAIL: EagerAssociationTest#test_eager_loaded_has_one_association_with_references_does_not_run_additional_queries (unexpected)
--- FAIL: BasicsTest#test_select_symbol (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_get_ids_for_loaded_associations (unexpected)
--- FAIL: BasicsTest#test_equality_of_new_records (unexpected)
--- FAIL: InheritanceTest#test_alt_inheritance_condition (unexpected)
--- FAIL: BasicsTest#test_all (unexpected)
--- FAIL: BasicsTest#test_cannot_call_connected_to_on_the_abstract_class_that_did_not_establish_the_connection (unexpected)
--- FAIL: EagerAssociationTest#test_limited_eager_with_order (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_default_scope_as_block (unexpected)
--- FAIL: InheritanceTest#test_should_store_demodulized_class_name_with_store_full_sti_class_option_disabled (unexpected)
--- FAIL: BasicsTest#test_touch_should_raise_error_on_a_new_object (unexpected)
--- FAIL: BasicsTest#test_ignored_columns_are_not_present_in_columns_hash (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_single_has_many_through_association_with_unpersisted_parent_instance (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_with_a_polymorphic_association_and_using_the_existential_predicate_but_also_using_a_select (unexpected)
--- FAIL: InheritanceTest#test_descends_from_active_record (unexpected)
--- FAIL: EagerAssociationTest#test_eager_has_many_through_multiple_with_order (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_save_fails_for_invalid_has_one (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_join_model_with_conditions (unexpected)
--- FAIL: BasicsTest#test_many_mutations (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_replace_order_is_preserved (unexpected)
--- FAIL: InheritanceTest#test_alt_becomes_bang_resets_inheritance_type_column (unexpected)
--- FAIL: BasicsTest#test_sql_injection_via_find (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_marshal_dump (unexpected)
--- FAIL: BasicsTest#test_initialize_with_attributes (unexpected)
--- FAIL: BasicsTest#test_#connected_to_many_cannot_be_called_on_anything_but_ActiveRecord::Base (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_association_proxy_transaction_method_starts_transaction_in_association_class (unexpected)
--- FAIL: EagerAssociationTest#test_include_has_one_using_primary_key (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_through_empty_belongs_to (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_has_one_reflection (unexpected)
--- FAIL: BasicsTest#test_create_after_initialize_with_array_param (unexpected)
--- FAIL: BasicsTest#test_when_ignored_attribute_is_loaded,_cast_type_should_be_preferred_over_DB_type (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_and_belongs_to_many_and_limit (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_limit (unexpected)
--- FAIL: BasicsTest#test_equality_with_blank_ids (unexpected)
--- FAIL: BasicsTest#test_cannot_call_connects_to_on_non-abstract_or_non-ActiveRecord::Base_classes (unexpected)
--- FAIL: InheritanceTest#test_where_create_with_invalid_type (unexpected)
--- FAIL: EagerAssociationTest#test_exceptions_have_suggestions_for_fix (unexpected)
--- FAIL: InheritanceTest#test_compute_type_nonexistent_constant (unexpected)
--- FAIL: BasicsTest#test_clone_preserves_subtype (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15773127/activerecord/1719035457914/1719043469286)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15782445?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15782445?buildTab=artifacts#/activerecord) on master @ [c0749a3e9d45fb57f8dfa0936626437b887cda56](https://github.com/cockroachdb/cockroach/commits/c0749a3e9d45fb57f8dfa0936626437b887cda56):


```
--- FAIL: PersistenceTest#test_update! (unexpected)
--- FAIL: PersistenceTest#test_raises_error_when_validations_failed (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_escaped_characters (unexpected)
--- FAIL: DirtyTest#test_attribute_will_change!_doesn't_try_to_save_non-persistable_attributes (unexpected)
--- FAIL: RelationTest#test_references_with_blank_value (unexpected)
--- FAIL: StoreTest#test_saved_changes_tracking_for_accessors_with_json_column (unexpected)
--- FAIL: StoreTest#test_convert_store_attributes_from_any_format_other_than_Hash_or_HashWithIndifferentAccess_losing_the_data (unexpected)
--- FAIL: FinderTest#test_joins_with_string_array (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_annotate_raise_errors (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_includes_in_through_association_scope (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_both_parent_ids_set_when_saving_new (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_incorrectly_ordered_through_associations (unexpected)
--- FAIL: FinderTest#test_last_with_irreversible_order (unexpected)
--- FAIL: FinderTest#test_find_with_ids_returning_ordered (unexpected)
--- FAIL: AutomaticInverseFindingTests#test_has_many_with_scoped_belongs_to_does_not_find_inverse_automatically (unexpected)
--- FAIL: RelationTest#test_scoped (unexpected)
--- FAIL: RelationTest#test_finding_with_assoc_order (unexpected)
--- FAIL: FinderTest#test_find_by!_doesn't_have_implicit_ordering (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_a_time_zone-aware_time_with_DST (unexpected)
--- FAIL: UniquenessValidationTest#test_validate_uniqueness_with_conditions_with_record_arg (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_attributes_not_backed_by_database_columns_keep_their_type_when_a_default_value_is_configured_separately (unexpected)
--- FAIL: RelationTest#test_find_by!_with_non-hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: PersistenceTest#test_decrement_with_touch_an_attribute_updates_timestamps (unexpected)
--- FAIL: FinderTest#test_find_by_one_attribute_that_is_an_alias (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_association_through_a_belongs_to_association_where_the_association_doesnt_exist (unexpected)
--- FAIL: RelationTest#test_support_upper_and_lower_case_directions (unexpected)
--- FAIL: RelationTest#test_find_by_requires_at_least_one_argument (unexpected)
--- FAIL: FinderTest#test_find_by_one_attribute_that_is_an_aggregate (unexpected)
--- FAIL: DirtyTest#test_partial_insert_off_with_unchanged_default_function_attribute (unexpected)
--- FAIL: PersistenceTest#test_update_many_with_duplicated_ids! (unexpected)
--- FAIL: PostgresqlIntervalTest#test_average_interval_type (unexpected)
--- FAIL: CockroachDB::EagerLoadPolyAssocsTest#test_include_query (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_overloaded_properties_save (unexpected)
--- FAIL: FinderTest#test_find_some_message_with_custom_primary_key (unexpected)
--- FAIL: FinderTest#test_eager_load_for_no_has_many_with_limit_and_joins_for_has_many (unexpected)
--- FAIL: ReflectionTest#test_validate_association_if_autosave (unexpected)
--- FAIL: PersistenceTest#test_build_many_through_factory_with_block (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_array (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_two_levels_with_condition (unexpected)
--- FAIL: TestAutosaveAssociationsInGeneral#test_autosave_works_even_when_other_callbacks_update_the_parent_model (unexpected)
--- FAIL: RelationTest#test_unscope_with_blank_value (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_dynamic_find_should_respect_association_include (unexpected)
--- FAIL: ReflectionTest#test_belongs_to_reflection_with_array_fk_raises (unexpected)
--- FAIL: DirtyTest#test_mutating_and_then_assigning_doesn't_remove_the_change (unexpected)
--- FAIL: RelationTest#test_first_or_create_with_array (unexpected)
--- FAIL: DirtyTest#test_save_should_store_serialized_attributes_even_with_partial_writes (unexpected)
--- FAIL: RelationTest#test_to_sql_on_eager_join (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15782445/activerecord/1719121926689/1719125840563)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15792338?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15792338?buildTab=artifacts#/activerecord) on master @ [f28d796e9710067c76d335efee3a45c85552d2bb](https://github.com/cockroachdb/cockroach/commits/f28d796e9710067c76d335efee3a45c85552d2bb):


```
--- FAIL: BelongsToAssociationsTest#test_create_bang_with_conditions (unexpected)
--- FAIL: AttributeMethodsTest#test_write_attribute_raises_ActiveModel::MissingAttributeError_when_the_attribute_does_not_exist (unexpected)
--- FAIL: BelongsToAssociationsTest#test_dependent_delete_and_destroy_with_belongs_to (unexpected)
--- FAIL: SchemaWithDotsTest#test_Active_Record_basics (unexpected)
--- FAIL: BelongsToAssociationsTest#test_assignment_before_child_saved (unexpected)
--- FAIL: AttributeMethodsTest#test_overridden_write_attribute (unexpected)
--- FAIL: BelongsToAssociationsTest#test_attributes_are_being_set_when_initialized_from_belongs_to_association_with_where_clause (unexpected)
--- FAIL: AttributeMethodsTest#test_non-attribute_read_and_write (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_writers_respect_access_control (unexpected)
--- FAIL: AttributeMethodsTest#test_on_the_fly_super_invokable_generated_attribute_methods_via_method_missing (unexpected)
--- FAIL: BelongsToAssociationsTest#test_reloading_association_with_key_change (unexpected)
--- FAIL: AttributeMethodsTest#test_declared_prefixed_attribute_method_affects_respond_to?_and_method_missing (unexpected)
--- FAIL: AttributeMethodsTest#test_instance_methods_should_be_defined_on_the_base_class (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_a_time_zone-aware_attribute_to_a_blank_string_returns_nil (unexpected)
--- FAIL: AttributeMethodsTest#test_converted_values_are_returned_after_assignment (unexpected)
--- FAIL: AttributeMethodsTest#test_boolean_attribute_predicate (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_names_with_a_custom_select (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_touch_option_on_touch_without_updated_at_attributes (unexpected)
--- FAIL: AttributeMethodsTest#test_accessed_fields (unexpected)
--- FAIL: BelongsToAssociationsTest#test_destroying_child_with_unloaded_parent_and_foreign_key_and_touch_is_possible_with_has_many_inversing (unexpected)
--- FAIL: BelongsToAssociationsTest#test_counter_cache_double_destroy (unexpected)
--- FAIL: ActiveRecord::Migration::ForeignKeyChangeColumnTest#test_remove_foreign_key_by_column (unexpected)
--- FAIL: AttributeMethodsTest#test_boolean_attributes (unexpected)
--- FAIL: BelongsToAssociationsTest#test_default_with_lambda (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_readers_respect_access_control (unexpected)
--- FAIL: AttributeMethodsTest#test_#id_value_alias_returns_id_column_only_for_composite_primary_key_models (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_reassignment_of_associated_type_updates_the_object (unexpected)
--- FAIL: AttributeMethodsTest#test_read_overridden_attribute_with_predicate_respects_override (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_children_can_override_parents (unexpected)
--- FAIL: AttributeMethodsTest#test_set_attributes_with_a_block (unexpected)
--- FAIL: DDLTest#test_column_defaults (unexpected)
--- FAIL: BelongsToAssociationsTest#test_id_assignment (unexpected)
--- FAIL: TransactionCallbacksTest#test_only_call_after_commit_on_create_and_doesnt_leaky (unexpected)
--- FAIL: ActiveRecord::Migration::ColumnsTest#test_rename_column (unexpected)
--- FAIL: AttributeMethodsTest#test_undeclared_attribute_method_does_not_affect_respond_to?_and_method_missing (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::ConnectionHandlersShardingDbTest#test_sharding_separation (unexpected)
--- FAIL: BelongsToAssociationsTest#test_with_polymorphic_and_condition (unexpected)
--- FAIL: ValidationsTest#test_valid_using_special_context (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attribute_can_read_aliased_attributes_as_well (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_enum_method_issues_a_deprecation_warning (unexpected)
--- FAIL: AttributeMethodsTest#test_should_unserialize_attributes_for_frozen_records (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::ConnectionHandlersShardingDbTest#test_same_shards_across_clusters (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_overloaded_properties_save (unexpected)
--- FAIL: ValidationsTest#test_validate_with_bang_and_context (unexpected)
--- FAIL: BelongsToAssociationsTest#test_eager_loading_wont_mutate_owner_record (unexpected)
--- FAIL: BelongsToAssociationsTest#test_custom_attribute_with_select (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_touch_option_on_destroy (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15792338/activerecord/1719208401459/1719215863739)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15811757?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15811757?buildTab=artifacts#/activerecord) on master @ [9af051e4151254a8ef10ce50a0990f7c26605226](https://github.com/cockroachdb/cockroach/commits/9af051e4151254a8ef10ce50a0990f7c26605226):


```
--- FAIL: InheritanceTest#test_new_without_storing_full_sti_class (unexpected)
--- FAIL: PersistenceTest#test_delete_isnt_affected_by_scoping (unexpected)
--- FAIL: FinderTest#test_with_limiting_with_custom_select (unexpected)
--- FAIL: RelationTest#test_primary_key (unexpected)
--- FAIL: PersistenceTest#test_update_many_with_duplicated_ids (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attribute (unexpected)
--- FAIL: PersistenceTest#test_update_many_with_invalid_id (unexpected)
--- FAIL: CalculationsTest#test_should_return_zero_if_sum_conditions_return_nothing (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attribute_when_false (unexpected)
--- FAIL: RelationTest#test_find_all_using_where_with_relation_with_joins (unexpected)
--- FAIL: RelationTest#test_multivalue_where (unexpected)
--- FAIL: CalculationsTest#test_ids_for_a_composite_primary_key_on_loaded_relation (unexpected)
--- FAIL: TestDestroyAsPartOfAutosaveAssociation#test_should_run_remove_callback_methods_for_habtm (unexpected)
--- FAIL: AttributeMethodsTest#test_typecast_attribute_from_select_to_false (unexpected)
--- FAIL: RelationTest#test_count (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_many_should_return_true_if_more_than_one (unexpected)
--- FAIL: FinderTest#test_last_bang_missing (unexpected)
--- FAIL: BasicsTest#test_create_without_prepared_statement (unexpected)
--- FAIL: IntegrationTest#test_cache_key_for_existing_record_is_not_timezone_dependent (unexpected)
--- FAIL: PersistenceTest#test_update_columns_with_default_scope (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_composite_primary_key_autosave (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_grafts_stashed_associations_to_correct_parent (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_nil_with_aggregate_having_one_mapping (unexpected)
--- FAIL: HasManyAssociationsTest#test_get_ids_for_ordered_association (unexpected)
--- FAIL: RelationTest#test_first_or_create_bang_with_valid_array (unexpected)
--- FAIL: BasicsTest#test_when_ignored_attribute_is_loaded,_cast_type_should_be_preferred_over_DB_type (unexpected)
--- FAIL: AssociationsTest#test_querying_by_whole_associated_records_using_query_constraints (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_association (unexpected)
--- FAIL: AttributeMethodsTest#test_case-sensitive_attributes_hash (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_for_inspect_with_a_non-primary_key_id_attribute (unexpected)
--- FAIL: EagerAssociationTest#test_eager_load_has_many_with_string_keys (unexpected)
--- FAIL: PersistenceTest#test_update_many_with_duplicated_ids! (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_with_or_doesnt_set_inverse_instance_key (unexpected)
--- FAIL: BasicsTest#test_find_symbol_ordered_last (unexpected)
--- FAIL: AttributeMethodsTest#test_generated_attribute_methods_ancestors_have_correct_module (unexpected)
--- FAIL: PersistenceTest#test_destroyed_returns_boolean (unexpected)
--- FAIL: EagerAssociationTest#test_including_duplicate_objects_from_belongs_to (unexpected)
--- FAIL: BasicsTest#test_connection_in_utc_time (unexpected)
--- FAIL: PersistenceTest#test_update_all_with_hash (unexpected)
--- FAIL: CalculationsTest#test_pluck_joined_with_polymorphic_relation (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_has_many_sti (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_destroy_on_association_clears_scope (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_many_should_count_instead_of_loading_association (unexpected)
--- FAIL: BasicsTest#test_bignum (unexpected)
--- FAIL: RelationTest#test_default_scoping_finder_methods (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_collection_delete_with_nonstandard_primary_key_on_belongs_to (unexpected)
--- FAIL: CalculationsTest#test_ids_with_includes_and_table_scope (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15811757/activerecord/1719294647608/1719304846271)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15826395?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15826395?buildTab=artifacts#/activerecord) on master @ [0e5781f3e2004ef6222e1ed4f5311b771936bd6a](https://github.com/cockroachdb/cockroach/commits/0e5781f3e2004ef6222e1ed4f5311b771936bd6a):


```
--- FAIL: CalculationsTest#test_should_resolve_aliased_attributes (unexpected)
--- FAIL: ActiveRecord::Encryption::EncryptableRecordApiTest#test_encrypted_attribute?_returns_false_for_regular_attributes (unexpected)
--- FAIL: UniquenessValidationWithIndexTest#test_new_record (unexpected)
--- FAIL: HasManyAssociationsTest#test_select_with_block_and_dirty_target (unexpected)
--- FAIL: BelongsToAssociationsTest#test_destroying_child_with_unloaded_parent_and_foreign_key_and_touch_is_possible_with_has_many_inversing (unexpected)
--- FAIL: RelationTest#test_multivalue_where (unexpected)
--- FAIL: BasicsTest#test_equality_of_collection_proxy_and_association_relation (unexpected)
--- FAIL: FinderTest#test_find_by_ids_with_limit_and_offset (unexpected)
--- FAIL: BasicsTest#test_dup_of_saved_object_marks_as_dirty_only_changed_attributes (unexpected)
--- FAIL: BelongsToAssociationsTest#test_optional_relation_can_be_set_per_model (unexpected)
--- FAIL: HasManyAssociationsTest#test_select_with_block (unexpected)
--- FAIL: FinderTest#test_find_with_string (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_escaped_characters (unexpected)
--- FAIL: FinderTest#test_second_with_offset (unexpected)
--- FAIL: RelationTest#test_first_or_create (unexpected)
--- FAIL: CalculationsTest#test_limit_is_kept (unexpected)
--- FAIL: RelationTest#test_order_with_hash_and_symbol_generates_the_same_sql (unexpected)
--- FAIL: UniquenessValidationWithIndexTest#test_expression_index (unexpected)
--- FAIL: BasicsTest#test_assert_queries (unexpected)
--- FAIL: BasicsTest#test_#present?_and_#blank?_on_ActiveRecord::Base_classes (unexpected)
--- FAIL: FinderTest#test_find_with_group_and_sanitized_having_method (unexpected)
--- FAIL: ActiveRecord::CockroachDB::Migration::ColumnsTest#test_change_column_default (unexpected)
--- FAIL: CallbacksOnDestroyUpdateActionRaceTest#test_trigger_on_update_where_row_was_deleted (unexpected)
--- FAIL: AssociationsTest#test_bad_collection_keys (unexpected)
--- FAIL: HasManyAssociationsTest#test_ids_reader_cache_not_used_for_size_when_association_is_dirty (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_default_pk_and_sequence (unexpected)
--- FAIL: UniquenessValidationWithIndexTest#test_non_unique_index (unexpected)
--- FAIL: CalculationsTest#test_count_takes_attribute_type_precedence_over_database_type (unexpected)
--- FAIL: FinderTest#test_find_with_hash_parameter (unexpected)
--- FAIL: InheritanceTest#test_find_first_within_inheritance (unexpected)
--- FAIL: BasicsTest#test_failed_comparison_of_unlike_class_records (unexpected)
--- FAIL: HasManyAssociationsTest#test_three_levels_of_dependence (unexpected)
--- FAIL: ActiveRecord::Encryption::EncryptableRecordTest#test_deterministic_attributes_can_be_searched_with_Active_Record_queries (unexpected)
--- FAIL: CalculationsTest#test_ids_with_includes_limit_and_empty_result (unexpected)
--- FAIL: CockroachDB::AdapterTestWithoutTransaction#test_reset_empty_table_with_custom_pk_sequence (unexpected)
--- FAIL: RelationTest#test_count_with_block (unexpected)
--- FAIL: BooleanTest#test_boolean (unexpected)
--- FAIL: CalculationsTest#test_ids_with_includes_and_scope (unexpected)
--- FAIL: HasManyAssociationsTest#test_unscopes_the_default_scope_of_associated_model_when_used_with_include (unexpected)
--- FAIL: FinderTest#test_include_on_unloaded_relation_with_match (unexpected)
--- FAIL: FinderTest#test_exists_with_includes_limit_and_empty_result (unexpected)
--- FAIL: HasManyAssociationsTest#test_transactions_when_adding_to_new_record (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_many_should_defer_to_collection_if_using_a_block (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_on_association_clears_scope (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_is_precise_enough (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_replace_on_new_object (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15826395/activerecord/1719381200907/1719386458296)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15840902?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15840902?buildTab=artifacts#/activerecord) on master @ [d2a5af43be05ade40531e6e0ef23c5880faa4494](https://github.com/cockroachdb/cockroach/commits/d2a5af43be05ade40531e6e0ef23c5880faa4494):


```
--- FAIL: FinderTest#test_find_by_with_range_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: FinderTest#test_first_bang_missing (unexpected)
--- FAIL: FinderTest#test_find_by_sql_with_sti_on_joined_table (unexpected)
--- FAIL: FinderTest#test_find_by_title_and_id_with_hash (unexpected)
--- FAIL: FinderTest#test_find_by!_doesn't_have_implicit_ordering (unexpected)
--- FAIL: FinderTest#test_find_passing_active_record_object_is_not_permitted (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: FinderTest#test_find_with_ids_with_limit_and_order_clause (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
--- FAIL: FinderTest#test_with_limiting_with_custom_select (unexpected)
--- FAIL: FinderTest#test_fifth (unexpected)
--- FAIL: FinderTest#test_exists_with_eager_load (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_nil_with_aggregate_having_one_mapping (unexpected)
--- FAIL: FinderTest#test_find_by_two_attributes_with_one_being_an_aggregate (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_hashed_table_name (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions (unexpected)
--- FAIL: FinderTest#test_last_with_integer_and_order_should_keep_the_order (unexpected)
--- FAIL: FinderTest#test_#find_by_with_composite_primary_key (unexpected)
--- FAIL: FinderTest#test_first (unexpected)
--- FAIL: FinderTest#test_model_class_responds_to_second_bang (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_create_fixtures_resets_sequences_when_not_cached (unexpected)
--- FAIL: FinderTest#test_nth_to_last_with_order_uses_limit (unexpected)
--- FAIL: FinderTest#test_third_to_last (unexpected)
--- FAIL: FinderTest#test_find_by_one_attribute_with_conditions (unexpected)
--- FAIL: FinderTest#test_#first_for_a_model_with_composite_query_constraints (unexpected)
--- FAIL: FinderTest#test_model_class_responds_to_third_to_last_bang (unexpected)
--- FAIL: FinderTest#test_find_by_ids (unexpected)
--- FAIL: FinderTest#test_condition_local_time_interpolation_with_default_timezone_utc (unexpected)
--- FAIL: FinderTest#test_find_by_one_attribute_bang (unexpected)
--- FAIL: FinderTest#test_hash_condition_utc_time_interpolation_with_default_timezone_local (unexpected)
--- FAIL: FinderTest#test_find_with_large_number (unexpected)
--- FAIL: FinderTest#test_find_by_bang_on_relation_with_large_number (unexpected)
--- FAIL: FinderTest#test_include_on_unloaded_relation_with_offset (unexpected)
--- FAIL: FinderTest#test_find_by_records (unexpected)
--- FAIL: FinderTest#test_find_on_a_scope_does_not_perform_statement_caching (unexpected)
--- FAIL: FinderTest#test_exists_with_string (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_default_pk_and_sequence (unexpected)
--- FAIL: FinderTest#test_model_class_responds_to_last_bang (unexpected)
--- FAIL: FinderTest#test_include_on_unloaded_relation_with_having_referencing_aliased_select (unexpected)
--- FAIL: FinderTest#test_find_without_primary_key (unexpected)
--- FAIL: FinderTest#test_take (unexpected)
--- FAIL: FinderTest#test_exists_with_distinct_association_includes_and_limit (unexpected)
--- FAIL: FinderTest#test_exists_returns_true_with_one_record_and_no_args (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_aggregate_having_three_mappings_array (unexpected)
--- FAIL: AssociationCallbacksTest#test_has_many_callbacks_for_destroy_on_parent (unexpected)
--- FAIL: FinderTest#test_exists_with_nil_arg (unexpected)
--- FAIL: FinderTest#test_eager_load_for_no_has_many_with_limit_and_left_joins_for_has_many (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15840902/activerecord/1719467436848/1719476695446)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15853573?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15853573?buildTab=artifacts#/activerecord) on master @ [b4402507fdebe5682f77e348331a308d378c91e1](https://github.com/cockroachdb/cockroach/commits/b4402507fdebe5682f77e348331a308d378c91e1):


```
--- FAIL: ReflectionTest#test_columns (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_hmt_does_not_table_name_collide_when_joining_associations (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_where_first_level_returns_nil (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_default_scope_as_lambda (unexpected)
--- FAIL: PersistenceTest#test_becomes_includes_errors (unexpected)
--- FAIL: HasManyAssociationsTest#test_attributes_are_set_when_initialized_from_has_many_null_relationship (unexpected)
--- FAIL: EagerAssociationTest#test_polymorphic_type_condition (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_two_levels_and_self_table_reference (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_invalid_adding (unexpected)
--- FAIL: BelongsToAssociationsTest#test_should_set_foreign_key_on_create_association! (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_names_on_a_new_record (unexpected)
--- FAIL: RelationTest#test_order_by_relation_attribute (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attribute_when_false (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_deleting (unexpected)
--- FAIL: BasicsTest#test_dup_for_a_composite_primary_key_model (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnNewRecord#test_autosave_new_record_on_has_many_can_be_disabled_per_relationship (unexpected)
--- FAIL: RelationTest#test_having_with_binds_for_both_where_and_having (unexpected)
--- FAIL: RelationTest#test_explicit_create_with (unexpected)
--- FAIL: AttributeMethodsTest#test_bulk_update_raises_ActiveRecord::UnknownAttributeError (unexpected)
--- FAIL: HasManyAssociationsTest#test_has_many_associations_on_new_records_use_null_relations (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_limit_and_offset_and_conditions (unexpected)
--- FAIL: BasicsTest#test_equality (unexpected)
--- FAIL: RelationTest#test_references_doesnt_trigger_eager_loading_if_reference_not_included (unexpected)
--- FAIL: RelationTest#test_joins_with_select_custom_attribute (unexpected)
--- FAIL: HasManyAssociationsTest#test_select_query_method (unexpected)
--- FAIL: EagerAssociationTest#test_eager_count_performed_on_a_has_many_through_association_with_multi_table_conditional (unexpected)
--- FAIL: EagerAssociationTest#test_eager-loading_non-readonly_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_grouped (unexpected)
--- FAIL: InheritanceTest#test_alt_becomes_bang_resets_inheritance_type_column (unexpected)
--- FAIL: BelongsToAssociationsTest#test_association_assignment_sticks (unexpected)
--- FAIL: BasicsTest#test_distinct_delegates_to_scoped (unexpected)
--- FAIL: HasManyAssociationsTest#test_updates_counter_cache_when_default_scope_is_given (unexpected)
--- FAIL: RelationTest#test_first_or_create_bang_with_valid_array (unexpected)
--- FAIL: AssociationsTest#test_belongs_to_a_model_with_composite_foreign_key_finds_associated_record (unexpected)
--- FAIL: PersistenceTest#test_increment_nil_attribute (unexpected)
--- FAIL: PersistenceTest#test_update_column (unexpected)
--- FAIL: BasicsTest#test_clone_of_new_object_marks_as_dirty_only_changed_attributes (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_dynamic_find_all_should_respect_readonly_access (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_should_append_to_association_order (unexpected)
--- FAIL: BasicsTest#test_touch_should_raise_error_on_a_new_object (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_has_one_cpk_has_one_autosave_with_id (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_new_with_values_in_collection (unexpected)
--- FAIL: PersistenceTest#test_persisted_returns_boolean (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_for_inspect_with_a_long_array (unexpected)
--- FAIL: RelationTest#test_loaded_first (unexpected)
--- FAIL: CallbackOrderTest#test_callbacks_run_in_order_defined_in_model_if_using_run_after_transaction_callbacks_in_order_defined (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15853573/activerecord/1719553851875/1719562804118)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15867122?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15867122?buildTab=artifacts#/activerecord) on master @ [047a7ed79756eef53b8b9ab4c9dd9c5a463496c9](https://github.com/cockroachdb/cockroach/commits/047a7ed79756eef53b8b9ab4c9dd9c5a463496c9):


```
--- FAIL: DefaultBinaryTest#test_default_varbinary_string_that_looks_like_hex (unexpected)
--- FAIL: CalculationsTest#test_aggregate_attribute_on_enum_type (unexpected)
--- FAIL: BelongsToAssociationsTest#test_eager_loading_with_primary_key_as_symbol (unexpected)
--- FAIL: CalculationsTest#test_pluck_type_cast_with_joins_without_table_name_qualified_column (unexpected)
--- FAIL: BasicsTest#test_equality (unexpected)
--- FAIL: BasicsTest#test_default_in_local_time (unexpected)
--- FAIL: CalculationsTest#test_count_selected_arel_attribute_in_joined_table (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_invalid_dependent_option_raises_exception (unexpected)
--- FAIL: CalculationsTest#test_distinct_joins_count_with_order_and_offset (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
--- FAIL: BasicsTest#test_non_valid_identifier_column_name (unexpected)
--- FAIL: BelongsToAssociationsTest#test_should_set_foreign_key_on_save (unexpected)
--- FAIL: SignedIdTest#test_fail_to_work_without_when_signed_id_verifier_secret_lambda_is_nil (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_qualified_column_name (unexpected)
--- FAIL: CalculationsTest#test_count_with_block (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_primary_key_joins_on_correct_column (unexpected)
--- FAIL: BasicsTest#test_utc_as_time_zone_and_new (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_on_relation (unexpected)
--- FAIL: CalculationsTest#test_pluck_auto_table_name_prefix (unexpected)
--- FAIL: BelongsToAssociationsTest#test_association_assignment_sticks (unexpected)
--- FAIL: CalculationsTest#test_select_avg_with_group_by_as_virtual_attribute_with_sql (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_with_custom_foreign_type (unexpected)
--- FAIL: CalculationsTest#test_pick_loaded_relation (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_for_single_table_inheritance_(STI_Models) (unexpected)
--- FAIL: BasicsTest#test_equality_of_relation_and_array (unexpected)
--- FAIL: CalculationsTest#test_minimum_and_maximum_on_non_numeric_type (unexpected)
--- FAIL: BasicsTest#test_limit_without_comma (unexpected)
--- FAIL: BasicsTest#test_readonly_attributes_when_configured_to_not_raise (unexpected)
--- FAIL: CalculationsTest#test_pick_delegate_to_all (unexpected)
--- FAIL: CalculationsTest#test_should_count_field_in_joined_table_with_group_by_when_tables_share_column_names (unexpected)
--- FAIL: BasicsTest#test_marshal_inspected_round_trip (unexpected)
--- FAIL: CalculationsTest#test_should_count_manual_select_with_include (unexpected)
--- FAIL: CalculationsTest#test_should_get_maximum_of_arel_attribute_with_include (unexpected)
--- FAIL: BasicsTest#test_sequence_name_with_abstract_class (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_assignment_with_primary_key_updates_foreign_id_field_for_new_and_saved_records (unexpected)
--- FAIL: BelongsToAssociationsTest#test_tracking_polymorphic_changes (unexpected)
--- FAIL: BelongsToAssociationsTest#test_tracking_change_from_one_persisted_record_to_another (unexpected)
--- FAIL: BasicsTest#test_find_ordered_last (unexpected)
--- FAIL: BasicsTest#test_quote (unexpected)
--- FAIL: BelongsToAssociationsTest#test_assignment_updates_foreign_id_field_for_new_and_saved_records (unexpected)
--- FAIL: CalculationsTest#test_should_return_float_average_if_db_returns_such (unexpected)
--- FAIL: BasicsTest#test_out_of_range_slugs (unexpected)
--- FAIL: BasicsTest#test_#connected_to_many_cannot_be_called_on_anything_but_ActiveRecord::Base (unexpected)
--- FAIL: CalculationsTest#test_count_with_block_and_column_name_raises_an_error (unexpected)
--- FAIL: BelongsToAssociationsTest#test_runs_parent_presence_check_if_parent_has_not_changed_and_belongs_to_required_validates_foreign_key_is_set (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_line_endings (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15867122/activerecord/1719640223401/1719649417746)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15875228?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15875228?buildTab=artifacts#/activerecord) on master @ [d13253527955eaa2da09394b8a2729627ab25c48](https://github.com/cockroachdb/cockroach/commits/d13253527955eaa2da09394b8a2729627ab25c48):


```
--- FAIL: BelongsToAssociationsTest#test_new_record_with_foreign_key_but_no_object (unexpected)
--- FAIL: CalculationsTest#test_pluck_not_auto_table_name_prefix_if_column_joined (unexpected)
--- FAIL: PersistenceTest#test_build (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_join_model_with_conditions_on_top_level (unexpected)
--- FAIL: DirtyTest#test_time_attributes_changes_without_time_zone (unexpected)
--- FAIL: InheritanceTest#test_finding_incorrect_type_data (unexpected)
--- FAIL: RelationTest#test_finding_with_subquery_with_binds (unexpected)
--- FAIL: CalculationsTest#test_no_queries_for_empty_relation_on_sum (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_a_time_zone-aware_time_in_the_current_time_zone (unexpected)
--- FAIL: HasOneThroughDisableJoinsAssociationsTest#test_preload_on_disable_joins_through (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_limit_and_multiple_associations (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_saving_multiple_relationships (unexpected)
--- FAIL: HasManyAssociationsTest#test_joins_with_namespaced_model_should_use_correct_type (unexpected)
--- FAIL: RelationTest#test_create_or_find_by (unexpected)
--- FAIL: RelationTest#test_to_json (unexpected)
--- FAIL: MultiParameterAttributeTest#test_multiparameter_attributes_on_time_with_time_zone_aware_attributes_false (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_optimizer_hints_raise_errors (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_multiple_stis_and_order (unexpected)
--- FAIL: IntegrationTest#test_to_param_for_a_composite_primary_key_model (unexpected)
--- FAIL: HasManyAssociationsTest#test_depends_and_nullify_on_polymorphic_assoc (unexpected)
--- FAIL: EagerAssociationTest#test_eager_loading_with_conditions_on_join_model_preloads (unexpected)
--- FAIL: JsonAttributeTest#test_select_array_json_value (unexpected)
--- FAIL: ActiveRecord::Migration::CheckConstraintTest#test_not_valid_check_constraint (unexpected)
--- FAIL: RelationTest#test_eager_load_with_blank_value (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_is_precise_enough (unexpected)
--- FAIL: RelationMergingTest#test_merging_compares_symbols_and_strings_as_equal (unexpected)
--- FAIL: ReflectionTest#test_join_table (unexpected)
--- FAIL: ModulesTest#test_module_table_name_suffix (unexpected)
--- FAIL: ValidationsTest#test_validates_acceptance_of_as_database_column (unexpected)
--- FAIL: ReflectionTest#test_scope_chain_does_not_interfere_with_hmt_with_polymorphic_case (unexpected)
--- FAIL: HasManyAssociationsTest#test_update_all_respects_association_scope (unexpected)
--- FAIL: MigrationTest#test_add_drop_table_with_prefix_and_suffix (unexpected)
--- FAIL: BasicsTest#test_quote (unexpected)
--- FAIL: InheritanceAttributeMappingTest#test_polymorphic_associations_custom_type (unexpected)
--- FAIL: HabtmDestroyOrderTest#test_not_destroying_a_student_with_lessons_leaves_student<=>lesson_association_intact (unexpected)
--- FAIL: CalculationsTest#test_pick_one (unexpected)
--- FAIL: ModulesTest#test_module_spanning_has_and_belongs_to_many_associations (unexpected)
--- FAIL: RelationTest#test_relations_limit_the_records_in_#inspect_at_10 (unexpected)
--- FAIL: EagerAssociationTest#test_eager_loading_with_select_on_joined_table_preloads (unexpected)
--- FAIL: RelationTest#test_to_sql_on_scoped_proxy (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_association_replaces_existing_with_dependent_option (unexpected)
--- FAIL: BasicsTest#test_reload (unexpected)
--- FAIL: UniquenessValidationWithIndexTest#test_new_record (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_multiple_associations_with_same_table_belongs_to (unexpected)
--- FAIL: ValidationsTest#test_invalid_record_exception (unexpected)
--- FAIL: CalculationsTest#test_offset_is_kept (unexpected)
--- FAIL: BasePreventWritesTest#test_preventing_writes_applies_to_all_connections_in_block (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15875228/activerecord/1719726926658/1719735200102)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15882733?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15882733?buildTab=artifacts#/activerecord) on master @ [d13253527955eaa2da09394b8a2729627ab25c48](https://github.com/cockroachdb/cockroach/commits/d13253527955eaa2da09394b8a2729627ab25c48):


```
--- FAIL: AttributeMethodsTest#test_came_from_user? (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_followed_by_save_does_not_load_target (unexpected)
--- FAIL: AttributeMethodsTest#test_attributes_for_database (unexpected)
--- FAIL: HasManyAssociationsTest#test_merging_with_custom_attribute_writer (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_a_time_zone-aware_datetime_in_the_current_time_zone (unexpected)
--- FAIL: HasManyAssociationsTest#test_counting_with_single_hash (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attributes_after_type_cast_on_a_date (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_ids_and_inverse_of (unexpected)
--- FAIL: AttributeMethodsTest#test_custom_field_attribute_predicate (unexpected)
--- FAIL: HasManyAssociationsTest#test_create_with_bang_on_has_many_raises_when_record_not_saved (unexpected)
--- FAIL: HasManyAssociationsTest#test_collection_not_empty_after_building (unexpected)
--- FAIL: HasManyAssociationsTest#test_key_ensuring_owner_was_is_not_valid_without_dependent_option (unexpected)
--- FAIL: AttributeMethodsTest#test_global_methods_are_overwritten_when_subclassing (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_first_after_reload (unexpected)
--- FAIL: AttributeMethodsTest#test_string_attribute_predicate (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_names_on_a_new_record (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_without_loading_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_by_integer_id (unexpected)
--- FAIL: HasManyAssociationsTest#test_default_select (unexpected)
--- FAIL: HasManyAssociationsTest#test_inverse_on_before_validate (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_default_pk_and_sequence (unexpected)
--- FAIL: HasManyAssociationsTest#test_create_from_association_should_respect_default_scope (unexpected)
--- FAIL: HasManyAssociationsTest#test_update_all_on_association_accessed_before_save (unexpected)
--- FAIL: HasManyAssociationsTest#test_include_uses_array_include_after_loaded (unexpected)
--- FAIL: HasManyAssociationsTest#test_subsequent_calls_to_many_should_use_query (unexpected)
--- FAIL: AttributeMethodsTest#test_user-defined_JSON_attribute_predicate (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroy_all (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_an_overridden_original_method_along_with_an_overridden_alias_method_doesn't_issue_a_deprecation (unexpected)
--- FAIL: HasManyAssociationsTest#test_associations_autosaves_when_object_is_already_persisted (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_with_extend_option_with_multiple_extensions (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_time_zone-aware_attribute_in_other_time_zone (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependence_with_transaction_support_on_failure (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_size_on_an_association_that_has_not_been_loaded_performs_a_query (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_with_rewhere_doesnt_set_inverse_instance_key (unexpected)
--- FAIL: HasManyAssociationsTest#test_collection_proxy_respects_default_scope (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroy_with_bang_bubbles_errors_from_associations (unexpected)
--- FAIL: HasManyAssociationsTest#test_joining_through_a_polymorphic_association_with_a_where_clause (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all_on_association_with_nil_dependency_is_the_same_as_not_loaded (unexpected)
--- FAIL: AttributeMethodsTest#test_time_attributes_are_retrieved_in_the_current_time_zone (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attribute_with_nil_should_not_asplode (unexpected)
--- FAIL: AttributeMethodsTest#test_respond_to?_with_a_custom_primary_key (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_protect_foreign_key (unexpected)
--- FAIL: HasManyAssociationsTest#test_counter_cache_updates_in_memory_after_create_with_array (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_names_with_a_custom_select (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_attributes_with_overridden_types_keep_their_type_when_a_default_value_is_configured_separately (unexpected)
--- FAIL: HasManyAssociationsTest#test_creating_using_primary_key (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15882733/activerecord/1719813195309/1719820642124)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #124901 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.1-rc]
- #123542 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.0-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15898281?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15898281?buildTab=artifacts#/activerecord) on master @ [9e1cd533828b7887a48db8635f705669287cf2d6](https://github.com/cockroachdb/cockroach/commits/9e1cd533828b7887a48db8635f705669287cf2d6):


```
--- FAIL: ReflectionTest#test_never_validate_association_if_explicit (unexpected)
--- FAIL: ReflectionTest#test_columns (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_build_many_before_save (unexpected)
--- FAIL: ReflectionTest#test_aggregation_reflection (unexpected)
--- FAIL: DefaultBinaryTest#test_default_varbinary_string_that_looks_like_hex (unexpected)
--- FAIL: ReflectionTest#test_columns_are_returned_in_the_order_they_were_declared (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_cascaded_eager_association_loading_with_duplicated_includes (unexpected)
--- FAIL: ReflectionTest#test_column_null_not_null (unexpected)
--- FAIL: ReflectionTest#test_join_table_can_be_overridden (unexpected)
--- FAIL: ReflectionTest#test_association_primary_key_raises_when_missing_primary_key (unexpected)
--- FAIL: DefaultBinaryTest#test_default_varbinary_string (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_has_many_sti (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_three_levels_by_ping_pong (unexpected)
--- FAIL: ReflectionTest#test_human_name (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_adding_unsavable_association (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_nulls_not_distinct (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_index_length (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_two_levels_and_self_table_reference (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_of_stis_with_multiple_references (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_assign_ids_with_belongs_to_cpk_model (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: TestDefaultAutosaveAssociationOnNewRecord#test_autosave_new_record_on_belongs_to_can_be_disabled_per_relationship (unexpected)
--- FAIL: ReflectionTest#test_scope_chain_of_polymorphic_association_does_not_leak_into_other_hmt_associations (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_where_first_level_returns_nil (unexpected)
--- FAIL: ReflectionTest#test_type (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_join_for_count (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_build_via_block_before_save (unexpected)
--- FAIL: ReflectionTest#test_foreign_type (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_default_pk_and_sequence (unexpected)
--- FAIL: ReflectionTest#test_association_primary_key_uses_explicit_primary_key_option_as_first_priority (unexpected)
--- FAIL: ReflectionTest#test_reflect_on_association_accepts_strings (unexpected)
--- FAIL: ReflectionTest#test_association_primary_key (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_recursive_cascading_four_levels_has_and_belongs_to_many (unexpected)
--- FAIL: ReflectionTest#test_reflect_on_association_accepts_symbols (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_invalid_adding_with_validate_false (unexpected)
--- FAIL: ReflectionTest#test_chain (unexpected)
--- FAIL: ReflectionTest#test_has_and_belongs_to_many_reflection (unexpected)
--- FAIL: ReflectionTest#test_has_many_reflection (unexpected)
--- FAIL: ReflectionTest#test_active_record_primary_key (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_parent_should_save_children_record_with_foreign_key_validation_set_in_before_save_callback (unexpected)
--- FAIL: ReflectionTest#test_foreign_key_is_inferred_from_model_name (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_two_levels_with_condition (unexpected)
--- FAIL: ReflectionTest#test_reflections_should_return_keys_as_strings (unexpected)
--- FAIL: ReflectionTest#test_human_name_for_column (unexpected)
--- FAIL: ReflectionTest#test_nested? (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_multiple_stis_and_order (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15898281/activerecord/1719899580420/1719907523377)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15916952?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15916952?buildTab=artifacts#/activerecord) on master @ [7461a85f839c50edbdea502fe09794739a99632e](https://github.com/cockroachdb/cockroach/commits/7461a85f839c50edbdea502fe09794739a99632e):


```
--- FAIL: FinderTest#test_find_by_ids (unexpected)
--- FAIL: HasOneAssociationsTest#test_build (unexpected)
--- FAIL: BasicsTest#test_primary_key_and_references_columns_should_be_identical_type (unexpected)
--- FAIL: PersistenceTest#test_update_many_with_invalid_id (unexpected)
--- FAIL: HasOneAssociationsTest#test_proxy_assignment (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates_after_parameterize_with_hyphens (unexpected)
--- FAIL: InheritanceTest#test_find_first_within_inheritance (unexpected)
--- FAIL: CalculationsTest#test_ids_with_preload (unexpected)
--- FAIL: CalculationsTest#test_should_reference_correct_aliases_while_joining_tables_of_has_many_through_association (unexpected)
--- FAIL: ReflectionTest#test_scope_chain_does_not_interfere_with_hmt_with_polymorphic_case_and_subclass_source (unexpected)
--- FAIL: RelationTest#test_find_by_with_non-hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: InheritanceTest#test_inheritance_with_default_scope (unexpected)
--- FAIL: CalculationsTest#test_should_sum_scoped_field (unexpected)
--- FAIL: CalculationsTest#test_select_avg_with_joins_and_group_by_as_virtual_attribute_with_ar (unexpected)
--- FAIL: CalculationsTest#test_distinct_count_with_order_and_limit (unexpected)
--- FAIL: RelationTest#test_create_or_find_by_should_not_raise_due_to_validation_errors (unexpected)
--- FAIL: BasicsTest#test_initialize_with_attributes (unexpected)
--- FAIL: ReflectionTest#test_type (unexpected)
--- FAIL: HasOneAssociationsTest#test_save_of_record_with_loaded_has_one (unexpected)
--- FAIL: HasOneThroughDisableJoinsAssociationsTest#test_has_one_through_with_belongs_to_on_disable_joins (unexpected)
--- FAIL: InheritanceTest#test_abstract_class (unexpected)
--- FAIL: RelationTest#test_to_sql_on_eager_join (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_field_having_condition_from_select (unexpected)
--- FAIL: ReflectionTest#test_irregular_reflection_class_name (unexpected)
--- FAIL: FinderTest#test_exists_with_includes_limit_and_empty_result (unexpected)
--- FAIL: BasicsTest#test_equality_with_blank_ids (unexpected)
--- FAIL: ReflectionTest#test_content_columns (unexpected)
--- FAIL: CalculationsTest#test_maximum_with_not_auto_table_name_prefix_if_column_included (unexpected)
--- FAIL: FinderTest#test_member_on_unloaded_relation_with_mismatched_class (unexpected)
--- FAIL: AssociationsTest#test_append_composite_foreign_key_has_many_association_with_autosave (unexpected)
--- FAIL: CalculationsTest#test_should_return_zero_if_sum_conditions_return_nothing (unexpected)
--- FAIL: BasicsTest#test_.columns_hash_raises_an_error_if_the_record_has_an_empty_table_name (unexpected)
--- FAIL: FinderTest#test_eager_load_for_no_has_many_with_limit_and_joins_for_has_many (unexpected)
--- FAIL: BasicsTest#test_new_record_returns_boolean (unexpected)
--- FAIL: BasicsTest#test_find_keeps_multiple_order_values (unexpected)
--- FAIL: ReflectionTest#test_human_name (unexpected)
--- FAIL: RelationTest#test_last (unexpected)
--- FAIL: HasOneAssociationsTest#test_build_respects_hash_condition (unexpected)
--- FAIL: CalculationsTest#test_pluck_not_auto_table_name_prefix_if_column_joined (unexpected)
--- FAIL: ReflectionTest#test_column_string_type_and_limit (unexpected)
--- FAIL: RelationTest#test_first_or_create_bang_with_no_parameters (unexpected)
--- FAIL: FinderTest#test_find_on_array_conditions (unexpected)
--- FAIL: AssociationsTest#test_loading_cpk_association_when_persisted_and_in_memory_differ (unexpected)
--- FAIL: RelationTest#test_joins_with_select (unexpected)
--- FAIL: RelationTest#test_build_with_array (unexpected)
--- FAIL: HasOneAssociationsTest#test_building_the_associated_object_with_an_invalid_type (unexpected)
--- FAIL: RelationTest#test_finding_with_order_concatenated (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15916952/activerecord/1719986195155/1719995085281)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15930183?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15930183?buildTab=artifacts#/activerecord) on master @ [291ce755a1692b23b986c2250a3209a1b8ca5428](https://github.com/cockroachdb/cockroach/commits/291ce755a1692b23b986c2250a3209a1b8ca5428):


```
--- FAIL: BelongsToAssociationsTest#test_create_with_conditions (unexpected)
--- FAIL: FinderTest#test_last_with_integer_and_order_should_keep_the_order (unexpected)
--- FAIL: IntegrationTest#test_param_delimiter_changes_delimiter_used_in_to_param (unexpected)
--- FAIL: IntegrationTest#test_cache_key_retains_version_when_custom_timestamp_is_used (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_eager_load_raise_errors (unexpected)
--- FAIL: CounterCacheTest#test_increment_counter (unexpected)
--- FAIL: RelationTest#test_create_bang_with_array (unexpected)
--- FAIL: EagerAssociationTest#test_finding_with_includes_on_has_one_association_with_same_include_includes_only_once (unexpected)
--- FAIL: RelationTest#test_one (unexpected)
--- FAIL: RelationTest#test_finding_with_arel_order (unexpected)
--- FAIL: RelationTest#test_finding_with_order_and_take (unexpected)
--- FAIL: RelationTest#test_extracted_association (unexpected)
--- FAIL: RelationTest#test_unscope_with_table_name_qualified_column (unexpected)
--- FAIL: I18nValidationTest#test_validates_uniqueness_of_on_generated_message_given_option_that_is_not_reserved (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_and_create_from_association_should_respect_passed_attributes_over_default_scope (unexpected)
--- FAIL: FinderTest#test_find_by_returns_nil_if_the_record_is_missing (unexpected)
--- FAIL: BelongsToAssociationsTest#test_runs_parent_presence_check_if_parent_has_not_changed_and_belongs_to_required_validates_foreign_key_is_set (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_with_touch_option_on_create (unexpected)
--- FAIL: CoreTest#test_inspect_instance (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_reassign_with_namespaced_models_and_counters (unexpected)
--- FAIL: ModulesTest#test_assign_ids (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_proxy_should_not_respond_to_private_methods (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_keys_bypass_attribute_protection (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_time_zone-aware_read_attribute (unexpected)
--- FAIL: BelongsToAssociationsTest#test_self_referential_belongs_to_with_counter_cache_assigning_nil (unexpected)
--- FAIL: FinderTest#test_find_passing_active_record_object_is_not_permitted (unexpected)
--- FAIL: HasManyAssociationsTest#test_collection_association_with_private_kernel_method (unexpected)
--- FAIL: TransactionTest#test_failing_on_exception (unexpected)
--- FAIL: FinderTest#test_exists_with_empty_hash_arg (unexpected)
--- FAIL: CounterCacheTest#test_reset_the_right_counter_if_two_have_the_same_class_name (unexpected)
--- FAIL: BelongsToAssociationsTest#test_should_set_foreign_key_on_save (unexpected)
--- FAIL: ReflectionTest#test_reflection_klass_requires_ar_subclass (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_by_integer_id (unexpected)
--- FAIL: HasOneAssociationsTest#test_create (unexpected)
--- FAIL: BelongsToAssociationsTest#test_reflect_the_most_recent_change (unexpected)
--- FAIL: SerializedAttributeTest#test_serialized_boolean_value_true (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_empty_array_with_aggregate_having_multiple_mappings (unexpected)
--- FAIL: HasOneAssociationsTest#test_natural_assignment (unexpected)
--- FAIL: FinderTest#test_#skip_query_cache!_for_#exists? (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_association_replaces_existing_without_dependent_option (unexpected)
--- FAIL: MultiParameterAttributeTest#test_multiparameter_attributes_on_time_will_ignore_hour_if_missing (unexpected)
--- FAIL: SerializationTest#test_read_attribute_for_serialization_with_format_after_find (unexpected)
--- FAIL: HasManyAssociationsTest#test_create_without_loading_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_load_target_respects_protected_attributes (unexpected)
--- FAIL: RelationTest#test_first_or_initialize (unexpected)
--- FAIL: TransactionTest#test_callback_rollback_in_create (unexpected)
--- FAIL: HasManyAssociationsTest#test_finding (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15930183/activerecord/1720041700453/1720049292362)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15932264?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15932264?buildTab=artifacts#/activerecord) on master @ [8f72a01e81d4751c6615a9a864b8e32076565b09](https://github.com/cockroachdb/cockroach/commits/8f72a01e81d4751c6615a9a864b8e32076565b09):


```
--- FAIL: BelongsToAssociationsTest#test_belongs_to_does_not_use_order_by (unexpected)
--- FAIL: BasicsTest#test_quoted_table_name_after_set_table_name (unexpected)
--- FAIL: RelationTest#test_to_sql_on_scoped_proxy (unexpected)
--- FAIL: PostgresqlBitStringTest#test_roundtrip (unexpected)
--- FAIL: HasManyAssociationsTest#test_default_select (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_none_should_return_false_if_any (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_find_in_association (unexpected)
--- FAIL: FinderTest#test_find_by_empty_ids (unexpected)
--- FAIL: PersistenceTest#test_update_columns_should_not_leave_the_object_dirty (unexpected)
--- FAIL: HasManyAssociationsTest#test_dangerous_association_name_raises_ArgumentError (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroying_by_integer_id (unexpected)
--- FAIL: BelongsToAssociationsTest#test_id_assignment (unexpected)
--- FAIL: FinderTest#test_find_by_with_range_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_associations_with_string_joins_and_order_references (unexpected)
--- FAIL: SerializedAttributeTestWithYamlSafeLoad#test_nil_not_serialized_without_class_constraint (unexpected)
--- FAIL: AssociationsTest#test_assign_composite_foreign_key_belongs_to_association_with_autosave (unexpected)
--- FAIL: AttributeMethodsTest#test_bulk_update_raises_ActiveRecord::UnknownAttributeError (unexpected)
--- FAIL: I18nGenerateMessageValidationTest#test_generate_message_taken_with_custom_message (unexpected)
--- FAIL: HasManyAssociationsTest#test_finder_bang_method_with_dirty_target (unexpected)
--- FAIL: RelationTest#test_automatically_added_order_references (unexpected)
--- FAIL: I18nValidationTest#test_validates_uniqueness_of_on_generated_message_given_custom_message (unexpected)
--- FAIL: SerializedAttributeTestWithYamlSafeLoad#test_serialized_class_attribute (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_reselect_raise_errors (unexpected)
--- FAIL: PersistenceTest#test_update_columns_not_equal_attributes (unexpected)
--- FAIL: PersistenceTest#test_create_many_through_factory_with_block (unexpected)
--- FAIL: UpdateAllTest#test_update_all_with_non_standard_table_name (unexpected)
--- FAIL: PostgresqlDateTest#test_save_infinity_and_beyond (unexpected)
--- FAIL: BasicsTest#test_sql_injection_via_find (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_updates_counter_cache_without_dependent_option (unexpected)
--- FAIL: FinderTest#test_find_with_hash_conditions_on_joined_table_and_with_range (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_within_expiration_duration (unexpected)
--- FAIL: PersistenceTest#test_becomes_after_reload_schema_from_cache (unexpected)
--- FAIL: FinderTest#test_find_an_empty_array (unexpected)
--- FAIL: HasOneThroughDisableJoinsAssociationsTest#test_nil_on_disable_joins_through (unexpected)
--- FAIL: DefaultScopingTest#test_create_attribute_overwrites_default_scoping (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_and_create_from_association_should_respect_unscope_over_default_scope (unexpected)
--- FAIL: HasManyAssociationsTest#test_merging_with_custom_attribute_writer (unexpected)
--- FAIL: FinderTest#test_find_with_hash_conditions_on_joined_table (unexpected)
--- FAIL: BelongsToAssociationsTest#test_natural_assignment_to_nil (unexpected)
--- FAIL: HasManyAssociationsTest#test_prevent_double_firing_the_before_save_callback_of_new_object_when_the_parent_association_saved_in_the_callback (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_self_type_mismatch (unexpected)
--- FAIL: HasManyAssociationsTest#test_attributes_are_being_set_when_initialized_from_has_many_association_with_multiple_where_clauses (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_collection_exists (unexpected)
--- FAIL: HasManyAssociationsTest#test_transaction_when_deleting_new_record (unexpected)
--- FAIL: PostgresqlInfinityTest#test_type_casting_string_on_a_float_column (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_readers_respect_access_control (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15932264/activerecord/1720072337115/1720080269891)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15943208?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15943208?buildTab=artifacts#/activerecord) on master @ [295d09a88895a69e5cc9149fb8165acf78e39e61](https://github.com/cockroachdb/cockroach/commits/295d09a88895a69e5cc9149fb8165acf78e39e61):


```
(orm_helpers.go:229).summarizeFailed: 
Tests run on Cockroach v24.2.0-alpha.2-dev-295d09a88895a69e5cc9149fb8165acf78e39e61
Tests run against activerecord 7.1.3
8108 Total Tests Run
8077 tests passed
31 tests failed
66 tests skipped
0 tests ignored
0 tests passed unexpectedly
5 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: DefaultBinaryTest#test_default_varbinary_string_that_looks_like_hex (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_create_fixtures_resets_sequences_when_not_cached (unexpected)
--- FAIL: DefaultBinaryTest#test_default_varbinary_string (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_default_pk_and_sequence (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15943208/activerecord/1720158768941/1720166221469)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15953434?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15953434?buildTab=artifacts#/activerecord) on master @ [c557fb59f6aec659d364e9002fc083c59c6392b6](https://github.com/cockroachdb/cockroach/commits/c557fb59f6aec659d364e9002fc083c59c6392b6):


```
--- FAIL: BasicsTest#test_clone_of_new_object_with_defaults (unexpected)
--- FAIL: PersistenceTest#test_raises_error_when_validations_failed (unexpected)
--- FAIL: PersistenceTest#test_reload_removes_custom_selects (unexpected)
--- FAIL: BasicsTest#test_scoped_can_take_a_values_hash (unexpected)
--- FAIL: HasOneAssociationsTest#test_polymorphic_has_one_with_touch_option_on_update_will_touch_record_by_fetching_from_database_if_needed (unexpected)
--- FAIL: BasicsTest#test_find_by_slug_with_array (unexpected)
--- FAIL: BasicsTest#test_default_values_are_deeply_dupped (unexpected)
--- FAIL: HasOneAssociationsTest#test_finding_using_primary_key (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_store_association_in_two_relations_with_one_save_in_existing_object_with_values (unexpected)
--- FAIL: EagerAssociationTest#test_including_associations_with_extensions_and_an_instance_dependent_scope_is_supported (unexpected)
--- FAIL: PersistenceTest#test_destroy (unexpected)
--- FAIL: InheritanceTest#test_should_store_demodulized_class_name_with_store_full_sti_class_option_disabled (unexpected)
--- FAIL: PersistenceTest#test_update_attribute_for_updated_at_on (unexpected)
--- FAIL: InheritanceTest#test_update_all_within_inheritance (unexpected)
--- FAIL: HasOneAssociationsTest#test_dependence (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_with_scope_with_joins (unexpected)
--- FAIL: PersistenceTest#test_update_columns (unexpected)
--- FAIL: ModulesTest#test_module_table_name_suffix_with_global_suffix (unexpected)
--- FAIL: PersistenceTest#test_increment_attribute_by (unexpected)
--- FAIL: BasicsTest#test_can_call_connected_to_with_role_and_shard_on_abstract_classes (unexpected)
--- FAIL: InheritanceTest#test_new_with_unrelated_type (unexpected)
--- FAIL: PersistenceTest#test_build_many_through_factory_with_block (unexpected)
--- FAIL: BasicsTest#test_#connected_to_many_cannot_be_called_with_classes_that_include_ActiveRecord::Base (unexpected)
--- FAIL: PersistenceTest#test_update_attribute (unexpected)
--- FAIL: HasOneAssociationsTest#test_building_the_associated_object_with_an_invalid_type (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_join_model_with_include (unexpected)
--- FAIL: BasicsTest#test_connection_in_local_time (unexpected)
--- FAIL: BasicsTest#test_set_table_name_with_inheritance (unexpected)
--- FAIL: HasOneAssociationsTest#test_association_enum_works_properly_with_nested_join (unexpected)
--- FAIL: InheritanceTest#test_compute_type_on_undefined_method (unexpected)
--- FAIL: BasicsTest#test_table_exists (unexpected)
--- FAIL: EagerAssociationTest#test_loading_with_one_association_with_non_preload (unexpected)
--- FAIL: HasOneAssociationsTest#test_nullification_on_destroyed_association (unexpected)
--- FAIL: EagerAssociationTest#test_preload_has_many_with_association_condition_and_default_scope (unexpected)
--- FAIL: EagerAssociationTest#test_with_two_tables_in_from_without_getting_double_quoted (unexpected)
--- FAIL: EagerAssociationTest#test_joins_with_includes_should_preload_via_joins (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_should_save_parent_but_not_invalid_child (unexpected)
--- FAIL: PersistenceTest#test_delete_all (unexpected)
--- FAIL: BasicsTest#test_successful_comparison_of_like_class_records (unexpected)
--- FAIL: EagerAssociationTest#test_eager_loaded_has_one_association_without_primary_key (unexpected)
--- FAIL: BasicsTest#test_ignored_columns_are_stored_as_an_array_of_string (unexpected)
--- FAIL: HasOneAssociationsTest#test_build_with_block (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_autosave_with_primary_key_manually_set (unexpected)
--- FAIL: HasOneAssociationsTest#test_reset_association (unexpected)
--- FAIL: InheritanceTest#test_inheritance_new_with_subclass (unexpected)
--- FAIL: HasOneAssociationsTest#test_replacement_failure_due_to_new_record_should_raise_error (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_cache_nils (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/cpu_arch=arm64/run_1
```

Parameters:
 - <code>ROACHTEST_arch=arm64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15953434/activerecord/1720245505381/1720255685771)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15962692?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15962692?buildTab=artifacts#/activerecord) on master @ [485975b3a824c68c07340e6a336c7864c00d3c6d](https://github.com/cockroachdb/cockroach/commits/485975b3a824c68c07340e6a336c7864c00d3c6d):


```
--- FAIL: BasicsTest#test_sequence_name_with_abstract_class (unexpected)
--- FAIL: HasManyAssociationsTest#test_adding (unexpected)
--- FAIL: BasicsTest#test_readonly_attributes (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_a_item_which_is_not_in_the_collection (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_nil (unexpected)
--- FAIL: HasManyAssociationsTest#test_first_or_initialize_adds_the_record_to_the_association (unexpected)
--- FAIL: BasicsTest#test_out_of_range_slugs (unexpected)
--- FAIL: InverseAssociationTests#test_this_inverse_stuff (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all_with_option_nullify (unexpected)
--- FAIL: CalculationsTest#test_minimum_and_maximum_on_non_numeric_type (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_models_with_composite_keys (unexpected)
--- FAIL: BasicsTest#test_marshal_between_processes (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependent_association_respects_optional_sanitized_conditions_on_delete (unexpected)
--- FAIL: PostgresqlTimestampWithTimeZoneTest#test_timestamp_with_zone_values_with_rails_time_zone_support_and_timestamptz_and_no_time_zone_set (unexpected)
--- FAIL: PersistenceTest#test_create_many_through_factory_with_block (unexpected)
--- FAIL: FinderTest#test_third_with_offset (unexpected)
--- FAIL: HasManyAssociationsTest#test_select_with_block_and_dirty_target (unexpected)
--- FAIL: PersistenceTest#test_destroy_many (unexpected)
--- FAIL: BasicsTest#test_connection_handler_can_be_overridden (unexpected)
--- FAIL: BasicsTest#test_create_after_initialize_without_block (unexpected)
--- FAIL: AttributeMethodsTest#test_user-defined_time_attribute_predicate (unexpected)
--- FAIL: CalculationsTest#test_apply_distinct_in_count (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependent_association_respects_optional_conditions_on_delete (unexpected)
--- FAIL: CalculationsTest#test_should_count_field_of_root_table_with_conflicting_group_by_column (unexpected)
--- FAIL: ReflectionTest#test_chain (unexpected)
--- FAIL: FinderTest#test_find_by_ids_missing_one (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_first_or_last_with_integer_on_association_should_not_load_association (unexpected)
--- FAIL: FinderTest#test_last_on_relation_with_limit_and_offset (unexpected)
--- FAIL: CalculationsTest#test_pluck_on_aliased_attribute (unexpected)
--- FAIL: IntegrationTest#test_to_param_returns_id_if_not_persisted_but_id_is_set (unexpected)
--- FAIL: FinderTest#test_find_doesnt_have_implicit_ordering (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_update_changing_ids_of_inversed_association_changes_the_counter_cache (unexpected)
--- FAIL: AttributeMethodsTest#test_user-defined_text_attribute_predicate (unexpected)
--- FAIL: BasicsTest#test_auto_id (unexpected)
--- FAIL: CalculationsTest#test_should_sum_arel_attribute (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_field_with_conditions_and_having (unexpected)
--- FAIL: HasManyAssociationsTest#test_restrict_with_exception (unexpected)
--- FAIL: FinderTest#test_exists_with_left_joins (unexpected)
--- FAIL: HasManyAssociationsTest#test_depends_and_nullify_on_polymorphic_assoc (unexpected)
--- FAIL: HasManyAssociationsTest#test_get_ids (unexpected)
--- FAIL: CalculationsTest#test_should_not_perform_joined_include_by_default (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_updates_counter_cache_with_dependent_destroy (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_array_of_ranges (unexpected)
--- FAIL: HasManyAssociationsTest#test_dangerous_association_name_raises_ArgumentError (unexpected)
--- FAIL: AttributeMethodsTest#test_method_overrides_in_multi-level_subclasses (unexpected)
--- FAIL: PersistenceTest#test_increment_nil_attribute (unexpected)
--- FAIL: BasicsTest#test_#connected_to_many_cannot_be_called_on_anything_but_ActiveRecord::Base (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/cpu_arch=arm64/run_1
```

Parameters:
 - <code>ROACHTEST_arch=arm64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15962692/activerecord/1720331811596/1720343116129)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15971625?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15971625?buildTab=artifacts#/activerecord) on master @ [485975b3a824c68c07340e6a336c7864c00d3c6d](https://github.com/cockroachdb/cockroach/commits/485975b3a824c68c07340e6a336c7864c00d3c6d):


```
--- FAIL: HasManyAssociationsTest#test_destroy_on_association_clears_scope (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_indexes_for_existent_table (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_all (unexpected)
--- FAIL: CalculationsTest#test_count_with_column_parameter (unexpected)
--- FAIL: BasicsTest#test_table_name_based_on_model_name (unexpected)
--- FAIL: PersistenceTest#test_update_attribute_for_aborted_callback! (unexpected)
--- FAIL: RelationTest#test_except (unexpected)
--- FAIL: ReflectionTest#test_chain (unexpected)
--- FAIL: BelongsToAssociationsTest#test_dangerous_association_name_raises_ArgumentError (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_first_after_reload (unexpected)
--- FAIL: FinderTest#test_model_class_responds_to_second_to_last_bang (unexpected)
--- FAIL: BasicsTest#test_#connected_to_many_sets_prevent_writes_if_role_is_reading (unexpected)
--- FAIL: HasOneAssociationsTest#test_update_with_foreign_and_primary_keys (unexpected)
--- FAIL: RelationTest#test_finding_with_cross_table_order_and_limit (unexpected)
--- FAIL: BasicsTest#test_when_ignored_attribute_is_loaded,_cast_type_should_be_preferred_over_DB_type (unexpected)
--- FAIL: PersistenceTest#test_save_nil_string_attributes (unexpected)
--- FAIL: BasicsTest#test_preserving_time_objects_with_local_time_conversion_to_default_timezone_utc (unexpected)
--- FAIL: EagerAssociationTest#test_should_work_inverse_of_with_eager_load (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_none_should_return_false_if_any (unexpected)
--- FAIL: PersistenceTest#test_update_uses_query_constraints_config (unexpected)
--- FAIL: BasicsTest#test_columns_should_obey_set_primary_key (unexpected)
--- FAIL: BelongsToAssociationsTest#test_missing_attribute_error_is_raised_when_no_foreign_key_attribute (unexpected)
--- FAIL: PersistenceTest#test_destroy_for_a_failed_to_destroy_cpk_record (unexpected)
--- FAIL: FinderTest#test_find_by_bang_on_relation_with_large_number (unexpected)
--- FAIL: BasicsTest#test_default_char_types (unexpected)
--- FAIL: CalculationsTest#test_ids_for_a_composite_primary_key (unexpected)
--- FAIL: RelationTest#test_find_ids (unexpected)
--- FAIL: HasManyAssociationsTest#test_in_memory_replacements_sets_inverse_instance (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates_case_shown_in_doc (unexpected)
--- FAIL: HasManyAssociationsTest#test_create_followed_by_save_does_not_load_target (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_type_cast_does_not_corrupt_the_query_cache (unexpected)
--- FAIL: PersistenceTest#test_increment_aliased_attribute (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_preloaded_records_are_not_duplicated (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all_with_option_delete_all (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_with_scope_with_joins (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_joining_has_many_through_with_distinct (unexpected)
--- FAIL: BelongsToAssociationsTest#test_natural_assignment_with_primary_key (unexpected)
--- FAIL: PersistenceTest#test_update_all_with_hash (unexpected)
--- FAIL: CalculationsTest#test_limit_is_kept (unexpected)
--- FAIL: PersistenceTest#test_update_many_with_invalid_id (unexpected)
--- FAIL: EagerAssociationTest#test_associations_loaded_for_all_records (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_build_then_save_with_has_one_inverse (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates_edge_case (unexpected)
--- FAIL: EagerAssociationTest#test_string_id_column_joins (unexpected)
--- FAIL: BasicsTest#test_quoting_arrays (unexpected)
--- FAIL: RelationTest#test_default_reverse_order_on_table_without_primary_key (unexpected)
--- FAIL: BasicsTest#test_abstract_class_table_name (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15971625/activerecord/1720417916384/1720426162081)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15990049?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15990049?buildTab=artifacts#/activerecord) on master @ [6b22014dec8c5e8e6006d710ed05bab5b01b4ae2](https://github.com/cockroachdb/cockroach/commits/6b22014dec8c5e8e6006d710ed05bab5b01b4ae2):


```
--- FAIL: InheritanceTest#test_inheritance_find_all (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_one_condition_being_aggregate_and_another_not (unexpected)
--- FAIL: FinderTest#test_find_by!_with_non-hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_assignment_with_primary_key_foreign_type_field_updating (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_readonly_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_ids_and_inverse_of (unexpected)
--- FAIL: BasicsTest#test_ignored_columns_not_included_in_SELECT (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_many_via_block (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_with_touch_option_on_nonpersisted_built_associations_doesnt_update_parent (unexpected)
--- FAIL: InheritanceTest#test_new_with_ar_base (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_conditions_string_with_unquoted_table_name (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_attributes_are_available_to_after_initialize (unexpected)
--- FAIL: BasicsTest#test_readonly_attributes (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_not_resaved_when_unchanged (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_relationship_cannot_have_a_counter_cache (unexpected)
--- FAIL: HasManyAssociationsTest#test_counter_cache_updates_in_memory_after_create_with_array (unexpected)
--- FAIL: HasManyAssociationsTest#test_with_polymorphic_has_many_with_custom_columns_name (unexpected)
--- FAIL: FinderTest#test_find_with_a_multiple_sets_of_composite_primary_key (unexpected)
--- FAIL: ReflectionTest#test_includes_accepts_strings (unexpected)
--- FAIL: BelongsToAssociationsTest#test_default_with_lambda (unexpected)
--- FAIL: BelongsToAssociationsTest#test_building_the_belonging_object_with_sti_subclass (unexpected)
--- FAIL: FinderTest#test_third_to_last_have_primary_key_order_by_default (unexpected)
--- FAIL: HasManyAssociationsTest#test_create_followed_by_save_does_not_load_target (unexpected)
--- FAIL: HasManyAssociationsTest#test_create_with_bang_on_has_many_when_parent_is_new_raises (unexpected)
--- FAIL: BasicsTest#test_default_in_local_time (unexpected)
--- FAIL: HasManyAssociationsTest#test_building_the_associated_object_with_an_unrelated_type (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_two_levels_and_self_table_reference (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_that_should_not_be_fully_merged (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_with_touch_option_on_empty_update (unexpected)
--- FAIL: BelongsToAssociationsTest#test_tracking_polymorphic_changes (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_does_not_cache_has_many_association_subset_when_preloaded_with_a_through_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_first_nth_or_last_on_existing_record_with_build_should_load_association (unexpected)
--- FAIL: InheritanceTest#test_class_with_blank_sti_name (unexpected)
--- FAIL: ReflectionTest#test_read_attribute_names (unexpected)
--- FAIL: HasOneAssociationsTest#test_with_polymorphic_has_one_with_custom_columns_name (unexpected)
--- FAIL: HasManyAssociationsTest#test_counting (unexpected)
--- FAIL: HasManyAssociationsTest#test_depends_and_nullify (unexpected)
--- FAIL: InheritanceTest#test_inheritance_new_with_base_class (unexpected)
--- FAIL: BelongsToAssociationsTest#test_reloading_association_with_key_change (unexpected)
--- FAIL: HasManyAssociationsTest#test_no_sql_should_be_fired_if_association_already_loaded (unexpected)
--- FAIL: FinderTest#test_named_bind_variables (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroy_does_not_raise_when_association_errors_on_destroy (unexpected)
--- FAIL: BelongsToAssociationsTest#test_dependency_should_halt_parent_destruction_with_cascaded_three_levels (unexpected)
--- FAIL: FinderTest#test_find_by_association_subquery (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_through_has_one_association_with_order (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_build_then_save_with_has_many_inverse (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_two_levels (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15990049/activerecord/1720504456555/1720512819541)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16005104?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16005104?buildTab=artifacts#/activerecord) on master @ [0b918d1dc3a9ce1f04975202be3b04ec375a816e](https://github.com/cockroachdb/cockroach/commits/0b918d1dc3a9ce1f04975202be3b04ec375a816e):


```
--- FAIL: HasManyAssociationsTest#test_calling_first_nth_or_last_on_existing_record_with_build_should_load_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_empty_on_an_association_that_has_been_loaded_does_not_performs_query (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependence_for_associations_with_hash_condition (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_first_nth_or_last_on_new_record_should_not_run_queries (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_many_on_loaded_association_should_not_use_query (unexpected)
--- FAIL: HasManyAssociationsTest#test_building_the_associated_object_with_an_invalid_type (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_self_type_mismatch (unexpected)
--- FAIL: HasManyAssociationsTest#test_select_with_block (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all_on_association_with_nil_dependency_is_the_same_as_not_loaded (unexpected)
--- FAIL: HasManyAssociationsTest#test_dangerous_association_name_raises_ArgumentError (unexpected)
--- FAIL: HasManyAssociationsTest#test_collection_size_twice_for_regressions (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroy_with_bang_bubbles_errors_from_associations (unexpected)
--- FAIL: HasManyAssociationsTest#test_dynamic_find_should_respect_association_order_for_through (unexpected)
--- FAIL: HasManyAssociationsTest#test_get_ids (unexpected)
--- FAIL: HasManyAssociationsTest#test_taking (unexpected)
--- FAIL: HasManyAssociationsTest#test_counting_with_column_name_and_hash (unexpected)
--- FAIL: HasManyAssociationsTest#test_finding_array_compatibility (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_one_should_return_false_if_more_than_one (unexpected)
--- FAIL: HasManyAssociationsTest#test_create_with_bang_on_has_many_when_parent_is_new_raises (unexpected)
--- FAIL: HasManyAssociationsTest#test_first_or_initialize_adds_the_record_to_the_association (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: HasManyAssociationsTest#test_include_checks_if_record_exists_if_target_not_loaded (unexpected)
--- FAIL: HasManyAssociationsTest#test_counter_cache_on_unloaded_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependent_association_respects_optional_sanitized_conditions_on_delete (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all_with_option_delete_all (unexpected)
--- FAIL: HasManyAssociationsTest#test_extend_option_affects_per_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_collection_empty_with_dirty_target (unexpected)
--- FAIL: HasManyAssociationsTest#test_clearing_an_association_collection (unexpected)
--- FAIL: HasManyAssociationsTest#test_in_memory_replacement_executes_no_queries (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_size_on_an_association_that_has_been_loaded_does_not_perform_query (unexpected)
--- FAIL: HasManyAssociationsTest#test_replace_with_less_and_dependent_nullify (unexpected)
--- FAIL: HasManyAssociationsTest#test_counting_with_counter_sql (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_none_should_return_false_if_any (unexpected)
--- FAIL: HasManyAssociationsTest#test_transactions_when_replacing_on_new_record (unexpected)
--- FAIL: HasManyAssociationsTest#test_get_ids_for_loaded_associations (unexpected)
--- FAIL: HasManyAssociationsTest#test_prevent_double_insertion_of_new_object_when_the_parent_association_loaded_in_the_after_save_callback (unexpected)
--- FAIL: HasManyAssociationsTest#test_counter_cache_updates_in_memory_after_create (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_with_rewhere_doesnt_set_inverse_instance_key (unexpected)
--- FAIL: HasManyAssociationsTest#test_defining_has_many_association_with_nullify_dependency_lazily_evaluates_target_class (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependence (unexpected)
--- FAIL: HasManyAssociationsTest#test_subsequent_calls_to_one_should_use_query (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_empty_on_an_association_that_has_not_been_loaded_performs_a_query (unexpected)
--- FAIL: HasManyAssociationsTest#test_ids_reader_memoization (unexpected)
--- FAIL: HasManyAssociationsTest#test_depends_and_nullify (unexpected)
--- FAIL: HasManyAssociationsTest#test_update_all_on_association_accessed_before_save_with_explicit_foreign_key (unexpected)
--- FAIL: HasManyAssociationsTest#test_sharded_deleting_models (unexpected)
--- FAIL: HasManyAssociationsTest#test_create_with_bang_on_habtm_when_parent_is_new_raises (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/cpu_arch=arm64/run_1
```

Parameters:
 - <code>ROACHTEST_arch=arm64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16005104/activerecord/1720591057376/1720602120073)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16022013?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16022013?buildTab=artifacts#/activerecord) on master @ [b78574b4b75b6f9ab8841d96f350697030f5e4df](https://github.com/cockroachdb/cockroach/commits/b78574b4b75b6f9ab8841d96f350697030f5e4df):


```
--- FAIL: EagerAssociationTest#test_preloading_readonly_association (unexpected)
--- FAIL: HasOneAssociationsTest#test_clearing_an_association_clears_the_associations_inverse (unexpected)
--- FAIL: EagerAssociationTest#test_including_associations_with_extensions_and_an_instance_dependent_scope_is_supported (unexpected)
--- FAIL: EagerAssociationTest#test_eager_has_many_with_association_inheritance (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_multi_table_conditional_properly_counts_the_records_when_using_size (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_associations_with_string_joins_and_order_references (unexpected)
--- FAIL: HasOneAssociationsTest#test_dependence_with_nil_associate (unexpected)
--- FAIL: EagerAssociationTest#test_type_cast_in_where_references_association_name (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_the_same_association_twice_works (unexpected)
--- FAIL: EagerAssociationTest#test_eager_load_has_many_quotes_table_and_column_names (unexpected)
--- FAIL: EagerAssociationTest#test_loading_with_no_associations (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit_and_conditions_on_the_eagers (unexpected)
--- FAIL: HasOneAssociationsTest#test_polymorphic_has_one_with_touch_option_on_create_wont_cache_association_so_fetching_after_transaction_commit_works (unexpected)
--- FAIL: EagerAssociationTest#test_order_on_join_table_with_include_and_limit (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_default_scope_as_class_method (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_multiple_associations_with_same_table_belongs_to (unexpected)
--- FAIL: EagerAssociationTest#test_should_work_inverse_of_with_eager_load (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_empty_belongs_to_polymorphic (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_empty_belongs_to (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: EagerAssociationTest#test_three_level_nested_preloading_does_not_raise_exception_when_association_does_not_exist (unexpected)
--- FAIL: EagerAssociationTest#test_eager_habtm_with_association_inheritance (unexpected)
--- FAIL: HasOneAssociationsTest#test_proxy_assignment (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to (unexpected)
--- FAIL: HasOneAssociationsTest#test_building_the_associated_object_with_explicit_sti_base_class (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_default_scope_as_callable (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_join_model_with_conditions_on_top_level (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_does_not_use_order_by (unexpected)
--- FAIL: EagerAssociationTest#test_eager_loading_with_conditions_on_join_model_preloads (unexpected)
--- FAIL: EagerAssociationTest#test_association_loading_notification (unexpected)
--- FAIL: EagerAssociationTest#test_eager_load_has_many_through_with_string_keys (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_of_instance_dependent_associations_is_supported (unexpected)
--- FAIL: EagerAssociationTest#test_associations_with_extensions_are_not_instance_dependent (unexpected)
--- FAIL: HasOneAssociationsTest#test_composite_primary_key_malformed_association_owner_class (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_assignment_triggers_save_on_change_on_replacing_object (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_with_conditions_join_model_with_include (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_through_a_polymorphic_association_doesn't_require_the_association_to_exist (unexpected)
--- FAIL: HasOneAssociationsTest#test_build_with_block (unexpected)
--- FAIL: HasOneAssociationsTest#test_create (unexpected)
--- FAIL: HasOneAssociationsTest#test_dangerous_association_name_raises_ArgumentError (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_through_has_one_association_with_conditions_on_association (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_through_has_one_association_with_order_on_nested_association (unexpected)
--- FAIL: EagerAssociationTest#test_finding_with_includes_on_has_one_association_with_same_include_includes_only_once (unexpected)
--- FAIL: EagerAssociationTest#test_load_with_sti_sharing_association (unexpected)
--- FAIL: EagerAssociationTest#test_preconfigured_includes_with_habtm (unexpected)
--- FAIL: EagerAssociationTest#test_preload_with_invalid_argument (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16022013/activerecord/1720677199127/1720685654391)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16040622?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16040622?buildTab=artifacts#/activerecord) on master @ [3afb6935d0ef3de7c0d44cfb3cd54f312752c186](https://github.com/cockroachdb/cockroach/commits/3afb6935d0ef3de7c0d44cfb3cd54f312752c186):


```
--- FAIL: AttributeMethodsTest#test_declared_prefixed_attribute_method_affects_respond_to?_and_method_missing (unexpected)
--- FAIL: InheritanceTest#test_inheritance_base_class (unexpected)
--- FAIL: AssociationsTest#test_append_composite_foreign_key_has_many_association_with_autosave (unexpected)
--- FAIL: DateTimeTest#test_assign_empty_time (unexpected)
--- FAIL: I18nGenerateMessageValidationTest#test_translation_for_'taken'_can_be_overridden_in_activerecord_attributes_scope (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attribute_for_database (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_preloading_across_has_one_constrains_loaded_records (unexpected)
--- FAIL: AssociationsTest#test_assign_belongs_to_cpk_model_by_id_attribute (unexpected)
--- FAIL: ReflectionTest#test_chain (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_two_levels_with_condition (unexpected)
--- FAIL: ReflectionTest#test_reflection_klass_not_found_with_pointer_to_non_existent_class_name (unexpected)
--- FAIL: HasOneAssociationsTest#test_reload_association (unexpected)
--- FAIL: MultiParameterAttributeTest#test_multiparameter_attributes_on_time (unexpected)
--- FAIL: TouchLaterTest#test_touch_touches_immediately (unexpected)
--- FAIL: StrictLoadingTest#test_strict_loading_with_has_one_through_does_not_prevent_creation_of_association (unexpected)
--- FAIL: HasOneAssociationsTest#test_association_change_calls_delete (unexpected)
--- FAIL: HasOneAssociationsTest#test_replacement_failure_due_to_new_record_should_raise_error (unexpected)
--- FAIL: FixturesWithoutInstantiationTest#test_accessor_methods (unexpected)
--- FAIL: MultiParameterAttributeTest#test_multiparameter_attributes_on_time_with_time_zone_aware_attributes (unexpected)
--- FAIL: SerializedAttributeTestWithYamlSafeLoad#test_nil_not_serialized_without_class_constraint (unexpected)
--- FAIL: InheritanceTest#test_alt_finding_incorrect_type_data (unexpected)
--- FAIL: ActiveRecord::InvertibleMigrationTest#test_up_only (unexpected)
--- FAIL: AttributeMethodsTest#test_user-defined_time_attribute_predicate (unexpected)
--- FAIL: FixturesTest#test_complete_instantiation (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_partial_indices (unexpected)
--- FAIL: AssociationsTest#test_subselect (unexpected)
--- FAIL: HasOneAssociationsTest#test_successful_build_association (unexpected)
--- FAIL: IntegrationTest#test_cache_version_format_is_not_too_precise (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_uses_default_if_blank (unexpected)
--- FAIL: CallbacksOnMultipleInstancesInATransactionTest#test_destroyed_callback_called_on_destroyed_instance_when_preceded_in_transaction_by_save_from_separate_instance (unexpected)
--- FAIL: AttributeMethodsTest#test_#undefine_attribute_methods_undefines_alias_attribute_methods (unexpected)
--- FAIL: InheritanceTest#test_compute_type_success (unexpected)
--- FAIL: SerializedAttributeTestWithYamlSafeLoad#test_serialized_column_should_unserialize_after_update_column (unexpected)
--- FAIL: ReflectionTest#test_collection_association (unexpected)
--- FAIL: AssociationsTest#test_loading_cpk_association_when_persisted_and_in_memory_differ (unexpected)
--- FAIL: AttributeMethodsTest#test_read_overridden_attribute (unexpected)
--- FAIL: SerializedAttributeTestWithYamlSafeLoad#test_serialized_attribute_in_base_class (unexpected)
--- FAIL: TestAutosaveAssociationsInGeneral#test_autosave_works_even_when_other_callbacks_update_the_parent_model (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_preload_through_missing_records (unexpected)
--- FAIL: SignedIdTest#test_fail_to_find_signed_record_with_purpose (unexpected)
--- FAIL: IntegrationTest#test_cache_version_format_is_precise_enough (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_preloading_across_has_one_through_constrains_loaded_records (unexpected)
--- FAIL: ReflectionTest#test_scope_chain_does_not_interfere_with_hmt_with_polymorphic_and_subclass_source_2 (unexpected)
--- FAIL: AttributeMethodsTest#test_overridden_read_attribute (unexpected)
--- FAIL: IntegrationTest#test_cache_key_retains_version_when_custom_timestamp_is_used (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_with_block (unexpected)
--- FAIL: AttributeMethodsTest#test_number_attribute_predicate (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/cpu_arch=arm64/run_1
```

Parameters:
 - <code>ROACHTEST_arch=arm64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16040622/activerecord/1720763784437/1720774727272)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16058260?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16058260?buildTab=artifacts#/activerecord) on master @ [dc3ebba01015d2c7275aa27af2a54fe88b87e901](https://github.com/cockroachdb/cockroach/commits/dc3ebba01015d2c7275aa27af2a54fe88b87e901):


```
(orm_helpers.go:229).summarizeFailed: 
Tests run on Cockroach v24.2.0-alpha.2-dev-dc3ebba01015d2c7275aa27af2a54fe88b87e901
Tests run against activerecord 7.1.3
8108 Total Tests Run
8074 tests passed
34 tests failed
66 tests skipped
0 tests ignored
0 tests passed unexpectedly
8 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: SchemaDumperTest#test_schema_dumps_index_sort_order (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_default_pk_and_sequence (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_index_columns_in_right_order (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_create_fixtures_resets_sequences_when_not_cached (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_nulls_not_distinct (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_partial_indices (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_index_length (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16058260/activerecord/1720850085313/1720858909415)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16065509?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16065509?buildTab=artifacts#/activerecord) on master @ [1ebd9e1e105cd90ef0bcff47a47369681d165d14](https://github.com/cockroachdb/cockroach/commits/1ebd9e1e105cd90ef0bcff47a47369681d165d14):


```
--- FAIL: BasicsTest#test_cannot_call_connected_to_with_role_and_shard_on_non-abstract_classes (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_multibyte_character (unexpected)
--- FAIL: BasicsTest#test_find_reverse_ordered_last (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attribute_when_true (unexpected)
--- FAIL: AttributeMethodsTest#test_user-defined_time_attribute_predicate (unexpected)
--- FAIL: BasicsTest#test_ignored_columns_are_stored_as_an_array_of_string (unexpected)
--- FAIL: AttributeMethodsTest#test_#undefine_attribute_methods_undefines_alias_attribute_methods (unexpected)
--- FAIL: BasicsTest#test_singular_table_name_guesses (unexpected)
--- FAIL: AttributeMethodsTest#test_declared_affixed_attribute_method_affects_respond_to?_and_method_missing (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates_after_parameterize (unexpected)
--- FAIL: BasicsTest#test_equality_with_blank_ids (unexpected)
--- FAIL: AttributeMethodsTest#test_on_the_fly_super_invokable_generated_attribute_methods_via_method_missing (unexpected)
--- FAIL: BasicsTest#test_create_after_initialize_without_block (unexpected)
--- FAIL: BasicsTest#test_attributes (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates_words_properly (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_method? (unexpected)
--- FAIL: AttributeMethodsTest#test_typecast_attribute_from_select_to_true (unexpected)
--- FAIL: BasicsTest#test_invalid_limit (unexpected)
--- FAIL: IntegrationTest#test_param_delimiter_changes_delimiter_used_in_to_param (unexpected)
--- FAIL: AttributeMethodsTest#test_number_attribute_predicate (unexpected)
--- FAIL: BasicsTest#test_set_table_name_symbol_converted_to_string (unexpected)
--- FAIL: AttributeMethodsTest#test_respond_to?_with_a_custom_primary_key (unexpected)
--- FAIL: BasicsTest#test_find_by_slug_with_array (unexpected)
--- FAIL: IntegrationTest#test_cache_version_format_is_not_too_precise (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_a_time_zone-aware_time_in_the_current_time_zone (unexpected)
--- FAIL: TestAutosaveAssociationValidationsOnAHABTMAssociation#test_should_not_automatically_validate_associations_without_:validate_=>_true (unexpected)
--- FAIL: AttributeMethodsTest#test_write_attribute_does_not_raise_when_the_attribute_isn't_selected (unexpected)
--- FAIL: BasicsTest#test_dup_with_aggregate_of_same_name_as_attribute (unexpected)
--- FAIL: BasicsTest#test_non_valid_identifier_column_name (unexpected)
--- FAIL: BasicsTest#test_columns_should_obey_set_primary_key (unexpected)
--- FAIL: BasicsTest#test_connection_in_utc_time (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_present_with_booleans (unexpected)
--- FAIL: BasicsTest#test_bignum (unexpected)
--- FAIL: BasicsTest#test_primary_key_and_references_columns_should_be_identical_type (unexpected)
--- FAIL: BasicsTest#test_limit_should_sanitize_sql_injection_for_limit_without_commas (unexpected)
--- FAIL: BasicsTest#test_equality_of_collection_proxy_and_association_relation (unexpected)
--- FAIL: AttributeMethodsTest#test_overridden_write_attribute (unexpected)
--- FAIL: BasicsTest#test_when_#reload_called,_ignored_columns'_attribute_methods_are_not_defined (unexpected)
--- FAIL: BasicsTest#test_preserving_time_objects_with_utc_time_conversion_to_default_timezone_local (unexpected)
--- FAIL: BasicsTest#test_find_by_slug_with_range (unexpected)
--- FAIL: BasicsTest#test_touch_should_raise_error_on_a_new_object (unexpected)
--- FAIL: BasicsTest#test_distinct_delegates_to_scoped (unexpected)
--- FAIL: BasicsTest#test_#connecting_to_with_role_and_shard (unexpected)
--- FAIL: BasicsTest#test_set_table_name_with_inheritance (unexpected)
--- FAIL: BasicsTest#test_default_in_utc_with_time_zone (unexpected)
--- FAIL: BasicsTest#test_#connected_to_many_with_a_multiple_classes_without_brackets_works (unexpected)
--- FAIL: BasicsTest#test_#connected_to_many_sets_prevent_writes_if_role_is_reading (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16065509/activerecord/1720936427201/1720944482880)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16072133?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16072133?buildTab=artifacts#/activerecord) on master @ [67aa2613446c3c6b53bad21f01286c98290cc0a3](https://github.com/cockroachdb/cockroach/commits/67aa2613446c3c6b53bad21f01286c98290cc0a3):


```
--- FAIL: AssociationsTest#test_has_many_with_foreign_key_as_an_array_raises (unexpected)
--- FAIL: NumericalityValidationTest#test_virtual_attribute_with_precision_round_half_even (unexpected)
--- FAIL: UniquenessValidationTest#test_validate_uniqueness_when_integer_out_of_range_show_order_does_not_matter (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_primary_key_option (unexpected)
--- FAIL: InheritanceTest#test_alt_update_all_within_inheritance (unexpected)
--- FAIL: FinderTest#test_find_by_returns_nil_if_the_record_is_missing (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_fields_with_table_alias (unexpected)
--- FAIL: CalculationsTest#test_should_count_manual_select_with_group_with_count_all (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_with_custom_foreign_type (unexpected)
--- FAIL: RelationTest#test_includes_with_select (unexpected)
--- FAIL: RelationTest#test_find_all_with_join (unexpected)
--- FAIL: CalculationsTest#test_should_return_decimal_average_of_integer_field (unexpected)
--- FAIL: SerializedAttributeTest#test_serialized_boolean_value_true (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates (unexpected)
--- FAIL: RelationTest#test_relation_join_method (unexpected)
--- FAIL: RelationTest#test_unscope_with_table_name_qualified_hash (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_with_block (unexpected)
--- FAIL: EnumTest#test_enums_are_distinct_per_class (unexpected)
--- FAIL: TimestampTest#test_touching_an_attribute_updates_it (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_association_class (unexpected)
--- FAIL: AssociationsTest#test_nullify_composite_foreign_key_has_many_association (unexpected)
--- FAIL: TransactionTest#test_using_named_savepoints (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_through_a_polymorphic_association_doesn't_require_the_association_to_exist (unexpected)
--- FAIL: BelongsToAssociationsTest#test_should_set_foreign_key_on_create_association_with_unpersisted_owner (unexpected)
--- FAIL: FinderTest#test_exists_with_eager_load (unexpected)
--- FAIL: SpatialQueriesTest#test_query_multi_point (unexpected)
--- FAIL: SchemaWithDotsTest#test_Active_Record_basics (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
--- FAIL: PersistenceTest#test_delete (unexpected)
--- FAIL: TestAutosaveAssociationValidationsOnAHasManyAssociation#test_should_automatically_validate_associations (unexpected)
--- FAIL: RelationTest#test_reorder_deduplication (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_find_in_association (unexpected)
--- FAIL: PersistenceTest#test_update_column_with_model_having_primary_key_other_than_id (unexpected)
--- FAIL: RelationTest#test_support_upper_and_lower_case_directions (unexpected)
--- FAIL: PersistenceTest#test_update_columns_returns_boolean (unexpected)
--- FAIL: FinderTest#test_exists_with_empty_hash_arg (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_with_false (unexpected)
--- FAIL: IntegrationTest#test_param_delimiter_is_defined_per_class (unexpected)
--- FAIL: AttributeMethodsTest#test_declared_suffixed_attribute_method_affects_respond_to?_and_method_missing (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_both_parent_ids_set_when_saving_new (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_pk_is_not_required_for_join (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_symbols_as_keys (unexpected)
--- FAIL: ReflectionTest#test_belongs_to_inferred_foreign_key_from_assoc_name (unexpected)
--- FAIL: FinderTest#test_find_by_doesn't_have_implicit_ordering (unexpected)
--- FAIL: PersistenceTest#test_update_column_should_raise_exception_if_new_record (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_update_counter_caches_on_destroy_with_indestructible_through_record (unexpected)
--- FAIL: DefaultScopingTest#test_create_with_using_both_string_and_symbol (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16072133/activerecord/1721022724999/1721030918177)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16088498?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16088498?buildTab=artifacts#/activerecord) on master @ [73ecc8b494c4e7f757bb35d68b3587a51af93b02](https://github.com/cockroachdb/cockroach/commits/73ecc8b494c4e7f757bb35d68b3587a51af93b02):


```
--- FAIL: ValidationsTest#test_numericality_validation_checks_against_raw_value (unexpected)
--- FAIL: RelationTest#test_reload (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- FAIL: BasicsTest#test_initialize_with_invalid_attribute (unexpected)
--- FAIL: HasOneAssociationsTest#test_update_with_foreign_and_primary_keys (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_primary_key_option (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_empty_on_an_association_that_has_been_loaded_does_not_performs_query (unexpected)
--- FAIL: HasManyAssociationsTest#test_set_ids_for_association_on_new_record_applies_association_correctly (unexpected)
--- FAIL: InheritanceTest#test_abstract_class (unexpected)
--- FAIL: HasManyAssociationsTest#test_default_select (unexpected)
--- FAIL: EnumTest#test_deserialize_enum_value_to_original_hash_key (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_update_on_id_changes_the_counter_cache (unexpected)
--- FAIL: TransactionTest#test_using_named_savepoints (unexpected)
--- FAIL: FinderTest#test_find_by_returns_nil_if_the_record_is_missing (unexpected)
--- FAIL: AssociationsTest#test_belongs_to_a_model_with_composite_foreign_key_finds_associated_record (unexpected)
--- FAIL: RelationTest#test_create_or_find_by_with_bang_with_non_unique_attributes (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_hashed_table_name (unexpected)
--- FAIL: FinderTest#test_fourth_have_primary_key_order_by_default (unexpected)
--- FAIL: FinderTest#test_find_ignores_previously_inserted_record (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_concat (unexpected)
--- FAIL: RelationTest#test_many_with_limits (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_update_changing_ids_changes_the_counter_cache (unexpected)
--- FAIL: RelationTest#test_one (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_left_joined_same_table_with_through_table (unexpected)
--- FAIL: AssociationsTest#test_association_with_references (unexpected)
--- FAIL: RelationTest#test_find_by_id_with_list_of_ar (unexpected)
--- FAIL: CalculationsTest#test_ids_with_includes (unexpected)
--- FAIL: BasicsTest#test_dup_with_aggregate_of_same_name_as_attribute (unexpected)
--- FAIL: HasManyAssociationsTest#test_create_from_association_should_respect_default_scope (unexpected)
--- FAIL: RelationTest#test_already-loaded_relations_don't_perform_a_new_query_in_#pretty_print (unexpected)
--- FAIL: CalculationsTest#test_calculation_with_query_cache (unexpected)
--- FAIL: RelationTest#test_unscoped_block_style (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_assign_ids_with_belongs_to_cpk_model (unexpected)
--- FAIL: FinderTest#test_find_by_two_attributes_but_passing_only_one (unexpected)
--- FAIL: RelationTest#test_where_id_with_delegated_ar_object (unexpected)
--- FAIL: InheritanceTest#test_should_store_demodulized_class_name_with_store_full_sti_class_option_disabled (unexpected)
--- FAIL: CockroachDB::LeftOuterJoinAssociationTest#test_does_not_override_select (unexpected)
--- FAIL: HasManyAssociationsTest#test_building_the_associated_object_with_implicit_sti_base_class (unexpected)
--- FAIL: FinderTest#test_take_failing (unexpected)
--- FAIL: RelationTest#test_pluck_with_subquery_in_from_uses_original_table_name (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_present_with_booleans (unexpected)
--- FAIL: IntegrationTest#test_to_param_for_a_composite_primary_key_model (unexpected)
--- FAIL: HasManyAssociationsTest#test_counter_cache_updates_in_memory_after_create (unexpected)
--- FAIL: RelationTest#test_relations_limit_the_records_in_#pretty_print_at_10 (unexpected)
--- FAIL: RelationTest#test_except (unexpected)
--- FAIL: RelationTest#test_find_all_using_where_twice_should_or_the_relation (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16088498/activerecord/1721109281561/1721118278753)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16125740?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16125740?buildTab=artifacts#/activerecord) on master @ [46854026bc11b3870c8bbefc976f5550b18dd82c](https://github.com/cockroachdb/cockroach/commits/46854026bc11b3870c8bbefc976f5550b18dd82c):


```
--- FAIL: CalculationsTest#test_select_avg_with_joins_and_group_by_as_virtual_attribute_with_sql (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_enum_method_issues_a_deprecation_warning (unexpected)
--- FAIL: HasManyAssociationsTest#test_collection_not_empty_after_building (unexpected)
--- FAIL: HasManyAssociationsTest#test_finding_array_compatibility (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_collection_build_with_nonstandard_primary_key_on_belongs_to (unexpected)
--- FAIL: AttributeMethodsTest#test_boolean_attribute_predicate (unexpected)
--- FAIL: AttributeMethodsTest#test_allocated_objects_can_be_inspected (unexpected)
--- FAIL: HasOneAssociationsTest#test_dependence_with_missing_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_first_after_reset (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_collection_singular_ids_setter_raises_exception_when_invalid_ids_set (unexpected)
--- FAIL: DefaultScopingTest#test_create_attribute_overwrites_default_scoping (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_index_columns_in_right_order (unexpected)
--- FAIL: CalculationsTest#test_should_generate_valid_sql_with_joins_and_group (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all_association_with_primary_key_deletes_correct_records (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::TypesTest#test_attributes_which_are_invalid_for_database_can_still_be_reassigned (unexpected)
--- FAIL: BasicsTest#test_#connecting_to_with_role (unexpected)
--- FAIL: HasOneAssociationsTest#test_restrict_with_exception (unexpected)
--- FAIL: FinderTest#test_hash_condition_utc_time_interpolation_with_default_timezone_local (unexpected)
--- FAIL: StrictLoadingTest#test_strict_loading_n_plus_one_only_mode (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_multiple_columns_and_selection_clause (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_replace_association_with_duplicates (unexpected)
--- FAIL: ReflectionTest#test_collection_association (unexpected)
--- FAIL: BasicsTest#test_previously_new_record_on_destroyed_record (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_interdependent_one_level_and_two_levels (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_unscope_should_affect_to_through_scope (unexpected)
--- FAIL: FinderTest#test_include_on_unloaded_relation_without_match (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_qualified_attribute_dot_notation_symbol (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_keys_bypass_attribute_protection (unexpected)
--- FAIL: CalculationsTest#test_async_pluck_none_relation (unexpected)
--- FAIL: DefaultScopingTest#test_default_scope_as_block_referencing_scope (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_destroy_association (unexpected)
--- FAIL: FinderTest#test_find_by_id_with_large_number (unexpected)
--- FAIL: AttributeMethodsTest#test_write_nil_to_time_attribute (unexpected)
--- FAIL: AttributeMethodsTest#test_converted_values_are_returned_after_assignment (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_missing_first_record (unexpected)
--- FAIL: CalculationsTest#test_pluck_functions_with_alias (unexpected)
--- FAIL: BasicsTest#test_find_symbol_ordered_last (unexpected)
--- FAIL: BasicsTest#test_primary_key_and_references_columns_should_be_identical_type (unexpected)
--- FAIL: CockroachDB::EagerLoadPolyAssocsTest#test_include_query (unexpected)
--- FAIL: HasManyAssociationsTest#test_depends_and_nullify_on_polymorphic_assoc (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_without_loading_association (unexpected)
--- FAIL: BasicsTest#test_default_values_are_deeply_dupped (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_against_given_relation (unexpected)
--- FAIL: HasOneAssociationsTest#test_with_select (unexpected)
--- FAIL: BasicsTest#test_ignored_columns_are_stored_as_an_array_of_string (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attribute_for_database_with_aliased_attribute (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::ConnectionHandlersShardingDbTest#test_sharding_separation (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16125740/activerecord/1721282074237/1721291697032)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #127333 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.3-rc]
- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
- #125730 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.2-rc]
- #121343 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16145920?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16145920?buildTab=artifacts#/activerecord) on master @ [59be0708627bb27f49d8976182062c6886d64282](https://github.com/cockroachdb/cockroach/commits/59be0708627bb27f49d8976182062c6886d64282):


```
--- FAIL: HasManyThroughAssociationsTest#test_update_counter_caches_on_destroy (unexpected)
--- FAIL: AttributeMethodsTest#test_instance_methods_should_be_defined_on_the_base_class (unexpected)
--- FAIL: PersistenceTest#test_becomes_after_reload_schema_from_cache (unexpected)
--- FAIL: EagerAssociationTest#test_exceptions_have_suggestions_for_fix (unexpected)
--- FAIL: EagerAssociationTest#test_loading_from_an_association_that_has_a_hash_of_conditions (unexpected)
--- FAIL: FinderTest#test_find_by_ids_missing_one (unexpected)
--- FAIL: HasManyAssociationsTest#test_can_unscope_and_where_the_default_scope_of_the_associated_model (unexpected)
--- FAIL: AssociationsTest#test_eager_loading_should_not_change_count_of_children (unexpected)
--- FAIL: RelationTest#test_relations_limit_the_records_in_#inspect_at_10 (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_attributes_are_being_set_when_initialized_from_habtm_association_with_where_clause (unexpected)
--- FAIL: AttributeMethodsTest#test_on_the_fly_super_invokable_generated_attribute_methods_via_method_missing (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_a_regular_association_through_a_polymorphic_association_doesn't_require_the_association_to_exist_on_all_types (unexpected)
--- FAIL: PersistenceTest#test_destroy_uses_query_constraints_config (unexpected)
--- FAIL: EagerAssociationTest#test_loading_polymorphic_association_with_mixed_table_conditions (unexpected)
--- FAIL: CalculationsTest#test_minimum_and_maximum_on_non_numeric_type (unexpected)
--- FAIL: PersistenceTest#test_update_columns_changing_id (unexpected)
--- FAIL: HasManyAssociationsTest#test_depends_and_nullify (unexpected)
--- FAIL: InheritanceAttributeTest#test_inheritance_new_with_subclass_as_default (unexpected)
--- FAIL: FinderTest#test_include_on_unloaded_relation_with_match (unexpected)
--- FAIL: BasicsTest#test_generated_association_methods_module_name (unexpected)
--- FAIL: HasManyAssociationsTest#test_building_the_associated_object_with_implicit_sti_base_class (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit_and_scoped_conditions_on_the_eagers (unexpected)
--- FAIL: BasicsTest#test_find_ordered_last (unexpected)
--- FAIL: BasicsTest#test_marshal_new_record_round_trip (unexpected)
--- FAIL: RelationTest#test_find_all_using_where_with_relation_with_bound_values (unexpected)
--- FAIL: HasManyAssociationsTest#test_create_from_association_should_respect_default_scope (unexpected)
--- FAIL: FinderTest#test_exists_with_distinct_and_offset_and_eagerload_and_order (unexpected)
--- FAIL: FinderTest#test_find_by!_doesn't_have_implicit_ordering (unexpected)
--- FAIL: AttributeMethodsTest#test_user-defined_JSON_attribute_predicate (unexpected)
--- FAIL: RelationTest#test_select_takes_an_aliased_attribute (unexpected)
--- FAIL: ModulesTest#test_module_spanning_has_and_belongs_to_many_associations (unexpected)
--- FAIL: StrictLoadingTest#test_strict_loading_n_plus_one_only_mode (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_nested_has_many_through_association_with_unpersisted_parent_instance (unexpected)
--- FAIL: CalculationsTest#test_group_by_with_quoted_count_and_order_by_alias (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_belongs_to_should_update_when_the_through_foreign_key_changes (unexpected)
--- FAIL: AttributeMethodsTest#test_update_array_content (unexpected)
--- FAIL: InheritanceTest#test_where_create_with_subclass (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_association_through_a_has_many_association_to_self (unexpected)
--- FAIL: RelationTest#test_count_explicit_columns (unexpected)
--- FAIL: InheritanceTest#test_eager_load_belongs_to_something_inherited (unexpected)
--- FAIL: HasManyAssociationsTest#test_add_record_to_collection_should_change_its_updated_at (unexpected)
--- FAIL: CalculationsTest#test_ids_with_includes (unexpected)
--- FAIL: FinderTest#test_take_and_first_and_last_with_integer_should_return_an_array (unexpected)
--- FAIL: AttributeMethodsTest#test_#id_value_alias_returns_id_column_only_for_composite_primary_key_models (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_store_association_in_two_relations_with_one_save_in_existing_object_with_values (unexpected)
--- FAIL: HasOneAssociationsTest#test_polymorphic_has_one_with_touch_option_on_update_will_touch_record_by_fetching_from_database_if_needed (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_grouped_association_with_invalid_field (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16145920/activerecord/1721368353341/1721376067287)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16162243?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16162243?buildTab=artifacts#/activerecord) on master @ [7a8ce2447eb023e855cb85b3ce65a07a70a39334](https://github.com/cockroachdb/cockroach/commits/7a8ce2447eb023e855cb85b3ce65a07a70a39334):


```
--- FAIL: BasicsTest#test_sql_injection_via_find (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_the_same_association_twice_works (unexpected)
--- FAIL: CalculationsTest#test_pluck_auto_table_name_prefix (unexpected)
--- FAIL: BasicsTest#test_arel_attribute_normalization (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_default_pk_and_sequence (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_marshal_dump_and_load_via_disk (unexpected)
--- FAIL: TransactionCallbacksTest#test_only_call_after_commit_on_create_after_transaction_commits_for_new_record_if_create_succeeds_creating_through_association (unexpected)
--- FAIL: CalculationsTest#test_no_queries_for_empty_relation_on_minimum (unexpected)
--- FAIL: BelongsToAssociationsTest#test_create! (unexpected)
--- FAIL: BasicsTest#test_previously_new_record_returns_boolean (unexpected)
--- FAIL: BasicsTest#test_#connected_to_many_with_a_single_argument_for_classes (unexpected)
--- FAIL: EagerAssociationTest#test_duplicate_middle_objects (unexpected)
--- FAIL: BasicsTest#test_all (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_multiple_fields (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_for_existing_record_with_updated_at_and_custom_cache_timestamp_format (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_with_polymorphic_has_many_does_not_allow_to_override_type_and_id (unexpected)
--- FAIL: BasicsTest#test_table_name_with_2_abstract_subclasses (unexpected)
--- FAIL: EagerAssociationTest#test_eager-loading_a_polymorphic_association_with_references_to_the_associated_table (unexpected)
--- FAIL: CalculationsTest#test_sum_should_return_valid_values_for_decimals (unexpected)
--- FAIL: AttributeMethodsTest#test_generated_attribute_methods_ancestors_have_correct_module (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_conditions_string_with_quoted_table_name (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_join (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_preload_through_missing_records (unexpected)
--- FAIL: HasManyAssociationsTest#test_has_many_without_counter_cache_option (unexpected)
--- FAIL: AttributeMethodsTest#test_bulk_updates_respect_access_control (unexpected)
--- FAIL: AttributeMethodsTest#test_read_overridden_attribute (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attributes_after_type_cast_on_a_date (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_has_many_through_association_avoids_calling_association.reader (unexpected)
--- FAIL: HasManyAssociationsTest#test_cant_save_has_many_readonly_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_associations_autosaves_when_object_is_already_persisted (unexpected)
--- FAIL: BasicsTest#test_find_by_slug_with_range (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_with_invalid_field (unexpected)
--- FAIL: InheritanceTest#test_where_create_with_unrelated_type (unexpected)
--- FAIL: HasManyAssociationsTest#test_unscopes_the_default_scope_of_associated_model_when_used_with_include (unexpected)
--- FAIL: HasManyAssociationsTest#test_in_memory_replacement_maintains_order (unexpected)
--- FAIL: AttributeMethodsTest#test_removing_time_zone-aware_types (unexpected)
--- FAIL: CalculationsTest#test_select_avg_with_group_by_as_virtual_attribute_with_ar (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates_edge_case (unexpected)
--- FAIL: CalculationsTest#test_#skip_query_cache!_for_#ids (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_marshal_dump_and_load_with_gzip (unexpected)
--- FAIL: AssociationsTest#test_cpk_model_has_many_records_by_id_attribute (unexpected)
--- FAIL: BasicsTest#test_find_keeps_multiple_group_values (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_invalid_association_reference (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_store_association_with_a_polymorphic_relationship (unexpected)
--- FAIL: InheritanceTest#test_where_new_with_invalid_type (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_models_with_composite_keys (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_one_dependent_does_not_destroy_dependent (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16162243/activerecord/1721454825027/1721460226719)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16171801?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16171801?buildTab=artifacts#/activerecord) on master @ [05052df0b4d89dd1147714431163cd8a9b63c589](https://github.com/cockroachdb/cockroach/commits/05052df0b4d89dd1147714431163cd8a9b63c589):


```
(orm_helpers.go:229).summarizeFailed: 
Tests run on Cockroach v24.2.0-alpha.2-dev-05052df0b4d89dd1147714431163cd8a9b63c589
Tests run against activerecord 7.1.3
8108 Total Tests Run
8079 tests passed
29 tests failed
66 tests skipped
0 tests ignored
0 tests passed unexpectedly
3 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_default_pk_and_sequence (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_create_fixtures_resets_sequences_when_not_cached (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16171801/activerecord/1721541363604/1721550363700)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16180363?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16180363?buildTab=artifacts#/activerecord) on master @ [05052df0b4d89dd1147714431163cd8a9b63c589](https://github.com/cockroachdb/cockroach/commits/05052df0b4d89dd1147714431163cd8a9b63c589):


```
--- FAIL: HasManyAssociationsTest#test_update_all_on_association_accessed_before_save_with_explicit_foreign_key (unexpected)
--- FAIL: HasManyAssociationsTest#test_add_record_to_collection_should_change_its_updated_at (unexpected)
--- FAIL: TransactionTest#test_rollback_dirty_changes (unexpected)
--- FAIL: ReflectionTest#test_reflection_klass_not_found_with_no_class_name_option (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_counter_cache (unexpected)
--- FAIL: HasOneAssociationsTest#test_replacement_failure_due_to_existing_record_should_raise_error (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_store_association_with_a_polymorphic_relationship (unexpected)
--- FAIL: EagerAssociationTest#test_eager_habtm_with_association_inheritance (unexpected)
--- FAIL: PersistenceTest#test_delete_new_record (unexpected)
--- FAIL: TestAutosaveAssociationOnAHasManyAssociationDefinedInSubclassWithAcceptsNestedAttributes#test_should_update_children_when_association_redefined_in_subclass (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_user_provided_defaults_are_persisted_even_if_unchanged (unexpected)
--- FAIL: StoreTest#test_writing_store_attributes_does_not_update_unchanged_value (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_marshal_dump_and_load_with_ignored_tables (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all,_when_not_loaded,_doesn't_load_the_records (unexpected)
--- FAIL: AttributeMethodsTest#test_respond_to? (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_does_not_use_order_by (unexpected)
--- FAIL: ReflectionTest#test_has_and_belongs_to_many_reflection (unexpected)
--- FAIL: ModulesTest#test_module_spanning_has_and_belongs_to_many_associations (unexpected)
--- FAIL: HasManyAssociationsTest#test_adding_array_and_collection (unexpected)
--- FAIL: PersistenceTest#test_update_columns_should_not_modify_updated_at (unexpected)
--- FAIL: DirtyTest#test_changed_attributes_should_be_preserved_if_save_failure (unexpected)
--- FAIL: DirtyTest#test_attribute_changes (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroying_a_collection (unexpected)
--- FAIL: PersistenceTest#test_destroy! (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_first_nth_or_last_on_existing_record_with_create_should_not_load_association (unexpected)
--- FAIL: StrictLoadingTest#test_strict_loading_with_has_many_through_cascade_down_to_middle_records (unexpected)
--- FAIL: HasManyAssociationsTest#test_has_many_through_respects_hash_conditions (unexpected)
--- FAIL: DirtyTest#test_datetime_attribute_doesnt_change_if_zone_is_modified_in_string (unexpected)
--- FAIL: MultiParameterAttributeTest#test_multiparameter_attributes_on_time_with_utc (unexpected)
--- FAIL: PersistenceTest#test_reload_uses_query_constraints_config (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_order_string_with_unquoted_table_name (unexpected)
--- FAIL: PersistenceTest#test_save_nil_string_attributes (unexpected)
--- FAIL: JsonSerializationTest#test_uses_serializable_hash_with_except_option (unexpected)
--- FAIL: ReflectionTest#test_active_record_primary_key_raises_when_missing_primary_key (unexpected)
--- FAIL: HasManyAssociationsTest#test_create_children_could_be_rolled_back_by_after_save (unexpected)
--- FAIL: PersistenceTest#test_update_column (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_an_sti_join_model_with_conditions_on_both (unexpected)
--- FAIL: EagerAssociationTest#test_eager_loading_with_select_on_joined_table_preloads (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_validation_does_not_validate_stale_association_target (unexpected)
--- FAIL: AttributeMethodsTest#test_update_array_content (unexpected)
--- FAIL: EagerAssociationTest#test_loading_from_an_association (unexpected)
--- FAIL: DirtyTest#test_time_attributes_changes_with_time_zone (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_one_should_defer_to_collection_if_using_a_block (unexpected)
--- FAIL: BelongsToAssociationsTest#test_type_mismatch (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnNewRecord#test_autosave_new_record_on_has_one_can_be_disabled_per_relationship (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_each (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_a_collection (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16180363/activerecord/1721627669108/1721635408645)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16190962?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16190962?buildTab=artifacts#/activerecord) on master @ [7c29b0c71f3ee8ccc436b5f039955bf6d2add914](https://github.com/cockroachdb/cockroach/commits/7c29b0c71f3ee8ccc436b5f039955bf6d2add914):


```
--- FAIL: HasManyAssociationsTest#test_destroy_all (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_many_with_merged_options (unexpected)
--- FAIL: HasManyAssociationsTest#test_clear_collection_should_not_change_updated_at (unexpected)
--- FAIL: ModulesTest#test_compute_type_can_infer_class_name_of_sibling_inside_module (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_does_not_raise_exception_when_association_does_not_exist (unexpected)
--- FAIL: EagerAssociationTest#test_loading_from_an_association_that_has_a_hash_of_conditions (unexpected)
--- FAIL: EagerAssociationTest#test_loading_association_with_string_joins (unexpected)
--- FAIL: HasOneAssociationsTest#test_exclusive_dependence (unexpected)
--- FAIL: HasManyAssociationsTest#test_depends_and_nullify (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit_and_conditions_array (unexpected)
--- FAIL: HasOneAssociationsTest#test_natural_assignment_to_already_associated_record (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_proxy_should_not_respond_to_private_methods (unexpected)
--- FAIL: HasManyAssociationsTest#test_get_ids_for_unloaded_associations_does_not_load_them (unexpected)
--- FAIL: BelongsToAssociationsTest#test_building_the_belonging_object_for_composite_primary_key (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_proxy_transaction_method_starts_transaction_in_association_class (unexpected)
--- FAIL: ModulesTest#test_module_spanning_has_and_belongs_to_many_associations (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_has_one_using_reorder (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_touch_option_on_update (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_build_and_then_save_parent_should_not_reload_target (unexpected)
--- FAIL: BelongsToAssociationsTest#test_create_with_block (unexpected)
--- FAIL: HasOneAssociationsTest#test_build_and_create_should_not_happen_within_scope (unexpected)
--- FAIL: BelongsToAssociationsTest#test_should_set_foreign_key_on_save! (unexpected)
--- FAIL: HasManyAssociationsTest#test_replace_failure (unexpected)
--- FAIL: HasManyAssociationsTest#test_counting_with_column_name_and_hash (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_association_class (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_belongs_to_association_associated_by_a_composite_query_constraints (unexpected)
--- FAIL: HasManyAssociationsTest#test_counter_cache_updates_in_memory_after_create (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_default_scope_as_class_method_using_find_method (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all_on_association_with_nil_dependency_is_the_same_as_not_loaded (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_with_block (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one (unexpected)
--- FAIL: BelongsToAssociationsTest#test_natural_assignment_to_nil (unexpected)
--- FAIL: BelongsToAssociationsTest#test_stale_tracking_doesn't_care_about_the_type (unexpected)
--- FAIL: StrictLoadingTest#test_strict_loading_with_has_one_through_does_not_prevent_creation_of_association (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_counter_after_touch (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_default_scope_as_class_method (unexpected)
--- FAIL: HasManyAssociationsTest#test_dangerous_association_name_raises_ArgumentError (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_composite_primary_key_autosave (unexpected)
--- FAIL: HasManyAssociationsTest#test_has_many_without_counter_cache_option (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_updates_counter_cache_with_dependent_destroy (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_through_has_one_association_with_conditions (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit_and_high_offset_and_multiple_array_conditions (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_and_create_should_not_happen_within_scope (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_attributes_are_available_to_after_initialize (unexpected)
--- FAIL: HasManyAssociationsTest#test_counting_with_single_hash (unexpected)
--- FAIL: BelongsToAssociationsTest#test_setting_foreign_key_after_nil_target_loaded (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_counter_when_update_columns (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16190962/activerecord/1721714092659/1721721094546)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16199253?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16199253?buildTab=artifacts#/activerecord) on master @ [4af6b13bf95a52fe8f6857ffd32e8c1156c36950](https://github.com/cockroachdb/cockroach/commits/4af6b13bf95a52fe8f6857ffd32e8c1156c36950):


```
--- FAIL: AttributeMethodsTest#test_write_attribute (unexpected)
--- FAIL: AttributeMethodsTest#test_boolean_attribute_predicate (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_a_through_association_twice_does_not_reset_it (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_with_inexistent_foreign_key_failing (unexpected)
--- FAIL: EagerAssociationTest#test_finding_with_includes_on_has_one_association_with_same_include_includes_only_once (unexpected)
--- FAIL: AttributeMethodsTest#test_attributes_without_primary_key (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit_and_conditions_array_on_the_eagers (unexpected)
--- FAIL: EagerAssociationTest#test_scoping_with_a_circular_preload (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_conditions_string_with_quoted_table_name (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_multiple_stis_and_order (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_default_scope_as_class_method_using_find_by_method (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_with_a_bang_for_single_table_inheritance_(STI_Models) (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_an_association_method_issues_a_deprecation_warning (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_proxy_should_not_respond_to_private_methods (unexpected)
--- FAIL: AttributeMethodsTest#test_aliasing_``id``_attribute_allows_reading_the_column_value (unexpected)
--- FAIL: EagerAssociationTest#test_loading_association_with_intersection_joins (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_with_has_one_through_an_sti_with_after_initialize (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_one_through_join_model_with_conditions_on_the_through (unexpected)
--- FAIL: AttributeMethodsTest#test_hashes_are_not_mangled (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_through_has_one_association_with_conditions_on_association (unexpected)
--- FAIL: UniquenessValidationTest#test_validate_uniqueness_with_columns_which_are_sql_keywords (unexpected)
--- FAIL: AttributeMethodsTest#test_caching_a_nil_primary_key (unexpected)
--- FAIL: EagerAssociationTest#test_preconfigured_includes_with_has_one (unexpected)
--- FAIL: EagerAssociationTest#test_with_two_tables_in_from_without_getting_double_quoted (unexpected)
--- FAIL: AttributeMethodsTest#test_aliasing_``id``_attribute_allows_reading_the_column_value_for_a_CPK_model (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_on_relation (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_enum_method_issues_a_deprecation_warning (unexpected)
--- FAIL: EagerAssociationTest#test_dont_create_temporary_active_record_instances (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_cascaded_eager_association_loading_with_twice_includes_edge_cases (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_transaction (unexpected)
--- FAIL: PostgresqlTimestampWithTimeZoneTest#test_timestamp_with_zone_values_with_rails_time_zone_support_and_timestamptz_and_no_time_zone_set (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_an_overridden_original_method_issues_a_deprecation (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_respects_hash_condition (unexpected)
--- FAIL: AttributeMethodsTest#test_bulk_updates_respect_access_control (unexpected)
--- FAIL: HasOneAssociationsTest#test_assignment_before_child_saved (unexpected)
--- FAIL: ForbiddenAttributesProtectionTest#test_permitted_attributes_can_be_used_for_sti_inheritance_column (unexpected)
--- FAIL: AttributeMethodsTest#test_came_from_user? (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_of_stis_with_multiple_references (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attribute_for_database_with_aliased_attribute (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_belongs_to_sti (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attribute_when_true (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attribute_when_false (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_within_expiration_duration (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_hmt_does_not_table_name_collide_when_joining_associations (unexpected)
--- FAIL: HasOneAssociationsTest#test_natural_assignment_to_nil (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attribute_for_database (unexpected)
--- FAIL: EagerAssociationTest#test_loading_with_one_association_with_non_preload (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16199253/activerecord/1721800582159/1721809643623)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16208824?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16208824?buildTab=artifacts#/activerecord) on master @ [da85c982fa391ef34fd14c29006b1f629217ff42](https://github.com/cockroachdb/cockroach/commits/da85c982fa391ef34fd14c29006b1f629217ff42):


```
--- FAIL: RelationTest#test_size_with_eager_loading_and_custom_order_and_distinct (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_multiple_associations_with_same_table_has_one (unexpected)
--- FAIL: SignedIdTest#test_finding_signed_record_outside_expiration_duration_raises_on_the_bang (unexpected)
--- FAIL: RelationTest#test_create_with_block (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_delete_all_for_with_dependent_option_destroy (unexpected)
--- FAIL: RelationTest#test_support_upper_and_lower_case_directions (unexpected)
--- FAIL: RelationTest#test_create_or_find_by (unexpected)
--- FAIL: PersistenceTest#test_becomes_includes_errors (unexpected)
--- FAIL: PersistenceTest#test_destroy_with_multiple_composite_primary_keys (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_primary_key_option (unexpected)
--- FAIL: ModulesTest#test_find_account_and_include_company (unexpected)
--- FAIL: EagerAssociationTest#test_calculate_with_string_in_from_and_eager_loading (unexpected)
--- FAIL: EagerAssociationTest#test_eager_loading_of_instance_dependent_associations_is_not_supported (unexpected)
--- FAIL: EagerAssociationTest#test_eager_loaded_has_many_association_without_primary_key (unexpected)
--- FAIL: PersistenceTest#test_save_touch_false (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_store_association_in_two_relations_with_one_save_in_existing_object (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_limit_and_offset_and_conditions (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_multi_table_conditional_properly_counts_the_records_when_using_size (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_delete_on_association_clears_scope (unexpected)
--- FAIL: RelationTest#test_finding_with_assoc_reorder_by_aliased_attributes (unexpected)
--- FAIL: EagerAssociationTest#test_loading_association_with_intersection_joins (unexpected)
--- FAIL: PersistenceTest#test_increment_attribute (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_assign_ids (unexpected)
--- FAIL: EagerAssociationTest#test_including_association_based_on_sql_condition_and_no_database_column (unexpected)
--- FAIL: RelationTest#test_eager_load_with_blank_value (unexpected)
--- FAIL: PersistenceTest#test_update_many! (unexpected)
--- FAIL: BasicsTest#test_count_with_join (unexpected)
--- FAIL: BasicsTest#test_many_mutations (unexpected)
--- FAIL: RelationTest#test_where_with_take_memoization (unexpected)
--- FAIL: RelationTest#test_find_with_included_associations (unexpected)
--- FAIL: PersistenceTest#test_preserve_original_sti_type (unexpected)
--- FAIL: HasOneAssociationsTest#test_association_enum_works_properly_with_nested_join (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_through_record_is_built_when_created_with_where (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_adding_from_the_project_fixed_timestamp (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_conditions_string_with_unquoted_table_name (unexpected)
--- FAIL: EagerAssociationTest#test_load_with_sti_sharing_association (unexpected)
--- FAIL: HasOneAssociationsTest#test_association_attributes_are_available_to_after_initialize (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_destroy_associations_destroys_multiple_associations (unexpected)
--- FAIL: RelationTest#test_loaded_first_with_limit (unexpected)
--- FAIL: RelationTest#test_many (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_collection_singular_ids_getter_with_string_primary_keys (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_include_uses_array_include_after_loaded (unexpected)
--- FAIL: SignedIdTest#test_fail_to_find_signed_record_within_expiration_time (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_limit_and_multiple_associations (unexpected)
--- FAIL: RelationTest#test_find_by!_with_multi-arg_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: SignedIdTest#test_finding_signed_record_that_has_been_destroyed_raises_on_the_bang (unexpected)
--- FAIL: HasOneAssociationsTest#test_cant_save_readonly_association (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16208824/activerecord/1721886913410/1721894244797)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16217301?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16217301?buildTab=artifacts#/activerecord) on master @ [7fb362dd5aa6e85d65c4c89f208c5bed51ab9692](https://github.com/cockroachdb/cockroach/commits/7fb362dd5aa6e85d65c4c89f208c5bed51ab9692):


```
--- FAIL: FinderTest#test_hash_condition_find_with_one_condition_being_aggregate_and_another_not (unexpected)
--- FAIL: BelongsToAssociationsTest#test_where_on_polymorphic_association_with_nil (unexpected)
--- FAIL: BasicsTest#test_#connecting_to_with_role (unexpected)
--- FAIL: FinderTest#test_exists_returns_false_with_false_arg (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_uses_default_if_not_persisted (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_include_method_in_association_through_should_return_true_for_instance_added_with_nested_builds (unexpected)
--- FAIL: RelationTest#test_find_with_readonly_option (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::ConnectionHandlersMultiDbTest#test_loading_relations_with_multi_db_connections (unexpected)
--- FAIL: PostgresqlByteaTest#test_write_value (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_association (unexpected)
--- FAIL: BasicsTest#test_attributes_on_dummy_time_with_invalid_time (unexpected)
--- FAIL: InheritanceAttributeMappingTest#test_sti_with_custom_type (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_create_on_new_record (unexpected)
--- FAIL: BasicsTest#test_switching_between_table_name (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attribute_when_true (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_through_scope_with_includes (unexpected)
--- FAIL: FinderTest#test_find_with_nil_inside_set_passed_for_attribute (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_find_in_association (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_with_reflection_using_class_name_and_fixtures (unexpected)
--- FAIL: RelationTest#test_none? (unexpected)
--- FAIL: AssociationsTest#test_subselect (unexpected)
--- FAIL: ModulesTest#test_eager_loading_in_modules (unexpected)
--- FAIL: EagerAssociationTest#test_eager_loading_with_conditions_on_string_joined_table_preloads (unexpected)
--- FAIL: HasManyAssociationsTest#test_collection_not_empty_after_building (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_default_scope (unexpected)
--- FAIL: FinderTest#test_member_on_loaded_relation_with_composite_primary_key (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_dynamic_find_should_respect_association_include (unexpected)
--- FAIL: RelationTest#test_select_takes_an_aliased_attribute (unexpected)
--- FAIL: PostgresqlByteaTest#test_write_nil (unexpected)
--- FAIL: CalculationsTest#test_count_with_eager_loading_and_custom_order_and_distinct (unexpected)
--- FAIL: BasicsTest#test_distinct_delegates_to_scoped (unexpected)
--- FAIL: BasicsTest#test_columns_should_obey_set_primary_key (unexpected)
--- FAIL: RelationTest#test_unscope_specific_where_value (unexpected)
--- FAIL: EagerAssociationTest#test_loading_association_with_same_table_joins (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_does_not_use_order_by (unexpected)
--- FAIL: HasManyAssociationsTest#test_cant_save_has_many_readonly_association (unexpected)
--- FAIL: PersistenceTest#test_persist_inherited_class_with_different_table_name (unexpected)
--- FAIL: HasManyAssociationsTest#test_restrict_with_exception (unexpected)
--- FAIL: PersistenceTest#test_decrement_with_touch_updates_timestamps (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_reassignment_of_associated_type_updates_the_object (unexpected)
--- FAIL: BasicsTest#test_comparison_with_different_objects (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_has_one_with_cpk (unexpected)
--- FAIL: CalculationsTest#test_should_get_minimum_of_arel_attribute (unexpected)
--- FAIL: BelongsToAssociationsTest#test_building_the_belonging_object_with_an_unrelated_type (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociationWithAcceptsNestedAttributes#test_indexed_errors_on_base_attribute_should_be_properly_translated (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_destruction_does_not_error_without_primary_key (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_reserved_words (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16217301/activerecord/1721973269049/1721980985943)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16229827?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16229827?buildTab=artifacts#/activerecord) on master @ [1893ac1d7188dc87dbb2b379b30172bf83ca4645](https://github.com/cockroachdb/cockroach/commits/1893ac1d7188dc87dbb2b379b30172bf83ca4645):


```
--- FAIL: HasManyAssociationsTest#test_set_ids_for_association_on_new_record_applies_association_correctly (unexpected)
--- FAIL: HasManyAssociationsTest#test_first_or_create_adds_the_record_to_the_association (unexpected)
--- FAIL: HasOneThroughDisableJoinsAssociationsTest#test_nil_on_disable_joins_through (unexpected)
--- FAIL: EagerAssociationTest#test_finding_with_includes_on_empty_polymorphic_type_column (unexpected)
--- FAIL: FinderTest#test_named_bind_variables_with_quotes (unexpected)
--- FAIL: FinderTest#test_any_with_scope_on_hash_includes (unexpected)
--- FAIL: AttributeMethodsTest#test_write_attribute_raises_ActiveModel::MissingAttributeError_when_the_attribute_does_not_exist (unexpected)
--- FAIL: HasManyAssociationsTest#test_get_ids_for_through (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_through_has_one_association_with_conditions_on_nested_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_has_many_preloading_with_duplicate_records (unexpected)
--- FAIL: PersistenceTest#test_build_many (unexpected)
--- FAIL: EagerAssociationTest#test_eager_has_many_through_with_order (unexpected)
--- FAIL: OptimisticLockingTest#test_lock_with_custom_column_without_default_should_work_with_null_in_the_database (unexpected)
--- FAIL: HasManyAssociationsTest#test_adding (unexpected)
--- FAIL: BelongsToAssociationsTest#test_with_polymorphic_and_condition (unexpected)
--- FAIL: AttributeMethodsTest#test_read_overridden_attribute_with_predicate_respects_override (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_select_chosen_fields_only (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_insert_records_via_has_many_through_association_with_scope (unexpected)
--- FAIL: FinderTest#test_second_to_last (unexpected)
--- FAIL: FinderTest#test_find_by_ids (unexpected)
--- FAIL: FinderTest#test_member_on_unloaded_relation_with_limit (unexpected)
--- FAIL: TransactionTest#test_restore_frozen_state_after_double_destroy (unexpected)
--- FAIL: EagerAssociationTest#test_exceptions_have_suggestions_for_fix (unexpected)
--- FAIL: HasManyAssociationsTest#test_find_ids (unexpected)
--- FAIL: BelongsToAssociationsTest#test_where_with_custom_primary_key (unexpected)
--- FAIL: PersistenceTest#test_update_many_with_invalid_id (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_for_inspect_with_a_long_array (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_deleting_all (unexpected)
--- FAIL: HasManyAssociationsTest#test_replace_failure (unexpected)
--- FAIL: SerializedAttributeTestWithYamlSafeLoad#test_serialized_attribute_works_under_concurrent_initial_access (unexpected)
--- FAIL: FinderTest#test_find_by_nil_attribute (unexpected)
--- FAIL: InheritanceTest#test_complex_inheritance (unexpected)
--- FAIL: PersistenceTest#test_save_valid_record (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_multiple_associations_with_same_table_has_one (unexpected)
--- FAIL: FinderTest#test_find_an_empty_array (unexpected)
--- FAIL: HasManyAssociationsTest#test_can_unscope_and_where_the_default_scope_of_the_associated_model (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_with_reflection_using_class_name_and_fixtures (unexpected)
--- FAIL: PersistenceTest#test_update! (unexpected)
--- FAIL: HasManyAssociationsTest#test_replace_with_new (unexpected)
--- FAIL: SerializedAttributeTestWithYamlSafeLoad#test_serialized_attribute_on_alias_attribute (unexpected)
--- FAIL: PersistenceTest#test_save_touch_false (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_save_succeeds_for_invalid_belongs_to_with_validate_false (unexpected)
--- FAIL: EagerAssociationTest#test_loading_association_with_string_joins (unexpected)
--- FAIL: FinderTest#test_find_with_ids_with_id_out_of_range (unexpected)
--- FAIL: UniquenessValidationTest#test_validate_uniqueness_without_primary_key (unexpected)
--- FAIL: PostgresqlTimestampTest#test_timestamp_with_zone_values_without_rails_time_zone_support (unexpected)
--- FAIL: PersistenceTest#test_destroy_uses_query_constraints_config (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16229827/activerecord/1722059653099/1722066667279)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16236776?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16236776?buildTab=artifacts#/activerecord) on master @ [fb8032d12d2a787b54076812595e9d7fb22f925a](https://github.com/cockroachdb/cockroach/commits/fb8032d12d2a787b54076812595e9d7fb22f925a):


```
--- FAIL: CalculationsTest#test_calculation_with_query_cache (unexpected)
--- FAIL: PersistenceTest#test_decrement_attribute (unexpected)
--- FAIL: CalculationsTest#test_aggregate_attribute_on_enum_type (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_belongs_to_sti (unexpected)
--- FAIL: CalculationsTest#test_pick_loaded_relation (unexpected)
--- FAIL: RelationTest#test_group_with_subquery_in_from_does_not_use_original_table_name (unexpected)
--- FAIL: CalculationsTest#test_count_selected_arel_attribute (unexpected)
--- FAIL: HasManyAssociationsTest#test_passes_custom_context_validation_to_validate_children (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_dynamic_find_should_respect_association_include (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_update_counter_caches_on_delete_with_dependent_nullify (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_explicit_join (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_get_ids_for_unloaded_associations_does_not_load_them (unexpected)
--- FAIL: HasManyAssociationsTest#test_subsequent_calls_to_one_should_use_query (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_can_update_through_association (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_proxy_transaction_method_starts_transaction_in_association_class (unexpected)
--- FAIL: HasManyAssociationsTest#test_key_ensuring_owner_was_is_valid_when_dependent_option_is_destroy_async (unexpected)
--- FAIL: HasManyAssociationsTest#test_custom_named_counter_cache (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_adding_from_the_project (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_default_scope_on_the_target (unexpected)
--- FAIL: EagerAssociationTest#test_eager-loading_readonly_association (unexpected)
--- FAIL: AssociationsTest#test_belongs_to_with_foreign_key_as_an_array_raises (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_grouped_with_longer_field (unexpected)
--- FAIL: InheritanceTest#test_abstract_class (unexpected)
--- FAIL: ReflectionTest#test_automatic_inverse_suppresses_name_error_for_association (unexpected)
--- FAIL: RelationTest#test_build_with_array (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_self_type_mismatch (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_has_many_through_with_distinct (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_has_and_belongs_to_many_in_a_namespaced_model_pointing_to_a_non_namespaced_model (unexpected)
--- FAIL: AssociationsTest#test_assign_persisted_composite_foreign_key_belongs_to_association (unexpected)
--- FAIL: CalculationsTest#test_sum_with_from_option (unexpected)
--- FAIL: ReflectionTest#test_non_existent_types_are_identity_types (unexpected)
--- FAIL: RelationTest#test_find_or_initialize_by_with_block (unexpected)
--- FAIL: RelationTest#test_find_by!_requires_at_least_one_argument (unexpected)
--- FAIL: CalculationsTest#test_should_order_by_grouped_field (unexpected)
--- FAIL: HasManyAssociationsTest#test_collection_size_twice_for_regressions (unexpected)
--- FAIL: EagerAssociationTest#test_including_duplicate_objects_from_belongs_to (unexpected)
--- FAIL: EagerAssociationTest#test_loading_with_multiple_associations (unexpected)
--- FAIL: HasManyAssociationsTest#test_create (unexpected)
--- FAIL: CalculationsTest#test_should_count_field_in_joined_table_with_group_by_when_tables_share_column_names (unexpected)
--- FAIL: HasManyAssociationsTest#test_replace_with_same_content (unexpected)
--- FAIL: HasManyAssociationsTest#test_in_memory_replacement_maintains_order (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_preload_raise_errors (unexpected)
--- FAIL: CalculationsTest#test_pluck_loaded_relation_aliased_attribute (unexpected)
--- FAIL: IntegrationTest#test_to_param_for_a_composite_primary_key_model (unexpected)
--- FAIL: ReflectionTest#test_includes_accepts_strings (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_missing_first_record (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_association_through_a_belongs_to_association_where_the_association_doesnt_exist (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16236776/activerecord/1722145999148/1722154188055)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16244040?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16244040?buildTab=artifacts#/activerecord) on master @ [fb8032d12d2a787b54076812595e9d7fb22f925a](https://github.com/cockroachdb/cockroach/commits/fb8032d12d2a787b54076812595e9d7fb22f925a):


```
--- FAIL: AttributeMethodsTest#test_attribute_names_on_a_queried_record (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_merge_join_association_with_has_many_through_association_proxy (unexpected)
--- FAIL: InheritanceTest#test_new_with_invalid_type (unexpected)
--- FAIL: CalculationsTest#test_should_sum_field_with_conditions (unexpected)
--- FAIL: ModulesTest#test_module_table_name_prefix (unexpected)
--- FAIL: BasicsTest#test_cannot_call_connected_to_on_the_abstract_class_that_did_not_establish_the_connection (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_polymorphic_with_rewhere (unexpected)
--- FAIL: BasicsTest#test_auto_id (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_many_via_block (unexpected)
--- FAIL: RelationTest#test_to_json (unexpected)
--- FAIL: PersistenceTest#test_build_many (unexpected)
--- FAIL: ActiveRecord::Migration::ColumnsTest#test_change_column_null_false (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_find_should_append_to_association_order (unexpected)
--- FAIL: HasManyAssociationsTest#test_finding_with_foreign_key (unexpected)
--- FAIL: HasManyAssociationsTest#test_transaction_when_deleting_persisted (unexpected)
--- FAIL: RelationTest#test_reload (unexpected)
--- FAIL: AssociationsTest#test_loading_the_association_target_should_load_most_recent_attributes_for_child_records_marked_for_destruction (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_destroy_all_on_composite_primary_key_model (unexpected)
--- FAIL: PersistenceTest#test_delete_all (unexpected)
--- FAIL: ActiveRecord::Migration::ForeignKeyChangeColumnWithPrefixTest#test_remove_reference_column_of_child_table (unexpected)
--- FAIL: HasManyAssociationsTest#test_set_ids_for_association_on_new_record_applies_association_correctly (unexpected)
--- FAIL: AttributeMethodsTest#test_declared_prefixed_attribute_method_affects_respond_to?_and_method_missing (unexpected)
--- FAIL: InheritanceTest#test_update_all_within_inheritance (unexpected)
--- FAIL: ViewWithoutPrimaryKeyTest#test_reading (unexpected)
--- FAIL: CalculationsTest#test_sum_takes_attribute_type_precedence_over_database_type (unexpected)
--- FAIL: InheritanceTest#test_alt_find_first_within_inheritance (unexpected)
--- FAIL: HasManyAssociationsTest#test_building_the_associated_object_with_an_unrelated_type (unexpected)
--- FAIL: HasOneAssociationsTest#test_create_association_replaces_existing_with_dependent_option (unexpected)
--- FAIL: BelongsToAssociationsTest#test_optional_relation (unexpected)
--- FAIL: CallbacksOnMultipleInstancesInATransactionTest#test_updated_callback_called_on_first_to_save_when_followed_in_transaction_by_destroy_from_separate_instance_with_old_configuration (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_multiple_fields_when_table_name_is_too_long (unexpected)
--- FAIL: BelongsToAssociationsTest#test_should_set_foreign_key_on_save (unexpected)
--- FAIL: HasManyAssociationsTest#test_create_many (unexpected)
--- FAIL: InheritanceTest#test_becomes_and_change_tracking_for_inheritance_columns (unexpected)
--- FAIL: PostgresqlByteaTest#test_write_value (unexpected)
--- FAIL: IntegrationTest#test_param_delimiter_changes_delimiter_used_in_to_param (unexpected)
--- FAIL: BasicsTest#test_dont_clear_inheritance_column_when_setting_explicitly (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_select_chosen_fields_only (unexpected)
--- FAIL: FinderTest#test_find_by_title_and_id_with_hash (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_assignment_updates_foreign_id_field_for_new_and_saved_records (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_for_existing_record_with_nil_updated_timestamps (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_delete_on_association_clears_scope (unexpected)
--- FAIL: InheritanceTest#test_a_bad_type_column (unexpected)
--- FAIL: ActiveRecord::Migration::ForeignKeyChangeColumnWithPrefixTest#test_remove_foreign_key_by_column_in_change_table (unexpected)
--- FAIL: HasManyAssociationsTest#test_get_ids_ignores_include_option (unexpected)
--- FAIL: TestAutosaveAssociationsInGeneral#test_autosave_does_not_pass_through_non_custom_validation_contexts (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_for_inspect_with_a_date (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16244040/activerecord/1722232665857/1722240211572)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16258192?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16258192?buildTab=artifacts#/activerecord) on master @ [5d9405ff2e16622c055f94208992a5347c4c273e](https://github.com/cockroachdb/cockroach/commits/5d9405ff2e16622c055f94208992a5347c4c273e):


```
--- FAIL: RelationTest#test_count_explicit_columns (unexpected)
--- FAIL: PersistenceTest#test_destroy_record_with_associations (unexpected)
--- FAIL: PersistenceTest#test_update_raises_record_not_found_exception (unexpected)
--- FAIL: RelationTest#test_unscope_with_blank_value (unexpected)
--- FAIL: RelationTest#test_count (unexpected)
--- FAIL: PersistenceTest#test_decrement_attribute_by (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: RelationTest#test_extracted_association (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_preload_raise_errors (unexpected)
--- FAIL: RelationTest#test_eager_load_with_blank_value (unexpected)
--- FAIL: RelationTest#test_references_doesnt_trigger_eager_loading_if_reference_not_included (unexpected)
--- FAIL: PersistenceTest#test_reload_via_querycache (unexpected)
--- FAIL: RelationTest#test_unscope_with_subquery (unexpected)
--- FAIL: PersistenceTest#test_update_columns_changing_id (unexpected)
--- FAIL: PersistenceTest#test_becomes_after_reload_schema_from_cache (unexpected)
--- FAIL: RelationTest#test_create (unexpected)
--- FAIL: RelationTest#test_count_with_distinct (unexpected)
--- FAIL: RelationTest#test_reverse_arel_order_with_function (unexpected)
--- FAIL: RelationTest#test_unscope_with_double_dot_where (unexpected)
--- FAIL: PersistenceTest#test_delete_record_with_associations (unexpected)
--- FAIL: RelationTest#test_create_bang_with_array (unexpected)
--- FAIL: RelationTest#test_finding_with_cross_table_order_and_limit (unexpected)
--- FAIL: RelationTest#test_create_or_find_by_with_non_unique_attributes (unexpected)
--- FAIL: RelationTest#test_create_or_find_by_with_bang_within_transaction (unexpected)
--- FAIL: RelationTest#test_first_or_create_bang_with_invalid_block (unexpected)
--- FAIL: RelationTest#test_order_by_relation_attribute (unexpected)
--- FAIL: RelationTest#test_first_or_create_bang_with_invalid_options (unexpected)
--- FAIL: PersistenceTest#test_update_many_with_active_record_base_object (unexpected)
--- FAIL: RelationTest#test_respond_to_dynamic_finders (unexpected)
--- FAIL: RelationTest#test_locked_should_not_build_arel (unexpected)
--- FAIL: RelationTest#test_unscope_with_table_name_qualified_hash (unexpected)
--- FAIL: PersistenceTest#test_update_uses_query_constraints_config (unexpected)
--- FAIL: PersistenceTest#test_update_columns_with_one_changed_and_one_updated (unexpected)
--- FAIL: RelationTest#test_size (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_reselect_raise_errors (unexpected)
--- FAIL: RelationTest#test_any (unexpected)
--- FAIL: PersistenceTest#test_destroy_new_record (unexpected)
--- FAIL: RelationTest#test_new_with_array (unexpected)
--- FAIL: PersistenceTest#test_create (unexpected)
--- FAIL: RelationTest#test_includes_with_blank_value (unexpected)
--- FAIL: RelationTest#test_reverse_order_with_function_other_predicates (unexpected)
--- FAIL: RelationTest#test_find_by_with_non-hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: RelationTest#test_create_with_array (unexpected)
--- FAIL: RelationTest#test_find_by_with_hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: RelationTest#test_one (unexpected)
--- FAIL: RelationTest#test_empty_complex_chained_relations (unexpected)
--- FAIL: PersistenceTest#test_update_sti_type (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16258192/activerecord/1722318811600/1722326519698)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16270317?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16270317?buildTab=artifacts#/activerecord) on master @ [c7d1ceed692d36e9d7c5a71e7122408e2e3e3b8b](https://github.com/cockroachdb/cockroach/commits/c7d1ceed692d36e9d7c5a71e7122408e2e3e3b8b):


```
--- FAIL: HasManyAssociationsTest#test_clearing_an_exclusively_dependent_association_collection (unexpected)
--- FAIL: FoxyFixturesTest#test_supports_timestamps_in_join_tables (unexpected)
--- FAIL: InheritanceAttributeMappingTest#test_polymorphic_associations_custom_type (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_overloaded_properties_save (unexpected)
--- FAIL: HasOneAssociationsTest#test_polymorphic_has_one_with_touch_option_on_update_will_touch_record_by_fetching_from_database_if_needed (unexpected)
--- FAIL: RelationTest#test_find_by_id_with_list_of_ar (unexpected)
--- FAIL: HasManyAssociationsTest#test_replace_failure (unexpected)
--- FAIL: AttributeMethodsTest#test_should_unserialize_attributes_for_frozen_records (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_where_first_level_returns_nil (unexpected)
--- FAIL: IntegrationTest#test_cache_key_is_stable_with_versioning_on (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_reorder_raise_errors (unexpected)
--- FAIL: DelegatedTypeTest#test_delegated_type_predicates_with_custom_foreign_type (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_build_then_save_with_has_one_inverse (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_on_association_clears_scope (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_associations_with_string_joins_and_order_references (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_with_instance_dependent_scope (unexpected)
--- FAIL: CockroachDB::FixturesTest#test_multiple_clean_fixtures (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_proxy_transaction_method_starts_transaction_in_association_class (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_fields_with_table_alias (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_for_inspect_with_a_date (unexpected)
--- FAIL: AssociationsTest#test_loading_the_association_target_should_load_most_recent_attributes_for_child_records_marked_for_destruction (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_truncates_after_parameterize (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_none_should_return_false_if_any (unexpected)
--- FAIL: HasManyAssociationsTest#test_dependent_association_respects_optional_hash_conditions_on_delete (unexpected)
--- FAIL: PersistenceTest#test_update_many_with_duplicated_ids! (unexpected)
--- FAIL: CalculationsTest#test_pluck_type_cast_with_joins_without_table_name_qualified_column (unexpected)
--- FAIL: RelationTest#test_count_complex_chained_relations (unexpected)
--- FAIL: EagerAssociationTest#test_eager_count_performed_on_a_has_many_association_with_multi_table_conditional (unexpected)
--- FAIL: PersistenceTest#test_update_columns_should_raise_exception_if_new_record (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_attributes_are_available_to_after_initialize (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_through_scope_isnt_affected_by_scoping (unexpected)
--- FAIL: HasOneAssociationsTest#test_finding_using_primary_key (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_existing_with_nonstandard_primary_key_on_belongs_to (unexpected)
--- FAIL: ModulesTest#test_module_spanning_has_and_belongs_to_many_associations (unexpected)
--- FAIL: HasOneAssociationsTest#test_dependence_with_missing_association_and_nullify (unexpected)
--- FAIL: TransactionCallbacksTest#test_only_call_after_commit_on_create_after_transaction_commits_for_new_record_if_create_succeeds_creating_through_association (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_nested_has_many_through_association_with_unpersisted_parent_instance (unexpected)
--- FAIL: HasManyAssociationsTest#test_add_record_to_collection_should_change_its_updated_at (unexpected)
--- FAIL: HabtmDestroyOrderTest#test_not_destroying_a_lesson_with_students_leaves_student<=>lesson_association_intact (unexpected)
--- FAIL: EagerAssociationTest#test_eager_load_has_many_through_quotes_table_and_column_names (unexpected)
--- FAIL: RelationTest#test_count_with_block (unexpected)
--- FAIL: SerializedAttributeTest#test_serialized_boolean_value_true (unexpected)
--- FAIL: HasOneAssociationsTest#test_restrict_with_exception (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_regroup_raise_errors (unexpected)
--- FAIL: PersistenceTest#test_delete_record_with_associations (unexpected)
--- FAIL: ActiveRecord::CacheKeyTest#test_cache_version_calls_updated_at_when_the_value_is_generated_at_create_time (unexpected)
--- FAIL: HasManyAssociationsTest#test_can_rewhere_the_default_scope_of_the_associated_model (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16270317/activerecord/1722405075841/1722413445054)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16284955?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16284955?buildTab=artifacts#/activerecord) on master @ [7a1bf52daa3f3dcaa434fd66d2b3c960a3c9c2a5](https://github.com/cockroachdb/cockroach/commits/7a1bf52daa3f3dcaa434fd66d2b3c960a3c9c2a5):


```
--- FAIL: HasManyThroughAssociationsTest#test_update_counter_caches_on_delete_with_dependent_nullify (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_get_ids_for_unloaded_associations_does_not_load_them (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_association_through_a_has_many_association_to_self (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_collection_singular_ids_through_setter_raises_exception_when_invalid_ids_set (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_polymorphic_with_rewhere (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_delete_all_for_with_dependent_option_nullify (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_through_association_with_through_scope_and_nested_where (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_do_not_cache_association_reader_if_the_though_method_has_default_scopes (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_delete_association (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_dynamic_find_should_respect_association_include (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_collection_singular_ids_getter_with_string_primary_keys (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_build_then_remove_then_save (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_that_should_not_be_fully_merged (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_count_has_many_through_with_named_scope (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_duplicated_has_many_through_with_through_scope_with_joins (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_replace_association_with_duplicates (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_tags_has_manu_posts_through_association_with_composite_query_constraints (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_existing_with_nonstandard_primary_key_on_belongs_to (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_yaml_dump_and_load (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_push_with_invalid_record (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_update_counter_caches_on_delete_with_dependent_destroy (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_clear_associations (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_belongs_to_should_update_when_the_through_foreign_key_changes (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_default_scope_on_the_target (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_delete_on_association_clears_scope (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_joining_has_many_through_with_distinct (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_through_association_with_left_joins (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_collection_delete_with_nonstandard_primary_key_on_belongs_to (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_delete_through_belongs_to_with_dependent_delete_all (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_primary_key_option (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_association_through_a_belongs_to_association_where_the_association_doesnt_exist (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_no_pk_join_table_append (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_replace_order_is_preserved (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_unscope_should_affect_to_through_scope (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_associations_on_new_records_use_null_relations (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_through_scope_isnt_affected_by_scoping (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_marshal_dump_and_load_with_gzip (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_marshal_dump_and_load (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_associations_sum_on_columns (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_destroy_association (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_joining_has_many_through_belongs_to (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_replace_association (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_get_ids_for_has_many_through_with_conditions_should_not_preload (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_association_callback_ordering (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_include_method_in_association_through_should_return_true_for_instance_added_with_build (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_polymorphic_with_primary_key_option (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16284955/activerecord/1722491575506/1722500676131)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16298784?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16298784?buildTab=artifacts#/activerecord) on master @ [78fa126dfc749e620c28531c1659e35bf98034ec](https://github.com/cockroachdb/cockroach/commits/78fa126dfc749e620c28531c1659e35bf98034ec):


```
--- FAIL: BasicsTest#test_previously_new_record_returns_boolean (unexpected)
--- FAIL: RelationTest#test_create_or_find_by_with_bang_within_transaction (unexpected)
--- FAIL: FinderTest#test_implicit_order_set_to_primary_key (unexpected)
--- FAIL: HasManyAssociationsTest#test_regular_create_on_has_many_when_parent_is_new_raises (unexpected)
--- FAIL: BasicsTest#test_primary_key_and_references_columns_should_be_identical_type (unexpected)
--- FAIL: RelationTest#test_empty_complex_chained_relations (unexpected)
--- FAIL: RelationTest#test_find_by_with_multi-arg_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: ReflectionTest#test_content_columns (unexpected)
--- FAIL: FinderTest#test_find_doesnt_have_implicit_ordering (unexpected)
--- FAIL: BasicsTest#test_bignum_pk (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_collection_singular_ids_setter (unexpected)
--- FAIL: HasManyAssociationsTest#test_counting_with_counter_sql (unexpected)
--- FAIL: HasOneAssociationsTest#test_destroyed_by_association_set_in_child_destroy_callback_on_replace (unexpected)
--- FAIL: CalculationsTest#test_sum_takes_attribute_type_precedence_over_database_type (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_no_pk_join_table_append (unexpected)
--- FAIL: FinderTest#test_exists_uses_existing_scope (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_adding_a_collection (unexpected)
--- FAIL: FinderTest#test_include_on_unloaded_relation_with_having_referencing_aliased_select (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_save_succeeds_for_invalid_belongs_to_with_validate_false (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_has_many_association_associated_by_a_composite_query_constraints (unexpected)
--- FAIL: HasManyAssociationsTest#test_set_ids_for_association_on_new_record_applies_association_correctly (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_belongs_to_should_update_when_the_through_foreign_key_changes (unexpected)
--- FAIL: PersistenceTest#test_update_many_with_array_of_active_record_base_objects (unexpected)
--- FAIL: PersistenceTest#test_update_all_with_hash (unexpected)
--- FAIL: FinderTest#test_find_by_and_where_consistency_with_active_record_instance (unexpected)
--- FAIL: FinderTest#test_first_failing (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all_with_option_delete_all (unexpected)
--- FAIL: FinderTest#test_include_when_non_AR_object_passed_on_unloaded_relation (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_has_and_belongs_to_many_with_belongs_to (unexpected)
--- FAIL: HasManyAssociationsTest#test_create_the_association_with_an_array (unexpected)
--- FAIL: FinderTest#test_find (unexpected)
--- FAIL: ForbiddenAttributesProtectionTest#test_forbidden_attributes_cannot_be_used_for_sti_inheritance_column (unexpected)
--- FAIL: CalculationsTest#test_pick_one (unexpected)
--- FAIL: HasManyAssociationsTest#test_defining_has_many_association_with_nullify_dependency_lazily_evaluates_target_class (unexpected)
--- FAIL: EagerAssociationTest#test_has_many_association_ignores_the_scoping (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_belongs_to_association_associated_by_a_composite_query_constraints (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: RelationTest#test_order_with_blank_value (unexpected)
--- FAIL: RelationTest#test_unscope_with_table_name_qualified_hash (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_conditions_string_with_unquoted_table_name (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_default_scope (unexpected)
--- FAIL: InheritanceTest#test_inheritance_find (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_composite_primary_key_autosave (unexpected)
--- FAIL: RelationTest#test_find_all_using_where_with_relation_and_alternate_primary_key (unexpected)
--- FAIL: SignedIdTest#test_fail_to_find_record_from_broken_signed_id (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_deleting_all (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16298784/activerecord/1722577744633/1722586151818)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16313376?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16313376?buildTab=artifacts#/activerecord) on master @ [356e1cc24a67d7e9a1e637b98d1664794f70fa03](https://github.com/cockroachdb/cockroach/commits/356e1cc24a67d7e9a1e637b98d1664794f70fa03):


```
--- FAIL: PostgresqlTimestampTest#test_timestamp_with_zone_values_without_rails_time_zone_support (unexpected)
--- FAIL: InheritanceTest#test_alt_inheritance_find_all (unexpected)
--- FAIL: FinderTest#test_find_with_a_multiple_sets_of_composite_primary_key_wrapped_in_an_array (unexpected)
--- FAIL: AssociationsTest#test_loading_the_association_target_should_load_most_recent_attributes_for_child_records_marked_for_destruction (unexpected)
--- FAIL: PersistenceTest#test_create_with_custom_timestamps (unexpected)
--- FAIL: InheritanceTest#test_new_with_ar_base (unexpected)
--- FAIL: PersistenceTest#test_destroy_with_single_composite_primary_key (unexpected)
--- FAIL: InheritanceAttributeMappingTest#test_polymorphic_associations_custom_type (unexpected)
--- FAIL: FinderTest#test_joins_with_string_array (unexpected)
--- FAIL: FinderTest#test_nth_to_last_with_order_uses_limit (unexpected)
--- FAIL: FinderTest#test_find_ignores_previously_inserted_record (unexpected)
--- FAIL: InheritanceTest#test_inheritance_new_with_default_class (unexpected)
--- FAIL: FinderTest#test_condition_hash_interpolation (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_with_custom_primary_key (unexpected)
--- FAIL: PersistenceTest#test_delete_doesnt_run_callbacks (unexpected)
--- FAIL: FinderTest#test_take_bang_present (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_enum_method_issues_a_deprecation_warning (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_time_zone-aware_attribute_with_a_string (unexpected)
--- FAIL: AttributeMethodsTest#test_set_attributes_without_a_hash (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_attributes_not_backed_by_database_columns_return_the_default_on_models_loaded_from_database (unexpected)
--- FAIL: AssociationsTest#test_nullify_composite_foreign_key_belongs_to_association (unexpected)
--- FAIL: PersistenceTest#test_persisted_returns_boolean (unexpected)
--- FAIL: PersistenceTest#test_create_many_through_factory_with_block (unexpected)
--- FAIL: FinderTest#test_find_by_two_attributes_with_one_being_an_aggregate (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_overloaded_properties_save (unexpected)
--- FAIL: FinderTest#test_include_when_non_AR_object_passed_on_unloaded_relation (unexpected)
--- FAIL: AttributeMethodsTest#test_update_array_content (unexpected)
--- FAIL: InheritanceTest#test_class_with_blank_sti_name (unexpected)
--- FAIL: FinderTest#test_find_with_a_multiple_sets_of_composite_primary_key_wrapped_in_an_array_ordered (unexpected)
--- FAIL: CheckEscapedYamlFixturesTest#test_proper_escaped_fixture (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_save_succeeds_for_invalid_has_one_with_validate_false (unexpected)
--- FAIL: PersistenceTest#test_save_null_string_attributes (unexpected)
--- FAIL: FinderTest#test_find_with_order_on_included_associations_with_construct_finder_sql_for_association_limiting_and_is_distinct (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_method? (unexpected)
--- FAIL: EnumTest#test_validate_inclusion_of_value_in_array (unexpected)
--- FAIL: HabtmDestroyOrderTest#test_should_not_raise_error_if_have_foreign_key_in_the_join_table (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_attributes_not_backed_by_database_columns_are_not_dirty_when_unchanged (unexpected)
--- FAIL: InheritanceTest#test_where_create_with_unrelated_type (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_an_overridden_original_method_from_a_module_issues_a_deprecation (unexpected)
--- FAIL: FinderTest#test_exists_with_distinct_association_includes_limit_and_order (unexpected)
--- FAIL: AttributeMethodsTest#test_hash_content (unexpected)
--- FAIL: FinderTest#test_find_by_sql_with_sti_on_joined_table (unexpected)
--- FAIL: AttributeMethodsTest#test_boolean_attributes (unexpected)
--- FAIL: FinderTest#test_find_with_nil_inside_set_passed_for_attribute (unexpected)
--- FAIL: PersistenceTest#test_update (unexpected)
--- FAIL: AssociationsTest#test_belongs_to_a_model_with_composite_foreign_key_finds_associated_record (unexpected)
--- FAIL: ReloadAssociationCacheTest#test_reload_sets_correct_owner_for_association_cache (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16313376/activerecord/1722664257779/1722672420020)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128153 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.0-rc release-blocker]
- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16321152?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16321152?buildTab=artifacts#/activerecord) on master @ [a0a52309cbbe5cb68d576fb366411fa7d7cf3574](https://github.com/cockroachdb/cockroach/commits/a0a52309cbbe5cb68d576fb366411fa7d7cf3574):


```
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_assign_ids (unexpected)
--- FAIL: BasicsTest#test_quote (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_destroy_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_create_the_association_with_an_array (unexpected)
--- FAIL: BelongsToAssociationsTest#test_should_set_foreign_key_on_create_association (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_has_many_through_with_implicit_source (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_preloading_empty_through_association_via_joins (unexpected)
--- FAIL: DirtyTest#test_attribute_should_be_compared_with_type_cast (unexpected)
--- FAIL: CalculationsTest#test_no_limit_no_offset (unexpected)
--- FAIL: HasManyAssociationsTest#test_adding_using_create (unexpected)
--- FAIL: NumericalityValidationTest#test_on_abstract_class (unexpected)
--- FAIL: RelationTest#test_pluck_with_from_includes_original_table_name (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_type_cast_does_not_corrupt_the_query_cache (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_attributes_not_backed_by_database_columns_properly_interact_with_mutation_and_dirty (unexpected)
--- FAIL: StrictLoadingTest#test_raises_on_lazy_loading_a_strict_loading_habtm_relation (unexpected)
--- FAIL: RelationTest#test_order_by_relation_attribute (unexpected)
--- FAIL: CalculationsTest#test_select_avg_with_group_by_as_virtual_attribute_with_sql (unexpected)
--- FAIL: ActiveRecord::StatementCacheTest#test_statement_cache_with_complex_statement (unexpected)
--- FAIL: PersistenceTest#test_reload_uses_query_constraints_config (unexpected)
--- FAIL: PersistenceTest#test_increment_attribute_by (unexpected)
--- FAIL: RelationTest#test_to_yaml (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_the_given_default_value_is_cast_from_user (unexpected)
--- FAIL: RelationTest#test_unscope_with_unknown_column (unexpected)
--- FAIL: EagerAssociationTest#test_eager-loading_non-readonly_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroying_a_collection (unexpected)
--- FAIL: AssociationsTest#test_preloads_model_with_query_constraints_by_explicitly_configured_fk_and_pk (unexpected)
--- FAIL: NamedScopingTest#test_model_class_should_respond_to_many (unexpected)
--- FAIL: BelongsToAssociationsTest#test_reflect_the_most_recent_change (unexpected)
--- FAIL: RelationTest#test_finding_with_subquery (unexpected)
--- FAIL: CalculationsTest#test_count_with_order (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_the_association_with_an_array (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_associations_with_conditions (unexpected)
--- FAIL: TestAutosaveAssociationOnAHasAndBelongsToManyAssociation#test_should_not_use_default_invalid_error_on_associated_models (unexpected)
--- FAIL: RelationTest#test_many (unexpected)
--- FAIL: HasManyAssociationsTest#test_create (unexpected)
--- FAIL: HasManyAssociationsTest#test_raises_RecordNotDestroyed_when_replaced_child_can't_be_destroyed (unexpected)
--- FAIL: AssociationsTest#test_querying_by_relation_with_composite_key (unexpected)
--- FAIL: BasicsTest#test_touch_should_raise_error_on_a_new_object (unexpected)
--- FAIL: CalculationsTest#test_should_return_zero_if_sum_conditions_return_nothing (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_touch_option_on_touch_and_reassigned_parent (unexpected)
--- FAIL: RelationTest#test_order_triggers_eager_loading_when_ordering_using_hash_syntax (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_of_instance_dependent_associations_is_supported (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_conditions_string_with_quoted_table_name (unexpected)
--- FAIL: BelongsToAssociationsTest#test_attributes_are_being_set_when_initialized_from_belongs_to_association_with_where_clause (unexpected)
--- FAIL: RelationTest#test_automatically_added_where_not_references (unexpected)
--- FAIL: PersistenceTest#test_destroy_raises_record_not_found_exception (unexpected)
--- FAIL: TestAutosaveAssociationWithTouch#test_autosave_with_touch_should_not_raise_system_stack_error (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16321152/activerecord/1722750631063/1722755704392)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128153 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.0-rc release-blocker]
- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16328334?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16328334?buildTab=artifacts#/activerecord) on master @ [5f7709cf41a3996c4bfd7eae1e231e75b024db47](https://github.com/cockroachdb/cockroach/commits/5f7709cf41a3996c4bfd7eae1e231e75b024db47):


```
--- FAIL: PersistenceTest#test_populates_non_primary_key_autoincremented_column_for_a_cpk_model (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_updates_counter_cache_with_dependent_delete_all (unexpected)
--- FAIL: BasicsTest#test_quoted_table_name_after_set_table_name (unexpected)
--- FAIL: RelationTest#test_finding_with_complex_order (unexpected)
--- FAIL: SignedIdTest#test_finding_record_from_broken_signed_id_raises_on_the_bang (unexpected)
--- FAIL: CalculationsTest#test_should_return_decimal_average_of_integer_field (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_predicates_respect_access_control (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_models_with_composite_keys (unexpected)
--- FAIL: RelationTest#test_automatically_added_reorder_references (unexpected)
--- FAIL: HasManyAssociationsTest#test_three_levels_of_dependence (unexpected)
--- FAIL: RelationTest#test_blank_like_arguments_to_query_methods_dont_raise_errors (unexpected)
--- FAIL: NamedScopingTest#test_should_build_on_top_of_scope (unexpected)
--- FAIL: HasManyAssociationsTest#test_select_with_block (unexpected)
--- FAIL: AttributeMethodsTest#test_boolean_attributes (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroying (unexpected)
--- FAIL: BelongsToAssociationsTest#test_dont_find_target_when_saving_foreign_key_after_stale_association_loaded (unexpected)
--- FAIL: CalculationsTest#test_should_average_arel_attribute (unexpected)
--- FAIL: RelationTest#test_relations_with_cached_arel_can't_be_mutated_[internal_API] (unexpected)
--- FAIL: BasicsTest#test_marshal_inspected_round_trip (unexpected)
--- FAIL: HasManyAssociationsTest#test_replace_with_same_content (unexpected)
--- FAIL: TransactionTest#test_rollback_on_ruby_timeout (unexpected)
--- FAIL: PersistenceTest#test_save_nil_string_attributes (unexpected)
--- FAIL: ForbiddenAttributesProtectionTest#test_forbidden_attributes_cannot_be_used_for_sti_inheritance_column (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_uses_default_if_blank (unexpected)
--- FAIL: BasicsTest#test_comparison_with_different_objects (unexpected)
--- FAIL: BasicsTest#test_using_table_name_qualified_column_names_unless_having_SELECT_list_explicitly (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_updates_counter_cache_without_dependent_option (unexpected)
--- FAIL: CalculationsTest#test_pluck_loaded_relation_aliased_attribute (unexpected)
--- FAIL: CalculationsTest#test_count_with_from_option (unexpected)
--- FAIL: AttributeMethodsTest#test_update_array_content (unexpected)
--- FAIL: PostgresqlUUIDTest#test_uuid_change_case_does_not_mark_dirty (unexpected)
--- FAIL: RelationTest#test_create_or_find_by_with_bang_should_raise_due_to_validation_errors (unexpected)
--- FAIL: PersistenceTest#test_instantiate_creates_a_new_instance (unexpected)
--- FAIL: CalculationsTest#test_should_return_nil_as_average (unexpected)
--- FAIL: PersistenceTest#test_update_column_with_default_scope (unexpected)
--- FAIL: CallbackOrderTest#test_callbacks_run_in_order_defined_in_model_if_not_using_run_after_transaction_callbacks_in_order_defined (unexpected)
--- FAIL: NamedScopingTest#test_define_scope_for_reserved_words (unexpected)
--- FAIL: PersistenceTest#test_update_many_with_array_of_active_record_base_objects (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_left_joins_raise_errors (unexpected)
--- FAIL: BelongsToAssociationsTest#test_stale_tracking_doesn't_care_about_the_type (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_invalid_string_to_a_zone-aware_time_attribute (unexpected)
--- FAIL: CockroachDB::DefaultNumbersTest#test_default_decimal_zero_with_large_scale (unexpected)
--- FAIL: BelongsToAssociationsTest#test_natural_assignment_with_primary_key (unexpected)
--- FAIL: HasManyAssociationsTest#test_taking_with_inverse_of (unexpected)
--- FAIL: IntegrationTest#test_cache_key_for_newer_updated_at (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_touch_on_multiple_records (unexpected)
--- FAIL: RelationTest#test_finding_with_arel_order (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16328334/activerecord/1722837037397/1722843771875)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128153 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.0-rc release-blocker]
- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16343379?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16343379?buildTab=artifacts#/activerecord) on master @ [dbc62db07a1f149d305cff463b350588086f1b08](https://github.com/cockroachdb/cockroach/commits/dbc62db07a1f149d305cff463b350588086f1b08):


```
--- FAIL: PersistenceTest#test_becomes_includes_changed_attributes (unexpected)
--- FAIL: PersistenceTest#test_becomes_initializes_missing_attributes (unexpected)
--- FAIL: PersistenceTest#test_update_attribute_uses_query_constraints_config (unexpected)
--- FAIL: BasicsTest#test_create_without_prepared_statement (unexpected)
--- FAIL: AssociationsTest#test_nullify_composite_foreign_key_has_many_association (unexpected)
--- FAIL: PersistenceTest#test_update_column_should_not_use_setter_method (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_open_ended_range (unexpected)
--- FAIL: PersistenceTest#test_instantiate_creates_a_new_instance (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_index_sort_order (unexpected)
--- FAIL: BasicsTest#test_dup (unexpected)
--- FAIL: RelationTest#test_find_by_requires_at_least_one_argument (unexpected)
--- FAIL: FinderTest#test_find_with_order_on_included_associations_with_construct_finder_sql_for_association_limiting_and_is_distinct (unexpected)
--- FAIL: BasicsTest#test_marshal_round_trip (unexpected)
--- FAIL: RelationTest#test_count (unexpected)
--- FAIL: RelationTest#test_find_by!_doesn't_have_implicit_ordering (unexpected)
--- FAIL: FinderTest#test_include_on_unloaded_relation_with_limit (unexpected)
--- FAIL: RelationTest#test_first_or_create_bang_with_valid_block (unexpected)
--- FAIL: BasicsTest#test_load_with_condition (unexpected)
--- FAIL: FinderTest#test_find_by_with_hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: RelationTest#test_finding_with_subquery_without_select_does_not_change_the_select (unexpected)
--- FAIL: CalculationsTest#test_count_with_reverse_order (unexpected)
--- FAIL: PersistenceTest#test_update_uses_query_constraints_config (unexpected)
--- FAIL: FinderTest#test_#skip_query_cache!_for_#exists? (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_scoped_field (unexpected)
--- FAIL: SignedIdTest#test_fail_to_work_without_a_signed_id_verifier_secret (unexpected)
--- FAIL: CalculationsTest#test_pluck_not_auto_table_name_prefix_if_column_joined (unexpected)
--- FAIL: FinderTest#test_unexisting_record_exception_handling (unexpected)
--- FAIL: FinderTest#test_find_with_large_number (unexpected)
--- FAIL: BasicsTest#test_new_threads_get_default_the_default_connection_handler (unexpected)
--- FAIL: FinderTest#test_select_rows (unexpected)
--- FAIL: RelationTest#test_where_id_with_delegated_ar_object (unexpected)
--- FAIL: RelationTest#test_relations_limit_the_records_in_#inspect_at_10 (unexpected)
--- FAIL: PersistenceTest#test_becomes_default_sti_subclass (unexpected)
--- FAIL: PersistenceTest#test_update_raises_record_not_found_exception (unexpected)
--- FAIL: RelationTest#test_joins_with_string_array (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_join (unexpected)
--- FAIL: RelationTest#test_count_with_block (unexpected)
--- FAIL: BasicsTest#test_find_symbol_ordered_last (unexpected)
--- FAIL: FinderTest#test_find_by_one_attribute_with_several_options (unexpected)
--- FAIL: FinderTest#test_find_by_with_associations (unexpected)
--- FAIL: PersistenceTest#test_class_level_update_without_ids! (unexpected)
--- FAIL: RelationTest#test_reverse_order_with_multiargument_function (unexpected)
--- FAIL: CalculationsTest#test_should_get_maximum_of_arel_attribute (unexpected)
--- FAIL: BasicsTest#test_previously_new_record_on_destroyed_record (unexpected)
--- FAIL: ReflectionTest#test_active_record_primary_key_raises_when_missing_primary_key (unexpected)
--- FAIL: ReflectionTest#test_never_validate_association_if_explicit (unexpected)
--- FAIL: RelationTest#test_ordering_with_extra_spaces (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16343379/activerecord/1722923381800/1722931321858)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128153 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.0-rc]
- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16356864?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16356864?buildTab=artifacts#/activerecord) on master @ [7917e91eadb2f622faadf635a3e13c384e42070e](https://github.com/cockroachdb/cockroach/commits/7917e91eadb2f622faadf635a3e13c384e42070e):


```
--- FAIL: InheritanceTest#test_find_first_within_inheritance (unexpected)
--- FAIL: DelegatedTypeTest#test_touch_account (unexpected)
--- FAIL: PersistenceTest#test_create_model_with_custom_named_uuid_pk_populates_id (unexpected)
--- FAIL: OptimisticLockingWithSchemaChangeTest#test_destroy_existing_object_with_locking_column_value_null_in_the_database (unexpected)
--- FAIL: IntegrationTest#test_cache_key_for_existing_record_is_not_timezone_dependent (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_nil_with_aggregate_having_one_mapping (unexpected)
--- FAIL: EagerAssociationTest#test_finding_with_includes_on_has_many_association_with_same_include_includes_only_once (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_with_or_doesnt_set_inverse_instance_key (unexpected)
--- FAIL: IntegrationTest#test_cache_key_format_for_existing_record_with_updated_at (unexpected)
--- FAIL: HasManyAssociationsTest#test_has_many_build_with_options (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_through_has_one_association_with_order (unexpected)
--- FAIL: InheritanceTest#test_a_bad_type_column (unexpected)
--- FAIL: SerializedAttributeTest#test_serialized_no_default_class_for_object (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_with_a_polymorphic_association_and_using_the_existential_predicate_but_also_using_a_select (unexpected)
--- FAIL: FinderTest#test_model_class_responds_to_last_bang (unexpected)
--- FAIL: FinderTest#test_exists_with_large_number (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_with_create_and_invalid_options (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_with_scope_with_joins (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_one_on_loaded_association_should_not_use_query (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_update_counter_caches_on_destroy (unexpected)
--- FAIL: DelegatedTypeTest#test_accessor (unexpected)
--- FAIL: HasManyAssociationsTest#test_collection_association_with_private_kernel_method (unexpected)
--- FAIL: PersistenceTest#test_update_columns_should_not_use_setter_method (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_the_same_association_twice_works (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_collection_build_with_nonstandard_primary_key_on_belongs_to (unexpected)
--- FAIL: HasManyAssociationsTest#test_dynamic_find_should_respect_association_order_for_through (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_with_has_one_through_an_sti_with_after_initialize (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_create_bang_should_raise_exception_when_join_record_has_errors (unexpected)
--- FAIL: ReflectionTest#test_content_columns (unexpected)
--- FAIL: PersistenceTest#test_update_attribute_for_readonly_attribute (unexpected)
--- FAIL: HasManyAssociationsTest#test_finding_default_orders (unexpected)
--- FAIL: FinderTest#test_last_with_integer_and_reorder_should_use_sql_limit (unexpected)
--- FAIL: HasManyAssociationsTest#test_in_memory_replacements_sets_inverse_instance (unexpected)
--- FAIL: FinderTest#test_include_on_unloaded_relation_without_match (unexpected)
--- FAIL: FinderTest#test_find_by_one_attribute_with_several_options (unexpected)
--- FAIL: AttributeMethodsTest#test_respond_to? (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_create_fixtures_resets_sequences_when_not_cached (unexpected)
--- FAIL: ReflectionTest#test_has_many_reflection (unexpected)
--- FAIL: EagerAssociationTest#test_loading_with_multiple_associations (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_limit_and_multiple_associations (unexpected)
--- FAIL: HasManyAssociationsTest#test_do_not_call_callbacks_for_delete_all (unexpected)
--- FAIL: AttributeMethodsTest#test_number_attribute_predicate (unexpected)
--- FAIL: EagerAssociationTest#test_loading_association_with_same_table_joins (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit_and_conditions_on_the_eagers (unexpected)
--- FAIL: FinderTest#test_exists_with_polymorphic_relation (unexpected)
--- FAIL: PersistenceTest#test_increment_attribute (unexpected)
--- FAIL: PersistenceTest#test_create_with_custom_timestamps (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16356864/activerecord/1723009848863/1723018433471)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128153 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2.0-rc]
- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16387742?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16387742?buildTab=artifacts#/activerecord) on master @ [2558f3f3a1dab0907db3dafd510ec5bc2464073f](https://github.com/cockroachdb/cockroach/commits/2558f3f3a1dab0907db3dafd510ec5bc2464073f):


```
--- FAIL: RelationTest#test_new_with_array (unexpected)
--- FAIL: BelongsToAssociationsTest#test_reloading_the_belonging_object (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_grouped_with_invalid_field (unexpected)
--- FAIL: FinderTest#test_exists_returns_true_with_one_record_and_no_args (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_build_for_has_many_through_association (unexpected)
--- FAIL: AttributeMethodsTest#test_inherited_custom_accessors (unexpected)
--- FAIL: BelongsToAssociationsTest#test_where_on_polymorphic_association_with_nil (unexpected)
--- FAIL: RelationTest#test_build_with_array (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_cascaded_eager_association_loading_with_join_for_count (unexpected)
--- FAIL: EagerAssociationTest#test_including_associations_with_extensions_and_an_instance_dependent_scope_is_supported (unexpected)
--- FAIL: PersistenceTest#test_save_nil_string_attributes (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_with_create_exclamation_and_no_options (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_assignment_updates_foreign_id_field_for_new_and_saved_records (unexpected)
--- FAIL: PersistenceTest#test_destroy_many_with_invalid_id (unexpected)
--- FAIL: CalculationsTest#test_group_by_attribute_with_custom_type (unexpected)
--- FAIL: EagerAssociationTest#test_nested_loading_through_has_one_association (unexpected)
--- FAIL: EagerAssociationTest#test_eager-loading_readonly_association (unexpected)
--- FAIL: AttributeMethodsTest#test_YAML_dumping_a_record_with_time_zone-aware_attribute (unexpected)
--- FAIL: PostgresqlByteaTest#test_write_value (unexpected)
--- FAIL: RelationTest#test_select_takes_an_aliased_attribute (unexpected)
--- FAIL: RelationTest#test_reverse_arel_assoc_order_with_multiargument_function (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_conditions_string_with_unquoted_table_name (unexpected)
--- FAIL: FinderTest#test_implicit_order_set_to_primary_key (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_aggregate_attribute_having_same_name_as_field_and_key_value_being_attribute_value (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_polymorphic_with_custom_foreign_type (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_indexes_for_existent_table (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_with_reflection_using_class_name_and_fixtures (unexpected)
--- FAIL: AttributeMethodsTest#test_number_attribute_predicate (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_singleton_has_many_through (unexpected)
--- FAIL: CalculationsTest#test_pluck_joined_with_polymorphic_relation (unexpected)
--- FAIL: CalculationsTest#test_sum_takes_attribute_type_precedence_over_database_type (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_caching_of_columns (unexpected)
--- FAIL: RelationTest#test_loading_query_is_annotated_in_#pretty_print (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_multiple_fields_when_table_name_is_too_long (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_obeys_order_on_through_association (unexpected)
--- FAIL: RelationTest#test_automatically_added_having_references (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_grafts_stashed_associations_to_correct_parent (unexpected)
--- FAIL: CalculationsTest#test_should_count_with_manual_distinct_select_and_distinct (unexpected)
--- FAIL: CalculationsTest#test_select_avg_with_group_by_as_virtual_attribute_with_sql (unexpected)
--- FAIL: CalculationsTest#test_distinct_count_with_order_and_offset (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_delete_all_for_with_dependent_option_destroy (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_group_raise_errors (unexpected)
--- FAIL: PersistenceTest#test_save_for_record_with_only_primary_key_that_is_provided (unexpected)
--- FAIL: FinderTest#test_include_on_loaded_relation_with_match (unexpected)
--- FAIL: RelationTest#test_finding_with_arel_assoc_order (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_new_by_building (unexpected)
--- FAIL: EagerAssociationTest#test_limited_eager_with_numeric_in_association (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16387742/activerecord/1723182686372/1723190473348)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128153 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2.0-rc]
- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16405022?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16405022?buildTab=artifacts#/activerecord) on master @ [27468a22f82fb1e500bf7b35a62ca58d80abac39](https://github.com/cockroachdb/cockroach/commits/27468a22f82fb1e500bf7b35a62ca58d80abac39):


```
--- FAIL: TestDefaultAutosaveAssociationOnNewRecord#test_autosave_new_record_on_has_one_can_be_disabled_per_relationship (unexpected)
--- FAIL: AssociationsTest#test_loading_the_association_target_should_keep_child_records_marked_for_destruction (unexpected)
--- FAIL: AssociationsTest#test_querying_by_single_associated_record_works_using_query_constraints (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_parent_should_save_children_record_with_foreign_key_validation_set_in_before_save_callback (unexpected)
--- FAIL: SchemaIndexIncludeColumnsTest#test_schema_dumps_index_included_columns (unexpected)
--- FAIL: AssociationsTest#test_using_limitable_reflections_helper (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_assign_ids_with_belongs_to_cpk_model (unexpected)
--- FAIL: AssociationsTest#test_belongs_to_a_model_with_composite_primary_key_uses_composite_pk_in_sql (unexpected)
--- FAIL: AssociationsTest#test_assign_persisted_composite_foreign_key_belongs_to_association (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_replace_on_duplicated_object (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: AssociationsTest#test_append_composite_has_many_through_association (unexpected)
--- FAIL: AssociationsTest#test_belongs_to_a_model_with_composite_foreign_key_finds_associated_record (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: ModulesTest#test_associations_spanning_cross_modules (unexpected)
--- FAIL: AssociationsTest#test_belongs_to_a_cpk_model_by_id_attribute (unexpected)
--- FAIL: ModulesTest#test_module_table_name_prefix (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_build_many_via_block_before_save (unexpected)
--- FAIL: HasOneThroughDisableJoinsAssociationsTest#test_preload_on_disable_joins_through (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_adding_before_save (unexpected)
--- FAIL: ModulesTest#test_table_name (unexpected)
--- FAIL: AssociationsTest#test_preloads_model_with_query_constraints_by_explicitly_configured_fk_and_pk (unexpected)
--- FAIL: AssociationsTest#test_eager_loading_should_not_change_count_of_children (unexpected)
--- FAIL: AssociationsTest#test_append_composite_foreign_key_has_many_association (unexpected)
--- FAIL: AssociationsTest#test_bad_collection_keys (unexpected)
--- FAIL: HasOneThroughDisableJoinsAssociationsTest#test_has_one_through_with_belongs_to_on_disable_joins (unexpected)
--- FAIL: AssociationsTest#test_belongs_to_with_foreign_key_as_an_array_raises (unexpected)
--- FAIL: AssociationsTest#test_model_with_composite_query_constraints_has_many_association_sql (unexpected)
--- FAIL: AssociationsTest#test_assign_belongs_to_cpk_model_by_id_attribute (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_replace_on_new_object (unexpected)
--- FAIL: AssociationsTest#test_loading_the_association_target_should_load_most_recent_attributes_for_child_records_marked_for_destruction (unexpected)
--- FAIL: AssociationsTest#test_should_construct_new_finder_sql_after_create (unexpected)
--- FAIL: AssociationsTest#test_append_composite_foreign_key_has_many_association_with_autosave (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnNewRecord#test_autosave_new_record_on_has_many_can_be_disabled_per_relationship (unexpected)
--- FAIL: HasOneThroughDisableJoinsAssociationsTest#test_disable_joins_through_with_enum_type (unexpected)
--- FAIL: HasOneThroughDisableJoinsAssociationsTest#test_counting_on_disable_joins_through (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnNewRecord#test_autosave_new_record_on_belongs_to_can_be_disabled_per_relationship (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_build_before_save (unexpected)
--- FAIL: AssociationsTest#test_query_constraints_that_dont_include_the_primary_key_raise (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_assign_ids_with_cpk_for_two_models (unexpected)
--- FAIL: HasOneThroughDisableJoinsAssociationsTest#test_nil_on_disable_joins_through (unexpected)
--- FAIL: ModulesTest#test_eager_loading_in_modules (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_assign_ids (unexpected)
--- FAIL: ModulesTest#test_find_account_and_include_company (unexpected)
--- FAIL: TestAutosaveAssociationOnAHasManyAssociationDefinedInSubclassWithAcceptsNestedAttributes#test_should_update_children_when_association_redefined_in_subclass (unexpected)
--- FAIL: AssociationsTest#test_include_with_order_works (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_adding_unsavable_association (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16405022/activerecord/1723269010536/1723276531079)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128153 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2.0-rc]
- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16414561?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16414561?buildTab=artifacts#/activerecord) on master @ [451cdbe687b3303665bfd490af59c5df777c8909](https://github.com/cockroachdb/cockroach/commits/451cdbe687b3303665bfd490af59c5df777c8909):


```
(orm_helpers.go:229).summarizeFailed: 
Tests run on Cockroach v24.2.0-alpha.2-dev-451cdbe687b3303665bfd490af59c5df777c8909
Tests run against activerecord 7.1.3
8108 Total Tests Run
8079 tests passed
29 tests failed
66 tests skipped
0 tests ignored
0 tests passed unexpectedly
3 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_default_pk_and_sequence (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_create_fixtures_resets_sequences_when_not_cached (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16414561/activerecord/1723355493350/1723363612107)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128153 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2.0-rc]
- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16423103?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16423103?buildTab=artifacts#/activerecord) on master @ [2f0b0399a5c9656bdf8a417d0ab57e4eb1ae0f64](https://github.com/cockroachdb/cockroach/commits/2f0b0399a5c9656bdf8a417d0ab57e4eb1ae0f64):


```
--- FAIL: CalculationsTest#test_should_group_by_multiple_fields (unexpected)
--- FAIL: BasicsTest#test_previously_changed_dup (unexpected)
--- FAIL: CalculationsTest#test_average_with_from_option (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_assignment_foreign_key_type_string (unexpected)
--- FAIL: RelationTest#test_finding_with_reorder (unexpected)
--- FAIL: RelationTest#test_first_or_create_with_array (unexpected)
--- FAIL: BasicsTest#test_incomplete_schema_loading (unexpected)
--- FAIL: BelongsToAssociationsTest#test_attributes_are_set_without_error_when_initialized_from_belongs_to_association_with_array_in_where_clause (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_destruction_does_not_error_without_primary_key (unexpected)
--- FAIL: SignedIdTest#test_find_signed_record_within_expiration_time (unexpected)
--- FAIL: CalculationsTest#test_pluck_type_cast_with_joins_without_table_name_qualified_column (unexpected)
--- FAIL: PersistenceTest#test_create_model_with_custom_named_uuid_pk_populates_id (unexpected)
--- FAIL: CalculationsTest#test_pluck_on_aliased_attribute (unexpected)
--- FAIL: RelationTest#test_finding_with_complex_order_and_limit (unexpected)
--- FAIL: HasOneAssociationsTest#test_reload_association (unexpected)
--- FAIL: ReflectionTest#test_always_validate_association_if_explicit (unexpected)
--- FAIL: CalculationsTest#test_should_return_nil_as_average (unexpected)
--- FAIL: BelongsToAssociationsTest#test_creating_the_belonging_object_with_primary_key (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_update_columns_after_push_without_duplicate_join_table_rows (unexpected)
--- FAIL: RelationTest#test_size_with_eager_loading_and_manual_distinct_select_and_custom_order (unexpected)
--- FAIL: RelationTest#test_select_with_subquery_in_from_uses_original_table_name (unexpected)
--- FAIL: BelongsToAssociationsTest#test_create! (unexpected)
--- FAIL: BasicsTest#test_equality_of_destroyed_records (unexpected)
--- FAIL: CalculationsTest#test_distinct_count_with_order_and_limit (unexpected)
--- FAIL: RelationTest#test_reverse_order_with_reorder_nil_removes_the_order (unexpected)
--- FAIL: PersistenceTest#test_find_via_reload (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_replace_with_less (unexpected)
--- FAIL: RelationTest#test_build_with_array (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_new_with_values_in_collection (unexpected)
--- FAIL: BasicsTest#test_ignored_columns_don't_prevent_explicit_declaration_of_attribute_methods (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_touch_option_on_touch_and_reassigned_parent (unexpected)
--- FAIL: CalculationsTest#test_no_limit_no_offset (unexpected)
--- FAIL: CalculationsTest#test_ids_on_loaded_relation (unexpected)
--- FAIL: HasOneAssociationsTest#test_proxy_assignment (unexpected)
--- FAIL: BasicsTest#test_auto_id (unexpected)
--- FAIL: BasicsTest#test_previously_new_record_on_destroyed_record (unexpected)
--- FAIL: RelationTest#test_select_takes_an_aliased_attribute (unexpected)
--- FAIL: PersistenceTest#test_becomes_after_reload_schema_from_cache (unexpected)
--- FAIL: CalculationsTest#test_distinct_joins_count_with_order_and_limit (unexpected)
--- FAIL: BasicsTest#test_marshalling_new_record_round_trip_with_associations (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_adding_multiple (unexpected)
--- FAIL: PersistenceTest#test_delete_many (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_index_columns_in_right_order (unexpected)
--- FAIL: RelationTest#test_find_by!_requires_at_least_one_argument (unexpected)
--- FAIL: RelationTest#test_to_sql_on_eager_join (unexpected)
--- FAIL: RelationTest#test_any (unexpected)
--- FAIL: BasicsTest#test_readonly_attributes_on_a_new_record (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16423103/activerecord/1723441821276/1723448964844)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128153 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2.0-rc]
- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16441740?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16441740?buildTab=artifacts#/activerecord) on master @ [9765a2cc0d7b4d6dacff07ea5668d1b59a36cfc2](https://github.com/cockroachdb/cockroach/commits/9765a2cc0d7b4d6dacff07ea5668d1b59a36cfc2):


```
(orm_helpers.go:229).summarizeFailed: 
Tests run on Cockroach v24.3.0-alpha.00000000-dev-9765a2cc0d7b4d6dacff07ea5668d1b59a36cfc2
Tests run against activerecord 7.1.3
8108 Total Tests Run
8074 tests passed
34 tests failed
66 tests skipped
0 tests ignored
0 tests passed unexpectedly
8 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: SchemaDumperTest#test_schema_dumps_index_sort_order (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_partial_indices (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_index_columns_in_right_order (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_create_fixtures_resets_sequences_when_not_cached (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_nulls_not_distinct (unexpected)
--- FAIL: SchemaDumperTest#test_schema_dumps_index_length (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_default_pk_and_sequence (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/cpu_arch=arm64/run_1
```

Parameters:
 - <code>ROACHTEST_arch=arm64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16441740/activerecord/1723528379457/1723539241119)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128153 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2.0-rc]
- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16458698?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16458698?buildTab=artifacts#/activerecord) on master @ [abe3d27bee5120afa287df2f8cc81c82ab6864da](https://github.com/cockroachdb/cockroach/commits/abe3d27bee5120afa287df2f8cc81c82ab6864da):


```
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_callbacks_firing_order_on_save (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_collection_size_from_build (unexpected)
--- FAIL: PersistenceTest#test_populates_non_primary_key_autoincremented_column (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_with_instance_dependent_scope (unexpected)
--- FAIL: PersistenceTest#test_class_level_destroy (unexpected)
--- FAIL: CalculationsTest#test_count_with_order (unexpected)
--- FAIL: PersistenceTest#test_destroy_record_with_associations (unexpected)
--- FAIL: RelationTest#test_finding_with_desc_order_with_string (unexpected)
--- FAIL: AssociationsTest#test_nullify_composite_foreign_key_belongs_to_association (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_loading_cpk_association_with_unpersisted_owner (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_create_on_new_record (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_can_update_through_association (unexpected)
--- FAIL: HasManyAssociationsTest#test_has_many_without_counter_cache_option (unexpected)
--- FAIL: HasManyAssociationsTest#test_associations_order_should_be_priority_over_throughs_order (unexpected)
--- FAIL: RelationTest#test_count_on_association_relation (unexpected)
--- FAIL: FinderTest#test_exists_with_joins (unexpected)
--- FAIL: RelationTest#test_size_with_eager_loading_and_manual_distinct_select_and_custom_order (unexpected)
--- FAIL: ValidationsTest#test_numericality_validation_with_mutation (unexpected)
--- FAIL: RelationTest#test_first_get_more_than_available (unexpected)
--- FAIL: HasManyAssociationsTest#test_modifying_a_through_a_has_many_should_raise (unexpected)
--- FAIL: RelationTest#test_order_triggers_eager_loading_when_ordering_using_hash_syntax (unexpected)
--- FAIL: HasManyAssociationsTest#test_build_with_polymorphic_has_many_does_not_allow_to_override_type_and_id (unexpected)
--- FAIL: BasicsTest#test_default_values (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_replace_association (unexpected)
--- FAIL: FinderTest#test_#last_for_a_model_with_composite_query_constraints (unexpected)
--- FAIL: BelongsToAssociationsTest#test_building_the_belonging_object_with_implicit_sti_base_class (unexpected)
--- FAIL: BelongsToAssociationsTest#test_with_polymorphic_and_condition (unexpected)
--- FAIL: HasManyAssociationsTest#test_defining_has_many_association_with_nullify_dependency_lazily_evaluates_target_class (unexpected)
--- FAIL: RelationTest#test_finding_with_arel_order (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_field_with_conditions_and_having (unexpected)
--- FAIL: RelationTest#test_unscope_with_table_name_qualified_hash (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: HasManyAssociationsTest#test_dont_call_save_callbacks_twice_on_has_many (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_preload_with_nested_association (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_multiple_fields_when_table_name_is_too_long (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroying (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_explicit_join (unexpected)
--- FAIL: RelationTest#test_optimizer_hints_with_blank_value (unexpected)
--- FAIL: FinderTest#test_find_with_a_multiple_sets_of_composite_primary_key (unexpected)
--- FAIL: InheritanceComputeTypeTest#test_sti_type_from_attributes_disabled_in_non_sti_class (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_replace_with_less (unexpected)
--- FAIL: HasManyAssociationsTest#test_taking_with_a_number (unexpected)
--- FAIL: ReflectionTest#test_class_for_class_name (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_get_collection_singular_ids_on_has_many_through_with_conditions_and_include (unexpected)
--- FAIL: HasManyAssociationsTest#test_does_not_duplicate_associations_when_used_with_natural_primary_keys (unexpected)
--- FAIL: PersistenceTest#test_create_model_with_custom_named_uuid_pk_populates_id (unexpected)
--- FAIL: HasManyAssociationsTest#test_joining_through_a_polymorphic_association_with_a_where_clause (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/cpu_arch=arm64/run_1
```

Parameters:
 - <code>ROACHTEST_arch=arm64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16458698/activerecord/1723614761970/1723625376532)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128153 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2.0-rc]
- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16475250?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16475250?buildTab=artifacts#/activerecord) on master @ [575cdd4696dfcac8f311d1ea546683271102f73e](https://github.com/cockroachdb/cockroach/commits/575cdd4696dfcac8f311d1ea546683271102f73e):


```
--- FAIL: HasOneAssociationsTest#test_building_the_associated_object_with_an_unrelated_type (unexpected)
--- FAIL: FinderTest#test_take (unexpected)
--- FAIL: InheritanceTest#test_abstract_class (unexpected)
--- FAIL: FinderTest#test_find_by_two_attributes_with_one_being_an_aggregate (unexpected)
--- FAIL: RelationTest#test_select_with_subquery_in_from_does_not_use_original_table_name (unexpected)
--- FAIL: TransactionTest#test_raising_exception_in_callback_rollbacks_in_save (unexpected)
--- FAIL: HasManyAssociationsTest#test_collection_empty_with_dirty_target (unexpected)
--- FAIL: MultiParameterAttributeTest#test_multiparameter_assigned_attributes_did_not_come_from_user (unexpected)
--- FAIL: DDLTest#test_create_simple_geography_using_shortcut (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all_association_with_primary_key_deletes_correct_records (unexpected)
--- FAIL: FinderTest#test_first_have_determined_order_by_default (unexpected)
--- FAIL: RelationMergingTest#test_relation_merging_with_joins (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_redefine_habtm (unexpected)
--- FAIL: HasManyAssociationsTest#test_delete_all_with_option_nullify (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_transaction (unexpected)
--- FAIL: DDLTest#test_create_point_geometry_using_shortcut (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_none_should_return_true_if_none (unexpected)
--- FAIL: ActiveRecord::Migration::ForeignKeyChangeColumnWithPrefixTest#test_rename_reference_column_of_child_table (unexpected)
--- FAIL: AttributeMethodsTest#test_time_attributes_are_retrieved_in_the_current_time_zone (unexpected)
--- FAIL: ReflectionTest#test_validate_association_if_autosave (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_includes_raise_errors (unexpected)
--- FAIL: TransactionTest#test_accessing_raw_connection_materializes_transaction (unexpected)
--- FAIL: MultiParameterAttributeTest#test_multiparameter_attributes_on_date_with_empty_month (unexpected)
--- FAIL: ActiveRecord::Encryption::ExtendedDeterministicQueriesTest#test_If_support_unencrypted_data_is_opted_in_at_the_attribute_level,_can_find_encrypted_data (unexpected)
--- FAIL: ActiveRecord::PredicateBuilderTest#test_build_from_hash_with_schema (unexpected)
--- FAIL: EnumTest#test_capital_characters_for_enum_names (unexpected)
--- FAIL: StrictLoadingTest#test_raises_on_lazy_loading_a_belongs_to_relation_if_strict_loading_by_default (unexpected)
--- FAIL: InversePolymorphicBelongsToTests#test_inversed_instance_should_load_after_autosave_if_it_is_not_already_loaded (unexpected)
--- FAIL: TransactionCallbacksTest#test_no_after_commit_on_destroy_after_transaction_commits_for_destroyed_new_record (unexpected)
--- FAIL: HasManyAssociationsTest#test_replace_with_new (unexpected)
--- FAIL: CalculationsTest#test_pluck_functions_with_alias (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_has_and_belongs_to_many_while_partial_inserts_false (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_open_ended_range (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_procs_for_default_values (unexpected)
--- FAIL: TestAutosaveAssociationOnAHasOneAssociation#test_changed_for_autosave_should_handle_cycles (unexpected)
--- FAIL: RelationTest#test_automatically_added_having_references (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_join (unexpected)
--- FAIL: ConcurrentTransactionTest#test_transaction_per_thread (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_of_stis_with_multiple_references (unexpected)
--- FAIL: HasOneAssociationsTest#test_attributes_are_being_set_when_initialized_from_has_one_association_with_where_clause (unexpected)
--- FAIL: TestNestedAttributesOnAHasAndBelongsToManyAssociation#test_should_take_a_hash_with_composite_id_keys_and_assign_the_attributes_to_the_associated_models (unexpected)
--- FAIL: HasManyAssociationsTest#test_add_record_to_collection_should_change_its_updated_at (unexpected)
--- FAIL: ActiveRecord::CustomPropertiesTest#test_children_inherit_custom_properties (unexpected)
--- FAIL: InheritanceTest#test_alt_update_all_within_inheritance (unexpected)
--- FAIL: TransactionTest#test_rollback_for_freshly_persisted_records (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_replace_on_new_object (unexpected)
--- FAIL: ActiveRecord::Encryption::EncryptableRecordApiTest#test_encrypted_attribute?_returns_false_for_regular_attributes (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16475250/activerecord/1723701171039/1723706368286)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128153 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2.0-rc]
- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16491798?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16491798?buildTab=artifacts#/activerecord) on master @ [ec66ae778100724b11f1b8978c915be1bcea390c](https://github.com/cockroachdb/cockroach/commits/ec66ae778100724b11f1b8978c915be1bcea390c):


```
--- FAIL: EagerAssociationTest#test_preloading_has_many_through_with_custom_scope (unexpected)
--- FAIL: EagerAssociationTest#test_order_on_join_table_with_include_and_limit (unexpected)
--- FAIL: EagerAssociationTest#test_associations_loaded_for_all_records (unexpected)
--- FAIL: EagerAssociationTest#test_count_eager_with_has_many_and_limit_and_high_offset (unexpected)
--- FAIL: EagerAssociationTest#test_eager_has_many_with_association_inheritance (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_create_fixtures_resets_sequences_when_not_cached (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_multiple_associations_with_same_table_has_one (unexpected)
--- FAIL: EagerAssociationTest#test_eager_load_has_many_quotes_table_and_column_names (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_does_not_cache_has_many_association_subset_when_preloaded_with_a_through_association (unexpected)
--- FAIL: BasicsTest#test_hashing (unexpected)
--- FAIL: BasicsTest#test_columns_should_obey_set_primary_key (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit_and_conditions_array_on_the_eagers (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_an_sti_join_model (unexpected)
--- FAIL: BasicsTest#test_limit_without_comma (unexpected)
--- FAIL: EagerAssociationTest#test_circular_preload_does_not_modify_unscoped (unexpected)
--- FAIL: BasicsTest#test_cannot_call_connected_to_with_role_and_shard_on_non-abstract_classes (unexpected)
--- FAIL: BasicsTest#test_quoting_arrays (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_associations_with_string_joins_and_order_references (unexpected)
--- FAIL: BasicsTest#test_attribute_names (unexpected)
--- FAIL: BasicsTest#test_failed_comparison_of_unlike_class_records (unexpected)
--- FAIL: BasicsTest#test_previously_new_record_returns_boolean (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_has_many_through_with_implicit_source (unexpected)
--- FAIL: BasicsTest#test_column_types_on_queries_on_postgresql (unexpected)
--- FAIL: BasicsTest#test_create_without_prepared_statement (unexpected)
--- FAIL: BasicsTest#test_default_in_utc (unexpected)
--- FAIL: BasicsTest#test_previously_new_record_on_destroyed_record (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_one_through_join_model_with_conditions_on_the_through (unexpected)
--- FAIL: BasicsTest#test_using_table_name_qualified_column_names_unless_having_SELECT_list_explicitly (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_has_many_with_cpk (unexpected)
--- FAIL: BasicsTest#test_preserving_time_objects_with_local_time_conversion_to_default_timezone_utc (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_a_polymorphic_association_with_references_to_the_associated_table (unexpected)
--- FAIL: BasicsTest#test_custom_mutator (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_through_join_model_ignores_default_includes (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_multi_table_conditional_properly_counts_the_records_when_using_size (unexpected)
--- FAIL: EagerAssociationTest#test_eager_load_belongs_to_quotes_table_and_column_names (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_empty_belongs_to (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_floating_point_numbers (unexpected)
--- FAIL: BasicsTest#test_attributes_on_dummy_time (unexpected)
--- FAIL: EagerAssociationTest#test_join_eager_with_empty_order_should_generate_valid_sql (unexpected)
--- FAIL: EagerAssociationTest#test_eager_load_has_one_quotes_table_and_column_names (unexpected)
--- FAIL: BasicsTest#test_dont_clear_inheritance_column_when_setting_explicitly (unexpected)
--- FAIL: BasicsTest#test_switching_between_table_name (unexpected)
--- FAIL: EagerAssociationTest#test_base_messages (unexpected)
--- FAIL: BasicsTest#test_cannot_call_connected_to_on_the_abstract_class_that_did_not_establish_the_connection (unexpected)
--- FAIL: BasicsTest#test_can_call_connected_to_with_role_and_shard_on_abstract_classes (unexpected)
--- FAIL: BasicsTest#test_readonly_attributes_on_a_new_record (unexpected)
--- FAIL: EagerAssociationTest#test_eager_with_has_many_and_limit_and_conditions_array (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16491798/activerecord/1723787611133/1723796490892)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128153 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2.0-rc]
- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16507379?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16507379?buildTab=artifacts#/activerecord) on master @ [1993fc04b5116f20f4814d637c7ce87b003687e4](https://github.com/cockroachdb/cockroach/commits/1993fc04b5116f20f4814d637c7ce87b003687e4):


```
--- FAIL: InheritanceComputeTypeTest#test_instantiation_doesnt_try_to_require_corresponding_file (unexpected)
--- FAIL: NumericalityValidationTest#test_virtual_attribute_with_precision_round_down (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_with_custom_primary_key (unexpected)
--- FAIL: DirtyTest#test_changed?_in_after_callbacks_returns_false (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to (unexpected)
--- FAIL: RelationTest#test_unscope_with_arel_sql (unexpected)
--- FAIL: HasManyAssociationsTest#test_association_with_instance_dependent_scope (unexpected)
--- FAIL: ValidationsTest#test_exception_on_create_bang_many (unexpected)
--- FAIL: HasManyAssociationsTest#test_transactions_when_replacing_on_new_record (unexpected)
--- FAIL: DirtyTest#test_reload_should_clear_changed_attributes (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_count (unexpected)
--- FAIL: PersistenceTest#test_update_column_for_readonly_attribute (unexpected)
--- FAIL: AssociationsTest#test_bad_collection_keys (unexpected)
--- FAIL: InheritanceTest#test_compute_type_on_undefined_method (unexpected)
--- FAIL: BelongsToAssociationsTest#test_natural_assignment_with_primary_key (unexpected)
--- FAIL: RelationTest#test_no_arguments_to_group_raise_errors (unexpected)
--- FAIL: HasManyAssociationsTest#test_get_ids_for_unloaded_associations_does_not_load_them (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_scope_can_unscope (unexpected)
--- FAIL: HasManyAssociationsTest#test_counter_cache_on_unloaded_association (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_save_bang_should_raise_exception_when_join_record_has_errors (unexpected)
--- FAIL: PersistenceTest#test_delete_record_with_associations (unexpected)
--- FAIL: BasicsTest#test_scoped_can_take_a_values_hash (unexpected)
--- FAIL: RelationTest#test_left_outer_joins_with_blank_value (unexpected)
--- FAIL: InheritanceTest#test_alt_update_all_within_inheritance (unexpected)
--- FAIL: FinderTest#test_hash_condition_utc_time_interpolation_with_default_timezone_local (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_save_returns_falsy_when_join_record_has_errors (unexpected)
--- FAIL: HasManyAssociationsTest#test_raises_RecordNotDestroyed_when_replaced_child_can't_be_destroyed (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_yaml_loads_5_1_dump_without_indexes_still_queries_for_indexes (unexpected)
--- FAIL: PersistenceTest#test_save_nil_string_attributes (unexpected)
--- FAIL: BasicsTest#test_generated_association_methods_module_name (unexpected)
--- FAIL: RelationTest#test_size (unexpected)
--- FAIL: HasManyAssociationsTest#test_destroying_a_collection (unexpected)
--- FAIL: AssociationCallbacksTest#test_has_many_callbacks_for_destroy_on_parent (unexpected)
--- FAIL: BasicsTest#test_#connected_to_many_cannot_be_called_with_classes_that_include_ActiveRecord::Base (unexpected)
--- FAIL: FinderTest#test_take (unexpected)
--- FAIL: PersistenceTest#test_class_level_update_without_ids! (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_new_aliased_to_build (unexpected)
--- FAIL: HasManyAssociationsTest#test_creating_using_primary_key (unexpected)
--- FAIL: InheritanceTest#test_where_create_bang_with_invalid_type (unexpected)
--- FAIL: BelongsToAssociationsTest#test_cant_save_readonly_association (unexpected)
--- FAIL: PersistenceTest#test_update_column (unexpected)
--- FAIL: PersistenceTest#test_update_columns_with_default_scope (unexpected)
--- FAIL: BasicsTest#test_ignored_columns_are_not_present_in_columns_hash (unexpected)
--- FAIL: InheritanceTest#test_where_create_with_unrelated_type (unexpected)
--- FAIL: RelationTest#test_using_a_custom_table_affects_the_wheres (unexpected)
--- FAIL: BasicsTest#test_equality_of_collection_proxy_and_association_relation (unexpected)
--- FAIL: InheritanceTest#test_inherits_custom_primary_key (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16507379/activerecord/1723874036496/1723882846150)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128153 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2.0-rc]
- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16516775?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16516775?buildTab=artifacts#/activerecord) on master @ [1993fc04b5116f20f4814d637c7ce87b003687e4](https://github.com/cockroachdb/cockroach/commits/1993fc04b5116f20f4814d637c7ce87b003687e4):


```
--- FAIL: HasAndBelongsToManyAssociationsTest#test_has_and_belongs_to_many_with_belongs_to (unexpected)
--- FAIL: ModulesTest#test_associations_spanning_cross_modules (unexpected)
--- FAIL: CalculationsTest#test_sum_takes_attribute_type_precedence_over_database_type (unexpected)
--- FAIL: CalculationsTest#test_pick_loaded_relation_aliased_attribute (unexpected)
--- FAIL: CalculationsTest#test_ids_on_loaded_relation (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_grouped_by_function (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_adding_single (unexpected)
--- FAIL: CalculationsTest#test_should_average_arel_attribute (unexpected)
--- FAIL: ReflectionTest#test_scope_chain_does_not_interfere_with_hmt_with_polymorphic_and_subclass_source_2 (unexpected)
--- FAIL: CalculationsTest#test_should_reference_correct_aliases_while_joining_tables_of_has_many_through_association (unexpected)
--- FAIL: ReflectionTest#test_reflect_on_association_accepts_symbols (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_updating_attributes_on_non_rich_associations (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_join (unexpected)
--- FAIL: CalculationsTest#test_pluck_in_relation (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_new_with_values_in_collection (unexpected)
--- FAIL: CalculationsTest#test_offset_is_kept (unexpected)
--- FAIL: CalculationsTest#test_sum_uses_enumerable_version_when_block_is_given (unexpected)
--- FAIL: ReflectionTest#test_column_string_type_and_limit (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_field_through_association_and_having (unexpected)
--- FAIL: CalculationsTest#test_no_queries_for_empty_relation_on_minimum (unexpected)
--- FAIL: ModulesTest#test_module_table_name_prefix (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_field_having_condition_from_select (unexpected)
--- FAIL: ReflectionTest#test_integer_columns (unexpected)
--- FAIL: CalculationsTest#test_pluck_multiple_columns (unexpected)
--- FAIL: ModulesTest#test_module_table_name_suffix (unexpected)
--- FAIL: ReflectionTest#test_default_association_validation (unexpected)
--- FAIL: CalculationsTest#test_count_selected_arel_attribute (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_alternate_database (unexpected)
--- FAIL: CalculationsTest#test_pluck_type_cast_with_conflict_column_names (unexpected)
--- FAIL: CalculationsTest#test_should_resolve_aliased_attributes (unexpected)
--- FAIL: AssociationsTest#test_assign_composite_foreign_key_belongs_to_association (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_attributes_are_being_set_when_initialized_from_habtm_association_with_multiple_where_clauses (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::SchemaCacheTest#test_marshal_dump_and_load (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_consider_type (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_distinct_order_preserved (unexpected)
--- FAIL: AssociationsTest#test_polymorphic_belongs_to_uses_parent_query_constraints (unexpected)
--- FAIL: ReflectionTest#test_human_name (unexpected)
--- FAIL: CalculationsTest#test_should_return_type_casted_values_with_group_and_expression (unexpected)
--- FAIL: CalculationsTest#test_ids_for_a_composite_primary_key_on_loaded_relation (unexpected)
--- FAIL: CalculationsTest#test_select_avg_with_joins_and_group_by_as_virtual_attribute_with_sql (unexpected)
--- FAIL: CalculationsTest#test_should_count_manual_select_with_group_with_count_all (unexpected)
--- FAIL: AssociationsTest#test_has_many_with_foreign_key_as_an_array_raises (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_grouped_with_longer_field (unexpected)
--- FAIL: CalculationsTest#test_pluck_loaded_relation_multiple_columns (unexpected)
--- FAIL: CockroachDB::FixturesTest#test_clean_fixtures (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_join_with_group (unexpected)
--- FAIL: ReflectionTest#test_association_primary_key_uses_explicit_primary_key_option_as_first_priority (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16516775/activerecord/1723960243763/1723968554445)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128153 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2.0-rc]
- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16526778?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16526778?buildTab=artifacts#/activerecord) on master @ [08bb256bc15218deaa8884bc89bddb2766514598](https://github.com/cockroachdb/cockroach/commits/08bb256bc15218deaa8884bc89bddb2766514598):


```
--- FAIL: BelongsToAssociationsTest#test_association_assignment_sticks (unexpected)
--- FAIL: RelationTest#test_size_with_limit (unexpected)
--- FAIL: RelationTest#test_reorder_with_first (unexpected)
--- FAIL: PersistenceTest#test_destroy_record_with_associations (unexpected)
--- FAIL: EagerAssociationTest#test_order_on_join_table_with_include_and_limit (unexpected)
--- FAIL: CalculationsTest#test_should_get_maximum_of_arel_attribute (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_include? (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_method? (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_counter (unexpected)
--- FAIL: FinderTest#test_select_rows (unexpected)
--- FAIL: HasOneAssociationsTest#test_finding_using_primary_key (unexpected)
--- FAIL: IntegrationTest#test_cache_key_for_newer_updated_at (unexpected)
--- FAIL: PersistenceTest#test_update_raises_record_not_found_exception (unexpected)
--- FAIL: ReflectionTest#test_chain (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_the_same_association_twice_works (unexpected)
--- FAIL: RelationTest#test_to_yaml (unexpected)
--- FAIL: CalculationsTest#test_ids_with_includes_offset (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_create_with_conditions_hash_on_through_association (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attribute (unexpected)
--- FAIL: PersistenceTest#test_delete (unexpected)
--- FAIL: FinderTest#test_find_by_with_hash_conditions_returns_the_first_matching_record (unexpected)
--- FAIL: FinderTest#test_hash_condition_find_with_array (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_build_before_save (unexpected)
--- FAIL: CalculationsTest#test_count_selected_arel_attribute (unexpected)
--- FAIL: RelationTest#test_order_with_blank_value (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_association_class (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_marshal_dump (unexpected)
--- FAIL: RelationTest#test_first_or_initialize_with_no_parameters (unexpected)
--- FAIL: I18nGenerateMessageValidationTest#test_generate_message_invalid_with_default_message (unexpected)
--- FAIL: HasManyAssociationsTest#test_finding_with_condition_hash (unexpected)
--- FAIL: FinderTest#test_find_with_string (unexpected)
--- FAIL: RelationTest#test_find_all_using_where_twice_should_or_the_relation (unexpected)
--- FAIL: RelationTest#test_loading_with_one_association_with_non_preload (unexpected)
--- FAIL: PersistenceTest#test_update_columns_changing_id (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_callbacks_firing_order_on_update (unexpected)
--- FAIL: HasManyAssociationsTest#test_deleting_models_with_composite_keys (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_association_with_non_numeric_foreign_key (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_for_inspect_with_a_non-primary_key_id_attribute (unexpected)
--- FAIL: AssociationsTest#test_has_many_with_foreign_key_as_an_array_raises (unexpected)
--- FAIL: HasManyAssociationsTest#test_taking_not_found (unexpected)
--- FAIL: EagerAssociationTest#test_preloading_polymorphic_with_custom_foreign_type (unexpected)
--- FAIL: PersistenceTest#test_destroyed_returns_boolean (unexpected)
--- FAIL: FinderTest#test_symbols_table_ref (unexpected)
--- FAIL: CalculationsTest#test_ids_with_includes (unexpected)
--- FAIL: PersistenceTest#test_update_attribute_for_updated_at_on (unexpected)
--- FAIL: HasManyAssociationsTest#test_reloading_unloaded_associations_with_query_cache (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_reserved_words (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16526778/activerecord/1724049919155/1724056090576)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128153 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2.0-rc]
- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16536452?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16536452?buildTab=artifacts#/activerecord) on master @ [08bb256bc15218deaa8884bc89bddb2766514598](https://github.com/cockroachdb/cockroach/commits/08bb256bc15218deaa8884bc89bddb2766514598):


```
--- FAIL: EagerAssociationTest#test_preloading_a_regular_association_through_a_polymorphic_association_doesn't_require_the_association_to_exist_on_all_types (unexpected)
--- FAIL: EagerAssociationTest#test_include_has_one_using_primary_key (unexpected)
--- FAIL: BasicsTest#test_preserving_time_objects (unexpected)
--- FAIL: PersistenceTest#test_reload_uses_query_constraints_config (unexpected)
--- FAIL: EagerAssociationTest#test_eager-loading_with_a_polymorphic_association_won't_work_consistently (unexpected)
--- FAIL: CalculationsTest#test_pluck_replaces_select_clause (unexpected)
--- FAIL: PersistenceTest#test_save_for_record_with_only_primary_key_that_is_provided (unexpected)
--- FAIL: BasicsTest#test_reload (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_join_with_group (unexpected)
--- FAIL: FinderTest#test_exists_with_polymorphic_relation (unexpected)
--- FAIL: ReflectionTest#test_has_many_reflection (unexpected)
--- FAIL: SignedIdTest#test_fail_to_find_signed_record_with_purpose (unexpected)
--- FAIL: ActiveRecord::Migration::ForeignKeyChangeColumnWithPrefixTest#test_remove_foreign_key_by_column_in_change_table (unexpected)
--- FAIL: CalculationsTest#test_pluck (unexpected)
--- FAIL: HasManyAssociationsTest#test_calling_one_should_return_true_if_one (unexpected)
--- FAIL: BasicsTest#test_clear_cache! (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_through_scope_with_includes (unexpected)
--- FAIL: BasicsTest#test_equality_of_new_records (unexpected)
--- FAIL: BasicsTest#test_marshal_between_processes (unexpected)
--- FAIL: HasOneAssociationsTest#test_creation_failure_due_to_new_record_should_raise_error (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_a_time_zone-aware_attribute_to_a_blank_string_returns_nil (unexpected)
--- FAIL: EagerAssociationTest#test_eager_association_loading_with_belongs_to_and_limit_and_offset_and_multiple_associations (unexpected)
--- FAIL: PersistenceTest#test_class_level_delete_is_affected_by_scoping (unexpected)
--- FAIL: RelationTest#test_scoped (unexpected)
--- FAIL: BasicsTest#test_ignored_columns_are_stored_as_an_array_of_string (unexpected)
--- FAIL: BelongsToAssociationsTest#test_reloading_association_with_key_change (unexpected)
--- FAIL: EagerAssociationTest#test_deep_including_through_habtm (unexpected)
--- FAIL: CalculationsTest#test_should_return_decimal_average_of_integer_field (unexpected)
--- FAIL: PersistenceTest#test_update_does_not_run_sql_if_record_has_not_changed (unexpected)
--- FAIL: BasicsTest#test_initialize_with_invalid_attribute (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_for_inspect_with_a_non-primary_key_id_attribute (unexpected)
--- FAIL: RelationTest#test_empty (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_counter_when_update_columns (unexpected)
--- FAIL: FinderTest#test_joins_with_string_array (unexpected)
--- FAIL: RelationTest#test_joins_with_blank_value (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_joining_has_many_through_belongs_to (unexpected)
--- FAIL: HasManyAssociationsTest#test_collection_not_empty_after_building (unexpected)
--- FAIL: SerializedAttributeTestWithYamlSafeLoad#test_serialized_attribute_on_alias_attribute (unexpected)
--- FAIL: CallbacksTest#test_create (unexpected)
--- FAIL: RelationTest#test_select_with_blank_value (unexpected)
--- FAIL: BasicsTest#test_previously_changed (unexpected)
--- FAIL: ActiveRecord::Migration::ColumnsTest#test_change_column_default_to_null (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_replace_order_is_preserved (unexpected)
--- FAIL: FinderTest#test_exists_uses_existing_scope (unexpected)
--- FAIL: CockroachDB::AdapterTestWithoutTransaction#test_reset_empty_table_with_custom_pk_sequence (unexpected)
--- FAIL: FinderTest#test_find_on_hash_conditions_with_qualified_attribute_dot_notation_string (unexpected)
--- FAIL: AssociationsTest#test_should_construct_new_finder_sql_after_create (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16536452/activerecord/1724089567579/1724096988236)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128153 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2.0-rc]
- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16543981?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16543981?buildTab=artifacts#/activerecord) on master @ [ba7c94a6ffe5f0a990000188ecfd3a23f7c8b4a6](https://github.com/cockroachdb/cockroach/commits/ba7c94a6ffe5f0a990000188ecfd3a23f7c8b4a6):


```
--- FAIL: BelongsToAssociationsTest#test_belongs_to (unexpected)
--- FAIL: BelongsToAssociationsTest#test_create_bang_with_conditions (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_counter_with_reassigning (unexpected)
--- FAIL: RelationTest#test_reverse_order_with_function_other_predicates (unexpected)
--- FAIL: RelationTest#test_finding_with_subquery_with_eager_loading_in_where (unexpected)
--- FAIL: RelationTest#test_none? (unexpected)
--- FAIL: InheritanceTest#test_compute_type_on_undefined_method (unexpected)
--- FAIL: InheritanceTest#test_inheritance_new_with_default_class (unexpected)
--- FAIL: RelationTest#test_order_triggers_eager_loading_when_ordering_using_symbols (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_should_save_parent_but_not_invalid_child (unexpected)
--- FAIL: TransactionTest#test_thow_from_transaction_7_1_behavior (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_touch_option_on_destroy_with_destroyed_parent (unexpected)
--- FAIL: RelationTest#test_left_outer_joins_with_blank_value (unexpected)
--- FAIL: IntegrationTest#test_to_param_class_method_squishes (unexpected)
--- FAIL: CockroachDB::LeftOuterJoinAssociationTest#test_does_not_override_select (unexpected)
--- FAIL: TransactionTest#test_many_savepoints (unexpected)
--- FAIL: AssociationsTest#test_has_many_with_foreign_key_as_an_array_raises (unexpected)
--- FAIL: RelationTest#test_order_doesnt_trigger_eager_loading_when_ordering_using_the_owner_table_and_hash_syntax (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnABelongsToAssociation#test_store_association_in_two_relations_with_one_save_in_existing_object (unexpected)
--- FAIL: AttributeMethodsTest#test_#id_value_alias_returns_the_value_in_the_id_column,_when_id_column_exists (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_assignment_before_parent_saved (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_with_touch_option_on_touch (unexpected)
--- FAIL: ReflectionTest#test_never_validate_association_if_explicit (unexpected)
--- FAIL: BelongsToAssociationsTest#test_polymorphic_association_class (unexpected)
--- FAIL: TransactionTest#test_callback_rollback_in_create_with_rollback_exception (unexpected)
--- FAIL: NonFullStiClassNamesTest#test_class_names_with_eager_load (unexpected)
--- FAIL: RelationTest#test_to_yaml (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_preloading_across_has_one_through_constrains_loaded_records (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_with_cascaded_interdependent_one_level_and_two_levels (unexpected)
--- FAIL: RelationTest#test_unscoped_block_style (unexpected)
--- FAIL: InheritanceTest#test_inherits_custom_primary_key (unexpected)
--- FAIL: AssociationsTest#test_preloads_model_with_query_constraints_by_explicitly_configured_fk_and_pk (unexpected)
--- FAIL: AttributeMethodsTest#test_read_attributes_before_type_cast_on_a_datetime (unexpected)
--- FAIL: RelationTest#test_respond_to_dynamic_finders (unexpected)
--- FAIL: MultiParameterAttributeTest#test_multiparameter_attributes_on_date_with_empty_month (unexpected)
--- FAIL: AttributeMethodsTest#test_setting_invalid_string_to_a_zone-aware_time_attribute (unexpected)
--- FAIL: ReflectionTest#test_human_name (unexpected)
--- FAIL: IntegrationTest#test_to_param_returns_nil_if_not_persisted (unexpected)
--- FAIL: AttributeMethodsTest#test_converted_values_are_returned_after_assignment (unexpected)
--- FAIL: StrictLoadingTest#test_strict_loading_with_reflection_is_ignored_in_validation_context (unexpected)
--- FAIL: AttributeMethodsTest#test_attribute_for_inspect_with_a_string (unexpected)
--- FAIL: CascadedEagerLoadingTest#test_eager_association_loading_of_stis_with_multiple_references (unexpected)
--- FAIL: IntegrationTest#test_to_param_should_return_string (unexpected)
--- FAIL: PostgresqlUUIDTest#test_invalid_uuid_dont_modify_before_type_cast (unexpected)
--- FAIL: IntegrationTest#test_cache_key_for_newer_updated_on (unexpected)
--- FAIL: RelationTest#test_build_with_array (unexpected)
--- FAIL: AttributeMethodsTest#test_user-defined_date_attribute_predicate (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16543981/activerecord/1724133123062/1724138413454)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128153 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2.0-rc]
- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16573707?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16573707?buildTab=artifacts#/activerecord) on master @ [18c4e23306bb99cc28c83bb3e5726f6f4279b798](https://github.com/cockroachdb/cockroach/commits/18c4e23306bb99cc28c83bb3e5726f6f4279b798):


```
--- FAIL: RelationTest#test_first_or_create_bang_with_no_parameters (unexpected)
--- FAIL: InheritanceTest#test_alt_destroy_all_within_inheritance (unexpected)
--- FAIL: AssociationsTest#test_append_composite_foreign_key_has_many_association (unexpected)
--- FAIL: AttributeMethodsTest#test_#alias_attribute_with_an_overridden_original_method_along_with_an_overridden_alias_method_in_a_parent_class_doesn't_issue_a_deprecation (unexpected)
--- FAIL: CalculationsTest#test_should_calculate_grouped_by_function_with_table_alias (unexpected)
--- FAIL: FinderTest#test_include_on_loaded_relation_with_composite_primary_key (unexpected)
--- FAIL: RelationTest#test_create_or_find_by (unexpected)
--- FAIL: CalculationsTest#test_should_reference_correct_aliases_while_joining_tables_of_has_many_through_association (unexpected)
--- FAIL: HasOneAssociationsTest#test_finding_with_interpolated_condition (unexpected)
--- FAIL: PersistenceTest#test_update_all (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_assign_ids_ignoring_blanks (unexpected)
--- FAIL: RelationTest#test_automatically_added_having_references (unexpected)
--- FAIL: BelongsToAssociationsTest#test_dont_find_target_when_foreign_key_is_null (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_has_and_belongs_to_many_in_a_namespaced_model_pointing_to_a_non_namespaced_model (unexpected)
--- FAIL: FinderTest#test_last_with_integer_and_reorder_should_use_sql_limit (unexpected)
--- FAIL: SignedIdTest#test_can_get_a_signed_ID_in_an_after_create (unexpected)
--- FAIL: RelationTest#test_loaded_relations_cannot_be_mutated_by_multi_value_methods (unexpected)
--- FAIL: CalculationsTest#test_#skip_query_cache!_for_a_grouped_calculation (unexpected)
--- FAIL: CalculationsTest#test_minimum_and_maximum_on_time_attributes (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasManyAssociation#test_parent_should_save_children_record_with_foreign_key_validation_set_in_before_save_callback (unexpected)
--- FAIL: CalculationsTest#test_pluck_type_cast_with_joins_without_table_name_qualified_column (unexpected)
--- FAIL: HasOneAssociationsTest#test_nullification_on_destroyed_association (unexpected)
--- FAIL: FinderTest#test_member_on_unloaded_relation_with_mismatched_class (unexpected)
--- FAIL: RelationTest#test_last (unexpected)
--- FAIL: RelationTest#test_extracted_association (unexpected)
--- FAIL: BasicsTest#test_using_table_name_qualified_column_names_unless_having_SELECT_list_explicitly (unexpected)
--- FAIL: RelationTest#test_where_with_take_memoization (unexpected)
--- FAIL: BelongsToAssociationsTest#test_belongs_to_touch_with_reassigning (unexpected)
--- FAIL: ActiveRecord::Migration::ColumnAttributesTest#test_native_types (unexpected)
--- FAIL: BelongsToAssociationsTest#test_build_with_block (unexpected)
--- FAIL: InheritanceTest#test_should_store_full_class_name_with_store_full_sti_class_option_enabled (unexpected)
--- FAIL: FinderTest#test_first_have_primary_key_order_by_default (unexpected)
--- FAIL: RelationTest#test_first_or_create_with_array (unexpected)
--- FAIL: BelongsToAssociationsTest#test_clearing_an_association_clears_the_associations_inverse (unexpected)
--- FAIL: FinderTest#test_find_by_nil_and_not_nil_attributes (unexpected)
--- FAIL: PersistenceTest#test_build_through_factory_with_block (unexpected)
--- FAIL: BelongsToAssociationsTest#test_building_the_belonging_object_with_implicit_sti_base_class (unexpected)
--- FAIL: CalculationsTest#test_maximum_with_not_auto_table_name_prefix_if_column_included (unexpected)
--- FAIL: FinderTest#test_first_have_determined_order_by_default (unexpected)
--- FAIL: CalculationsTest#test_sum_with_not_auto_table_name_prefix_if_column_included (unexpected)
--- FAIL: CalculationsTest#test_pluck_with_qualified_column_name (unexpected)
--- FAIL: InheritanceTest#test_alt_becomes_bang_resets_inheritance_type_column (unexpected)
--- FAIL: CalculationsTest#test_should_group_by_summed_association (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_habtm_selects_all_columns_by_default (unexpected)
--- FAIL: CalculationsTest#test_group_by_attribute_with_custom_type (unexpected)
--- FAIL: BasicsTest#test_.columns_hash_raises_an_error_if_the_record_has_an_empty_table_name (unexpected)
--- FAIL: HasOneAssociationsTest#test_association_enum_works_properly (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_runtimeAssertionsBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16573707/activerecord/1724305896221/1724313139480)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #129307 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2.1-rc]
- #128153 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2.0-rc]
- #126984 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/cmd/roachtest/tests/activerecord.go
+++ b/pkg/cmd/roachtest/tests/activerecord.go
@@ -27,8 +27,8 @@ var railsReleaseTagRegex = regexp.MustCompile(`^v(?P<major>\d+)\.(?P<minor>\d+)\
 
 // WARNING: DO NOT MODIFY the name of the below constant/variable without approval from the docs team.
 // This is used by docs automation to produce a list of supported versions for ORM's.
-var supportedRailsVersion = "7.1.3"
-var activerecordAdapterVersion = "v7.1.0"
+var supportedRailsVersion = "7.2.1"
+var activerecordAdapterVersion = "v7.2.0"
 
 // This test runs activerecord's full test suite against a single cockroach node.
 

